<!DOCTYPE HTML>
<html>
<head><meta name="generator" content="Hexo 3.8.0">
  <meta charset="utf-8">
  
  <title>扩增子测序 QIIME1 | Xizhihui&#39;s Notes</title>
  <meta name="author" content="Xizhihui">
  
  <meta name="description" content="1 构建mapping file,并验证是否有误mapping file记录着样品对应的barcode、primer、treatment group等信息，以tab作为列分割符。各列名如下：



column name
Description 




SampleID
样品名，数字、字母、点号


BarcodeSequence
barcode序列，区分样本


LinkerPrimerSequence
5’端引物


ReversePrimerSequence
3’端反向引物,如果有的话


Treatment
分组信息


Description
样品详细注释


DOB
日期或其他信息



下机reads组成：AdapterA - BarcodeSequence - LinkerPrimerSequence - Target - ReversePrimer - AdapterB">
  
  
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">

  <meta property="og:title" content="扩增子测序 QIIME1">
  <meta property="og:site_name" content="Xizhihui&#39;s Notes">

  
    <meta property="og:image" content="">
  

  
  
    <link href="/favicon.png" rel="icon">
  

  <!-- CSS -->
  <link rel="stylesheet" href="/css/themes/cosmo.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/font-awesome.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/style.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/responsive.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight-default.min.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/google-fonts.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/comment.css" media="screen" type="text/css">
  <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
	<script src="https://cdnjs.cloudflare.com/ajax/libs/es5-shim/4.5.9/es5-shim.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/es5-shim/4.5.7/es5-sham.min.js"></script>
  <![endif]-->

  <script src="/js/jquery-2.0.3.min.js"></script>
  
    <script src="/js/marked.js"></script>
    <script src="/js/comment.js"></script>
    <script src="/js/timeago.min.js"></script>
    <script src="/js/highlight.min.js"></script>
	<script src="/js/spin.min.js"></script>
  
  <!-- analytics -->
  



</head>
</html>
 <body>  
  <nav id="main-nav" class="navbar  navbar-default navbar-fixed-top" role="navigation">
    <div class="container">
      <button type="button" class="navbar-header navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
	<span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
       <a class="navbar-brand" href="/">Xizhihui&#39;s Notes</a>
      <div class="collapse navbar-collapse nav-menu">
		<ul class="nav navbar-nav">
		  
		  <li>
			<a href="/archives" title="All the articles.">
			  <i class="fa fa-archive"></i>Archives
			</a>
		  </li>
		  
		  <li>
			<a href="/categories" title="All the categories.">
			  <i class="fa fa-folder"></i>Categories
			</a>
		  </li>
		  
		  <li>
			<a href="/tags" title="All the tags.">
			  <i class="fa fa-tags"></i>Tags
			</a>
		  </li>
		  
		  <li>
			<a href="/about" title="About me.">
			  <i class="fa fa-user"></i>About
			</a>
		  </li>
		  
		</ul>
      </div>
    </div> <!-- container -->
</nav>
<div class="clearfix"></div>

  <div class="container">
  	<div class="content">
    	 


	
		<div class="page-header ">		
			<h1 class="title "> 扩增子测序 QIIME1</h1>
		</div>		
	






<div class="row post">
	<!-- cols -->
	
	<div id="top_meta"></div>
	<div class="col-md-9">
	

	<!-- content -->
	<div class="mypage">		
	  		

	  <h2 id="1-构建mapping-file-并验证是否有误"><a href="#1-构建mapping-file-并验证是否有误" class="headerlink" title="1 构建mapping file,并验证是否有误"></a>1 构建mapping file,并验证是否有误</h2><p>mapping file记录着样品对应的barcode、primer、treatment group等信息，以tab作为列分割符。各列名如下：</p>
<table>
<thead>
<tr>
<th>column name</th>
<th>Description </th>
</tr>
</thead>
<tbody>
<tr>
<td>SampleID</td>
<td>样品名，数字、字母、点号</td>
</tr>
<tr>
<td>BarcodeSequence</td>
<td>barcode序列，区分样本</td>
</tr>
<tr>
<td>LinkerPrimerSequence</td>
<td>5’端引物</td>
</tr>
<tr>
<td>ReversePrimerSequence</td>
<td>3’端反向引物,如果有的话</td>
</tr>
<tr>
<td>Treatment</td>
<td>分组信息</td>
</tr>
<tr>
<td>Description</td>
<td>样品详细注释</td>
</tr>
<tr>
<td>DOB</td>
<td>日期或其他信息</td>
</tr>
</tbody>
</table>
<p>下机reads组成：AdapterA - BarcodeSequence - LinkerPrimerSequence - Target - ReversePrimer - AdapterB</p>
<a id="more"></a>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># mapping.tsv</span></span><br><span class="line"><span class="comment"># SampleID	BarcodeSequence	LinkerPrimerSequence	Treatment	ReversePrimer 	Description</span></span><br><span class="line"><span class="comment"># Example mapping file for the QIIME analysis package.</span></span><br><span class="line">PC.354	AGCACGAGCCTA	YATGCTGCCTCCCGTAGGAGT	Control 	GCGCACGGGTGAGTA		Control_mouse__I.D._354</span><br><span class="line">PC.355	AACTCGTCGATG	YATGCTGCCTCCCGTAGGAGT	Control 	GCGCACGGGTGAGTA		Control_mouse__I.D._355</span><br></pre></td></tr></table></figure>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">validate_mapping_file -m mapping.tsv -o 01_map_out</span><br></pre></td></tr></table></figure>
<h2 id="2-拆分和质控-demultiplex-and-QC"><a href="#2-拆分和质控-demultiplex-and-QC" class="headerlink" title="2 拆分和质控(demultiplex and QC)"></a>2 拆分和质控(demultiplex and QC)</h2><p>扩增子测序通常是混池测序，所以我们要对序列拆分到各自所属样品，并进行质控。执行这一步的脚本为split_libaries_fastq.py，质控在里面默认进行。输入的文件格式可以是 fastq.gz 也可以是 fastq。</p>
<h3 id="2-0-只进行质控"><a href="#2-0-只进行质控" class="headerlink" title="2.0 只进行质控"></a>2.0 只进行质控</h3><p>把split_libaries_fastq.py的参数–barcode_type 设置为’not-barcoded’即可。</p>
<h3 id="2-1-Illumina-paired-end-fastq"><a href="#2-1-Illumina-paired-end-fastq" class="headerlink" title="2.1 Illumina paired end fastq"></a>2.1 Illumina paired end fastq</h3><p>使用joined_paired_ends.py合并双端数据，如果后续进行拆分序列的话，你需要提供barcodes.fastq文件。这个由extract_barcodes来实现，见3.2.8.<br>joined_paired_ends内部使用fast-join或SeqPrep来进行，默认为前者，-m SeqPrep可以指定后者。<br>后续拆分同Illumina single end fastq，见3.2.2.</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">join_paired_ends.py -f forward_reads.fastq -r reverse_reads.fastq -b barcodes.fastq -o join_fastq/</span><br></pre></td></tr></table></figure>
<h3 id="2-2-Illumina-single-end-fastq"><a href="#2-2-Illumina-single-end-fastq" class="headerlink" title="2.2 Illumina single end fastq"></a>2.2 Illumina single end fastq</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 如果有多个的话，用逗号分隔输入, 指定--rev_comp_mapping_barcodes是使用barcode.fastq.gz与mapping里面的方向反向互补了(3‘端的单端barcode)。</span></span><br><span class="line">split_libraries_fastq.py -i read1.fastq.gz -b barcode.fastq.gz --rev_comp_mapping_barcodes -o slout_q20/ -m map.txt -q 19</span><br></pre></td></tr></table></figure>
<h3 id="2-3-Illumina-fastq：multiple-lane"><a href="#2-3-Illumina-fastq：multiple-lane" class="headerlink" title="2.3 Illumina fastq：multiple lane"></a>2.3 Illumina fastq：multiple lane</h3><p>若想同时拆分Illumina的多个lane的数据，可以按下列命令进行，但是要指定每个lane对应的barcode和mapping file.所有的拆分结果将写入seqs.fna中，但log和histogram file是按lane区分. split_libraries_log.txt记录了运行的各项参数，如果结果不满意可调整参数重新运行。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">split_libraries_fastq.py -i s_6_sequence.fastq,s_7_sequence.fastq,s_8_sequence.fastq \</span><br><span class="line">        -o sl_out/ -b s_6_barcode.fastq,s_7_barcode.fastq,s_8_barcode.fastq \</span><br><span class="line">        -m s_6_map.txt,s_7_map.txt,s_8_map.txt</span><br></pre></td></tr></table></figure>
<h3 id="2-4-Illumina-qseq"><a href="#2-4-Illumina-qseq" class="headerlink" title="2.4 Illumina qseq"></a>2.4 Illumina qseq</h3><p>首先你需要知道你的barcode位于哪个reads上，长度多少。可以查看一个样品的不同reads文件。<br>然后使用process_qseq.py生成fastq文件，每个lane生成一个single fastq文件。<br>最后才是使用split_libraries_fastq.py进行拆分，类似于2.2.3。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 假设文件为s_1_1_0001_qseq.txt，s_1_2_0001_qseq.txt</span></span><br><span class="line">head -n 1 s_1_1_0001_qseq.txt</span><br><span class="line"><span class="comment"># 如果结果类似如下，第八列表示是read2, 且barcode长度为12bases。</span></span><br><span class="line"><span class="comment"># 如果不是，查看s_1_2_0001_qseq.txt</span></span><br><span class="line">M10     68      1       1       28680   29475   0       2       ACTCACGGTATTA   \_J\Sa^Y[ZYK`   0</span><br><span class="line">M10     68      1       1       19607   29475   0       2       AGACTGAGTACTA   PP\JJ\JQ`\RK^   1</span><br></pre></td></tr></table></figure>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 使用process_qseq.py生成fastq</span></span><br><span class="line"><span class="comment"># read1</span></span><br><span class="line">process_qseq.py -i input_dir/ -o output_dir -r 1</span><br><span class="line"><span class="comment"># read2,并且截断了12个bases,12是因为你的barcode长12 bases；只处理barcode序列</span></span><br><span class="line">process_qseq.py -i input_dir/ -o output_dir -r 2 -b 12</span><br></pre></td></tr></table></figure>
<h3 id="2-5-iseq"><a href="#2-5-iseq" class="headerlink" title="2.5 iseq"></a>2.5 iseq</h3><p>使用process_iseq.py进行处理, 每个lane将产生一个single read；后续使用split_libraries_fastq.py进行拆分，类似于2.2.3。</p>
<ul>
<li>如果barcode与序列在一起(较为常见)</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># s_6_1_sequences.txt,s_7_1_sequences.txt都是含barcode的；barcode长度为12</span></span><br><span class="line"><span class="comment"># HWI-ST753_50:6:1101:15435:9071#0/1:ACCAGACGATGCTACGGAGGGAGCTAG...</span></span><br><span class="line">process_iseq.py -i s_6_1_sequences.txt,s_7_1_sequences.txt -o out_dir/ -b 12</span><br></pre></td></tr></table></figure>
<ul>
<li>如果barcode在header里面</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># ACAGCTA长为7,但是你的barcode为ACAGCT为6，可以指定--barcode-length</span></span><br><span class="line"><span class="comment"># HWI-6X_9267:1:1:12:410#ACAGCTA/1:TACGTAGGGTGCGAGCGTTAATCGGAATTACTGGGCGTAAAGCGTGCG</span></span><br><span class="line">process_iseq.py -i s_6_1_sequences.txt,s_7_1_sequences.txt -o out_dir/ --barcode-length 6 --barcode_in_header</span><br></pre></td></tr></table></figure>
<h3 id="2-6-LEA-Seq数据"><a href="#2-6-LEA-Seq数据" class="headerlink" title="2.6 LEA-Seq数据"></a>2.6 LEA-Seq数据</h3><p>使用split_libraries_lea_seq.py来处理.</p>
<h3 id="2-7-处理已经拆分的数据"><a href="#2-7-处理已经拆分的数据" class="headerlink" title="2.7 处理已经拆分的数据"></a>2.7 处理已经拆分的数据</h3><ul>
<li>multiple_split_libraries_fastq.py: 进行质控</li>
<li>multiple_join_paired_ends.py:    合并已经拆分的paired ends</li>
<li>multiple_extract_barcodes.py: 去除已经拆分的fastq内部的barcode</li>
</ul>
<h3 id="2-8-处理不同的barcode策略"><a href="#2-8-处理不同的barcode策略" class="headerlink" title="2.8 处理不同的barcode策略"></a>2.8 处理不同的barcode策略</h3><ul>
<li>single fastq starts with the barcode sequence(单个fastq单个barcode)</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 假设in_seqs.fastq的前10个碱基是barcode</span></span><br><span class="line"><span class="comment"># 如果你想反向互补barcode的话，设置--rev_comp_bc1</span></span><br><span class="line">extract_barcodes.py -f in_seqs.fastq --bc1_len 10 -o parsed_barcodes/ --input_type barcode_single_end</span><br></pre></td></tr></table></figure>
<ul>
<li>two fastqs each starts with part of barcode: paired-end(双端fasta,单个barcode)</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 假设barcode部分分布在reads1,6 bases；部分分布在reads2,8 bases。</span></span><br><span class="line">extract_barcodes.py --input_type barcode_paired_end -f reads1.fastq -r reads2.fastq --bc1_len 6 --bc2_len 8 -o parsed_barcodes/</span><br><span class="line"><span class="comment"># 如果你想反向互补barcode的话，--rev_comp_bc1, --rev_comp_bc2</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 有时候，reads的方向不确定，你需要根据mapping file里面的primer来确定reads的方向</span></span><br><span class="line">extract_barcodes.py --input_type barcode_paired_end -m mapping_file.txt \</span><br><span class="line">        -a -f reads1.fastq -r reads2.fastq \</span><br><span class="line">        --bc1_len 6 --bc2_len 8 -o parsed_barcodes/</span><br></pre></td></tr></table></figure>
<ul>
<li>two index/barcode reads and two fastq reads (双端fastq, 2个barcode)</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 这里已经得到分离的barcode了，所以用不上read文件</span></span><br><span class="line"><span class="comment"># index1.fastq,index2.fastq是barcode文件。每个barcode长度为6.</span></span><br><span class="line">extract_barcodes.py --input_type barcode_paired_end -f index1.fastq \</span><br><span class="line">        -r index2.fastq --bc1_len 6 --bc2_len 6 -o parsed_barcodes/</span><br></pre></td></tr></table></figure>
<ul>
<li>single stitched read with barcodes on each end (单个fastq, 双端barcode)</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">extract_barcodes.py --input_type barcode_paired_stitched -f reads.fastq --bc1_len 6 --bc2_len 6 -o parsed_barcodes/</span><br></pre></td></tr></table></figure>
<ul>
<li>barcode在fastq文件的描述行内</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># @MCIC-SOLEXA_0051_FC:1:1:4065:1039#CGATGT/1</span></span><br><span class="line"><span class="comment"># NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</span></span><br><span class="line"><span class="comment"># +MCIC-SOLEXA_0051_FC:1:1:4065:1039#CGATGT/1</span></span><br><span class="line"><span class="comment"># KPPPQWWWWWQQ________BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB</span></span><br><span class="line"><span class="comment"># 这里barcode开头是#，长度为6</span></span><br><span class="line">extract_barcodes.py --input_type barcode_in_label --char_delineator <span class="string">"#"</span> -f in_seqs.fastq --bc1_len 6 -o parsed_barcodes/</span><br></pre></td></tr></table></figure>
<h3 id="2-9-处理Joint-Genome-Institute-fastq-文件"><a href="#2-9-处理Joint-Genome-Institute-fastq-文件" class="headerlink" title="2.9 处理Joint Genome Institute fastq 文件"></a>2.9 处理Joint Genome Institute fastq 文件</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># @MISEQ03:64:000000000-A2H3D:1:1101:14358:1530 1:N:0:TCCACAGGAGT</span></span><br><span class="line"><span class="comment"># TNCAGAGGGTGCAAGCGTTAATCGGAATTACTGGGCGTAAAGCGCGCGTAGGTGGTTTGTTA</span></span><br><span class="line"><span class="comment"># + </span></span><br><span class="line"><span class="comment"># .....</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># TCCACAGGAGT为barcode，长为11</span></span><br><span class="line">extract_reads_from_interleaved_file.py -i reads_to_extract.fastq -o extracted_reads</span><br><span class="line">extract_barcodes.py -f extracted_reads/forward_reads.fastq -o barcode_reads -c barcode_in_label --char_delineator <span class="string">'1:N:0:'</span> -l 11</span><br></pre></td></tr></table></figure>
<h2 id="3-去除嵌合体和OTU聚类"><a href="#3-去除嵌合体和OTU聚类" class="headerlink" title="3 去除嵌合体和OTU聚类"></a>3 去除嵌合体和OTU聚类</h2><p><a href="http://qiime.org/scripts/pick_otus.html?highlight=pick_otus" target="_blank" rel="noopener">pick_otus.py</a>整合十多种算法用于聚类OTU。<br>输出结果的第一列为OTU ID，第二列为该OTU的种子序列。以usearch为例,程序内部执行的基本过程如下：</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#1. 根据序列长度排序(rank)          2. 去除重复序列(去冗余)</span></span><br><span class="line"><span class="comment">#3. 根据序列丰度进行排序            4. 对序列进行去噪</span></span><br><span class="line"><span class="comment">#5. 去除嵌合体,先是de novo, 然后是ref-based</span></span><br><span class="line"><span class="comment">#6. 对序列进行合并                  7. 再根据丰度进行拍醋</span></span><br><span class="line"><span class="comment">#8. 基于排序进行聚类                9. 为聚类的结果添加对应的序列id</span></span><br><span class="line"><span class="comment">#10.对序列完成分类</span></span><br></pre></td></tr></table></figure>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><span class="line">pick_otus.py -i split_library_output/seqs.fna -o picked_otus_default</span><br><span class="line"><span class="comment"># 较为重要的参数</span></span><br><span class="line"><span class="comment"># -i &lt;input_fasta&gt;</span></span><br><span class="line"><span class="comment"># -m &lt;method&gt;		使用blast, uclust_ref, usearch_ref, usearch61_ref时需要指定-r参数.默认为uclust.</span></span><br><span class="line"><span class="comment"># -o &lt;out_dir&gt;, -r &lt;ref_seq&gt;</span></span><br><span class="line"><span class="comment"># -s &lt;similarity&gt;	默认为0.97</span></span><br><span class="line"><span class="comment"># -j &lt;ErrorRate&gt;</span></span><br><span class="line"><span class="comment"># -g &lt;min_count_in_cluster&gt; 当cluster的序列count数低于设定值时，该cluster被忽略</span></span><br><span class="line"><span class="comment"># -f &lt;chimeric_ref_seq&gt;		当-m usearch时需要指定</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># uclust-ref</span></span><br><span class="line">pick_otus.py -i seqs.fna -r refseqs.fasta -m uclust_ref --denovo_otu_id_prefix qiime_otu_</span><br><span class="line"></span><br><span class="line"><span class="comment"># cdhit</span></span><br><span class="line"><span class="comment"># -n 100将会执行预聚类，前100个bp完全相同的序列将被预聚类;使用-t参数则不指定-n num</span></span><br><span class="line">pick_otus.py -i seqs.fna -m cdhit -o cdhit_picked_otus/ -n 100</span><br><span class="line"></span><br><span class="line"><span class="comment"># blast</span></span><br><span class="line"><span class="comment"># 如果参考序列时预构建的blast序列,使用-b代替-r进行指定; -s指定相似度; -e指定错误发现率(blast专有)</span></span><br><span class="line">pick_otus.py -i seqs.fna -o blast_picked_otus_90_percent/ -m blast -r refseqs.fasta -s 0.90 -e 1e-30</span><br><span class="line"></span><br><span class="line"><span class="comment"># prefix-suffix</span></span><br><span class="line"><span class="comment"># 前50个bp后25个bp完全相同的序列将会被预聚类</span></span><br><span class="line">pick_otus.py -i seqs.fna -o prefix_suffix_picked_otus/ -m prefix_suffix -p 50 -u 25</span><br></pre></td></tr></table></figure>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># mothur</span></span><br><span class="line"><span class="comment"># -c 指定聚类算法, mothur专有,取值furtherest, neareast, average</span></span><br><span class="line">pick_otus.py -i seqs.aligned.fna -o mothur_picked_otus_nn/ -m mothur -c nearest</span><br><span class="line"></span><br><span class="line"><span class="comment"># usearch</span></span><br><span class="line"><span class="comment"># --suppress_reference_chimera_detection禁止基于参考序列的嵌合体发现</span></span><br><span class="line">pick_otus.py -i seqs.fna -m usearch --word_length 64 --db_filepath refseqs.fasta -o usearch_qf_results/ --minsize 2</span><br><span class="line"></span><br><span class="line"><span class="comment"># swarm</span></span><br><span class="line">pick_otus.py -i <span class="variable">$PWD</span>/seqs.fna -m swarm -o <span class="variable">$PWD</span>/picked_otus_swarm</span><br></pre></td></tr></table></figure>
<h2 id="4-选取OTU代表性序列"><a href="#4-选取OTU代表性序列" class="headerlink" title="4 选取OTU代表性序列"></a>4 选取OTU代表性序列</h2><p>输出为代表序列out_rep_seq.fasta。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">pick_rep_set.py -f split_library_output/seqs.fna -m most_abundant -i seqs_outs.txt -o out_rep_seq.fasta</span><br><span class="line"></span><br><span class="line"><span class="comment"># -i &lt;input_otus_file&gt;</span></span><br><span class="line"><span class="comment"># -f &lt;all_seqs_fna&gt;</span></span><br><span class="line"><span class="comment"># -m &lt;method&gt; 	random, longest, most_abundant, first(default)</span></span><br></pre></td></tr></table></figure>
<h2 id="5-进行物种分类"><a href="#5-进行物种分类" class="headerlink" title="5 进行物种分类"></a>5 进行物种分类</h2><p>使用assign_taxonomy.py进行物种分类；对于给定的序列集，该脚本会尝试对每条序列确定其分类，当前支持方法有BLAST、RDP分类器、RTAX、mothur和uclust。输出目录默认为methodname_assigned_taxonomy, 输出的结果文件第一列是序列ID，第二列是该序列所属物种分类，第三列是质量分数，以及其他信息，依据方法不同而不同。使用的参考序列最新版本可在qiime的resources页面下载。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># default：uclust</span></span><br><span class="line">assign_taxonomy.py -i repr_set_seqs.fasta -r ref_seq_set.fna -t id_to_taxonomy.txt</span><br><span class="line"><span class="comment"># SortMeRNA</span></span><br><span class="line">assign_taxonomy.py -i repr_set_seqs.fasta -r ref_seq_set.fna -t id_to_taxonomy.txt -m sortmerna</span><br><span class="line"><span class="comment"># Blast</span></span><br><span class="line"><span class="comment"># -e maximum e-value to record an assignment</span></span><br><span class="line">assign_taxonomy.py -i repr_set_seqs.fasta -r ref_seq_set.fna -t id_to_taxonomy.txt -e 0.01 -m blast</span><br><span class="line"><span class="comment"># RDP classifier</span></span><br><span class="line"><span class="comment"># 训练集默认为-t和-r指定的taxonomy和otus；如未指定，使用默认设置</span></span><br><span class="line"><span class="comment"># -c minimum confidence to record an assignment</span></span><br><span class="line">assign_taxonomy.py -i repr_set_seqs.fasta -m rdp -c 0.80</span><br><span class="line"><span class="comment"># RTAX</span></span><br><span class="line">assign_taxonomy.py -i rtax_repr_set_seqs.fasta -m rtax \</span><br><span class="line"> 		--read_1_seqs_fp read_1.seqs.fna \</span><br><span class="line"> 		--read_2_seqs_fp read_2.seqs.fna \</span><br><span class="line"> 		-r rtax_ref_seq_set.fna \</span><br><span class="line"> 		-t rtax_id_to_taxonomy.txt</span><br><span class="line"><span class="comment"># mothur</span></span><br><span class="line">assign_taxonomy.py -i mothur_repr_set_seqs.fasta -m mothur -r mothur_ref_seq_set.fna -t mothur_id_to_taxonomy.txt</span><br></pre></td></tr></table></figure>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># -o &lt;out_dir&gt; 				输出结果路径</span></span><br><span class="line"><span class="comment"># -i &lt;otu_rep_set.fasta&gt;	输入的代表序列</span></span><br><span class="line"><span class="comment"># -t &lt;id_to_taxonomy&gt; 		分类时参考序列ID对应的物种分类信息。</span></span><br><span class="line"><span class="comment"># -r &lt;ref_otu.fasta&gt;		分类时的参考序列</span></span><br><span class="line"><span class="comment"># -p &lt;rdp_train_set&gt; 		指定RDP的训练数据集，会覆盖-r,-t指定的参数</span></span><br><span class="line"><span class="comment"># -m &lt;method&gt; 				选取某个方法进行分类</span></span><br><span class="line"><span class="comment"># -b &lt;blast_db&gt; 			指定使用blast时的数据库, 与-r同义</span></span><br><span class="line"><span class="comment"># -e &lt;max_E_value&gt; 			使用blast时的最大错误值</span></span><br><span class="line"><span class="comment"># -c &lt;min_confidence&gt;		RDP分类时的最小置信值,超过该值才会认为属于该分类</span></span><br></pre></td></tr></table></figure>
<hr>
<h2 id="6-生成OTU表-biom文件"><a href="#6-生成OTU表-biom文件" class="headerlink" title="6 生成OTU表(biom文件)"></a>6 生成OTU表(biom文件)</h2><p>make_otu_table.py将对OTU进行汇总和注释，生成biom格式的OTU表。</p>
<h3 id="6-1-生成OTU表"><a href="#6-1-生成OTU表" class="headerlink" title="6.1 生成OTU表"></a>6.1 生成OTU表</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">make_otu_table.py -i otu_map.txt \</span><br><span class="line"> 		-t tax_assignments.txt \</span><br><span class="line"> 		-o otu_table.biom \</span><br><span class="line"> 		-m mapping_file.txt</span><br><span class="line"></span><br><span class="line"><span class="comment"># -i &lt;input_otu_map&gt;   	指定输入的OTU map，通常是pick_otus.py的结果文件</span></span><br><span class="line"><span class="comment"># -o &lt;out_biom_dir&gt; 	输出结果文件</span></span><br><span class="line"><span class="comment"># -m &lt;mapping_file&gt; 	指定mapping file,如果有的话</span></span><br><span class="line"><span class="comment"># -e &lt;id_file&gt;			指定排除OTU的identifiers文件</span></span><br></pre></td></tr></table></figure>
<h3 id="6-2-生成OTU表同时进行过滤"><a href="#6-2-生成OTU表同时进行过滤" class="headerlink" title="6.2 生成OTU表同时进行过滤"></a>6.2 生成OTU表同时进行过滤</h3><p>为了便于后续操作，一般会对OTU表进行过滤等操作.有两种方式，第一种方式是在构建的时候进行过滤，这通常是过滤嵌合体或未比对序列.另一种方式是生成otu表后，再进行过滤。其他过滤操作见3.7。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># identify_chimeric_seqs.py, align_seqs.py 生成不要的seqs对应的文件exclude.txt，然后在make_otu_table.py中以-e参数指定exclude.txt</span></span><br></pre></td></tr></table></figure>
<h2 id="7-对OTU表-biom-进行操作-过滤、拆分等"><a href="#7-对OTU表-biom-进行操作-过滤、拆分等" class="headerlink" title="7 对OTU表(biom)进行操作: 过滤、拆分等"></a>7 对OTU表(biom)进行操作: 过滤、拆分等</h2><h3 id="7-1-过滤"><a href="#7-1-过滤" class="headerlink" title="7.1 过滤"></a>7.1 过滤</h3><ul>
<li>Even sampling, rarefaction (稀释)</li>
</ul>
<p>这里的抽样方式有单次抽样、多次等差抽样(稀释抽样)、多次平均抽样。single_rareefaction.py只进行一次抽样, multiple_rarefactions.py进行多次抽样(每次抽样深度不同), multiple_rarefactions_even_depth.py进行多次抽样(每次抽样深度相同)。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 从otu-table中按每个样品100条序列随机抽取，并输出</span></span><br><span class="line">single_rarefaction.py -i otu_table.biom -o otu_table_even100.biom -d 100</span><br><span class="line"><span class="comment"># -k &lt;bool&gt;		指定是否保留全0的OTU,default:fasle</span></span><br><span class="line"><span class="comment"># --subsample_multinomial &lt;bool&gt; 是否是放回抽样,default:fasle</span></span><br><span class="line"></span><br><span class="line">multiple_rarefactions.py -i otu_table.biom -m 10 -x 140 -s 10 -n 2 -o rarefied_otu_tables/</span><br><span class="line"><span class="comment"># -m &lt;min_num&gt;		指定多次抽样的最低抽样数,seqs/sample, 抽样深度</span></span><br><span class="line"><span class="comment"># -x &lt;max_num&gt;		指定多次抽样的最大抽样数</span></span><br><span class="line"><span class="comment"># -s &lt;step_num&gt;		指定2次抽样的抽样次数的步长,差值</span></span><br><span class="line"><span class="comment"># -n &lt;num_reps&gt; 	每一步抽样的重复次数</span></span><br><span class="line"><span class="comment"># -k，--subsample_nultinomial 同单次抽样</span></span><br><span class="line"></span><br><span class="line">multiple_rarefactions_even_depth.py -i otu_table.biom -o rarefied_otu_tables/ -d 100 -n 10</span><br></pre></td></tr></table></figure>
<ul>
<li>过滤OTUs/observations</li>
</ul>
<p>过滤时可以根据mapping file的某列值进行过滤，也可以根据OTU的count数来过滤(singletons, doubletons…),丰度过滤。 根据run_number的切分和过滤有点相似，但是切分是所有run切分。过滤可以只过滤某个值。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 保留mapping file中某列的值等于指定值的OTUs, "Sample_Type:*,!Control_Blank"则表示删除</span></span><br><span class="line">filter_samples_from_otu_table.py -i split_otu_tables/otu_table_1.biom -o otu_table_run1_blank_samples.biom -m map.txt -s <span class="string">"Sample_Type:Control_Blank"</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 排除singleton</span></span><br><span class="line">filter_otus_from_otu_table.py -i otu_table.biom -o filtered_otu_table.biom -n 2</span><br><span class="line"></span><br><span class="line"><span class="comment"># -n &lt;min_count&gt;					如果该OTU的total counts低于指定值,排除该OTU</span></span><br><span class="line"><span class="comment"># -x &lt;max_count&gt; 					如果该OTU的total counts高于指定值,排除该OTU</span></span><br><span class="line"><span class="comment"># -s &lt;min_samples&gt;					如果属于该OTU的样品数低于指定值,排除该OTU</span></span><br><span class="line"><span class="comment"># -y &lt;max_samples&gt; 					如果属于该OTU的样品数超过指定值,排除该OTU</span></span><br><span class="line"><span class="comment"># -e &lt;out_ids_to_exclude_file&gt;		排除指定id的OTUs</span></span><br></pre></td></tr></table></figure>
<ul>
<li>过滤样本</li>
</ul>
<p>过滤样本可以基于丰度过滤，也可以基于metadata过滤，还可以基于ID过滤。过滤样本与过滤OTUs是相类似的.</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 保留total OTUs counts在[100,150]的样品</span></span><br><span class="line">filter_samples_from_otu_table.py -i otu_table.biom -o filtered_otu_table.biom -n 100 -x 150</span><br><span class="line"></span><br><span class="line"><span class="comment"># 保留mapping file中某列的值等于指定值的样品</span></span><br><span class="line">filter_samples_from_otu_table.py -i otu_table.biom -o filtered_otu_table.biom -m map.txt -s <span class="string">'Treatment:Control'</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 保留mapping file中某列的值==不等于==指定值的样品</span></span><br><span class="line">filter_samples_from_otu_table.py -i otu_table.biom -o filtered_otu_table.biom -m map.txt -s <span class="string">'Treatment:*,!Control'</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 保留在ids.txt的样品; 如果设定了--negate_sample_id_fp,则是删除在ids.txt的样品</span></span><br><span class="line">filter_samples_from_otu_table.py -i otu_table.biom -o filtered_otu_table.biom --sample_id_fp ids.txt</span><br></pre></td></tr></table></figure>
<ul>
<li>过滤分类群: 保留属于特定分类的OTU</li>
</ul>
<p>过滤分类群时，你可以有3种逻辑的过滤方式：or, not or, not but</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 保留属于__Bacteroidetes 或者 p__Firmicutes</span></span><br><span class="line">filter_taxa_from_otu_table.py -i otu_table.biom -o otu_table_bac_firm_only.biom -p p__Bacteroidetes,p__Firmicutes</span><br><span class="line"><span class="comment"># 保留不属于__Bacteroidetes和 p__Firmicutes</span></span><br><span class="line">filter_taxa_from_otu_table.py -i otu_table.biom -o otu_table_non_bac_firm.biom -n p__Bacteroidetes,p__Firmicutes</span><br><span class="line"><span class="comment"># 保留不属于c__Clostridia但是属于p__Firmicutes</span></span><br><span class="line">filter_taxa_from_otu_table.py -i otu_table.biom -o otu_table_all_firm_but_not_clos.biom -p p__Firmicutes -n c__Clostridia</span><br><span class="line"></span><br><span class="line"><span class="comment"># -p &lt;list, comma&gt; 保留</span></span><br><span class="line"><span class="comment"># -n &lt;list, comma&gt; 排除</span></span><br><span class="line"><span class="comment"># --metadata_field 指定用于过滤的metadata的列名</span></span><br></pre></td></tr></table></figure>
<h3 id="7-2-拆分和合并"><a href="#7-2-拆分和合并" class="headerlink" title="7.2 拆分和合并"></a>7.2 拆分和合并</h3><ul>
<li>基于mapping/metadata进行拆分</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">split_otu_table.py -i otu_table.biom -m Fasting_Map.txt -f Treatment -o per_study_otu_tables</span><br><span class="line">split_otu_table.py -i otu_table.biom -m Fasting_Map.txt -f Treatment,Color -o ./per_study_otu_tables/</span><br></pre></td></tr></table></figure>
<ul>
<li>基于taxonomy(物种分类)进行拆分</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 将otu_table依据Level3进行拆分</span></span><br><span class="line">split_otu_table_by_taxonomy.py -i otu_table.biom -L 3 -o ./L3/</span><br></pre></td></tr></table></figure>
<ul>
<li>合并</li>
</ul>
<p>在进行合并时，你要保证合并的两个OTUs表的OTU id和sample id是兼容的；如果两个表有重叠内容，将会计算重叠内容的值的和。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">merge_otu_tables.py -i otu_table1.biom,otu_table2.biom -o merged_otu_table.biom</span><br></pre></td></tr></table></figure>
<h3 id="7-3-统计汇总"><a href="#7-3-统计汇总" class="headerlink" title="7.3 统计汇总"></a>7.3 统计汇总</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 按样品进行统计汇总,取总的observations的所有counts</span></span><br><span class="line">biom summarize-table -i rich_sparse_otu_table.biom -o rich_sparse_otu_table_summary.txt</span><br><span class="line"><span class="comment"># 按样品进行汇总，但只取唯一的observations的counts</span></span><br><span class="line">biom summarize-table -i rich_sparse_otu_table.biom --qualitative-o rich_sparse_otu_table_summary.txt</span><br></pre></td></tr></table></figure>
<h3 id="7-4-对OTU表进行排序"><a href="#7-4-对OTU表进行排序" class="headerlink" title="7.4 对OTU表进行排序"></a>7.4 对OTU表进行排序</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 按照sample id进行排序, 字母排序</span></span><br><span class="line">sort_otu_table.py -i otu_table.biom -o sorted_otu_table.biom</span><br><span class="line"><span class="comment"># 按照mapping/metadata file的某个列排序</span></span><br><span class="line">sort_otu_table.py -i otu_table.biom -o dob_sorted_otu_table.biom -m Fasting_Map.txt -s Age</span><br><span class="line"><span class="comment"># 按照给定顺序的sample id进行排序</span></span><br><span class="line">sort_otu_table.py -i otu_table.biom -o sorted_otu_table.biom -l sample_id_list.txt</span><br></pre></td></tr></table></figure>
<h3 id="7-5-biom文件的其他处理"><a href="#7-5-biom文件的其他处理" class="headerlink" title="7.5 biom文件的其他处理"></a>7.5 biom文件的其他处理</h3><ul>
<li>添加metadata信息到biom文件中</li>
</ul>
<p>见<a href="http://biom-format.org/documentation/adding_metadata.html" target="_blank" rel="noopener">网页tutorial</a></p>
<ul>
<li>转换biom到其他格式</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># json/hdf5</span></span><br><span class="line">biom convert -i table.txt -o table.from_txt_json.biom --table-type=<span class="string">"OTU table"</span> --to-json</span><br><span class="line">biom convert -i table.txt -o table.from_txt_hdf5.biom --table-type=<span class="string">"OTU table"</span> --to-hdf5</span><br><span class="line"><span class="comment"># tsv, 并添加taxonomy物种分类信息; 指定--output-metadata-id "ConsensusLineage"可以把添加的taxonomy列名改为ConsensusLineage</span></span><br><span class="line">biom convert -i table.biom -o table.from_biom.txt --to-tsv --header-key taxonomy</span><br><span class="line"><span class="comment"># 把tsv转换回去</span></span><br><span class="line">biom convert -i otu_table.txt -o new_otu_table.biom --to-hdf5 --table-type=<span class="string">"OTU table"</span> --process-obs-metadata taxonomy</span><br></pre></td></tr></table></figure>
<h2 id="8-物种分类分析-群落组成分析-otu"><a href="#8-物种分类分析-群落组成分析-otu" class="headerlink" title="8 物种分类分析 /群落组成分析(otu)"></a>8 物种分类分析 /群落组成分析(otu)</h2><h3 id="8-1-OTU聚类分析和分类学分析"><a href="#8-1-OTU聚类分析和分类学分析" class="headerlink" title="8.1 OTU聚类分析和分类学分析"></a>8.1 OTU聚类分析和分类学分析</h3><p>此处是OTU聚类表(OTU table),OTU在每个样品中的分布信息就是OTU聚类分析；后续添加的分类注释信息就是分类学信息。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#1. 生成OTU聚类表, 见2.6</span></span><br><span class="line"><span class="comment">#2. 把表转换为tsv格式, 见2.7.5</span></span><br></pre></td></tr></table></figure>
<h3 id="8-2-OTU物种注释统计、OTU物种组成图"><a href="#8-2-OTU物种注释统计、OTU物种组成图" class="headerlink" title="8.2 OTU物种注释统计、OTU物种组成图"></a>8.2 OTU物种注释统计、OTU物种组成图</h3><p>使用summeriza_taxa_through_plot.py对OTU聚类表进行统计, 以各个分类水平进行统计每个样品在该分类水平上的OTU总数,并进行可视化. 该脚本由summarize_taxa.py和plot_taxa_summary.py组成。生成的otu_table_L*.txt文件是各个Level的水平物种丰度统计表。taxa_summary_plots文件夹则是物种丰度绘制图。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">summeriza_taxa_through_plots.py -o taxa_summary -i otu_table_w_tax.biom -m Fasting_Map.txt</span><br><span class="line"></span><br><span class="line"><span class="comment"># -i &lt;input_otu_table&gt; 	指定input文件, otu_table需要包含物种信息</span></span><br><span class="line"><span class="comment"># -p &lt;param_file&gt; 		指定特定中间过程脚本的参数</span></span><br><span class="line"><span class="comment"># -m &lt;mapping_file&gt;		如果指定了-c,-m必须指定</span></span><br><span class="line"><span class="comment"># -c &lt;mapping_category&gt;	指定进行统计的分组信息(按组统计)</span></span><br><span class="line"><span class="comment"># -s &lt;bool&gt; 			指定是否排序otu table,default: false</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># summeriza_taxa_through_plots.py内部依序执行的命令</span></span><br><span class="line">summarize_taxa.py -i otus/otu_table.biom -o taxa_summary</span><br><span class="line">plot_taxa_summary.py -i taxa_summary/otu_table_L2.txt,taxa_summary/otu_table_L3.txt,taxa_summary/otu_table_L4.txt,taxa_summary/otu_table_L5.txt,taxa_summary/otu_table_L6.txt -o taxa_summary/taxa_summary_plots/</span><br></pre></td></tr></table></figure>
<h3 id="8-3-OTU进化树分析-系统发育树"><a href="#8-3-OTU进化树分析-系统发育树" class="headerlink" title="8.3 OTU进化树分析(系统发育树)"></a>8.3 OTU进化树分析(系统发育树)</h3><p>构建OTU系统发育树分为三个步骤，分别使用三个脚本进行.生成的树可以用meta、Figtree等软件进行查看。系统发育树的构建有多种方法,可见<a href="http://www.biotrainee.com/thread-2253-1-1.html" target="_blank" rel="noopener">biotrainee</a></p>
<ol>
<li>序列比对</li>
</ol>
<p>可以选择多种方法进行序列比对,生成的文件夹命名为method_aligned的文件夹里,含有method_aligned.fasta, method_failures.fasta, method_log.txt</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">align_seqs.py -i otu_rep_set.fasta -o pynast_aligend/</span><br><span class="line"></span><br><span class="line"><span class="comment"># -m &lt;align_method&gt; 	指定比对序列的算法, pynast(default), infernal, clustalW, muscle, mafft</span></span><br><span class="line"><span class="comment"># -a &lt;pairwse_align&gt; 	指定在PyNAST中成对比对的算法, muscle, pair_hmm, clustal, blast, uclust(default), mafft</span></span><br><span class="line"><span class="comment"># -e &lt;min_length&gt;		进行比对的序列最短长度</span></span><br><span class="line"><span class="comment"># -p &lt;min_percent_id&gt; 	blast中最低的一致性占比(default:0.75)</span></span><br><span class="line"><span class="comment"># -d &lt;blast_db&gt;			当-m pynast时, 进行blast的数据库</span></span><br><span class="line"><span class="comment"># -t &lt;template_align&gt; 	指定预比对的文件;可在http://greengenes.lbl.gov/下载获得</span></span><br><span class="line"><span class="comment"># --muscle_max_memory</span></span><br><span class="line"></span><br><span class="line">align_seqs.py -i <span class="variable">$PWD</span>/unaligned.fna -t core_set_aligned.fasta.imputed -o <span class="variable">$PWD</span>/pynast_aligned/ -e 500 -p 95.0</span><br></pre></td></tr></table></figure>
<ol start="2">
<li>过滤比对结果中的高变异区</li>
</ol>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">filter_alignment.py -i pynast_aligned/otu_rep_set_aligned.fasta -o pynast_aligned/ --remove_outliers -g 0.95</span><br><span class="line"></span><br><span class="line"><span class="comment"># -m &lt;lane_mask_file&gt; 	指定在构建树时保留位点的lanemask文件</span></span><br><span class="line"><span class="comment"># -s &lt;suppress_lane_mask_filter&gt;</span></span><br><span class="line"><span class="comment"># -g &lt;allow_grap_frac&gt; 	指定某个位点所允许的最少gap数</span></span><br><span class="line"><span class="comment"># -t &lt;threshold&gt; 		指定认定为outlier的偏离标准偏差倍数, default=3</span></span><br><span class="line"><span class="comment"># -r &lt;remove_outliers&gt;	指定过滤outlier</span></span><br></pre></td></tr></table></figure>
<ol start="3">
<li>生成树</li>
</ol>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">make_phylogeny.py -i pynast_aligned/otu_rep_set_aligned_pfilter.fasta -o rep_set.tre</span><br><span class="line"></span><br><span class="line"><span class="comment"># -t &lt;tree_method&gt; 		指定进行建树的方法, clustalw, raxml_v730, muscle, fasttree(default), clearcut</span></span><br><span class="line"><span class="comment"># -l &lt;log_fp&gt;			log的存储位置, 不指定的话不会创建log</span></span><br><span class="line"><span class="comment"># -r &lt;root_method&gt; 		指定构建树的根(root)的方法, midpoint, tree_method_default(default)</span></span><br></pre></td></tr></table></figure>
<h3 id="8-4-OTU-Heatmap绘制（物种分类热图"><a href="#8-4-OTU-Heatmap绘制（物种分类热图" class="headerlink" title="8.4 OTU Heatmap绘制（物种分类热图)"></a>8.4 OTU Heatmap绘制（物种分类热图)</h3><p>使用make_otu_heatmap.py就对OTU表绘制热图，热图的每一行对应着一个OTU，每一列对应着一个样本。颜色代表了对应样本(列)的对应OTU(行)的丰度(counts).默认情况下，OTUs(行)将会使用UPGMA层次聚类，样品(列)将会按照在OTU表中出现的顺序进行排列。用户可以按照聚类树对行/列进行排序，也可以按照mapping file排序样品，对于后者，样品将会进行分组并聚类。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># default</span></span><br><span class="line">make_otu_heatmap.py -i otu_table.biom -o heatmap.pdf</span><br><span class="line"></span><br><span class="line"><span class="comment"># -t &lt;otu_tree&gt; 	指定用于OTUs排序的OTU tree</span></span><br><span class="line"><span class="comment"># -m &lt;mapping_file&gt; 指定用于样品排序的meta/mapping file</span></span><br><span class="line"><span class="comment"># -c &lt;category&gt;		指定用于样品排序的mapping file 列(分组)。</span></span><br><span class="line"><span class="comment"># -s &lt;sample_tree&gt; 	指定用于样品排序的sample_tree,可来源于upgma_cluster.py的输出</span></span><br><span class="line"><span class="comment"># -g &lt;image_type&gt; 	输出热图的格式, png, svg, pdf(default)</span></span><br><span class="line"><span class="comment"># --no_log_transform 不进行log转换, 0值被设为最小值的1/2</span></span><br><span class="line"><span class="comment"># --suppress_row/column_clustering 不进行upgma聚类,-t/-m设定了,不聚类的设定被忽略</span></span><br><span class="line"><span class="comment"># --absolute_abundance 不进行百分比化</span></span><br><span class="line"><span class="comment"># --color_scheme 	指定颜色主题</span></span><br><span class="line"><span class="comment"># --width,--height,--dpi 指定热图图片相关格式</span></span><br><span class="line"><span class="comment"># --obs_md_category 	observation metadata category to plot</span></span><br><span class="line"><span class="comment"># --obs_md_level 	The level of observation metadata to plot for hierarchical metadata</span></span><br></pre></td></tr></table></figure>
<h3 id="8-5-OTU-network分析"><a href="#8-5-OTU-network分析" class="headerlink" title="8.5 OTU network分析"></a>8.5 OTU network分析</h3><p>make_otu_network.py可以利用OTU和metadata/mapping file生成用于cytoscape可视化的网络文件，并对网络进行统计。进行网络分析是因为他可以展示和分析样品之间OTUs的分布，这对于展示大数据中样本的相似性和差别十分便捷。可视化结果展示的是根据样品之间共有OTUs数目的聚类结果，网络的加权值是每个OTU所含有的序列数。然后根据网络节点间的关联来统计其聚类模式，这其中用到G-test(我不懂啊)。</p>
<p>基于network分析的样品比较与基于tree的PCoA分析相得益彰。在多数研究中，这两个方法的结果总是相一致的，但是反映这数据的不同方面。network分析提供了系统发育的可视化呈现，PCoA-unifrac聚类提供了发育的聚类水平，这有可能在network观察到。从技术上讲，OTUs和样品是network中的两种节点, OTU-node通过edges同sample-node连接，这些edges就是samples在OTUs中的序列。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">make_otu_network.py -i otu_table.biom -m Fasting_Map.txt -o otu_network</span><br><span class="line"><span class="comment"># -b &lt;color_by&gt; 		指定依据mapping_file的特定列进行着色,可逗号分隔指定多个</span></span><br><span class="line"><span class="comment"># -k &lt;bg_color&gt; 		指定网络的背景色</span></span><br></pre></td></tr></table></figure>
<h2 id="9-α多样性分析-alpha-diversity-analysis"><a href="#9-α多样性分析-alpha-diversity-analysis" class="headerlink" title="9 α多样性分析 (alpha diversity analysis)"></a>9 α多样性分析 (alpha diversity analysis)</h2><p>α多样性的分析主要是通过计算<a href="http://scikit-bio.org/docs/latest/generated/skbio.diversity.alpha.html" target="_blank" rel="noopener">各项α多样性指数</a>来表征α多样性的, 比如Shannon指数、ACE指数、Chao1指数等。</p>
<h3 id="9-1-计算指数和对应的指数稀释曲线"><a href="#9-1-计算指数和对应的指数稀释曲线" class="headerlink" title="9.1 计算指数和对应的指数稀释曲线"></a>9.1 计算指数和对应的指数稀释曲线</h3><p>alpha diversity index and rarefaction curve. 在alpha_rarefaction.py中，可以生成稀释OTU table，基于稀释表计算α多样性矩阵，然后画出稀释曲线图。可计算的α多样性指数有：ace, berger_parker_d, brillouin_d, chao1, chao1_ci, dominance, doubles, enspie, equitability, esty_ci, fisher_alpha, gini_index, goods_coverage, heip_e, kempton_taylor_q, margalef, mcintosh_d, mcintosh_e, menhinick, michaelis_menten_fit, observed_otus, observed_species, osd, simpson_reciprocal, robbins, shannon, simpson, simpson_e, singles, strong, PD_whole_tree</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 指定需要计算的α多样性指数</span></span><br><span class="line"><span class="built_in">echo</span> <span class="string">'alpha_diversity:metrics simpson,shannon,PD_whole_tree,chao1,observed_species'</span> &gt; alpha_params.txt</span><br><span class="line">alpha_rarefaction.py -i otu_table.biom -o arare_max100/ -t rep_set.tre -m Fasting_Map.txt -e 100 -p alpha_params.txt</span><br><span class="line"><span class="comment"># -p &lt;parameter_fp&gt; 		指定过程脚本对应的参数</span></span><br><span class="line"><span class="comment"># -n &lt;num_steps&gt; 			指定进行抽样时的步数,不是步长, 是生成稀释OTU表的大小</span></span><br><span class="line"><span class="comment"># -w 						只打印过程脚本，不运行</span></span><br><span class="line"><span class="comment"># -a 						指定进行多线程</span></span><br><span class="line"><span class="comment"># -t &lt;tree_file&gt;			指定计算系统发育指数的tree</span></span><br><span class="line"><span class="comment"># --min_rare_depth &lt;num&gt;	指定抽样时的最小深度</span></span><br><span class="line"><span class="comment"># -e,--max_rare_depth &lt;num&gt; 指定抽样时的最大深度</span></span><br><span class="line"><span class="comment"># -O &lt;jobs_to_start&gt; 		指定跳转到第几步开始执行,重新执行时十分有用</span></span><br><span class="line"><span class="comment"># -f 						指定强制覆写文件</span></span><br><span class="line"><span class="comment"># --retain_intermediate_files 	指定保留中间过程文件</span></span><br><span class="line"></span><br><span class="line">multiple_rarefactions.py -i ../04_pick_open_ref_otus/otu_table_mc2.biom \</span><br><span class="line">		-m 10 -x 100 \</span><br><span class="line">		-s 9 -o ./alpha/rarefaction/</span><br><span class="line">alpha_diversity.py -i ./alpha/rarefaction/ </span><br><span class="line">		-o ./alpha/alpha_div/ \</span><br><span class="line">		--metrics simpson,shannon,PD_whole_tree,chao1,observed-species \</span><br><span class="line">		-t ../04_pick_open_ref_otus/rep_set.tre</span><br><span class="line">collate_alpha.py -i ./alpha/alpha_div/ \</span><br><span class="line">		-o ./alpha/alpha_div_collated/</span><br><span class="line">rm -r ./alpha/rarefaction/ ./alpha/alpha_div/</span><br><span class="line">make_rarefaction_plots.py -i ./alpha/alpha_div_collated/ \</span><br><span class="line">		-m ../map.tsv \</span><br><span class="line">		-o ./alpha/alpha_rarefaction_plots/</span><br></pre></td></tr></table></figure>
<h3 id="9-2-Rank-abundance曲线"><a href="#9-2-Rank-abundance曲线" class="headerlink" title="9.2 Rank-abundance曲线"></a>9.2 Rank-abundance曲线</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 比较多个样品的排序-丰度曲线图；如果想比较一个OTU表里面的所有样品, -s '*' 即可</span></span><br><span class="line">plot_rank_abundance_graph.py -i otu_table.biom -s <span class="string">'PC.354,PC.481,PC.364'</span> -x -v -o rank_abundance.pdf</span><br><span class="line"><span class="comment"># -s &lt;sample_name&gt;		画图的样品名</span></span><br><span class="line"><span class="comment"># -a,--absolute_counts 	指定图中使用的数据是绝对丰度,</span></span><br><span class="line"><span class="comment"># -n,--no_legend 		指定图中不画图例</span></span><br><span class="line"><span class="comment"># -x,--x_linear_scale 	指定x轴要进行线性缩放</span></span><br><span class="line"><span class="comment"># -y,--y_linear_scale 	指定y轴要进行线性缩放</span></span><br><span class="line"><span class="comment"># -f &lt;file_type&gt; 		指定输出图的类型, pdf(default), svg, png, eps</span></span><br><span class="line"><span class="comment"># -v 					输出过程信息</span></span><br></pre></td></tr></table></figure>
<h3 id="9-3-α多样性指数差异分析"><a href="#9-3-α多样性指数差异分析" class="headerlink" title="9.3 α多样性指数差异分析"></a>9.3 α多样性指数差异分析</h3><p>基于样品的多样性指数，可以检验组间样品的alpha diversity是否存在显著差异，检验方法可以是Wilcoxon秩和检验(2组样品)或者Kruskal-Willis秩和检验(多于两组).</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 汇总计算的所有alpha diversity index, 如果是执行了3.9.1,则不必执行本命令;</span></span><br><span class="line">collate_alpha.py -i alpha_div/ -o alpha_div_collated/</span><br><span class="line"></span><br><span class="line"><span class="comment"># 比较alpha diversity index，按SampleType作为分组，得到每个diversity index下的成对组间统计</span></span><br><span class="line">ls alpha/alpha_div_collated/*.txt | <span class="keyword">while</span> <span class="built_in">read</span> div;</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">name=`basename <span class="variable">$div</span> .txt`</span><br><span class="line">compare_alpha_diversity.py -i <span class="variable">$div</span> -m ../map.tsv -c SampleType -d 100 -o compare_alpha_div/<span class="variable">$name</span></span><br><span class="line"><span class="keyword">done</span></span><br><span class="line"><span class="comment"># 如果想把各个diversity index合在一张图,需要用额外的R代码</span></span><br></pre></td></tr></table></figure>
<h3 id="9-4-物种累积曲线-species-accumulation-curves"><a href="#9-4-物种累积曲线-species-accumulation-curves" class="headerlink" title="9.4 物种累积曲线  (species accumulation curves)"></a>9.4 物种累积曲线  (species accumulation curves)</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line"># 可以使用R的vegan包</span><br></pre></td></tr></table></figure>
<h2 id="10-β-多样性分析-beta-diversity-analysis"><a href="#10-β-多样性分析-beta-diversity-analysis" class="headerlink" title="10 β 多样性分析  (beta diversity analysis)"></a>10 β 多样性分析  (beta diversity analysis)</h2><p><a href="http://www.omicshare.com/forum/thread-3251-1-1.html" target="_blank" rel="noopener">微生物群落差异分析方法大揭秘</a></p>
<h3 id="10-0-一步法进行PCoA分析"><a href="#10-0-一步法进行PCoA分析" class="headerlink" title="10.0 一步法进行PCoA分析"></a>10.0 一步法进行PCoA分析</h3><p>beta_diversity_through_plot.py可以进行β多样性分析、主坐标轴分析，并画出3D PCoA图. 首先该脚本会随机对输入的otu_table.biom进行抽样，以达到抽平每个样品的序列数目的目的。然后计算各个表征β多样性的距离矩阵。最后基于这些距离矩阵进行主坐标轴分析，并对结果进行可视化。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">beta_diversity_through_plots.py -i otu_table.biom -m map.tsv -o beta_diversity -e 100</span><br><span class="line"><span class="comment"># -t &lt;tree_fp&gt; 		指定树的文件</span></span><br><span class="line"><span class="comment"># -p &lt;param_fp&gt; 	指定中间命令的参数文件</span></span><br><span class="line"><span class="comment"># -f,-w 			强制覆写和只打印命令不运行</span></span><br><span class="line"><span class="comment"># -a 				指定多线程运行</span></span><br><span class="line"><span class="comment"># -e &lt;seqs_per_sample&gt; 进行样品抽平的覆盖深度</span></span><br><span class="line"><span class="comment"># --supress_emperor_plots</span></span><br><span class="line"><span class="comment"># -O,--jobs_to_start 跳转到第几步开始运行。</span></span><br></pre></td></tr></table></figure>
<h3 id="10-1-计算β多样性距离矩阵"><a href="#10-1-计算β多样性距离矩阵" class="headerlink" title="10.1 计算β多样性距离矩阵"></a>10.1 计算β多样性距离矩阵</h3><p>beta_diversity.py支持计算多种距离矩阵, 使用-s参数查看支持的矩阵.由于unifrac，还需要提供发育树作为输入。一般来讲，由于unifrac有用到系统发育信息，所以一般推荐使用unifrac。定量的距离矩阵(weighted)更适合用于揭示由于群落物种丰度引起的群落差异,而定性的距离矩阵更适用于因环境不同所导致的群落差异。</p>
<p><strong>2.10.2-2.10.5等分析均基于本步骤生成的距离矩阵进行后续的分析。</strong></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 查看支持的矩阵类型,然后选择想要计算的距离矩阵</span></span><br><span class="line">beta_diversity.py -s</span><br><span class="line">metrics=unweighted_unifrac,weighted_unifrac,bray_curtis,binary_jaccard</span><br><span class="line">beta_diversity.py -i 04_pick_open_ref_otus/otu_table_mc2.biom \</span><br><span class="line">   		-o 05_beta_diversity \</span><br><span class="line">   		--metrics <span class="variable">$metrics</span> \</span><br><span class="line">   		-t 04_pick_open_ref_otus/rep_set.tre</span><br><span class="line"></span><br><span class="line">mv 05_beta_diversity/weighted_unifrac_otu_table_mc2.txt 05_beta_diversity/weighted_unifrac_distance_matrix.txt</span><br><span class="line"></span><br><span class="line"><span class="comment"># -i &lt;otu_table&gt; 	指定输入otu_table, 可以以文件夹的形式指定多个</span></span><br><span class="line"><span class="comment"># -r &lt;rows&gt; 		指定只计算指定行的距离矩阵,应该传入样品名,例如'S1,S3'之类的</span></span><br><span class="line"><span class="comment"># -m &lt;metrics&gt; 		指定要计算的beta diversity matrix类型, 可以用逗号分隔指定多个,默认: unweighted_unifrac,weighted_unifrac</span></span><br><span class="line"><span class="comment"># -s &lt;show_metrics&gt; 显示支持的距离矩阵类型</span></span><br><span class="line"><span class="comment"># -t &lt;tree_file&gt; 	指定输入的系统发育树</span></span><br><span class="line"><span class="comment"># -f &lt;full_tree&gt; 	指定使用全部的系统树;OTU table与系统树不对应的话,使用全部整个系统树会导致很多额外的结果.</span></span><br></pre></td></tr></table></figure>
<h3 id="10-2-PCoA分析"><a href="#10-2-PCoA分析" class="headerlink" title="10.2 PCoA分析"></a>10.2 PCoA分析</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># -i,-o还可以是文件夹用以进行批量计算</span></span><br><span class="line">principal_coordinates.py -i 05_beta_diversity/weighted_unifrac_dm.txt \</span><br><span class="line">		-o 05_beta_diversity/weighted_unifrac_pc.txt</span><br><span class="line"></span><br><span class="line"><span class="comment"># 画2d图</span></span><br><span class="line">make_2d_plots.py -i weighted_unifrac_pc.txt -m map.tsv -o 2d_plots</span><br><span class="line"><span class="comment"># -b &lt;color_by&gt;			以逗号分隔的metadata分类/组列名,用以指定着色分组。可以使用'col1&amp;&amp;col2'指定一致着色</span></span><br><span class="line"><span class="comment"># -k &lt;bg_color&gt;</span></span><br><span class="line"><span class="comment"># --ellipsoid_opacity 	指定透明度,只在jackknifed beta diversity中有用</span></span><br><span class="line"><span class="comment"># --ellipsoid_method	指定绘制的数据类型, IQR (default) 和 sdev, 只在jackknifed beta diversity中有用</span></span><br><span class="line"><span class="comment"># --master_pcoa			</span></span><br><span class="line"><span class="comment"># --scree 				绘制碎石图</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 画3d图</span></span><br><span class="line">make_emperor.py -i 05_beta_diversity/weighted_unifrac_pc.txt \</span><br><span class="line">		-o 05_beta_diversity/weighted_unifrac_emperor_pcoa_plot/ \</span><br><span class="line">		-m map.tsv</span><br><span class="line"></span><br><span class="line">make_emperor.py -i weighted_unifrac_pc.txt -m map.tsv \</span><br><span class="line">		-b <span class="string">'Treatment&amp;&amp;DOB,Treatment'</span> \ <span class="comment"># 依据metadata进行着色</span></span><br><span class="line">		-a DOB \ <span class="comment"># 指定一个轴为DOB</span></span><br><span class="line">		-x <span class="string">'DOB:20060000'</span> \ <span class="comment"># 当自定义轴DOB出现缺失值或值错误时,指定为20060000</span></span><br><span class="line">		-o emperor_colored</span><br><span class="line"><span class="comment"># 见https://biocore.github.io/emperor/build/html/scripts/make_emperor.html</span></span><br></pre></td></tr></table></figure>
<h3 id="10-3-PCA分析"><a href="#10-3-PCA分析" class="headerlink" title="10.3 PCA分析"></a>10.3 PCA分析</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">PCA分析基于OTU表，运用方差分解，将样本间的差异反映在二维坐标图上，坐标轴是两个主成分。样本组成越相似在图中则越聚集。</span><br><span class="line">待续.</span><br></pre></td></tr></table></figure>
<h3 id="10-4-NMDS分析"><a href="#10-4-NMDS分析" class="headerlink" title="10.4 NMDS分析"></a>10.4 NMDS分析</h3><p>与 PCoA 一样， 非度量多维尺度分析（ NMDS, nonmetric multidimensional scaling） 可以基于任何类型的距离/非相似性矩阵对对象（样本） 进行排序，其区别在于 NMDS 不再是特征根排序技术，也不再以排序轴承载更多的方差为目的，因此 NMDS 排序图可以任意旋转、中心化和倒置。 NMDS 分析在多维空间内构建对象的初始结构， 并用迭代程序不断地调整对象位置， 目标是尽可能的最小化应力函数（ stress function， 取值 0~1），应力函数是排序空间内对象结果与原始距离矩阵之间相异程度的度量。如果预先设定的排序轴数量比较少（如二维、三维），则在相同轴数的情况下， NMDS往往能够获得比 PCoA 更少失真的对象之间的关系。</p>
<p>R包vegan可以进行NMDS分析和作图。</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">nmds.py -i unweighted_unifrac_dm.txt -d 3 -o unweighted_unifrac_nmds.txt</span><br><span class="line"># -i,-o 			如果-i指定的是文件夹，那么就进行批量分析</span><br><span class="line"># -d &lt;dimensions&gt; 	指定NMDS分析的排序轴数量</span><br></pre></td></tr></table></figure>
<h3 id="10-5-ADONIS、ANOSIM分析"><a href="#10-5-ADONIS、ANOSIM分析" class="headerlink" title="10.5 ADONIS、ANOSIM分析"></a>10.5 ADONIS、ANOSIM分析</h3><p>通过compare_categories.py，我们可以比较不同组样品的聚类结果是否有显著性差异，该脚本提供多种方法，根据方法不同输出结果不同，但大多数是tsv文件。</p>
<p><strong>adonis</strong>, 又叫permanova, 是一种非参数检验方法。能够给出不同分组因素对样品差异的解释度（R值）与分组显著性（P值），它首先计算数据的相关性重心，然后计算这些重心点的平方偏差(R^2);最后基于原始数据的排列平方和进行F-test; 它以距离矩阵和mapping file作为输入，比如Unifrac聚类矩阵，通过指定分组列来进行比较。</p>
<p><strong>anosim</strong>,相似性分析，也是检验多组样本之间是否具有显著性差异，与adonis相似。但是它只对类别性变量有用，如果数据是连续型变量，推荐使用Mantel。它的计算公式为R=(rb-rw)/(N(N-1)/4)。其中rb指组间的所有距离的平均排名，rw指组内所有距离的平均排名。</p>
<p>adnois和anosim分析都只能衡量组与组之间整体是否有显著差异, 无法找到是哪个具体的OTU导致的该差异。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 看HOST_SUBJECT_ID分组的组间结果是否会有差异</span></span><br><span class="line">compare_categories.py --method adonis -i unweighted_unifrac_dm.txt -m map.txt -c HOST_SUBJECT_ID -o adonis_out -n 999</span><br><span class="line"><span class="comment"># --method &lt;method&gt; 		指定统计方法, adonis,anosim</span></span><br><span class="line"><span class="comment"># -i  &lt;distance_matrix&gt; 	指定输入的距离矩阵</span></span><br><span class="line"><span class="comment"># -m &lt;mapping_file&gt; 		指定输入的mapping file</span></span><br><span class="line"><span class="comment"># -c &lt;group/category&gt;		指定进行差异分析的分组方式</span></span><br><span class="line"><span class="comment"># -n &lt;permutations_num&gt;		指定进行计算p-value的排名数(前多少名)</span></span><br></pre></td></tr></table></figure>
<h3 id="10-6-样本聚类树图"><a href="#10-6-样本聚类树图" class="headerlink" title="10.6 样本聚类树图"></a>10.6 样本聚类树图</h3><p>样本聚类树可以从整体上描述和比较样本/分组见的相似性和差异性. 采用的聚类方法是UPGMA (unweighted pair group method with arithmetic mean, <a href="http://www.nmsr.org/upgma.htm)" target="_blank" rel="noopener">http://www.nmsr.org/upgma.htm)</a>, 这里以使用unweighted_unifrac聚类矩阵为例.</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"># unweighted_unifrac_dm.txt来源于3.10.1</span><br><span class="line">upgma_cluster.py -i unweighted_unifrac_dm.txt -o unweighted_unifrac_cluster.tre</span><br><span class="line"></span><br><span class="line"># 如果要探究生成的聚类树的健壮性的话，就进行jackknifed_beta_diversity分析</span><br><span class="line"># otu_table.biom是unweighted_unifrac_dm.txt的计算来源table</span><br><span class="line">jackknifed_beta_diversity.py -i otu_table.biom -o bdiv_jk100 -e 100 -m map.tsv -t rep_set.tre</span><br></pre></td></tr></table></figure>
<h3 id="10-7-Lefse分析"><a href="#10-7-Lefse分析" class="headerlink" title="10.7 Lefse分析"></a>10.7 Lefse分析</h3><p>线性判别分析（ LDA）效应量方法<br><a href="http://www.omicshare.com/forum/forum.php?mod=viewthread&amp;tid=563&amp;extra=page%3D1%26filter%3Dtypeid%26typeid%3D31" target="_blank" rel="noopener">http://www.omicshare.com/forum/forum.php?mod=viewthread&amp;tid=563&amp;extra=page%3D1%26filter%3Dtypeid%26typeid%3D31</a></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br></pre></td><td class="code"><pre><span class="line">conda install lefse -y</span><br><span class="line">conda install rpy2=2.8.6 -y</span><br><span class="line"><span class="comment"># 如果出现rpy2与matplotlib有conflict,删除matplotlib,先安装rpy2,后安装matplotlib</span></span><br><span class="line">conda remove matplotlib --force</span><br><span class="line"><span class="comment"># 如果有某个包总是断网下载不下来（得到该包的版本）,先安装这个包，再去安装rpy2</span></span><br><span class="line"><span class="comment"># https://anaconda.org/，下载对应包的对应版本</span></span><br><span class="line">conda install --use-local package_name</span><br><span class="line"></span><br><span class="line"><span class="comment"># 1. 先转化biom为TSV文件, 注意--header-key是taxonomy,非Taxonomy</span></span><br><span class="line">biom convert -i illumina/04_pick_open_ref_otus/otu_table_mc2_w_tax.biom -o otu_consensus.txt --to-tsv --header-key taxonomy --output-metadata-id <span class="string">"Consensus Lineage"</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 2. 转化为lefse的初始格式,设定分组、亚组和个体信息; 这里十分重要</span></span><br><span class="line">qiime2lefse.py --<span class="keyword">in</span> otu.txt \</span><br><span class="line"> 			   --md illumina/map.tsv \</span><br><span class="line"> 			   --out otu_lefse.txt \</span><br><span class="line"> 			   -c SampleType \</span><br><span class="line"> 			   -u Subject</span><br><span class="line"> 			   </span><br><span class="line"><span class="comment"># 3. 转换为lefse接受的格式</span></span><br><span class="line">lefse-format_input.py otu_lefse.txt otu_lefse.in -c 1 -s -1 -u 2</span><br><span class="line"></span><br><span class="line"><span class="comment"># 4. 走正常的分析流程</span></span><br><span class="line">run_lefse.py otu_lefse.in otu_lefse.res</span><br><span class="line">lefse-plot_res.py otu_lefse.res otu_lefse.png</span><br><span class="line">lefse-plot cladogram.py otu_lefse_res otu_lefse.cladogram.png --format png</span><br><span class="line">mkdir biomarkers_raw_image</span><br><span class="line">lefse-plot_features.py otu_lefse.in otu_lefse.res biomarkers_raw_images/</span><br></pre></td></tr></table></figure>
	  
	</div>

	<!-- recommended posts -->
	

	<div>
  	<center>
	<div class="pagination">
<ul class="pagination">
	 
				
    	<li class="prev"><a href="/2018/10/28/可变剪切-rMATS和rmats2sashimiplot/" class="alignleft prev"><i class="fa fa-arrow-circle-o-left"></i>Prev</a></li>
  		

        <li><a href="/archives"><i class="fa fa-archive"></i>Archive</a></li>

		
		   <li class="next"><a href="/2018/10/15/扩增子测序-QIIME2/" class="alignright next">Next<i class="fa fa-arrow-circle-o-right"></i></a></li>         
        
	
</ul>
</div>

    </center>
	</div>

    <!-- share -->
    
        
    <div class="bdsharebuttonbox">
        <a href="#" class="bds_more" data-cmd="more"></a>
        <a href="#" class="bds_weixin" data-cmd="weixin" title="分享到微信"></a>
        <a href="#" class="bds_tsina" data-cmd="tsina" title="分享到新浪微博"></a>
        <a href="#" class="bds_fbook" data-cmd="fbook" title="分享到Facebook"></a>
        <a href="#" class="bds_twi" data-cmd="twi" title="分享到Twitter"></a>
        <a href="#" class="bds_linkedin" data-cmd="linkedin" title="分享到linkedin"></a>
        <a href="#" class="bds_evernotecn" data-cmd="evernotecn" title="分享到印象笔记"></a>
        <a href="#" class="bds_youdao" data-cmd="youdao" title="分享到有道云笔记"></a>
        <a href="#" class="bds_copy" data-cmd="copy" title="分享到复制网址"></a>
    </div>
    <script>
        window._bd_share_config={"common":{"bdSnsKey":{},"bdText":"","bdMini":"2","bdMiniList":false,"bdPic":"","bdStyle":"1","bdSize":"24"},"share":{}};
        with(document)0[(getElementsByTagName('head')[0]||body).appendChild(createElement('script')).src='http://bdimg.share.baidu.com/static/api/js/share.js?v=89860593.js?cdnversion='+~(-new Date()/36e5)];
    </script>


        

    
	
	<!-- comment -->
	
<section id="comment">
  <h2 class="title">Comments</h2>
    	 
	 <div id="comment-thread"></div>
	 <div id="loading-spin"></div>
	 <script type="text/javascript">
	   getComments({
           type: "github" ? "github" : "github",       
	       user: "xizhihui",
	       repo: "",
		   client_id: "",
           client_secret: "",
		   no_comment: "No comments yet. Press the button and go to comment now!",
		   go_to_comment: "Go to comment",
		   no_issue: "no_issue",
		   issue_title: "扩增子测序 QIIME1",
		   issue_id: "",
		   btn_class: "btn btn-primary",
		   comments_target: "#comment-thread",
		   loading_target: "#loading_spin"
		   });
	 </script>
  
</section>


	</div> <!-- col-md-9/col-md-12 -->
		
	
	<div id="side_meta">
		<div class="col-md-3" id="post_meta"> 

	<!-- date -->
	
	<div class="meta-widget">
	<i class="fa fa-clock-o"></i>
	2018-10-15 
	</div>
	

	<!-- categories -->
    
	<div class="meta-widget">
	<a data-toggle="collapse" data-target="#categorys"><i class="fa fa-folder"></i></a>	
    <ul id="categorys" class="tag_box list-unstyled collapse in">
          
  <li>
    </li><li><a href="/Categories/Bioinformatics/">Bioinformatics<span>36</span></a></li>
  

    </ul>
	</div>
	

	<!-- tags -->
	
	<div class="meta-widget">
	<a data-toggle="collapse" data-target="#tags"><i class="fa fa-tags"></i></a>		  
    <ul id="tags" class="tag_box list-unstyled collapse in">	  
	    
  <li><a href="/Tags/软件和包/">软件和包<span>22</span></a></li> <li><a href="/Tags/扩增子/">扩增子<span>5</span></a></li>
    </ul>
	</div>
	

	<!-- toc -->
	<div class="meta-widget">
	
	   <a data-toggle="collapse" data-target="#toc"><i class="fa fa-bars"></i></a>
	   <div id="toc" class="toc collapse in">
			<ol class="toc-article"><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#1-构建mapping-file-并验证是否有误"><span class="toc-article-text">1 构建mapping file,并验证是否有误</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#2-拆分和质控-demultiplex-and-QC"><span class="toc-article-text">2 拆分和质控(demultiplex and QC)</span></a><ol class="toc-article-child"><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#2-0-只进行质控"><span class="toc-article-text">2.0 只进行质控</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#2-1-Illumina-paired-end-fastq"><span class="toc-article-text">2.1 Illumina paired end fastq</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#2-2-Illumina-single-end-fastq"><span class="toc-article-text">2.2 Illumina single end fastq</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#2-3-Illumina-fastq：multiple-lane"><span class="toc-article-text">2.3 Illumina fastq：multiple lane</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#2-4-Illumina-qseq"><span class="toc-article-text">2.4 Illumina qseq</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#2-5-iseq"><span class="toc-article-text">2.5 iseq</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#2-6-LEA-Seq数据"><span class="toc-article-text">2.6 LEA-Seq数据</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#2-7-处理已经拆分的数据"><span class="toc-article-text">2.7 处理已经拆分的数据</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#2-8-处理不同的barcode策略"><span class="toc-article-text">2.8 处理不同的barcode策略</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#2-9-处理Joint-Genome-Institute-fastq-文件"><span class="toc-article-text">2.9 处理Joint Genome Institute fastq 文件</span></a></li></ol></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#3-去除嵌合体和OTU聚类"><span class="toc-article-text">3 去除嵌合体和OTU聚类</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#4-选取OTU代表性序列"><span class="toc-article-text">4 选取OTU代表性序列</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#5-进行物种分类"><span class="toc-article-text">5 进行物种分类</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#6-生成OTU表-biom文件"><span class="toc-article-text">6 生成OTU表(biom文件)</span></a><ol class="toc-article-child"><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#6-1-生成OTU表"><span class="toc-article-text">6.1 生成OTU表</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#6-2-生成OTU表同时进行过滤"><span class="toc-article-text">6.2 生成OTU表同时进行过滤</span></a></li></ol></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#7-对OTU表-biom-进行操作-过滤、拆分等"><span class="toc-article-text">7 对OTU表(biom)进行操作: 过滤、拆分等</span></a><ol class="toc-article-child"><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#7-1-过滤"><span class="toc-article-text">7.1 过滤</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#7-2-拆分和合并"><span class="toc-article-text">7.2 拆分和合并</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#7-3-统计汇总"><span class="toc-article-text">7.3 统计汇总</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#7-4-对OTU表进行排序"><span class="toc-article-text">7.4 对OTU表进行排序</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#7-5-biom文件的其他处理"><span class="toc-article-text">7.5 biom文件的其他处理</span></a></li></ol></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#8-物种分类分析-群落组成分析-otu"><span class="toc-article-text">8 物种分类分析 /群落组成分析(otu)</span></a><ol class="toc-article-child"><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#8-1-OTU聚类分析和分类学分析"><span class="toc-article-text">8.1 OTU聚类分析和分类学分析</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#8-2-OTU物种注释统计、OTU物种组成图"><span class="toc-article-text">8.2 OTU物种注释统计、OTU物种组成图</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#8-3-OTU进化树分析-系统发育树"><span class="toc-article-text">8.3 OTU进化树分析(系统发育树)</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#8-4-OTU-Heatmap绘制（物种分类热图"><span class="toc-article-text">8.4 OTU Heatmap绘制（物种分类热图)</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#8-5-OTU-network分析"><span class="toc-article-text">8.5 OTU network分析</span></a></li></ol></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#9-α多样性分析-alpha-diversity-analysis"><span class="toc-article-text">9 α多样性分析 (alpha diversity analysis)</span></a><ol class="toc-article-child"><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#9-1-计算指数和对应的指数稀释曲线"><span class="toc-article-text">9.1 计算指数和对应的指数稀释曲线</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#9-2-Rank-abundance曲线"><span class="toc-article-text">9.2 Rank-abundance曲线</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#9-3-α多样性指数差异分析"><span class="toc-article-text">9.3 α多样性指数差异分析</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#9-4-物种累积曲线-species-accumulation-curves"><span class="toc-article-text">9.4 物种累积曲线  (species accumulation curves)</span></a></li></ol></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#10-β-多样性分析-beta-diversity-analysis"><span class="toc-article-text">10 β 多样性分析  (beta diversity analysis)</span></a><ol class="toc-article-child"><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#10-0-一步法进行PCoA分析"><span class="toc-article-text">10.0 一步法进行PCoA分析</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#10-1-计算β多样性距离矩阵"><span class="toc-article-text">10.1 计算β多样性距离矩阵</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#10-2-PCoA分析"><span class="toc-article-text">10.2 PCoA分析</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#10-3-PCA分析"><span class="toc-article-text">10.3 PCA分析</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#10-4-NMDS分析"><span class="toc-article-text">10.4 NMDS分析</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#10-5-ADONIS、ANOSIM分析"><span class="toc-article-text">10.5 ADONIS、ANOSIM分析</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#10-6-样本聚类树图"><span class="toc-article-text">10.6 样本聚类树图</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#10-7-Lefse分析"><span class="toc-article-text">10.7 Lefse分析</span></a></li></ol></li></ol>
		</div>
	
	</div>
	
    <hr>
	
</div><!-- col-md-3 -->

	</div>
		

</div><!-- row -->



	</div>
  </div>
  <div class="container-narrow">
  <footer> <p>
  &copy; 2018 Xizhihui
  
      with help from <a href="http://hexo.io/" target="_blank">Hexo</a> and <a href="http://getbootstrap.com/" target="_blank">Twitter Bootstrap</a>. Theme by <a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind</a>.    
</p> </footer>
</div> <!-- container-narrow -->
  


  
<a id="gotop" href="#">   
  <span>▲</span> 
</a>

<script src="/js/jquery.imagesloaded.min.js"></script>
<script src="/js/gallery.js"></script>
<script src="/js/bootstrap.min.js"></script>
<script src="/js/main.js"></script>
<script src="/js/search.js"></script> 


<link rel="stylesheet" href="/fancybox/jquery.fancybox.css" media="screen" type="text/css">
<script src="/fancybox/jquery.fancybox.pack.js"></script>
<script type="text/javascript">
(function($){
  $('.fancybox').fancybox();
})(jQuery);
</script>



   <script type="text/javascript">      
     var search_path = "search.xml";
	 if (search_path.length == 0) {
	 	search_path = "search.xml";
	 }
	 var path = "/" + search_path;
     searchFunc(path, 'local-search-input', 'local-search-result');
   </script>


<!-- syntax highlighting -->

  <script>
  marked.setOptions({
    highlight: function (code, lang) {
        return hljs.highlightAuto(code).value;
    }
  });
  function Highlighting(){
    var markdowns = document.getElementsByClassName('markdown');
    for(var i=0;i<markdowns.length;i++){
        if(markdowns[i].innerHTML) markdowns[i].innerHTML =marked(markdowns[i].innerHTML);
    }
  }
  window.addEventListener('DOMContentLoaded', Highlighting, false);
  window.addEventListener('load', Highlighting, false);
  </script>


</body>
   </html>
