<!DOCTYPE html>
<html>
<head>
	<title>XZH's Notes</title>
	<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
	<meta name="author" content="xizhihui" />
	<meta name="viewport" content="width=device-width, initial-scale=1" />
	<link rel="stylesheet" type="text/css" href="/statics/reset.css">
	<link rel="stylesheet" type="text/css" href="/statics/header_footer.css"></link>
	<link rel="stylesheet" type="text/css" href="/statics/default.css">
	<link rel="stylesheet" type="text/css" href="/statics/main.css">
</head>
<body>
	<header class="Header">
		<nav>
			<h3>XZH</h3>
			<ul>
				<li><a href="/index.html#Timeline">Timeline</a></li><li><a href="/index.html#Category">Category</a></li><li><a href="/index.html#Tags">Tags</a></li><li><a href="/index.html#About">About</a></li>
			</ul>
		</nav>
	</header>
	<div class="Motto">
		<h2>XZH's Notes</h2>
		<h4>The future comes for you ~ ~ ~</h4>
	</div>

	<main>
		<nav id="table"></nav>
		<content id="content" class="content">
			<h1 id="subread">subread</h1>
<p>subread是个套件,里面有subread aligner, subjunc aligner, featureCounts, exactSNP.</p>
<p>subread aligner可以用于DNA-seq和RNA-seq.当用于RNA-seq时,subread只适用于差异分析;对于检测基因组变异如可变剪接之类的,需要reads的完全比对,这时候可以使用subjunc进行比对.在比对RNA-seq数据时,subread不会取检测exon-exon junctions的存在,只会把exon-spanning eads的最大可比对区域作为比对结果.但是,如果只是进行差异分析的话,subread的结果足以进行.subread的比对上reads可能会比subjunc多.</p>
<h2 id="_1">安装</h2>
<div class="codehilite"><pre><span></span><span class="c1"># 下载源文件</span>
https://sourceforge.net/projects/subread/files/
tar zxvf subread-1.6.2.tar.gz
make -f Makefile.Linux

<span class="c1"># in R</span>
biocLite<span class="o">(</span><span class="s1">&#39;Rsubread&#39;</span><span class="o">)</span>
</pre></div>


<h2 id="subread-aligner">subread aligner</h2>
<div class="codehilite"><pre><span></span>subread-buildindex -o my_index ref.fa           <span class="c1">#允许单染色体文件</span>
<span class="c1"># -B                    把index进行分割多个文件;会覆盖-M的设置</span>
<span class="c1"># -c                    构建color-space索引</span>
<span class="c1"># -f &lt;int&gt;              去除高度重复序列的重复阈值,高于该阈值的重复序列将被去除</span>
<span class="c1"># -F                    构建一个完全索引,小鼠的完全索引达14G</span>
<span class="c1"># -M &lt;int&gt;              可以用int MB的RAM</span>
<span class="c1"># -o &lt;string&gt;           index的basename</span>

subread-align -t <span class="m">1</span> -i my_index -r reads.fastq -o subread_results.bam
<span class="c1"># -t &lt;int&gt;              进行比对的数据类型, 0是RNA-seq, 1是DNA-seq</span>
<span class="c1"># --multiMapping        允许多比对</span>
<span class="c1"># -B &lt;int&gt;              允许的多比对数指定为int</span>
<span class="c1"># -T &lt;int&gt;              指定用int个threads</span>
<span class="c1"># -I &lt;int&gt;              检测的indel长度最长为int bp</span>
<span class="c1"># -d &lt;int&gt;              minFragLength指定为50</span>
<span class="c1"># -D 600                maxFragLength指定为600</span>
<span class="c1"># -r fastq文件            输入的reads</span>
<span class="c1"># -R read2.fastq        如果是paired-end,由此指定read2</span>
<span class="c1"># -a &lt;string&gt;           指定注释文件</span>
<span class="c1"># -A &lt;string&gt;           指定参考基因组和注释之间的chr名对应关系的文件</span>
<span class="c1">#                       第一列是注释里面的染色体名,第二列是参考基因组对应的染色体名.不需要列名.</span>
</pre></div>


<div class="codehilite"><pre><span></span><span class="c1"># -b                    当比对color space文件时,输出的比对结果保持正常格式,而非color-space</span>
<span class="c1"># -F &lt;string&gt;           指定注释文件的格式, &#39;GTF&#39;, &#39;GFF&#39;, &#39;SAF&#39;(在Rsubread里默认这个)</span>
<span class="c1"># -m &lt;int&gt;              指定一致性阈值,当比对一致性超过该阈值则认为比对上</span>
<span class="c1"># -n &lt;int&gt;              允许的最大错配数</span>
<span class="c1"># -p &lt;int&gt;              当是paired-end时,两个reads的比对一致性最低值,应该小于-m的指定值</span>
<span class="c1"># -P 3/6/33/64          指定使用的phred质量值,3指phred+33; 在Rsubread中,33值phred+33.</span>

<span class="c1"># 其他有关genomic variance的设置见handbook.</span>

<span class="c1"># 比对microRNA-seq reads</span>
<span class="c1"># 注释下载: http://www.mirbase.org/</span>
subread-buildindex -F -B -o mm10_full_index mm10.fa
subread-align -t <span class="m">1</span> -i mm10_full_index <span class="se">\</span>
              -n <span class="m">35</span> -m <span class="m">4</span> -M <span class="m">3</span> -T <span class="m">10</span> -I <span class="m">0</span> <span class="se">\</span>
              --multiMapping -B <span class="m">10</span> <span class="se">\</span>
              -r miRNA_reads.fastq -o results.sam
</pre></div>


<h2 id="subjunc">subjunc</h2>
<div class="codehilite"><pre><span></span>subjunc -i my_index -r rnaseq-reads1.txt -R rnaseq-read2.txt -o subjunc_result
</pre></div>


<h2 id="featurecounts">featureCounts定量</h2>
<div class="codehilite"><pre><span></span>featureCounts -T 5 -a annotation.gtf -t exon -g gene_id \
              -o counts.txt mapping_results_SE.sam
              # 可以有多个bam文件
# paired-end
featureCounts -p -a annotation.gtf -t exon -g gene_id \
              # -P -d 50 -D 600     # 指定了fragment长度
              # -B  # 不考虑fragment,但两个read都要比对上,才计数
              # -C  # 排除嵌合(chimeric fragments)
              -o counts.txt mapping_results_PE.bam
</pre></div>


<h2 id="snp-calling">SNP calling</h2>
<div class="codehilite"><pre><span></span>exactSNP <span class="o">[</span>options<span class="o">]</span> -i input -g reference genome -o output

<span class="c1"># -a &lt;file&gt;             指定vcf格式的snp注释文件</span>
<span class="c1"># -b                    指定输入文件是bam文件</span>
<span class="c1"># -f &lt;float&gt;            指定含SNP位点区域的错配碱基的最小区域</span>
<span class="c1"># -g &lt;file&gt;             指定参考基因组文件, fasta格式</span>
<span class="c1"># -i &lt;file&gt;             指定输入文件,SAM或BAM,如果是BAM,指定-b选项</span>
<span class="c1"># -n &lt;int&gt;              指定出现错配碱基的最小数目</span>
<span class="c1"># -Q &lt;int&gt;              指定在50x测序深度是的SNPcalling的q-value的cutoff值</span>
<span class="c1"># -r &lt;int&gt;              判定为SNp时需要的最少比对上reads数目</span>
<span class="c1"># -s &lt;int&gt;              指定作为SNP位点的碱基质量值阈值</span>
<span class="c1"># -t &lt;int&gt;              read两端切除的碱基数</span>
<span class="c1"># -T &lt;int&gt;              Threads</span>
<span class="c1"># -x &lt;int&gt;              判断为SNP的位点的最大测序深度</span>
</pre></div>


<h2 id="r">在R里面</h2>
<div class="codehilite"><pre><span></span><span class="kn">library</span><span class="p">(</span>Rsubread<span class="p">)</span>
buildindex<span class="p">(</span>basename<span class="o">=</span><span class="s">&#39;my_index&#39;</span><span class="p">,</span> reference<span class="o">=</span><span class="s">&#39;genome.fa&#39;</span><span class="p">)</span>
align<span class="p">(</span>index<span class="o">=</span><span class="s">&#39;my_index&#39;</span><span class="p">,</span> type<span class="o">=</span><span class="s">&#39;dna&#39;</span><span class="p">,</span>
      readfile1<span class="o">=</span><span class="s">&#39;reads.txt.gz&#39;</span><span class="p">,</span>
      output_file<span class="o">=</span><span class="s">&#39;rsubread.bam&#39;</span><span class="p">,</span>
      nthreads<span class="o">=</span><span class="m">5</span><span class="p">,</span>
      indels<span class="o">=</span><span class="m">16</span><span class="p">,</span>
      unique<span class="o">=</span><span class="bp">F</span><span class="p">,</span> nBestLocations<span class="o">=</span><span class="m">3</span><span class="p">)</span>
<span class="c1"># pairedd-end</span>
align<span class="p">(</span>index<span class="o">=</span><span class="s">&#39;my_index&#39;</span><span class="p">,</span>
      readfile1<span class="o">=</span><span class="s">&#39;reads1.fq.fz&#39;</span><span class="p">,</span> readfile2<span class="o">=</span><span class="s">&#39;reads2.fq.gz&#39;</span><span class="p">,</span>
      type<span class="o">=</span><span class="s">&#39;dna&#39;</span><span class="p">,</span>
      output_file<span class="o">=</span><span class="s">&#39;rsubread.bam&#39;</span><span class="p">,</span>
      minFragLength<span class="o">=</span><span class="m">50</span><span class="p">,</span> maxFragLength<span class="o">=</span><span class="m">600</span><span class="p">)</span>

subjunc<span class="p">(</span>index<span class="o">=</span><span class="s">&#39;my_index&#39;</span><span class="p">,</span> readfile1<span class="o">=</span><span class="s">&#39;rnaseq-reads.txt.gz&#39;</span><span class="p">,</span> output_file<span class="o">=</span><span class="s">&#39;subjunc_results.bam&#39;</span><span class="p">)</span>

featureCounts<span class="p">(</span>files<span class="o">=</span><span class="s">&quot;mapping_results_SE.sam&quot;</span><span class="p">,</span> nthreads<span class="o">=</span><span class="m">5</span><span class="p">)</span>
<span class="c1"># 提供参考注释的话</span>
featureCounts<span class="p">(</span>files<span class="o">=</span><span class="s">&quot;mapping_results_SE.sam&quot;</span><span class="p">,</span>annot.ext<span class="o">=</span><span class="s">&quot;annotation.gtf&quot;</span><span class="p">,</span>
isGTFAnnotationFile<span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span>GTF.featureType<span class="o">=</span><span class="s">&quot;exon&quot;</span><span class="p">,</span>GTF.attrType<span class="o">=</span><span class="s">&quot;gene_id&quot;</span><span class="p">)</span>

<span class="c1"># paired-end</span>
featureCounts<span class="p">(</span>files<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="s">&#39;read1_mapped.bam&#39;</span><span class="p">,</span> <span class="s">&#39;read2_mapped.bam&#39;</span><span class="p">))</span>
featureCounts<span class="p">(</span>files<span class="o">=</span><span class="s">&#39;PE_mapped.bam&#39;</span><span class="p">,</span> isPairedEnd<span class="o">=</span><span class="bp">T</span><span class="p">,</span>
             requireBothEndsMapped<span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span> <span class="c1"># 要两个reads都比对上才计数</span>
             countChimericFragments<span class="o">=</span><span class="kc">FALSE</span><span class="p">)</span> <span class="c1"># 不考虑chimeric fragment</span>
</pre></div>


<h2 id="_2">其他小功能</h2>
<table>
<tr><td>repair</td><td>把paired-end.bam文件里面的reads成对放置</td></tr>
<tr><td>coverageCount</td><td>计算基因组中某个区域的read coverage</td></tr>
<tr><td>flattenGTF</td><td>把GTF里面的某个feature的行提取成SAF文件.如提取所有的exons注释</td></tr>
<tr><td>promoterRegions</td><td>只在Rsubread里面有,提取出每个基因的启动子区域位置.</td></tr>
<tr><td>removeDunp</td><td>去除SAM文件里面的重复reads</td></tr>
<tr><td>subread-fullscan</td><td>提取目标序列在染色体上的高度同源序列位置</td></tr>
</table>
		</content>

	</main>
	<footer>
		<p> Powered by xizhihui </p>
		<p> Thanks to <a href="https://www.hifreud.com/">hifreud</a> the theme is learned from his site. </p>
	</footer>

<script type="text/javascript" src="/js/index.json"></script>
<script type="text/javascript" src="/js/category.json"></script>
<script type="text/javascript" src="/js/tags.json"></script>
<script type="text/javascript" src="/js/about.json"></script>
<script type="text/javascript" src="/js/index.js"></script>
<script type="text/javascript">
	// listen to click event on navigator in header
	// document
	// 	.querySelector("div#head_right ul")
	// 	.addEventListener("click", change_page)
	var table = [['h2', '_1', '安装'], ['h2', 'subread-aligner', 'subread aligner'], ['h2', 'subjunc', 'subjunc'], ['h2', 'featurecounts', 'featureCounts定量'], ['h2', 'snp-calling', 'SNP calling'], ['h2', 'r', '在R里面'], ['h2', '_2', '其他小功能']]
	var html = format_article_table(table)
	var table_ele = document.querySelector("main #table")
	table_ele.innerHTML = html
	// window.onload = evt => resize_nav(table_ele)
	adaption(document, window)

</script>
</body>
</html>