<!DOCTYPE html>
<html>
<head>
	<title>XZH's Notes</title>
	<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
	<meta name="author" content="xizhihui" />
	<meta name="viewport" content="width=device-width, initial-scale=1" />
	<link rel="stylesheet" type="text/css" href="/statics/reset.css">
	<link rel="stylesheet" type="text/css" href="/statics/header_footer.css"></link>
	<link rel="stylesheet" type="text/css" href="/statics/default.css">
	<link rel="stylesheet" type="text/css" href="/statics/main.css">
</head>
<body>
	<header class="Header">
		<nav>
			<h3>XZH</h3>
			<ul>
				<li><a href="/index.html#Timeline">Timeline</a></li><li><a href="/index.html#Category">Category</a></li><li><a href="/index.html#Tags">Tags</a></li><li><a href="/index.html#About">About</a></li>
			</ul>
		</nav>
	</header>
	<div class="Motto">
		<h2>XZH's Notes</h2>
		<h4>The future comes for you ~ ~ ~</h4>
	</div>

	<main>
		<nav id="table"></nav>
		<content id="content" class="content">
			<h2 id="raw_data">测序raw_data质控和处理软件</h2>
<h2 id="1-fastqc">1. fastqc</h2>
<div class="codehilite"><pre><span></span>fastqc input_path/*.fastq<span class="o">(</span>.gz<span class="o">)</span> -o output_path -p <span class="m">10</span>
</pre></div>


<h2 id="2-cutadapt">2. cutadapt</h2>
<p>https://cutadapt.readthedocs.io/en/stable/guide.html#basic-usage</p>
<p>基本的用法如下，你需要将序列改成对应的3‘端adapter。支持压缩文件作为输入和输出。如果你的输出是fasta的话，cutadapt不会进行去除adapter，而是进行转化。</p>
<div class="codehilite"><pre><span></span>    <span class="c1"># 对于pair-end</span>
    cutadapt -a ADAPTER_FWD -A ADAPTER_REV -o out.1.fastq -p out.2.fastq reads.1.fastq reads.2.fastq

    <span class="c1"># 对于single-end</span>
    cutadapt -a AACCGGTT -o output.fastq input.fastq

    cutadapt -a AACCGGTT input.fastq &gt; output.fastq <span class="m">2</span>&gt;&lt;report <span class="nv">class</span><span class="o">=</span><span class="s2">&quot;txt&quot;</span>&gt;&lt;/report&gt;
    <span class="c1"># 使用管道的话，输入文件位置以-代替</span>
    tail -n <span class="m">4</span> input.fastq <span class="p">|</span> cutadapt -a AACCGGTT - &gt; output.fastq

    <span class="c1"># 双端时，-u指前一个，-U指后一个。指定切除特定长度的碱基, -5 表示切除3‘端（反向切除5个碱基），5表示切除5’端（正向切除5个碱基）</span>
    cutadapt -u <span class="m">5</span> -o trimmed.fastq input.fastq

    <span class="c1"># 单双端都可用. 切除碱基质量低于10的碱基（3‘端），使用的是phred quality+33.如果要使用+64版本，--quality-base=64指定</span>
    cutadapt -q <span class="m">10</span> -o output.fastq input.fastq
    <span class="c1"># 切除5‘端低于15的，3’端低于10的碱基</span>
    cutadapt -q <span class="m">15</span>,10 -o output.fastq input.fastq

    <span class="c1"># 把read切成指定长度，注意是从3‘端切掉.如果要从两端切掉，使用-u参数</span>
    cutadapt -l <span class="m">120</span> -o output.fastq input.fastq

    <span class="c1"># 给read name添加内容，下方指令把read1 改成read1 we found adapter1如果ACGT在该read1找到的话</span>
    cutadapt -a <span class="nv">adapter1</span><span class="o">=</span>ACGT -y <span class="s1">&#39; we found {names}&#39;</span> input.fastq

    <span class="c1"># 如果要添加上length字符的话. output.fastq =&gt; &quot;&gt;read1 length=120&quot;</span>
    cutadapt -l <span class="m">120</span> -o output.fastq --length-tag <span class="s1">&#39;length=&#39;</span> input.fastq

    <span class="c1"># -m length: 只输出长度大于length的reads</span>
    <span class="c1"># --too-short-output FILE: 把-m删除的reads输入到FILE</span>
    <span class="c1"># -M length：只输出短于length的reads</span>
</pre></div>


<div class="codehilite"><pre><span></span>    <span class="c1"># --too-long-output FILE: 把-M删除的reads出入到FILE</span>
    <span class="c1"># --untrimmed-output FILE</span>
    <span class="c1"># --discard-trimmed:去掉含有接头的reads</span>
    <span class="c1"># --discard-untrimmed：去掉不含接头的reads</span>
</pre></div>


<p>除了去除adapter，cutadapt可以做的事情有很多，比如3‘端低质量碱基的切除，reads过滤等等。下表是指定adapter类型的对应指令。</p>
<table>
<tr><td>接头类型</td><td>命令</td><td>接头类型</td><td>命令</td></tr>
<tr><td>3’adapter</td><td>-a ADAPTER</td><td>5'adapter</td><td>-g ADAPTER</td></tr>
<tr><td>Anchored 3'adapter</td><td>-a ADAPTER$</td><td>Anchored 5'adapter</td><td>-g ^ADAPTER</td></tr>
<tr><td>5'or3'(both possible)</td><td>-b ADAPTER</td><td>Linked adapter</td><td>-a ADAPTER1...ADAPTER2</td></tr>
</table>

<div class="codehilite"><pre><span></span><span class="c1">#   1. Unconditional base removal with --cut</span>
<span class="c1">#   2. Quality trimming (-q)</span>
<span class="c1">#   3. Adapter trimming (-a, -b, -g and uppercase versions)</span>
<span class="c1">#   4. Read shortening (--length)</span>
<span class="c1">#   5. N-end trimming (--trim-n)</span>
<span class="c1">#   6. Length tag modification (--length-tag)</span>
<span class="c1">#   7. Read name suffix removal (--strip-suffix)</span>
<span class="c1">#   8. Addition of prefix and suffix to read name (-x/--prefix and -y/--suffix)</span>
<span class="c1">#   9. Double-encode the sequence (only colorspace)</span>
<span class="c1">#   10.Replace negative quality values with zero (zero capping, only colorspace)</span>
</pre></div>


<h2 id="3-fastx-toolkit">3. Fastx-toolkit</h2>
<table>
<thead>
<tr>
<th>用途</th>
<th>用途</th>
<th>用途</th>
</tr>
</thead>
<tbody>
<tr>
<td>01.去除接头</td>
<td>02.去除低质量碱基</td>
<td>03.转换成fasta</td>
</tr>
<tr>
<td>04.碱基质量统计</td>
<td>05.生成碱基质量箱图</td>
<td>06.生成碱基质量分布图</td>
</tr>
<tr>
<td>07.重命名read</td>
<td>08.输出指定长度read(前后切除)</td>
<td>09.去重reads</td>
</tr>
<tr>
<td>10.去除人工reads(artifact)</td>
<td>11.生成反向互补序列</td>
<td>12.对fasta文件进行格式化</td>
</tr>
<tr>
<td>13.DNA序列与RNA互换</td>
<td>14.生成reads长度分布图</td>
<td>15.依据barcode进行reads分类成不同文件</td>
</tr>
</tbody>
</table>
<h3 id="31">3.1.去除接头</h3>
<div class="codehilite"><pre><span></span>fastx_clipper -a AGATCGGAAGAGCACACG -l <span class="m">25</span> -d <span class="m">0</span> -Q <span class="m">33</span> -i SRR306394_1.fastq -o 

 <span class="c1"># [-a ADAPTER] =接头序列（默认为CCTTAAGG）</span>
 <span class="c1"># [-l N]       = 忽略那些碱基数目少于N的reads，默认为5</span>
 <span class="c1"># [-d N]       = 保留接头序列后的N个碱基默认  -d 0</span>
 <span class="c1"># [-c]         = 放弃那些没有接头的序列.</span>
 <span class="c1"># [-C]         = 只保留没有接头的序列.</span>
 <span class="c1"># [-k]         = 报告只有接头的序列.</span>
 <span class="c1"># [-n]         = 保留有N多序列，默认不保留</span>
 <span class="c1"># [-v]         =详细-报告序列编号</span>
 <span class="c1"># [-z]         =压缩输出.</span>
 <span class="c1"># [-D]       = 输出调试结果.</span>
 <span class="c1"># [-M N]   =要求最小能匹配到接头的长度N，如果和接头匹配的长度小于N不修剪</span>
 <span class="c1"># [-i INFILE]  = 输入文件</span>
 <span class="c1"># [-o OUTFILE] = 输出文件</span>
</pre></div>


<h3 id="32">3.2. 去除低质量碱基</h3>
<div class="codehilite"><pre><span></span> fastq_quality_filter -q <span class="m">20</span> -p <span class="m">80</span> -Q <span class="m">33</span> -i input.fastq -o output.fastq

<span class="c1"># [-q N]       = 最小的需要留下的质量值</span>
<span class="c1"># [-p N]       = 每个reads中最少有百分之多少的碱基需要有-q的质量值</span>
<span class="c1"># [-z]         =压缩输出</span>
<span class="c1"># [-v]         =详细-报告序列编号，如果使用了-o则报告会直接在STDOUT，如果没有则输入到STDERR</span>
</pre></div>


<h3 id="33-fasta">3.3. 转换成fasta</h3>
<div class="codehilite"><pre><span></span>fastq_to_fasta <span class="o">[</span>-h<span class="o">]</span> <span class="o">[</span>-r<span class="o">]</span> <span class="o">[</span>-n<span class="o">]</span> <span class="o">[</span>-v<span class="o">]</span> <span class="o">[</span>-z<span class="o">]</span> <span class="o">[</span>-i INFILE<span class="o">]</span> <span class="o">[</span>-o OUTFILE<span class="o">]</span>
<span class="o">[</span>-h<span class="o">]</span>         显示帮助信息
<span class="o">[</span>-r<span class="o">]</span>         使用连续的数字来重命名序列
<span class="o">[</span>-n<span class="o">]</span>         保持序列中的未知碱基N, 默认为丢弃这样的序列
<span class="o">[</span>-v<span class="o">]</span>         打印出转换的reads数
<span class="o">[</span>-z<span class="o">]</span>         用GZIP压缩输出
<span class="o">[</span>-i INFILE<span class="o">]</span>  指定输入fastq
<span class="o">[</span>-o OUTFILE<span class="o">]</span> 指定输出fasta
</pre></div>


<h3 id="34">3.4. 碱基质量统计</h3>
<div class="codehilite"><pre><span></span>fastx_quality_stats <span class="o">[</span>-h<span class="o">]</span> <span class="o">[</span>-i INFILE<span class="o">]</span> <span class="o">[</span>-o OUTFILE<span class="o">]</span>
<span class="o">[</span>-h<span class="o">]</span>            显示帮助信息
<span class="o">[</span>-i INFILE<span class="o">]</span>     指定输入文件, 如果输入为fasta, 只统计碱基分布
<span class="o">[</span>-o OUTFILE<span class="o">]</span>    指定输出文件
</pre></div>


<h3 id="35">3.5. 生成碱基质量箱图</h3>
<div class="codehilite"><pre><span></span>fastq_quality_boxplot_graph.sh <span class="o">[</span>-i INPUT.TXT<span class="o">]</span> <span class="o">[</span>-t TITLE<span class="o">]</span> <span class="o">[</span>-p<span class="o">]</span> <span class="o">[</span>-o OUTPUT<span class="o">]</span>
<span class="o">[</span>-p<span class="o">]</span>            生成图片格式为Generate PostScript <span class="o">(</span>.PS<span class="o">)</span>, 默认PNG图片
<span class="o">[</span>-i INPUT.TXT<span class="o">]</span>  输入文件名, 应为fastx_quality_stats的结果
<span class="o">[</span>-o OUTPUT<span class="o">]</span>     输出文件名
<span class="o">[</span>-t TITLE<span class="o">]</span>      箱线图的标题
</pre></div>


<h3 id="36">3.6. 生成碱基质量分布图</h3>
<div class="codehilite"><pre><span></span>fastx_nucleotide_distribution_graph.sh <span class="o">[</span>-i INPUT.TXT<span class="o">]</span> <span class="o">[</span>-t TITLE<span class="o">]</span> <span class="o">[</span>-p<span class="o">]</span> <span class="o">[</span>-o OUTPUT<span class="o">]</span>
<span class="o">[</span>-p<span class="o">]</span>            生成图片格式为Generate PostScript <span class="o">(</span>.PS<span class="o">)</span>, 默认PNG图片
<span class="o">[</span>-i INPUT.TXT<span class="o">]</span>  输入文件名, 应为fastx_quality_stats的结果
<span class="o">[</span>-o OUTPUT<span class="o">]</span>     输出文件名
<span class="o">[</span>-t TITLE<span class="o">]</span>      柱状图的标题
</pre></div>


<h3 id="37read">3.7.重命名read</h3>
<div class="codehilite"><pre><span></span>fastx_renamer <span class="o">[</span>-n TYPE<span class="o">]</span> <span class="o">[</span>-h<span class="o">]</span> <span class="o">[</span>-z<span class="o">]</span> <span class="o">[</span>-v<span class="o">]</span> <span class="o">[</span>-i INFILE<span class="o">]</span> <span class="o">[</span>-o OUTFILE<span class="o">]</span>
<span class="o">[</span>-n TYPE<span class="o">]</span>    重命名类型:
             SEQ - 使用核苷酸序列作为名称
             COUNT - 使用简单的计数重命名
<span class="o">[</span>-h<span class="o">]</span>         显示帮助信息
<span class="o">[</span>-z<span class="o">]</span>         用GZIP压缩输出
<span class="o">[</span>-i INFILE<span class="o">]</span>  指定输入文件, 默认为标准输入
<span class="o">[</span>-o OUTFILE<span class="o">]</span> 指定输出文件, 默认为标准输出
</pre></div>


<h3 id="38-read">3.8. 输出指定长度read（前后切除）</h3>
<div class="codehilite"><pre><span></span>fastx_trimmer <span class="o">[</span>-h<span class="o">]</span> <span class="o">[</span>-f N<span class="o">]</span> <span class="o">[</span>-l N<span class="o">]</span> <span class="o">[</span>-z<span class="o">]</span> <span class="o">[</span>-v<span class="o">]</span> <span class="o">[</span>-i INFILE<span class="o">]</span> <span class="o">[</span>-o OUTFILE<span class="o">]</span>
<span class="o">[</span>-h<span class="o">]</span>            显示帮助信息
<span class="o">[</span>-f N<span class="o">]</span>          从前面第几个碱基开始保留, 默认值为1
<span class="o">[</span>-l N<span class="o">]</span>          保留几个碱基, 默认值为全部保留
<span class="o">[</span>-z<span class="o">]</span>            用GZIP压缩输出
<span class="o">[</span>-i INFILE<span class="o">]</span>     指定输入文件, 默认为标准输入
<span class="o">[</span>-o OUTFILE<span class="o">]</span>    指定输出文件, 默认为标准输出
</pre></div>


<h3 id="39reads">3.9.去重reads</h3>
<div class="codehilite"><pre><span></span>fastx_collapser <span class="o">[</span>-h<span class="o">]</span> <span class="o">[</span>-v<span class="o">]</span> <span class="o">[</span>-i INFILE<span class="o">]</span> <span class="o">[</span>-o OUTFILE<span class="o">]</span>
<span class="o">[</span>-h<span class="o">]</span>         显示帮助信息
<span class="o">[</span>-v<span class="o">]</span>         打印出输入/输出的统计摘要
<span class="o">[</span>-i INFILE<span class="o">]</span>  指定输入文件, 默认为标准输入
<span class="o">[</span>-o OUTFILE<span class="o">]</span> 指定输出文件, 默认为标准输出
</pre></div>


<h3 id="310readsartifact">3.10.去除人工reads（artifact）</h3>
<div class="codehilite"><pre><span></span>fastx_artifacts_filter <span class="o">[</span>-h<span class="o">]</span> <span class="o">[</span>-v<span class="o">]</span> <span class="o">[</span>-z<span class="o">]</span> <span class="o">[</span>-i INFILE<span class="o">]</span> <span class="o">[</span>-o OUTFILE<span class="o">]</span>
<span class="o">[</span>-h<span class="o">]</span>         显示帮助信息
<span class="o">[</span>-v<span class="o">]</span>         报告处理的序列数目
                 如果有<span class="s1">&#39;-o&#39;</span>, 在屏幕上标准输出
                 如果无<span class="s1">&#39;-o&#39;</span>, 信息在屏幕上输出, 则报告在标准错误上输出
<span class="o">[</span>-z<span class="o">]</span>         用GZIP压缩输出
<span class="o">[</span>-i INFILE<span class="o">]</span>  指定输入文件, 默认为标准输入
<span class="o">[</span>-o OUTFILE<span class="o">]</span> 指定输出文件, 默认为标准输出
</pre></div>


<h3 id="311">3.11. 生成反向互补序列</h3>
<div class="codehilite"><pre><span></span> fastx_reverse_complement -i input.fastq -o output.fastq
</pre></div>


<h3 id="312fasta">3.12.对fasta文件进行格式化</h3>
<div class="codehilite"><pre><span></span>fasta_formatter <span class="o">[</span>-h<span class="o">]</span> <span class="o">[</span>-i INFILE<span class="o">]</span> <span class="o">[</span>-o OUTFILE<span class="o">]</span> <span class="o">[</span>-w N<span class="o">]</span> <span class="o">[</span>-t<span class="o">]</span> <span class="o">[</span>-e<span class="o">]</span>
<span class="o">[</span>-w N<span class="o">]</span>          输出fasta每行最大的碱基数, 默认值为0, 也就是每条fasta占据一行序列, 容易脚本操作
<span class="o">[</span>-t<span class="o">]</span>            输出为制表符格式, 而不是fasta, 第一列为序列ID, 第二列为序列的单行显示
<span class="o">[</span>-e<span class="o">]</span>            输出空白序列, 默认值为丢弃, 具有序列ID但并没有任何碱基的为空序列
</pre></div>


<h3 id="313dnarna">3.13.DNA序列与RNA互换</h3>
<div class="codehilite"><pre><span></span>fasta_nucleotide_changer <span class="o">[</span>-h<span class="o">]</span> <span class="o">[</span>-z<span class="o">]</span> <span class="o">[</span>-v<span class="o">]</span> <span class="o">[</span>-i INFILE<span class="o">]</span> <span class="o">[</span>-o OUTFILE<span class="o">]</span> <span class="o">[</span>-r<span class="o">]</span> <span class="o">[</span>-d<span class="o">]</span>
<span class="o">[</span>-r<span class="o">]</span>         DNA to RNA, T to U
<span class="o">[</span>-d<span class="o">]</span>         RNA to DNA, U to T
<span class="o">[</span>-v<span class="o">]</span>         报告序列的数目
             如果有<span class="s1">&#39;-o&#39;</span>, 在屏幕上标准输出
             如果无<span class="s1">&#39;-o&#39;</span>, 则在标准错误输出
<span class="o">[</span>-z<span class="o">]</span>         用GZIP压缩输出
</pre></div>


<h3 id="314reads">3.14.生成reads长度分布图</h3>
<div class="codehilite"><pre><span></span>fasta_clipping_histogram.pl INPUT_FILE.FA OUTPUT_FILE.PNG
INPUT_FILE.FA   指定输入fasta文件, 可以为GZIP格式
OUTPUT_FILE.PNG 生成的柱状图
</pre></div>


<h3 id="315-barcodereads">3.15. 依据barcode进行reads分类成不同文件</h3>
<div class="codehilite"><pre><span></span>fastx_barcode_splitter.pl --bcfile FILE --prefix PREFIX <span class="o">[</span>--suffix SUFFIX<span class="o">]</span> <span class="o">[</span>--bol<span class="p">|</span>--eol<span class="o">]</span> <span class="o">[</span>--mismatches N<span class="o">]</span> <span class="o">[</span>--exact<span class="o">]</span> <span class="o">[</span>--partial N<span class="o">]</span> <span class="o">[</span>--quiet<span class="o">]</span> <span class="o">[</span>--debug<span class="o">]</span>
--bcfile FILE       Barcodes文件名, 解释如下
--prefix PREFIX     文件前缀, 会添加到输出文件, 可以用来辨别输出目录
--suffix            文件后缀, 可选, 可以用来辨别文件扩展名
--bol               尝试在序列的开始匹配barcodes, 也就是5‘端, 程序从0开始
--eol               尝试在序列的末尾匹配barcodes, 也就是3‘端, 程序从最后开始
                    --bol 与 --eo l必须有其一
--mismatches N      允许的最大错配数, 默认为1
--exact             等同于<span class="s1">&#39;--mismatches 0&#39;</span>, 如果与<span class="s1">&#39;--mismatches&#39;</span>同时存在, 那么<span class="s1">&#39;--exact&#39;</span>具有优先级
--partial N         允许部分barcodes匹配, 默认不允许, 解释如下
--quiet             运行结束不打印统计信息, 默认为打印
--debug             在标准错误上打印大量有用的debug信息
</pre></div>


<h2 id="4trimmomatic">4.Trimmomatic</h2>
<blockquote>
<p>你需要指定adapter的path,线程数,SRR的共有部分,及循环体.还要各种path</p>
</blockquote>
<ul>
<li>Paired end</li>
</ul>
<div class="codehilite"><pre><span></span><span class="nv">adapter</span><span class="o">=</span>
<span class="nv">threads</span><span class="o">=</span>
<span class="nv">SRR_base</span><span class="o">=</span>
<span class="k">for</span> <span class="o">((</span><span class="nv">id</span><span class="o">=</span><span class="m">4</span><span class="p">;</span>id&lt;<span class="o">=</span><span class="m">8</span><span class="p">;</span>id++<span class="o">))</span><span class="p">;</span>
<span class="k">do</span>
<span class="nb">time</span> trimmomatic PE ../00rawdata/<span class="si">${</span><span class="nv">SRR_base</span><span class="si">}${</span><span class="nv">id</span><span class="si">}</span>_1.fastq.gz  ../00rawdata/<span class="si">${</span><span class="nv">SRR_base</span><span class="si">}${</span><span class="nv">id</span><span class="si">}</span>_2.fq.gz <span class="se">\</span>
    ./trim/<span class="si">${</span><span class="nv">SRR_base</span><span class="si">}${</span><span class="nv">id</span><span class="si">}</span>_F.fastq.gz ./trim/<span class="si">${</span><span class="nv">SRR_base</span><span class="si">}${</span><span class="nv">id</span><span class="si">}</span>_UF.fastq.gz <span class="se">\</span>
    ./trim/<span class="si">${</span><span class="nv">SRR_base</span><span class="si">}${</span><span class="nv">id</span><span class="si">}</span>_R.fastq.gz ./trim/<span class="si">${</span><span class="nv">SRR_base</span><span class="si">}${</span><span class="nv">id</span><span class="si">}</span>_UR.fastq.gz <span class="se">\</span>
    ILLUMINACLIP:<span class="si">${</span><span class="nv">adapter</span><span class="si">}</span>:2:30:10 <span class="se">\</span>
    -threads <span class="si">${</span><span class="nv">threads</span><span class="si">}</span> <span class="se">\</span>
    LEADING:4 TRAILING:5 SLIDINGWINDOW:4:5 MINLEN:25
<span class="k">done</span>
</pre></div>


<ul>
<li>Single end</li>
</ul>
<div class="codehilite"><pre><span></span><span class="nv">adapter</span><span class="o">=</span>
<span class="nv">threads</span><span class="o">=</span>
<span class="nv">SRR_base</span><span class="o">=</span>
<span class="k">for</span> <span class="o">((</span><span class="nv">id</span><span class="o">=</span><span class="m">4</span><span class="p">;</span>id&lt;<span class="o">=</span><span class="m">8</span><span class="p">;</span>id++<span class="o">))</span><span class="p">;</span>
<span class="k">do</span>
<span class="nb">time</span> trimmomatic SE ../00rawdata/<span class="si">${</span><span class="nv">SRR_base</span><span class="si">}${</span><span class="nv">id</span><span class="si">}</span>.fastq.gz <span class="se">\</span>
    ./trim/<span class="si">${</span><span class="nv">SRR_base</span><span class="si">}${</span><span class="nv">id</span><span class="si">}</span>_F.fastq.gz <span class="se">\</span>
    ILLUMINACLIP:<span class="si">${</span><span class="nv">adapter</span><span class="si">}</span>:2:30:10 <span class="se">\</span>
    -threads <span class="si">${</span><span class="nv">threads</span><span class="si">}</span> <span class="se">\</span>
    LEADING:4 TRAILING:5 SLIDINGWINDOW:4:5 MINLEN:25
<span class="k">done</span>
</pre></div>
		</content>

	</main>
	<footer>
		<p> Powered by xizhihui </p>
		<p> Thanks to <a href="https://www.hifreud.com/">hifreud</a> the theme is learned from his site. </p>
	</footer>

<script type="text/javascript" src="/js/index.json"></script>
<script type="text/javascript" src="/js/category.json"></script>
<script type="text/javascript" src="/js/tags.json"></script>
<script type="text/javascript" src="/js/about.json"></script>
<script type="text/javascript" src="/js/index.js"></script>
<script type="text/javascript">
	// listen to click event on navigator in header
	// document
	// 	.querySelector("div#head_right ul")
	// 	.addEventListener("click", change_page)
	var table = [['h2', 'raw_data', '测序raw_data质控和处理软件'], ['h2', '1-fastqc', '1. fastqc'], ['h2', '2-cutadapt', '2. cutadapt'], ['h2', '3-fastx-toolkit', '3. Fastx-toolkit'], ['h3', '31', '3.1.去除接头'], ['h3', '32', '3.2. 去除低质量碱基'], ['h3', '33-fasta', '3.3. 转换成fasta'], ['h3', '34', '3.4. 碱基质量统计'], ['h3', '35', '3.5. 生成碱基质量箱图'], ['h3', '36', '3.6. 生成碱基质量分布图'], ['h3', '37read', '3.7.重命名read'], ['h3', '38-read', '3.8. 输出指定长度read（前后切除）'], ['h3', '39reads', '3.9.去重reads'], ['h3', '310readsartifact', '3.10.去除人工reads（artifact）'], ['h3', '311', '3.11. 生成反向互补序列'], ['h3', '312fasta', '3.12.对fasta文件进行格式化'], ['h3', '313dnarna', '3.13.DNA序列与RNA互换'], ['h3', '314reads', '3.14.生成reads长度分布图'], ['h3', '315-barcodereads', '3.15. 依据barcode进行reads分类成不同文件'], ['h2', '4trimmomatic', '4.Trimmomatic']]
	var html = format_article_table(table)
	var table_ele = document.querySelector("main #table")
	table_ele.innerHTML = html
	// window.onload = evt => resize_nav(table_ele)
	adaption(document, window)

</script>
</body>
</html>