<!DOCTYPE html>
<html>
<head>
	<title>XZH's Notes</title>
	<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
	<meta name="author" content="xizhihui" />
	<meta name="viewport" content="width=device-width, initial-scale=1" />
	<link rel="stylesheet" type="text/css" href="/statics/reset.css">
	<link rel="stylesheet" type="text/css" href="/statics/header_footer.css"></link>
	<link rel="stylesheet" type="text/css" href="/statics/default.css">
	<link rel="stylesheet" type="text/css" href="/statics/main.css">
</head>
<body>
	<header class="Header">
		<nav>
			<h3>XZH</h3>
			<ul>
				<li><a href="/index.html#Timeline">Timeline</a></li><li><a href="/index.html#Category">Category</a></li><li><a href="/index.html#Tags">Tags</a></li><li><a href="/index.html#About">About</a></li>
			</ul>
		</nav>
	</header>
	<div class="Motto">
		<h2>XZH's Notes</h2>
		<h4>The future comes for you ~ ~ ~</h4>
	</div>

	<main>
		<nav id="table"></nav>
		<content id="content" class="content">
			<h1 id="gene_id">常用gene_ID的转换(注释)</h1>
<h2 id="id">一、ID类型</h2>
<table>
    <thead><td>ID 示例</td><td>ID来源</td></thead>
    <tbody>
        <tr><td>ENSG00000116717</td><td>Ensemble ID</td></tr>
        <tr><td>GA45A_HUMAN </td><td>UniProtKB/Swiss-Prot, entry name</td></tr>
        <tr><td>A5PJB2_BOVIN    </td><td> UniProtKB/TrEMBL, entry name</td></tr>
        <tr><td>A2BC19, P12345, A0A022YWF9</td><td>UniProt, accession number</td></tr>
        <tr><td>GLA, GLB, UGT1A1    </td><td> HGNC Gene Symbol</td></tr>
        <tr><td>U12345, AF123456    </td><td> GenBank, NCBI, accession number</td></tr>
        <tr><td>NT_123456, NM_123456, NP_123456 </td><td> RefSeq, NCBI, accession number</td></tr>
        <tr><td>10598, 717  </td><td> Entrez ID, NCBI</td></tr>
        <tr><td>uc001ett, uc031tla.1</td><td> UCSC ID</td></tr>
    </tbody>
</table>

<h3 id="1-ensembl-stable-id">1. Ensembl stable ID</h3>
<p>Ensembl stable ID 的结构是根据不同物种设置的前缀, 加上数据所指的类型, 如基因蛋白质, 再加上一系列的数字. 有的时候可以有不同的版本, 则在 Ensembl ID 后面加上小数点和版本号.</p>
<p>Ensembl_gene_identifier是Ensembl ID里的一种, Enseml ID包括exon, protein family, gene, gene tree, protein, regulatory feature 和 transcript.</p>
<p>Ensembl ID的由5部分构成: ENS(species)(object type)(identifier).(version)
第一部分ENS代表这是一个Ensembl ID
第二部分代表物种, 如MUS代表小鼠(如果物种是人则此处为空)
第三部分代表ID的类型, 如G代表基因, T代表转录本, P代表蛋白, E代表外显子, S代表
第四部分是一个特殊的数字标志
第五部分代表版本号</p>
<p>比如:"ENSMUSG00000017167.6", 我们知道这是一个Ensembl ID (ENS), 物种为小鼠(MUS), 代表一个基因(G), 并且这是第6个版本(.6).</p>
<h4 id="_1">物种前缀</h4>
<p><table>
<tr><td>物种前缀</td><td>学名</td></tr>
<tr><td>ENSCEL</td><td>Caenorhabditis elegans (Caenorhabditis elegans)</td></tr>
<tr><td>ENSCAF</td><td>Canis lupus familiaris (Dog)</td></tr>
<tr><td>ENSDAR</td><td>Danio rerio (Zebrafish)</td></tr>
<tr><td>FB</td><td>Drosophila melanogaster (Fruitfly)</td></tr>
<tr><td>ENS</td><td>Homo sapiens (Human)</td></tr>
<tr><td>ENSMUS</td><td>Mus musculus (Mouse)</td></tr>
<tr><td>ENSRNO</td><td>Rattus norvegicus (Rat)</td></tr>
<tr><td>ENSXET</td><td>Xenopus tropicalis (Xenopus)</td></tr>
</table></p>
<h4 id="_2">类型前缀</h4>
<p><table>
<tr><td>类型前缀</td><td>类型</td></tr>
<tr><td>E</td><td>exon</td></tr>
<tr><td>FM</td><td>Ensembl protein family</td></tr>
<tr><td>G</td><td>gene</td></tr>
<tr><td>GT</td><td>gene tree</td></tr>
<tr><td>P</td><td>protein</td></tr>
<tr><td>R</td><td>regulatory feature</td></tr>
<tr><td>T</td><td>transcript</td></tr>
</table></p>
<h3 id="2-uniprot">2. UniProt</h3>
<p>UniProt中录入的数据都被分配了一个唯一的entry name.</p>
<ul>
<li>
<p>UniProtKB/Swiss-Prot entry name 是最多有 11 位包含大写字母的字符串, 一般有着 "X_Y" 的形式, 其中 "X" 是最多五个便于记忆的蛋白质编号, "_" 是下划线, "Y" 是最多五个便于记忆的物种编号.</p>
</li>
<li>
<p>UniProtKB/TrEMBL entry name 是最多 16 位包含大写字母的字符串, 一般有着 "X_Y" 的形式, 其中 "X" 是 6 到 10 个字符组成的 accession number, "_" 是下划线, "Y" 是最多五个便于记忆的物种编号.</p>
</li>
<li>
<p>UniProtKB 的 Accession Number 相当于数据库的主键, 由 6 到 10 个大写字母或者数字组成. 其构成规律为: [OPQ][0-9][A-Z0-9]{3}[0-9]|[A-NR-Z]<a 1_2="1,2" href="[A-Z][A-Z0-9]{2}[0-9]">0-9</a></p>
</li>
</ul>
<h4 id="_3">蛋白质编号示例</h4>
<p><table>
<tr><td>Code(X)</td><td> Recommended protein name</td><td> Gene name</td></tr>
<tr><td>B2MG</td><td> Beta-2-microglobulin</td><td> B2M</td></tr>
<tr><td>HBA</td><td> Hemoglobin subunit alpha</td><td> HBA1</td></tr>
<tr><td>INS</td><td> Insulin</td><td> INS</td></tr>
<tr><td>CAD17</td><td> Cadherin-17</td><td> CDH17</td></tr>
</table></p>
<h4 id="_4">物种编号</h4>
<p><table>
<tr><td>编号</td><td>物种</td></tr>
<tr><td>BOVIN</td><td>Bovine</td></tr>
<tr><td>CHICK</td><td>Chicken</td></tr>
<tr><td>ECOLI</td><td>Escherichia coli</td></tr>
<tr><td>HORSE</td><td>Horse</td></tr>
<tr><td>HUMAN</td><td>Homo sapiens</td></tr>
<tr><td>MAIZE</td><td>Maize (Zea mays)</td></tr>
<tr><td>MOUSE</td><td>Mouse</td></tr>
</table></p>
<h3 id="3-gene-symbol">3. Gene Symbol</h3>
<p>Gene Symbol 是用来表示基因的编码, 由大写字母构成, 或由大写字母和数字构成, 首字母均应该是字母.如: GLA "galactosidase, alpha"; GLB "galactosidase, beta"; UGT1A1 "UDP glycosyltransferase 1 family, polypeptide A1" 再到 UGT1A13 代表了 13 个不同的 gene symbol.</p>
<h3 id="4-ncbi">4. NCBI</h3>
<ul>
<li>GenBank的通用accession number通常是由一个大写字母加上 5 个数字的组合, 或者两个大写字母加上6个数字的组合.</li>
<li>RefSeq有一套特殊的Accesion Number.形式是: [A-Z]{2}[_][0-9]{6:},两个大写字母,一个下划线,6个或更多的数字.</li>
</ul>
<table>
<tr><td>Accession </<td> 前缀</td><td>类型</td><td>说明</td></tr>
<tr><td>AC_</td><td>Genomic</td><td>Complete genomic molecule, usually alternate assembly</td></tr>
<tr><td>NC_</td><td>Genomic</td><td>Complete genomic molecule, usually reference assembly</td></tr>
<tr><td>NG_</td><td>Genomic</td><td>Incomplete genomic region</td></tr>
<tr><td>NT_</td><td>Genomic</td><td>Contig or scaffold, clone-based or WGS</td></tr>
<tr><td>NW_</td><td>Genomic</td><td>Contig or scaffold, primarily WGS</td></tr>
<tr><td>NS_</td><td>Genomic</td><td>Environmental sequence</td></tr>
<tr><td>NZ_</td><td>Genomic</td><td>Unfinished WGS</td></tr>
<tr><td>NM_</td><td>mRNA</td><td>none</td></tr>
<tr><td>NR_</td><td>RNA</td><td>none</td></tr>
<tr><td>XM_</td><td>mRNA</td><td>Predicted model</td></tr>
<tr><td>XR_</td><td>RNA</td><td>Predicted model</td></tr>
<tr><td>AP_</td><td>Protein</td><td>Annotated on AC_ alternate assembly</td></tr>
<tr><td>NP_</td><td>Protein</td><td>Associated with an NM_ or NC_ accession</td></tr>
<tr><td>YP_</td><td>Protein</td><td>none</td></tr>
<tr><td>XP_</td><td>Protein</td><td>Predicted model, associated with an XM_ accession</td></tr>
<tr><td>ZP_</td><td>Protein</td><td>Predicted model, annotated on NZ_ genomic records</td></tr>
</table>

<h3 id="5-entrez-id">5. Entrez ID</h3>
<p>Entrez 是 NCBI 使用的能够对众多数据库进行联合搜索的搜索引擎, 其对不同的 Gene 进行了编号, 每个 gene 的编号就是 entrez gene id. 由于 entrez id 相对稳定, 所以也被众多其他数据库, 如 KEGG 等采用. Entrez Gene ID 就是一系列数字, 也比较容易辨识. R 或网站都有众多的工具可以帮助从不同的 ID 转换为 entrez id 或者反向转换.</p>
<h3 id="6ucsc-id">6.UCSC ID</h3>
<p>UCSC ID由小写字母和数字构成, 起始均为 uc, 然后是三位数字, 接着又是三位小写字母, 最后有小数点和数字构成版本号.
如: uc010qfk.3, uc010qfk.3.</p>
<h3 id="7rorghsegdbidorgxxegdb">7.使用R包org.Hs.eg.db转换ID(org.xx.eg.db)</h3>
<div class="codehilite"><pre><span></span><span class="c1">## 类似数据库查询一样</span>
<span class="kn">library</span><span class="p">(</span>org.Hs.eg.db<span class="p">)</span>
<span class="c1"># 显示可转换的ID类型</span>
keytypes<span class="p">(</span>org.Hs.eg.db<span class="p">)</span>
ids <span class="o">=</span> yourid_vector
<span class="c1"># 你要转换的选项,由keytypes里面选</span>
cols <span class="o">&lt;-</span> <span class="kt">c</span><span class="p">(</span><span class="s">&quot;SYMBOL&quot;</span><span class="p">,</span> <span class="s">&quot;GENENAME&quot;</span><span class="p">,</span> <span class="s">&#39;ENTREZID&#39;</span><span class="p">)</span>
getIDs <span class="o">=</span> select<span class="p">(</span>org.Hs.eg.db<span class="p">,</span> keys<span class="o">=</span>ensids<span class="p">,</span> columns<span class="o">=</span>cols<span class="p">,</span> keytype<span class="o">=</span><span class="s">&quot;ENSEMBL&quot;</span><span class="p">)</span>
</pre></div>


<h3 id="8biomart">8.使用biomaRt包转换</h3>
<div class="codehilite"><pre><span></span><span class="kn">source</span><span class="p">(</span><span class="s">&quot;https://bioconductor.org/biocLite.R&quot;</span><span class="p">)</span>
biocLite<span class="p">(</span><span class="s">&quot;biomaRt&quot;</span><span class="p">)</span>
listMarts<span class="p">()</span>

ensembl<span class="o">&lt;-</span>useMart<span class="p">(</span><span class="s">&quot;ENSEMBL_MART_ENSEMBL&quot;</span><span class="p">)</span>
all_datasets <span class="o">&lt;-</span> listDatasets<span class="p">(</span>ensembl<span class="p">)</span>

<span class="kn">library</span><span class="p">(</span>DT<span class="p">)</span>
datatable<span class="p">(</span>all_datasets<span class="p">,</span>
        options <span class="o">=</span> <span class="kt">list</span><span class="p">(</span>searching <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span>pageLength <span class="o">=</span> <span class="m">5</span><span class="p">,</span>lengthMenu <span class="o">=</span> <span class="kt">c</span><span class="p">(</span><span class="m">5</span><span class="p">,</span> <span class="m">10</span><span class="p">,</span> <span class="m">15</span><span class="p">,</span> <span class="m">20</span><span class="p">)))</span>

ensembl <span class="o">&lt;-</span> useMart<span class="p">(</span><span class="s">&quot;ensembl&quot;</span><span class="p">,</span> dataset <span class="o">=</span> <span class="s">&quot;hsapiens_gene_ensembl&quot;</span><span class="p">)</span>
filters <span class="o">=</span> listFilters<span class="p">(</span>ensembl<span class="p">)</span>
datatable<span class="p">(</span>filters<span class="p">,</span> options <span class="o">=</span> <span class="kt">list</span><span class="p">(</span>searching <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span>pageLength <span class="o">=</span> <span class="m">5</span><span class="p">,</span>lengthMenu <span class="o">=</span> <span class="kt">c</span><span class="p">(</span><span class="m">5</span><span class="p">,</span> <span class="m">10</span><span class="p">,</span> <span class="m">15</span><span class="p">,</span> <span class="m">20</span><span class="p">)))</span>
attributes <span class="o">=</span> listAttributes<span class="p">(</span>ensembl<span class="p">)</span>
datatable<span class="p">(</span><span class="kp">attributes</span><span class="p">,</span> options <span class="o">=</span> <span class="kt">list</span><span class="p">(</span>searching <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span>pageLength <span class="o">=</span> <span class="m">5</span><span class="p">,</span>lengthMenu <span class="o">=</span> <span class="kt">c</span><span class="p">(</span><span class="m">5</span><span class="p">,</span> <span class="m">10</span><span class="p">,</span> <span class="m">15</span><span class="p">,</span> <span class="m">20</span><span class="p">)))</span>
entrez.mart<span class="o">&lt;-</span>getBM<span class="p">(</span>attributes<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="s">&#39;ensembl_gene_id&#39;</span><span class="p">,</span> <span class="s">&#39;entrezgene&#39;</span><span class="p">),</span> filters <span class="o">=</span> <span class="s">&#39;ensembl_gene_id&#39;</span><span class="p">,</span> values <span class="o">=</span> <span class="kp">rownames</span><span class="p">(</span>fpkmtpens<span class="p">),</span> mart <span class="o">=</span> ensembl<span class="p">)</span>
</pre></div>


<h3 id="9">9. 其他数据库整合包</h3>
<p>http://www.bio-info-trainee.com/1399.html</p>
		</content>

	</main>
	<footer>
		<p> Powered by xizhihui </p>
		<p> Thanks to <a href="https://www.hifreud.com/">hifreud</a> the theme is learned from his site. </p>
	</footer>

<script type="text/javascript" src="/js/index.json"></script>
<script type="text/javascript" src="/js/category.json"></script>
<script type="text/javascript" src="/js/tags.json"></script>
<script type="text/javascript" src="/js/about.json"></script>
<script type="text/javascript" src="/js/index.js"></script>
<script type="text/javascript">
	// listen to click event on navigator in header
	// document
	// 	.querySelector("div#head_right ul")
	// 	.addEventListener("click", change_page)
	var table = [['h2', 'id', '一、ID类型'], ['h3', '1-ensembl-stable-id', '1. Ensembl stable ID'], ['h4', '_1', '物种前缀'], ['h4', '_2', '类型前缀'], ['h3', '2-uniprot', '2. UniProt'], ['h4', '_3', '蛋白质编号示例'], ['h4', '_4', '物种编号'], ['h3', '3-gene-symbol', '3. Gene Symbol'], ['h3', '4-ncbi', '4. NCBI'], ['h3', '5-entrez-id', '5. Entrez ID'], ['h3', '6ucsc-id', '6.UCSC ID'], ['h3', '7rorghsegdbidorgxxegdb', '7.使用R包org.Hs.eg.db转换ID(org.xx.eg.db)'], ['h3', '8biomart', '8.使用biomaRt包转换'], ['h3', '9', '9. 其他数据库整合包']]
	var html = format_article_table(table)
	var table_ele = document.querySelector("main #table")
	table_ele.innerHTML = html
	// window.onload = evt => resize_nav(table_ele)
	adaption(document, window)

</script>
</body>
</html>