<!DOCTYPE html>
<html>
<head>
	<title>XZH's Notes</title>
	<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
	<meta name="author" content="xizhihui" />
	<meta name="viewport" content="width=device-width, initial-scale=1" />
	<link rel="stylesheet" type="text/css" href="/statics/reset.css">
	<link rel="stylesheet" type="text/css" href="/statics/header_footer.css"></link>
	<link rel="stylesheet" type="text/css" href="/statics/default.css">
	<link rel="stylesheet" type="text/css" href="/statics/main.css">
</head>
<body>
	<header class="Header">
		<nav>
			<h3>XZH</h3>
			<ul>
				<li><a href="/index.html#Timeline">Timeline</a></li><li><a href="/index.html#Category">Category</a></li><li><a href="/index.html#Tags">Tags</a></li><li><a href="/index.html#About">About</a></li>
			</ul>
		</nav>
	</header>
	<div class="Motto">
		<h2>XZH's Notes</h2>
		<h4>The future comes for you ~ ~ ~</h4>
	</div>

	<main>
		<nav id="table"></nav>
		<content id="content" class="content">
			<h1 id="star">STAR</h1>
<h2 id="2">基本流程：2步</h2>
<ul>
<li>依据参考基因序列和参考注释，生成索引文件</li>
<li>进行比对</li>
</ul>
<h3 id="_1">生成索引</h3>
<div class="codehilite"><pre><span></span>STAR --runThreadN numberOfThreads <span class="se">\</span>
    --runMode genomeGenerata <span class="se">\</span>
    --genomeDir /path/to/output/index <span class="se">\</span>
    --genomeFastaFiles /path/to/ref.fasta1 /path/to/ref.fasta2... <span class="se">\</span>
    --sjdbGTFfile /path/to/annotations.gtf <span class="se">\</span>
    --sjdbOverhang ReadLength-1

<span class="c1"># 使用的参考基因组：一般来说，不应该包含patches和可变单倍体。可以使用的比如：</span>
<span class="c1"># EMSEMBL：ftp://ftp.ensembl.org/pub/release-77/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz</span>
<span class="c1"># NCBI：ftp://ftp.ncbi.nlm.nih.gov/genbank/genomes/Eukaryotes/vertebrates_mammals/Homo_sapiens/GRCh38/seqs_for_alignment_pipelines/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz</span>

<span class="c1"># annotation: 来源要和参考基因组匹配,比如注释和基因组都是来源ENSEMBL的,或者都是来自UCSC的;千万不要混用。这是因为它们对染色体的命名不一样。</span>
<span class="c1"># 如果你手动更改成一样的也可以使用。</span>


<span class="c1"># --sjdbGTFfile &lt;anno.gtf&gt;      选项默认只处理GTF中exon的行;</span>
<span class="c1"># --sddbGTFfeatureExon exon     默认值为exon,使只处理exon的行.</span>
<span class="c1"># --sjdbGTFtagExonParent        如果使用的是GTF3格式文件,需要指定本选项。</span>
<span class="c1"># --sjdbGTFtagExonParentTranscript &lt;transcript_id&gt;  如果指定,将只会处理能比对到指定transcript(transcript_id)的exon</span>
<span class="c1"># --sjdbFileChrStartEnd &lt;sjdbFile.txt&gt; 指定可变剪接的注释文件</span>
<span class="c1"># sjdbOverhang: 对于2x100b的Illumina双端测序来说, 是100-1=99.如果length是变化的，就用max(length)-1</span>
<span class="c1"># --genomeSAindexNbases &lt;int&gt;   对于较小的基因组,需要用此选项指定N的值.计算方式: min(14, log2(genomeLength)/2-1). 1Mb的基因组一般为7.</span>
<span class="c1"># --genomeChrBinNbits &lt;int&gt;     对于较大的基因组,需要用此选项指定N的值,计算方式: min(18, log2(genomeLength/NumberofReference)).对于3GB含100000个染色体/scaffolds的基因组来说,取15.</span>
</pre></div>


<h3 id="_2">进行比对</h3>
<div class="codehilite"><pre><span></span>STAR --runThreadN numberofthreads <span class="se">\</span>
    --genomeDir index_path <span class="se">\</span>
    --readFilesIn read1.fastq read2.fastq
    --outFileNamePrefix path/prefix

<span class="c1"># --readFilesCommand &lt;command&gt;      如果是压缩文件,可以提供解压缩命令.zcat, gzip -c, bzip2 -c.依据压缩文件的压缩形式而不同.</span>
<span class="c1"># --outSAMstrandField introMotif    指定值为introMotif将会为unstranded RNA-seq数据生成可用于cufflinks和cuffdiff的文件(含XS strand attribute)</span>
<span class="c1"># --outFilterIntroMotifs RemoveNoncanonical     后续进行cufflinks的话,推荐设定此选项.</span>
<span class="c1"># --outSAMtype BAM Unsorted         输出未排序的aligned.out.bam,可以直接用于HTSeq,不用进行name sorting</span>
<span class="c1"># --outSAMtype BAM SortedByCoordinate 输出根据坐标排序的aligned.sortedByCoord.out.bam,与samtools sort命令的输出一致</span>
<span class="c1"># --outSAMtype BAM Unsorted SortedByCoordinate  两种情况均输出.</span>
<span class="c1"># --quantMode TranscriptomeSAM      将会输出翻译成转录本坐标的bam文件,aligned.toTranscriptome.out.bam.</span>
                                    这个可以用于转录本定量软件的输入.比如RSEM, eXpress.
</pre></div>


<h3 id="2-pass-mapping-with-re-generated-genome">2-pass mapping with re-generated genome</h3>
<p>在初次比对后,为了提高比对到新的剪接位点的reads数,推荐再进行一次比对.新的比对将需要提供上一次比对产生的剪接位点文件.</p>
<div class="codehilite"><pre><span></span><span class="c1"># 1. 把SJ.out.tab文件合并,过滤掉在线粒体染色体上的junction、non-canonical junction。</span>
<span class="c1"># 1. 如果使用了注释的话,这里只需考虑新的剪接位点.</span>

<span class="c1"># 2. 把1.中得到的合并文件通过--sjdbFileChrStartEnd指定输入, --sjdbGTFile指定注释,重新生成索引</span>
<span class="c1"># 3. 使用新的索引再次进行比对</span>
</pre></div>


<h3 id="2-pass-mapping-without-re-generated-genome">2-pass mapping without re-generated genome</h3>
<p>要进行此项,在首次进行比对时,就要设置--twopass1readsN选项,值为所有reads数.此外，你还得设置--sjdbOverhang参数.</p>
		</content>

	</main>
	<footer>
		<p> Powered by xizhihui </p>
		<p> Thanks to <a href="https://www.hifreud.com/">hifreud</a> the theme is learned from his site. </p>
	</footer>

<script type="text/javascript" src="/js/index.json"></script>
<script type="text/javascript" src="/js/category.json"></script>
<script type="text/javascript" src="/js/tags.json"></script>
<script type="text/javascript" src="/js/about.json"></script>
<script type="text/javascript" src="/js/index.js"></script>
<script type="text/javascript">
	// listen to click event on navigator in header
	// document
	// 	.querySelector("div#head_right ul")
	// 	.addEventListener("click", change_page)
	var table = [['h2', '2', '基本流程：2步'], ['h3', '_1', '生成索引'], ['h3', '_2', '进行比对'], ['h3', '2-pass-mapping-with-re-generated-genome', '2-pass mapping with re-generated genome'], ['h3', '2-pass-mapping-without-re-generated-genome', '2-pass mapping without re-generated genome']]
	var html = format_article_table(table)
	var table_ele = document.querySelector("main #table")
	table_ele.innerHTML = html
	// window.onload = evt => resize_nav(table_ele)
	adaption(document, window)

</script>
</body>
</html>