<!DOCTYPE HTML>
<html>
<head><meta name="generator" content="Hexo 3.8.0">
  <meta charset="utf-8">
  
  <title>扩增子测序 QIIME2使用示例 | Xizhihui&#39;s Notes</title>
  <meta name="author" content="Xizhihui">
  
  <meta name="description" content="如果数据来源是分批次的话，需要分别导入，在去噪和生成特征表之后，把这些表进行合并，再进行后续分析。也就是说 1-3 对于分批次数据要分批次跑。">
  
  
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">

  <meta property="og:title" content="扩增子测序 QIIME2使用示例">
  <meta property="og:site_name" content="Xizhihui&#39;s Notes">

  
    <meta property="og:image" content="">
  

  
  
    <link href="/favicon.png" rel="icon">
  

  <!-- CSS -->
  <link rel="stylesheet" href="/css/themes/cosmo.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/font-awesome.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/style.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/responsive.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight-default.min.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/google-fonts.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/comment.css" media="screen" type="text/css">
  <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
	<script src="https://cdnjs.cloudflare.com/ajax/libs/es5-shim/4.5.9/es5-shim.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/es5-shim/4.5.7/es5-sham.min.js"></script>
  <![endif]-->

  <script src="/js/jquery-2.0.3.min.js"></script>
  
    <script src="/js/marked.js"></script>
    <script src="/js/comment.js"></script>
    <script src="/js/timeago.min.js"></script>
    <script src="/js/highlight.min.js"></script>
	<script src="/js/spin.min.js"></script>
  
  <!-- analytics -->
  



</head>
</html>
 <body>  
  <nav id="main-nav" class="navbar  navbar-default navbar-fixed-top" role="navigation">
    <div class="container">
      <button type="button" class="navbar-header navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
	<span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
       <a class="navbar-brand" href="/">Xizhihui&#39;s Notes</a>
      <div class="collapse navbar-collapse nav-menu">
		<ul class="nav navbar-nav">
		  
		  <li>
			<a href="/archives" title="All the articles.">
			  <i class="fa fa-archive"></i>Archives
			</a>
		  </li>
		  
		  <li>
			<a href="/Categories" title="All the categories.">
			  <i class="fa fa-folder"></i>Categories
			</a>
		  </li>
		  
		  <li>
			<a href="/Tags" title="All the tags.">
			  <i class="fa fa-tags"></i>Tags
			</a>
		  </li>
		  
		  <li>
			<a href="/About" title="About me.">
			  <i class="fa fa-user"></i>About
			</a>
		  </li>
		  
		</ul>
      </div>
    </div> <!-- container -->
</nav>
<div class="clearfix"></div>

  <div class="container">
  	<div class="content">
    	 


	
		<div class="page-header ">		
			<h1 class="title "> 扩增子测序 QIIME2使用示例</h1>
		</div>		
	






<div class="row post">
	<!-- cols -->
	
	<div id="top_meta"></div>
	<div class="col-md-9">
	

	<!-- content -->
	<div class="mypage">		
	  		

	  <p><strong>如果数据来源是分批次的话，需要分别导入，在去噪和生成特征表之后，把这些表进行合并，再进行后续分析。也就是说 1-3 对于分批次数据要分批次跑。</strong></p>
<a id="more"></a>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">wget -O sample_metadata.tsv https://data.qiime2.org/2018.6/tutorials/moving-pictures/sample_metadata.tsv</span><br><span class="line">mkdir emp-single-end-sequences</span><br><span class="line">wget -O emp-single-end-sequences/barcodes.fastq.gz https://data.qiime2.org/2018.6/tutorials/moving-pictures/barcodes.fastq.gz</span><br><span class="line">wget -O emp-single-end-sequences/sequences.fastq.gz https://data.qiime2.org/2018.6/tutorials/moving-pictures/sequences.fastq.gz</span><br></pre></td></tr></table></figure>
<h2 id="1-导入数据到QIIME2"><a href="#1-导入数据到QIIME2" class="headerlink" title="1 导入数据到QIIME2"></a>1 导入数据到QIIME2</h2><p>注意，metadata要放在input-path外部，同级存放。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">qiime tools import \</span><br><span class="line">		--<span class="built_in">type</span> EMPSingleEndSequences \</span><br><span class="line">		--input-path emp-single-end-sequences \</span><br><span class="line">		--output-path emp-single-end-sequences.qza</span><br></pre></td></tr></table></figure>
<h2 id="2-拆分序列"><a href="#2-拆分序列" class="headerlink" title="2 拆分序列"></a>2 拆分序列</h2><p>在QIIME1中，我们通常使用split_libraries.py / split_libraries_fastq.py进行拆分序列，并同时进行质量过滤。但是在QIIME2中，序列拆分和质量过滤分开进行。所以你既可以从未拆分序列开始，也可以从已拆分序列开始。这里我们从未拆分的序列开始。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 拆分序列</span></span><br><span class="line">qiime demux emp-single \</span><br><span class="line">		--i-seqs emp-single-end-sequences.qza \</span><br><span class="line">		--m-barcodes-file sample-metadata.tsv \</span><br><span class="line">		--m-barcodes-column BarcodeSequence \</span><br><span class="line">		--o-per-sample-sequences demux.qza</span><br><span class="line"></span><br><span class="line"><span class="comment"># 统计拆分后的序列，了解每个样品有多少序列和序列碱基质量值分布</span></span><br><span class="line">qiime demux summarize --i-data demux.qza --o-visualization demux.qzv</span><br></pre></td></tr></table></figure>
<p>在qiime2中，凡是通过–o-visualiztion参数生成的.qzv文件都可以通过qiime tools view来查看。如果是headless环境的话，你可以把.qzv文件解压，下载到本地来查看。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">qiime tools view demux.qzv</span><br></pre></td></tr></table></figure>
<h2 id="3-序列质控和构建特征表（feature-table）"><a href="#3-序列质控和构建特征表（feature-table）" class="headerlink" title="3 序列质控和构建特征表（feature  table）"></a>3 序列质控和构建特征表（feature  table）</h2><p>QIIME2中有许多中质控方法可供选择，比如DADA2、Deblur和基于质量分数过滤。这里我们使用DADA2和Deblur来进行质控，你可以选择任何一个方法。这些方法都会生成一个特征表( FeatureTable[Frequency])和特征数据( FeatureData[Frequency] )，前者包含每个样品中每条唯一序列的counts(frequences)，后者包含特征标识在特征表中对应的序列。FeatureTable与QIIME1中的OTU或biom表相对应，FeatureData同QIIME1中的rep_seqs.fna(代表序列)相对应。因为DADA2和Deblur的结果是通过聚类唯一序列得到的，所以这些结果与QIIME1得到的OTUs是一致的。但是在QIIME2中，获得的OTUs分辨率更高，这是因为在QIIME2中的质控效果更好。</p>
<h3 id="3-1-DADA2"><a href="#3-1-DADA2" class="headerlink" title="3.1 DADA2"></a>3.1 DADA2</h3><p>DADA2是一个检测和校正Illumina扩增测序数据的流程，里面的质控步骤会额外地过滤任何在测序数据中的phiX reads和嵌合序列。dada2 denoise-single方法需要2个参数用于质量过滤：–p-trim-left m, 会把序列的前m个碱基进行修剪, –p-trunc-len n, 会在第n个碱基处截断序列。这允许用户去除低质控的序列区域。具体的m和n值，你应该查看拆分序列后的序列质量统计结果才能确定。根据demux.qzv的结果，可以确定m=0，n=120.</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 去噪和构建</span></span><br><span class="line">qiime dada2 denoise-single \</span><br><span class="line">		--i-demultiplexed-seqs demux.qza \</span><br><span class="line">		--p-trim-left 0 \</span><br><span class="line">		--p-trunc-len 120 \</span><br><span class="line">		--o-representative-sequences rep-seqs-dada2.qza \</span><br><span class="line">		--o-table table-dada2.qza \</span><br><span class="line">		--o-denoising-stats stats-dada2.qza</span><br><span class="line"></span><br><span class="line"><span class="comment">## 把获得stats-dada2.qza统计数据变成可视化的结果</span></span><br><span class="line">qiime metadata tabulate \</span><br><span class="line">		--m-input-file stats-dada2.qza \</span><br><span class="line">		--o-visualization stats-dada2.qzv</span><br></pre></td></tr></table></figure>
<h3 id="3-2-Deblur"><a href="#3-2-Deblur" class="headerlink" title="3.2 Deblur"></a>3.2 Deblur</h3><p>Deblur使用序列错误分析来获得高质量的序列数据，这里面经历了2个步骤，第一步就是基于质量分数进行质量过滤。第二步是使用qiime deblur denoise-16s方法。这个方法需要–p-trim-length n来截断第n位碱基。通常推荐n值取质量值中值开始下降到较低值时的碱基位置。根据前面的demux.qzv中质量分布图，115-130的质量值显著下降。在有些情况下，比如同时分析多批次的测序数据时，你要保证所有的序列都是等长的，以避免出现study-specific bias。由于我们在dada2中使用了120的长度，这里同样使用120.</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 质量过滤</span></span><br><span class="line">qiime quality-filter q-score \</span><br><span class="line">		--i-demux demux.qza \</span><br><span class="line">		--o-filtered-sequences demux-filtered.qza \</span><br><span class="line">		--o-filter-stats demux-filter-stats.qza</span><br><span class="line"><span class="comment"># 过滤后可视化</span></span><br><span class="line">qiime metadata tabulate \</span><br><span class="line">		--m-input-file demux-filter-stats.qza \</span><br><span class="line">		--o-visualization demux-filter-stats.qzv</span><br><span class="line"></span><br><span class="line"><span class="comment"># 去噪和构建</span></span><br><span class="line">qiime deblur denoise-16S \</span><br><span class="line">		--i-demultiplexed-seqs demux-filtered.qza \</span><br><span class="line">		--p-trim-length 120 \</span><br><span class="line">		--o-representative-sequences rep-seqs-deblur.qza \</span><br><span class="line">		--o-table table-deblur.qza \</span><br><span class="line">		--p-sample-stats \</span><br><span class="line">		--o-stats deblur-stats.qza</span><br><span class="line"></span><br><span class="line"><span class="comment"># 可视化特种表</span></span><br><span class="line">qiime deblur visualize-stats \</span><br><span class="line">		--i-deblur-stats deblur-stats.qza \</span><br><span class="line">		--o-visualization deblur-stats.qzv</span><br></pre></td></tr></table></figure>
<h2 id="4-特种表和特征数据统计"><a href="#4-特种表和特征数据统计" class="headerlink" title="4 特种表和特征数据统计"></a>4 特种表和特征数据统计</h2><p>在完成质控和构建之后，你可能想探索一下生成的结果，feature-table summarize可以对特征表(‘OTUs’)做各种统计图；feature-table tabulate-seqs可以生成特征标识符和序列的映射文件，让你易于进行BLAST. 后续的可视化对于你了解特征表的各个细节大有裨益。</p>
<p>需要注意的是，正如开头所言，如果你的数据是分批次的话，那你应该先把各个特征表进行合并。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 可选，合并分批次的特征表, 这里的rep-seqs-1/2.qza就是4.3中--o-representative-sequences参数指定的输出</span></span><br><span class="line">qiime feature-table merge \</span><br><span class="line">		--i-tables table-1.qza \</span><br><span class="line">		--i-tables table-2.qza \</span><br><span class="line">		--o-merged-table table-merge.qza</span><br><span class="line">qiime feature-table merge-seqs \</span><br><span class="line">		--i-data rep-seqs-1.qza \</span><br><span class="line">		--i-data rep-seqs-2.qza \</span><br><span class="line">		--i-merged-data rep-seqs.qza</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment"># 这里使用deblur生成的表，如果要接上合并分批次的特征表，type=merge</span></span><br><span class="line"><span class="built_in">type</span>=<span class="string">'deblur'</span></span><br><span class="line">qiime feature-table summarize \</span><br><span class="line">		--i-table table-<span class="variable">$&#123;type&#125;</span>.qza \</span><br><span class="line">		--o-visualization table.qzv \</span><br><span class="line">		--m-sample-metadata-file sample-metadata.tsv</span><br><span class="line"></span><br><span class="line">qiime feature-table tabulate-seqs \</span><br><span class="line">		--i-data rep-seqs-<span class="variable">$&#123;type&#125;</span>.qza \</span><br><span class="line">		--o-visualization rep-seqs.qzv</span><br></pre></td></tr></table></figure>
<h2 id="5-生成系统发育树用于多样性分析"><a href="#5-生成系统发育树用于多样性分析" class="headerlink" title="5 生成系统发育树用于多样性分析"></a>5 生成系统发育树用于多样性分析</h2><p>QIIME2支持多个系统多样性矩阵，包含Faith’s Phylogenetic diversity和weighted/unweighted UniFrac.除了计算每个样品的counts (比如FeatureTable[Frequency]的数据) 以外, 这些多样性矩阵需要一个根系统发育树将各个特征联系起来。这个信息存储在 Phylogeny[Rooted]中，本步的目的就是生成这样一个Phylogeny[Rooted]文件。<br>首先我们需要使用qiime alignment mafft对代表序列进行比对。<br>然后我们用qiime alignment mask 把序列中的高度可变区屏蔽掉，因为这些区域会影响发育树的生成。<br>接着使用qiime phylogeny fasttree生成聚类树，这是一个无根树<br>最后，我们用qiime phylogeny midpoint-root，把无根树的中点节点作为其midpoint-root。<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><span class="line"># 这里仍然使用deblur的表</span><br><span class="line">type=&apos;deblur&apos;</span><br><span class="line"># 1. 比对</span><br><span class="line">qiime alignment mafft \</span><br><span class="line">		--i-sequences rep-seqs-$&#123;type&#125;.qza \</span><br><span class="line">		--o-alignment aligned-rep-seqs.qza</span><br><span class="line"># 2. 屏蔽高变区</span><br><span class="line">qiime alignment mask \</span><br><span class="line">		--i-alignment aligned-rep-seqs.qza \</span><br><span class="line">		--o-masked-alignment masked-aligned-rep-seqs.qza</span><br><span class="line"># 3. 生成无根树</span><br><span class="line">qiime phylogeny fasttree \</span><br><span class="line">		--i-alignment masked-aligned-rep-seqs.qza \</span><br><span class="line">		--o-tree unrooted-tree.qza</span><br><span class="line"># 4. 转化为有根树</span><br><span class="line">qiime phylogney midpoint-root \</span><br><span class="line">		--i-tree unrooted-tree.qza \</span><br><span class="line">		--o-rooted-tree rooted-tree.qza</span><br></pre></td></tr></table></figure></p>
<h2 id="6-alpha和beta多样性分析"><a href="#6-alpha和beta多样性分析" class="headerlink" title="6 alpha和beta多样性分析"></a>6 alpha和beta多样性分析</h2><p>qiime2的多样性分析可以通过q2-diversity插件实现，它支持计算alpha和beta多样性矩阵、相关的统计检验、生成交互可视化表等。我们首先使用core-metrics-phylogenetic方法，根据用户指定的深度(–p-sampling-depth)进行抽样，从而计算多样性矩阵，生成PCoA图。默认计算的矩阵类型有如下这些：</p>
<ul>
<li>alpha diversity</li>
<li><ul>
<li>Shannon’s diversity index : 群落丰富度的定量指标</li>
</ul>
</li>
<li><ul>
<li>Observed OTUs: 群落丰富度的定性指标</li>
</ul>
</li>
<li><ul>
<li>Faith’s Phylogenetic Diversity: 群落丰富度的定性指标，考虑各个特征之间的系统发育关系</li>
</ul>
</li>
<li><ul>
<li>Pielou’s Evenness: 群落均匀度的指标</li>
</ul>
</li>
<li><p>beta diversity</p>
</li>
<li><ul>
<li>Jaccard distanc： 群落相异性(群落多样性)的定性指标</li>
</ul>
</li>
<li><ul>
<li>Bray-Curits distance: 群落相异性(群落多样性)的定量指标</li>
</ul>
</li>
<li><ul>
<li>unweighted UniFrac distance: 群落相异性(群落多样性)的定性指标，考虑到各个特征之间的系统发育关系</li>
</ul>
</li>
<li><ul>
<li>weighted UniFrac distance：群落相异性(群落多样性)的定量指标，考虑到各个特征之间的系统发育关系</li>
</ul>
</li>
</ul>
<p>值得注意的是，–p-samping-depth的值是十分重要的，因为不同的抽样深度，最后统计出来的结果是不一样的，多样性矩阵对此很敏感。如果有样品的序列counts低于这个设定值时，这个样品将被丢弃。这个值的选择可以根据前面生成的特征表统计结果来选择（第四步的table.qzv)。如果数据量都很大，选最小的就好。但是有异常小的值，去掉这个值再选最小的。</p>
<h3 id="6-1-计算多样性矩阵"><a href="#6-1-计算多样性矩阵" class="headerlink" title="6.1 计算多样性矩阵"></a>6.1 计算多样性矩阵</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">qiime diversity core-metrics-phylogenetic \</span><br><span class="line">		--i-phylogeny rooted-tree.qza \</span><br><span class="line">		--i-table table.qza \</span><br><span class="line">		--p-sampling-depth 653 \</span><br><span class="line">		--m-metadata-file sample-metadata.tsv \</span><br><span class="line">		--output-dir core-metrics-results</span><br></pre></td></tr></table></figure>
<h3 id="6-2-alpha-diversity-analysis"><a href="#6-2-alpha-diversity-analysis" class="headerlink" title="6.2 alpha diversity analysis"></a>6.2 alpha diversity analysis</h3><p>在计算完多样性矩阵之后，我们可以探究各个样品中的微生物群落组成。<br>首先我们检验一下metadata中类别与alpha 多样性之间的联系，这里我们使用Faith Phylogenetic Diversity和evenness metrics来计算。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">qiime diversity alpha-group-significance \</span><br><span class="line">		--i-alpha-diversity core-metrics-results/faith_pd_vector.qza \</span><br><span class="line">		--m-metadata-file sample-metadata.tsv \</span><br><span class="line">		--o-visualization core-metrics-results/faith-pd-group-significance.qzv</span><br><span class="line"></span><br><span class="line">qiime diversity alpha-group-significance \</span><br><span class="line">		--i-alpha-diversity core-metrics-results/evenness_vector.qza \</span><br><span class="line">		--m-metadata-file sample-metadata.tsv \</span><br><span class="line">		--o-visualization core-metrics-results/evenness-group-significance.qzv</span><br></pre></td></tr></table></figure>
<p>如果想探究某个条件同群落多样性（alpha diversity relation）的相关性的话，可以使用qiime diversity alpha-correlation来检验。</p>
<h3 id="6-3-beta-diversity-analysis"><a href="#6-3-beta-diversity-analysis" class="headerlink" title="6.3 beta diversity analysis"></a>6.3 beta diversity analysis</h3><p>接下来我们分析特定分组下的样品组成，这里使用permanova统计方法，命令是qiime diversity beta-group-significance. 它将探究组内样品的距离与组间样品的距离的差异所在。–p-pairwise用于设定成对检验，比如本例中可以是gut vs. tongue。下面我们将检验不同组之间的unweighted UniFrac distance差异。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 分组为 BodySite, 统计unweighted_unifrace距离的组间是否有显著差异</span></span><br><span class="line">qiime diversity beta-group-significance \</span><br><span class="line">		--i-distance-matrix core-metrics-results/unweighted_unifrac_distance_matrix.qza \</span><br><span class="line">		--m-metadata-file sample-metadata.tsv \</span><br><span class="line">		--m-metadata-column BodySite \</span><br><span class="line">		--o-visualization core-metrics-results/unweighted-unifrac-body-site-significance.qzv \</span><br><span class="line">		--p-pairwise</span><br><span class="line"></span><br><span class="line"><span class="comment"># 分组为Subject, 统计unweighted_unifrace距离的组间是否有显著差异</span></span><br><span class="line">qiime diversity beta-group-significance \</span><br><span class="line">		--i-distance-matrix core-metrics-results/unweighted_unifrac_distance_matrix.qza \</span><br><span class="line">		--m-metadata-file sample-metadata.tsv \</span><br><span class="line">		--m-metadata-column Subject \</span><br><span class="line">		--o-visualization core-metrics-results/unweighted-unifrac-subject-significance.qzv \</span><br><span class="line">		--p-pairwise</span><br></pre></td></tr></table></figure>
<p>由于这里没有数据与样品的组成有关，我们就没有检验其中的关系。如果你对检验某个条件/分组对样品组成(sample composition)的相关性的话，可以结合使用qiime metadata distance-matrix, qiime diversity mantel, qiime diversity bioenv三个命令来实现。</p>
<h3 id="6-4-PCoA分析"><a href="#6-4-PCoA分析" class="headerlink" title="6.4 PCoA分析"></a>6.4 PCoA分析</h3><p>最后，分类是探究微生物群落组成的流行方法。我们可以使用Emperor工具来进行PCoA分析。虽然前面qiime diversity core-metrics-phylogenetic 生成了一些PCoA图，但是我们可以使用–p-custom-axes来个性化这个分析，这对探究时间序列数据十分有用。这里我们使用unweihted UniFrac和Bray-Curtis来进行PCoA分析，新的分析将加入daysSinceExperimentStart作为轴，以此来探究样品随时间的变化情况。</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line"># 三维可视化展示unweighted_unifrac的PCoA分析</span><br><span class="line">qiime emperor plot \</span><br><span class="line">		--i-pcoa core-metrics-results/unweighted_unifrac_pcoa_results.qza \</span><br><span class="line">		--m-metadata-file sample-metadata.tsv \</span><br><span class="line">		--p-custom-axes DaysSinceExperimentStart \</span><br><span class="line">		--o-visualization core-metrics-results/unweighted-unifrac-emperor-DaysSinceExperimentStart.qzv</span><br><span class="line"># 三维可视化展示unweighted_unifrac的PCoA分析</span><br><span class="line">qiime emperor plot \</span><br><span class="line">		--i-pcoa core-metrics-results/bray_curtis_pcoa_results.qza \</span><br><span class="line">		--m-metadata-file sample-metadata.tsv \</span><br><span class="line">		--p-custom-axes DaysSinceExperimentStart \</span><br><span class="line">		--o-visualization core-metrics-results/bray_curtis-emperor-DaysSinceExperimentStart.qzv</span><br></pre></td></tr></table></figure>
<h2 id="7-alpha稀释性曲线分析"><a href="#7-alpha稀释性曲线分析" class="headerlink" title="7 alpha稀释性曲线分析"></a>7 alpha稀释性曲线分析</h2><p>稀释性曲线(rarefaction curve)：一般是从样本中随机抽取一定数量的个体，统计出这些个体所代表物种数目，并以个体数与物种数来构建曲线。它可以用来比较测序数量不同的样本物种的丰富度，也可以用来说明样本的取样大小是否合理。分析采用对优化序列进行随机抽样的方法，以抽到的序列数与它们所能代表OTU的数目构建rarefaction curve。稀释性曲线图中，当曲线趋向平坦时，说明取样的数量合理，更多的取样只会产生少量新的OTU，反之则表明继续取样还可能产生较多新的OTU。因此，通过作稀释性曲线，可以得出样品的取样深度情况。<br>接下来我将使用qiime diversity alpha-rarefaction来探究抽样深度(sampling depth)与alpha diversity的关系。该命令将计算不同抽样深度下的一至多个α多样性矩阵，抽样深度由–p-min-depth/–p-max-depth指定。在每个抽样深度下，有10个多样性矩阵稀释表生成，稀释表迭代次数可由–p-iterations指定.你可以通过–me-metadata-file指定metadata进行样品分组。</p>
<p>生成的可视化文件有2个图。上面的那个图是alpha稀释曲线，注意用于判定样品的群落丰富度是否已经完全被观测/测序到。如果该曲线趋于平滑的话，表明对应x轴的抽样深度已经足以发现所有的群落种类（物种）。下面的那个图在有分组的时候很重要。它表明了抽样稀释时各个组剩余的样品数。如果抽样的深度d大于某样品的总计counts的话，往后的深度该样品就无法进行抽样了。这个图显示这在一定抽样深度下，稀释曲线的可信度。如果在某个抽样深度时，有某个组的剩余样品数为0，那么在该抽样深度下计算个多样性矩阵值就不可靠。</p>
<p>因此，你设定的–p-max-depth的值应该根据特征表统计的可视化图来选择，一般选择median frequency的值。如果稀释性曲线结果不好的话，可以适当提高或降低，但通常不能超出[low total frequency, max total frequency]的范围。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">qiime diversity alpha-rarefaction \</span><br><span class="line">		--i-table table.qza \</span><br><span class="line">		--i-phylogeny rooted-tree.qza \</span><br><span class="line">		--p-max-depth 4000 \</span><br><span class="line">		--m-metadata-file sample-metadata.tsv \</span><br><span class="line">		--o-visualization alpha-rarefaction.qzv</span><br></pre></td></tr></table></figure>
<h2 id="8-物种分类"><a href="#8-物种分类" class="headerlink" title="8 物种分类"></a>8 物种分类</h2><p>接下来，我们将探究样品的物种组成，并将其同样品表型信息联系起来。</p>
<p>第一步要对特征序列进行物种注释, 这里用到前面生成的特征数据(FeatureData[Sequence]).我们将使用一个预先训练过的朴素贝叶斯分类器(native bayes classifier)和q2-feature-classifier插件。这个分类器用Greengenes 13_8 99% OTUs (16s, 250 bases, v4 region, 515F/806R)的数据训练过。</p>
<p>值得注意的是，物种分类器在经过你的特有样品类似产生的数据训练后将表现更好，所以你可以用自己的数据完成<a href="https://docs.qiime2.org/2018.6/tutorials/feature-classifier/" target="_blank" rel="noopener">分类器的训练</a>。在<a href="https://docs.qiime2.org/2018.6/data-resources/" target="_blank" rel="noopener">资源页面</a>中，我们有提供其他数据类型训练过的分类器,你需要下载你数据的相同类型数据训练过的分类器来进行后续的分析。</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line"># 进行物种分类</span><br><span class="line">wget https://data.qiime2.org/2018.6/common/gg-13-8-99-515-806-nb-classifier.qza</span><br><span class="line">qiime feature-classifier classify-sklearn \</span><br><span class="line">		--i-classifier gg-13-8-99-515-806-nb-classifier.qza \</span><br><span class="line">		--i-reads rep-seqs.qza \</span><br><span class="line">		--o-classification taxonomy.qza</span><br><span class="line"># 可视化分类结果</span><br><span class="line">qiime metadata tabulate \</span><br><span class="line">		--m-input-file taxonomy.qza \</span><br><span class="line">		--o-visualization taxonomy.qzv</span><br><span class="line"># 物种分类柱状图</span><br><span class="line">qiime taxa barplot \</span><br><span class="line">		--i-table table.qza \</span><br><span class="line">		--i-taxonomy taxonomy.qza \</span><br><span class="line">		--m-metadata-file sample-metadata.tsv \</span><br><span class="line">		--o-visualization taxa-bar-plots.qzv</span><br></pre></td></tr></table></figure>
<h2 id="9-ANCOM-微生物组成-丰度差异分析"><a href="#9-ANCOM-微生物组成-丰度差异分析" class="headerlink" title="9 ANCOM (微生物组成/丰度差异分析)"></a>9 ANCOM (微生物组成/丰度差异分析)</h2><p>ANCOM，analysis of composition of microbiomes，可以用于确定不同样品间的微生物组成的丰度差异特征，在使用之前，你应该了解使用ANCOM的假设和局限性。丰度差异分析的研究比较活跃，目前已经有两个插件可以用于这个分析：q2-gneiss和q2-composition，这里我们使用q2-composition，如果你感兴趣的话，可以<a href="https://docs.qiime2.org/2018.6/tutorials/gneiss/" target="_blank" rel="noopener">在这里学习前者的使用</a>.</p>
<p>ANCOM假定只有少于25%的features会在样品之间发生改变。如果你确信你的研究有更多的features会发生改变，你应该舍弃ANCOM的使用，因为这将产生更多的错误，不论是一类还是二类错误。由于在各个身体部位的样品之间的丰度变化应该很大，所以我们只使用gut组样品进行这个分析。<br>在提取出gut组的特征表后，使用ANCOM分析该表。ANCOM基于每个样品的特征频率，但是不能容忍0值的存在。这里分为2步。首先要生成组成特征表(FeatureTable[Composition]),然后才能进行差异分析。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 对特征表进行过滤，只剩gut组</span></span><br><span class="line">qiime feature-table filter-samples \</span><br><span class="line">		--i-table table.qza \</span><br><span class="line">		--m-metadata-file sample-metadata.tsv \</span><br><span class="line">		--p-where <span class="string">"BodySite='gut'"</span> \</span><br><span class="line">		--o-filtered-table gut-table.qza</span><br><span class="line"></span><br><span class="line"><span class="comment"># 添加假的count（0值不容忍），生成组成特征表</span></span><br><span class="line">qiime composition add-pseudocount \</span><br><span class="line">		--i-table gut-table.qza \</span><br><span class="line">		--o-composition-table comp-gut-table.qza</span><br><span class="line"></span><br><span class="line"><span class="comment"># ANCOM分析</span></span><br><span class="line">qiime composition ancom \</span><br><span class="line">		--i-table comp-gut-table.qza \</span><br><span class="line">		--m-metadata-file sample-metadata.tsv \</span><br><span class="line">		--m-metadata-column Subject \</span><br><span class="line">		--o-visualization ancom-Subject.qzv</span><br></pre></td></tr></table></figure>
<p>有时候，我们会在特定的分类水平进行丰度差异分析，为了实现这个目标，我们可以对特征频数表进行聚合，然后再做ANCOM分析。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 在level 6的分类水平聚合数据</span></span><br><span class="line">qiime taxa collapse \</span><br><span class="line">		--i-table gut-table.qza \</span><br><span class="line">		--i-taxonomy taxonomy.qza \</span><br><span class="line">		--p-level 6</span><br><span class="line">		--o-collapsed-table gut-table-L6.qza</span><br><span class="line"></span><br><span class="line">qiime composition add-pseudocount \</span><br><span class="line">		--i-table gut-table-L6.qza \</span><br><span class="line">		--o-composition-table comp-gut-table-L6.qza</span><br><span class="line"></span><br><span class="line">qiime composition ancom \</span><br><span class="line">		--i-table comp-gut-table-L6.qza \</span><br><span class="line">		--m-metadata-file sample-metadata.tsv \</span><br><span class="line">		--m-metadata-column Subject \</span><br><span class="line">		--o-visualization L6-ancom-Subject.qzv</span><br></pre></td></tr></table></figure>
<h2 id="10-成对差异比较-时间序列分析"><a href="#10-成对差异比较-时间序列分析" class="headerlink" title="10 成对差异比较/时间序列分析"></a>10 成对差异比较/时间序列分析</h2><p>这部分是从fecal microbiota transplant教程里面整合的，主要使用的qiime longitudinal模块。要适应前面的数据，进行更改即可，具体的用法可以见<strong>第三节 3.7 使用q2-longitudinal进行纵向和成对差异比较</strong>。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 这里有2个坑</span></span><br><span class="line"><span class="comment"># 第一个是metadata里面donor的行在week列的值都为-1；而我们设定的值是0和18的比较。要把donor行都去掉</span></span><br><span class="line"><span class="comment"># 第二个是pairwise-difference里面的metadata，evenness_vector对应的metric列名是pielou_e。</span></span><br></pre></td></tr></table></figure>
	  
	</div>

	<!-- recommended posts -->
	

	<div>
  	<center>
	<div class="pagination">
<ul class="pagination">
	 
				
    	<li class="prev"><a href="/2018/10/15/扩增子测序-QIIME2/" class="alignleft prev"><i class="fa fa-arrow-circle-o-left"></i>Prev</a></li>
  		

        <li><a href="/archives"><i class="fa fa-archive"></i>Archive</a></li>

		
		   <li class="next"><a href="/2018/10/15/扩增子测序-理论基础/" class="alignright next">Next<i class="fa fa-arrow-circle-o-right"></i></a></li>         
        
	
</ul>
</div>

    </center>
	</div>

    <!-- share -->
    
        
    <div class="bdsharebuttonbox">
        <a href="#" class="bds_more" data-cmd="more"></a>
        <a href="#" class="bds_weixin" data-cmd="weixin" title="分享到微信"></a>
        <a href="#" class="bds_tsina" data-cmd="tsina" title="分享到新浪微博"></a>
        <a href="#" class="bds_fbook" data-cmd="fbook" title="分享到Facebook"></a>
        <a href="#" class="bds_twi" data-cmd="twi" title="分享到Twitter"></a>
        <a href="#" class="bds_linkedin" data-cmd="linkedin" title="分享到linkedin"></a>
        <a href="#" class="bds_evernotecn" data-cmd="evernotecn" title="分享到印象笔记"></a>
        <a href="#" class="bds_youdao" data-cmd="youdao" title="分享到有道云笔记"></a>
        <a href="#" class="bds_copy" data-cmd="copy" title="分享到复制网址"></a>
    </div>
    <script>
        window._bd_share_config={"common":{"bdSnsKey":{},"bdText":"","bdMini":"2","bdMiniList":false,"bdPic":"","bdStyle":"1","bdSize":"24"},"share":{}};
        with(document)0[(getElementsByTagName('head')[0]||body).appendChild(createElement('script')).src='http://bdimg.share.baidu.com/static/api/js/share.js?v=89860593.js?cdnversion='+~(-new Date()/36e5)];
    </script>


        

    
	
	<!-- comment -->
	
<section id="comment">
  <h2 class="title">Comments</h2>
    	 
	 <div id="comment-thread"></div>
	 <div id="loading-spin"></div>
	 <script type="text/javascript">
	   getComments({
           type: "github" ? "github" : "github",       
	       user: "xizhihui",
	       repo: "",
		   client_id: "",
           client_secret: "",
		   no_comment: "No comments yet. Press the button and go to comment now!",
		   go_to_comment: "Go to comment",
		   no_issue: "no_issue",
		   issue_title: "扩增子测序 QIIME2使用示例",
		   issue_id: "",
		   btn_class: "btn btn-primary",
		   comments_target: "#comment-thread",
		   loading_target: "#loading_spin"
		   });
	 </script>
  
</section>


	</div> <!-- col-md-9/col-md-12 -->
		
	
	<div id="side_meta">
		<div class="col-md-3" id="post_meta"> 

	<!-- date -->
	
	<div class="meta-widget">
	<i class="fa fa-clock-o"></i>
	2018-10-15 
	</div>
	

	<!-- categories -->
    
	<div class="meta-widget">
	<a data-toggle="collapse" data-target="#categorys"><i class="fa fa-folder"></i></a>	
    <ul id="categorys" class="tag_box list-unstyled collapse in">
          
  <li>
    </li><li><a href="/categories/Bioinformatics/">Bioinformatics<span>36</span></a></li>
  

    </ul>
	</div>
	

	<!-- tags -->
	
	<div class="meta-widget">
	<a data-toggle="collapse" data-target="#tags"><i class="fa fa-tags"></i></a>		  
    <ul id="tags" class="tag_box list-unstyled collapse in">	  
	    
  <li><a href="/tags/软件和包/">软件和包<span>22</span></a></li> <li><a href="/tags/扩增子/">扩增子<span>5</span></a></li>
    </ul>
	</div>
	

	<!-- toc -->
	<div class="meta-widget">
	
	   <a data-toggle="collapse" data-target="#toc"><i class="fa fa-bars"></i></a>
	   <div id="toc" class="toc collapse in">
			<ol class="toc-article"><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#1-导入数据到QIIME2"><span class="toc-article-text">1 导入数据到QIIME2</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#2-拆分序列"><span class="toc-article-text">2 拆分序列</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#3-序列质控和构建特征表（feature-table）"><span class="toc-article-text">3 序列质控和构建特征表（feature  table）</span></a><ol class="toc-article-child"><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#3-1-DADA2"><span class="toc-article-text">3.1 DADA2</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#3-2-Deblur"><span class="toc-article-text">3.2 Deblur</span></a></li></ol></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#4-特种表和特征数据统计"><span class="toc-article-text">4 特种表和特征数据统计</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#5-生成系统发育树用于多样性分析"><span class="toc-article-text">5 生成系统发育树用于多样性分析</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#6-alpha和beta多样性分析"><span class="toc-article-text">6 alpha和beta多样性分析</span></a><ol class="toc-article-child"><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#6-1-计算多样性矩阵"><span class="toc-article-text">6.1 计算多样性矩阵</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#6-2-alpha-diversity-analysis"><span class="toc-article-text">6.2 alpha diversity analysis</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#6-3-beta-diversity-analysis"><span class="toc-article-text">6.3 beta diversity analysis</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#6-4-PCoA分析"><span class="toc-article-text">6.4 PCoA分析</span></a></li></ol></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#7-alpha稀释性曲线分析"><span class="toc-article-text">7 alpha稀释性曲线分析</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#8-物种分类"><span class="toc-article-text">8 物种分类</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#9-ANCOM-微生物组成-丰度差异分析"><span class="toc-article-text">9 ANCOM (微生物组成/丰度差异分析)</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#10-成对差异比较-时间序列分析"><span class="toc-article-text">10 成对差异比较/时间序列分析</span></a></li></ol>
		</div>
	
	</div>
	
    <hr>
	
</div><!-- col-md-3 -->

	</div>
		

</div><!-- row -->



	</div>
  </div>
  <div class="container-narrow">
  <footer> <p>
  &copy; 2018 Xizhihui
  
      with help from <a href="http://hexo.io/" target="_blank">Hexo</a> and <a href="http://getbootstrap.com/" target="_blank">Twitter Bootstrap</a>. Theme by <a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind</a>.    
</p> </footer>
</div> <!-- container-narrow -->
  


  
<a id="gotop" href="#">   
  <span>▲</span> 
</a>

<script src="/js/jquery.imagesloaded.min.js"></script>
<script src="/js/gallery.js"></script>
<script src="/js/bootstrap.min.js"></script>
<script src="/js/main.js"></script>
<script src="/js/search.js"></script> 


<link rel="stylesheet" href="/fancybox/jquery.fancybox.css" media="screen" type="text/css">
<script src="/fancybox/jquery.fancybox.pack.js"></script>
<script type="text/javascript">
(function($){
  $('.fancybox').fancybox();
})(jQuery);
</script>



   <script type="text/javascript">      
     var search_path = "search.xml";
	 if (search_path.length == 0) {
	 	search_path = "search.xml";
	 }
	 var path = "/" + search_path;
     searchFunc(path, 'local-search-input', 'local-search-result');
   </script>


<!-- syntax highlighting -->

  <script>
  marked.setOptions({
    highlight: function (code, lang) {
        return hljs.highlightAuto(code).value;
    }
  });
  function Highlighting(){
    var markdowns = document.getElementsByClassName('markdown');
    for(var i=0;i<markdowns.length;i++){
        if(markdowns[i].innerHTML) markdowns[i].innerHTML =marked(markdowns[i].innerHTML);
    }
  }
  window.addEventListener('DOMContentLoaded', Highlighting, false);
  window.addEventListener('load', Highlighting, false);
  </script>


</body>
   </html>
