<!DOCTYPE html>
<html>
<head>
	<title>XZH's Notes</title>
	<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
	<meta name="author" content="xizhihui" />
	<meta name="viewport" content="width=device-width, initial-scale=1" />
	<link rel="stylesheet" type="text/css" href="/statics/reset.css">
	<link rel="stylesheet" type="text/css" href="/statics/header_footer.css"></link>
	<link rel="stylesheet" type="text/css" href="/statics/default.css">
	<link rel="stylesheet" type="text/css" href="/statics/main.css">
</head>
<body>
	<header class="Header">
		<nav>
			<h3>XZH</h3>
			<ul>
				<li><a href="/index.html#Timeline">Timeline</a></li><li><a href="/index.html#Category">Category</a></li><li><a href="/index.html#Tags">Tags</a></li><li><a href="/index.html#About">About</a></li>
			</ul>
		</nav>
	</header>
	<div class="Motto">
		<h2>XZH's Notes</h2>
		<h4>The future comes for you ~ ~ ~</h4>
	</div>

	<main>
		<nav id="table"></nav>
		<content id="content" class="content">
			<h2 id="qiime2">使用QIIME2进行扩增子分析</h2>
<h2 id="1-qiime2">1 qiime2中数据的导入与导出</h2>
<p>在QIIME2中，所有的输入数据都是以qiime2 artifacts的格式(.qza)存在，该格式有利于数据传递和生成路径追踪。在QIIME2中，你可以在分析的各个阶段导入数据，不论是原始的测序数据，还是经过处理产生的中间数据(eg. biom)，你都可以直接导入，接着进行后续分析。导入数据使用qiime tools import命令，你可以使用--show-importable-types和--show-importable-formats查看支持导入的数据类型，选择与你数据相应的导入类型。</p>
<p>你可以使用qiime tools export导出qiime的数据，用于R或者其他软件。这个命令与qiime tools extract命令的差别在于，导出命令只导出数据，与数据相关的生成追踪信息将被丢弃；但是提取命令不会，里面包含的provenance文件保存了这些追踪信息。</p>
<h3 id="11">1.1 导入含有测序质量的信息序列数据</h3>
<h4 id="111-emp-protocol-multiplexed-fastq">1.1.1 'EMP protocol' multiplexed fastq</h4>
<p>在EMP的格式结果中，由于它是混池测序的，有不同样品不同实验的测序结果，所以有2个fastq.gz文件，一个是测序序列文件，各个样品的测序结果混杂在一起(multiplexed)，另一个是对应的barcode文件.</p>
<p>这里的测序数据分别有single-end和pair-end，需要指定不同的导入类型</p>
<div class="codehilite"><pre><span></span>qiime tools import <span class="se">\</span>
        --type EMPSingleEndSequneces <span class="se">\</span>
        --input-path EMP_single <span class="se">\</span>
        --output-path EMP_single_sequences.qza

qiime tools import <span class="se">\</span>
        --type EMPPairEndSequences <span class="se">\</span>
        --input-path EMP_pair <span class="se">\</span>
        --output-path EMP_pair_sequences.qza
</pre></div>


<h4 id="112-fastq-manifest-formats">1.1.2 'Fastq manifest' formats</h4>
<p>在fastq manifest格式中，manifest文件指定了fastq.gz/fastq文件的绝对路径，比如TCGA中的manifest文件。你可以自己生成这类文件。它通常是逗号分隔的文件，每行的第一个域是样品标识符以供qiime2使用，第二个域是对应文件的绝对路径，第三个域是read的方向。注释行用#开头。文件的首行如果不是空行和注释行的话，则必须是表头行，内容为sample-id,absolute-filepath,direction。对于single-end reads，每个sample-id只能有1行信息。而对于pair-end reads，每个sample-id则必须有2行信息，分别对应着forward或reverse。在direction这一个域(列)的值只能是forward或者reverse。在absolute-filepath中，你可以使用环境变量$HOME/$PWD来指定路径。</p>
<p>在这个格式中，由于质量值不确定，我们需要选择自己数据质量值对应的输出格式。并且由于我们的manifest是根据sample-id来定义的，导入后的数据格式同EMP的数据demuplexed后的数据是一致的。</p>
<div class="codehilite"><pre><span></span><span class="c1"># pair-end reads</span>
sample-id,absolute-filepath,direction
sample-1,<span class="nv">$PWD</span>/path/sample1_R1.fastq.gz,forward
sample-1,<span class="nv">$PWD</span>/path/sample1_R2.fastq.gz,reverse
<span class="c1"># single-end reads</span>
sample-id,absolute-filepath,direction
sample-1,<span class="nv">$PWD</span>/path/sample1_R1.fastq.gz,forward
</pre></div>


<div class="codehilite"><pre><span></span><span class="c1"># 注意，在旧版本中--input-format对应着--source-format</span>
qiime tools import <span class="se">\</span>
        --type <span class="s1">&#39;SampleData[SequencesWithQuality]&#39;</span> <span class="se">\</span>
        --input-format SingleEndFastqManifestPhred33 <span class="se">\</span>
        --input-path ss-33-manifest <span class="se">\</span>
        --output-path single-end-demux.qza

qiime tools import <span class="se">\</span>
        --type <span class="s1">&#39;SampleData[PairedEndSequnecesWithQuality]&#39;</span>
        --input-format PairedEndFastqManifestPhred64
        --input-path pe-64-manifest <span class="se">\</span>
        --output-path paired-end-demux.qza
</pre></div>


<h4 id="113-casava-18-demultiplexed-fastq">1.1.3 Casava 1.8 demultiplexed fastq</h4>
<p>在这种格式中，每个样品只有1个fastq.gz文件，文件名就包含了样品标识符。对于single样品来说，可能是类似于‘L2S357_15_L001_R1_001.fastq.gz’的形式，含义是sample-id_barcode_lane-number_read-number_set-number。而对于paired样品来说，就额外多了一个‘L2S357_15_L001_R2_001.fastq.gz’，其中R1，R2表明是2个reads。</p>
<div class="codehilite"><pre><span></span><span class="c1"># single-end</span>
qiime tools import <span class="se">\</span>
        --type <span class="s1">&#39;SampleData[SequencesWithQuality]&#39;</span> <span class="se">\</span>
        --input-path casava-1.8 <span class="se">\</span>
        --input-format CasavaOneEightSingleLanePerSampleDirFmt <span class="se">\</span>
        --output-path demux-single-end.qza

<span class="c1"># paired-end</span>
qiime tools import <span class="se">\</span>
        --type <span class="s1">&#39;SampleData[PairedEndSequencesWithQuality]&#39;</span> <span class="se">\</span>
        --input-path casava-1.8 <span class="se">\</span>
        --input-format CasavaOneEightSingleLanePerSampleDirFmt <span class="se">\</span>
        --output-path demux-paired-end.qza
</pre></div>


<h3 id="12">1.2 导入特征表数据</h3>
<h4 id="121-biom">1.2.1 biom</h4>
<p>具体的格式描述见<a href="http://biom-format.org/documentation/format_versions/biom-1.0.html">biom v1.0.0</a>和<a href="http://biom-format.org/documentation/format_versions/biom-2.1.html">biom v2.1.0</a>。</p>
<div class="codehilite"><pre><span></span><span class="c1"># biom v1.0.0</span>
qiime tools import <span class="se">\</span>
        --input-path feature-table-v100.biom <span class="se">\</span>
        --type <span class="s1">&#39;FeatureTable[Frequency]&#39;</span> <span class="se">\</span>
        --input-format BIOMV100Format <span class="se">\</span>
        --output-path feature-table-v100.qza

<span class="c1"># biom v2.1.0</span>
qiime tools import <span class="se">\</span>
        --input-path feature-table-v210.biom <span class="se">\</span>
        --type <span class="s1">&#39;FeatureTable[Frequency]&#39;</span> <span class="se">\</span>
        --input-format BIOMV210Format <span class="se">\</span>
        --output-path feature-table-v210.qza
</pre></div>


<h3 id="13">1.3 导入特征序列数据</h3>
<p>这一类数据通常是代表序列数据，是fasta格式的DNA序列。有些没有经过比对的；所以不会含有-的gap形式。有些则是经过比对的代表序列数据，其中就会含有-，使得各个序列的长度一致。但是这2类数据可能有N这类简并碱基。有些qiime2命令不支持这类。</p>
<div class="codehilite"><pre><span></span><span class="c1"># unaligned representative sequences</span>
qiime tools import <span class="se">\</span>
        --type <span class="s1">&#39;FeatureData[Sequence]&#39;</span> <span class="se">\</span>
        --input-path sequences.fna <span class="se">\</span>
        --output-path sequences.qza

<span class="c1"># aligned representative sequences</span>
qiime tools import <span class="se">\</span>
        --type <span class="s1">&#39;FeatureData[AlignedSequence]&#39;</span> <span class="se">\</span>
        --input-path aligned-sequences.fna <span class="se">\</span>
        --output-path aligned-sequences.qza
</pre></div>


<h3 id="14">1.4 导入无根系统发育树</h3>
<div class="codehilite"><pre><span></span>qiime tools import <span class="se">\</span>
        --type <span class="s1">&#39;Phylogeny[Unrooted]&#39;</span>
        --input-path unrooted-tree.tre <span class="se">\</span>
        --output-path unrooted-tree.qza
</pre></div>


<h3 id="15">1.5 导出特征表</h3>
<div class="codehilite"><pre><span></span>qiime tools <span class="nb">export</span> <span class="se">\</span>
        --input-path feature-table.qza <span class="se">\</span>
        --output-path exported-feature-table
</pre></div>


<h3 id="16">1.6 导出发育树</h3>
<div class="codehilite"><pre><span></span>qiime tools <span class="nb">export</span> <span class="se">\</span>
        --input-path unrooted-tree.qza <span class="se">\</span>
        --output-path exproted-tree
</pre></div>


<h2 id="2-qiime2metadata">2 qiime2中的metadata</h2>
<p>qiime的sample meatadata存储了关于实验的元信息，诸如各类技术细节，barcod、run、Subject、time point等。 Feature metadata通常是一个特征注释，比如能域序列变化或OTU对应的分类信息。下面是一些关于metadata的操作。</p>
<h3 id="21-metadata">2.1 可视化metadata</h3>
<div class="codehilite"><pre><span></span>qiime metadata tabulate <span class="se">\</span>
        --m-input-file sample-metadata.tsv <span class="se">\</span>
        --o-visualization tabulated-sample-metadata.qzv

qiime metadata tabulate <span class="se">\</span>
        --m-input-file faith_pd_vector.qza <span class="se">\</span>
        --o-visualization tabulated-faith-pd-metadata.qzv
</pre></div>


<h3 id="22-metadata">2.2 合并两个metadata</h3>
<p>你可以联合几个metadata进行可视化</p>
<div class="codehilite"><pre><span></span>qiime metadata tabulate <span class="se">\</span>
  --m-input-file rep-seqs.qza <span class="se">\</span>
  --m-input-file taxonomy.qza <span class="se">\</span>
  --o-visualization tabulated-feature-metadata.qzv

qiime metadata tabulate <span class="se">\</span>
        --m-input-file sample-metadata.tsv <span class="se">\</span>
        --m-input-file faith_pd_vector.qza <span class="se">\</span>
        --o-visualization tabulated-combined.qzv

qiime emperor plot <span class="se">\</span>
        --i-pcoa unweighted_unifrac_pcoa_results.qza <span class="se">\</span>
        --m-metadata-file sample-metadata.tsv <span class="se">\</span>
        --m-metadata-file faith_pd_vector.qza <span class="se">\</span>
        --o-visualization unweighted-unifrac-emperor-with-alpha.qzv
</pre></div>


<h2 id="3-qiime2">3 qiime2中数据的过滤</h2>
<h3 id="31">3.1 过滤特征表</h3>
<p>特征表有两个维度：样品和特征，分别使用filter-samples和filter-features来实现。</p>
<ul>
<li>基于总体频率的过滤</li>
</ul>
<div class="codehilite"><pre><span></span><span class="c1"># 比如我们要过滤frequeny低于1500的样品</span>
qiime feature-table filter-samples <span class="se">\</span>
        --i-table table.qza <span class="se">\</span>
        --p-min-frequency <span class="m">1500</span> <span class="se">\</span>
        --o-filtered-table sample-frequency-filtered-table.qza

<span class="c1"># 过滤在所有样品中frequency总计少于10的features</span>
qiime feature-table filter-features <span class="se">\</span>
        --i-table table.qza <span class="se">\</span>
        --p-min-frequency <span class="m">10</span> <span class="se">\</span>
        --o-filtered-table feature-frequency-filtered-table.qza
</pre></div>


<ul>
<li>基于偶然性的过滤</li>
</ul>
<p>在进行计算特征表时或者特征序列时，有些样品个数或者某个特征仅仅为1，这些是需要进行过滤，因为可能是由测序引起的假阳性。</p>
<div class="codehilite"><pre><span></span>qiime feature-table filter-features <span class="se">\</span>
        --i-table table.qza <span class="se">\</span>
        --p-min-samples <span class="m">2</span> <span class="se">\</span>
        --o-filtered-table sample-contigency-filtered-table.qza

qiime feature-table filter-samples <span class="se">\</span>
        --i-table table.qza <span class="se">\</span>
        --p-min-features <span class="m">10</span> <span class="se">\</span>
        --o-filtered-table feature-contigency-filtered-table.qza
</pre></div>


<ul>
<li>基于id的过滤</li>
</ul>
<div class="codehilite"><pre><span></span>qiime feature-table filter-samples <span class="se">\</span>
        --i-table table.qza <span class="se">\</span>
        --m-metadata-file samples-to-keep.tsv <span class="se">\</span>
        --o-filtered-table id-filtered-table.qza
</pre></div>


<ul>
<li>基于metadata的过滤</li>
</ul>
<div class="codehilite"><pre><span></span><span class="c1"># 单个过滤</span>
qiime feature-table filter-samples <span class="se">\</span>
        --i-table table.qza <span class="se">\</span>
        --m-metadata-file sample-metadata.tsv <span class="se">\</span>
        --p-where <span class="s2">&quot;Subject=&#39;subject-1&#39;&quot;</span> <span class="se">\</span>
        --o-filtered-table subject-1-filtered-table.qza

<span class="c1"># 多个过滤</span>
qiime feature-table filter-samples <span class="se">\</span>
        --i-table table.qza <span class="se">\</span>
        --m-metadata-file sample-metadata.tsv <span class="se">\</span>
        --p-where <span class="s2">&quot;BodySite IN (&#39;left palm&#39;, &#39;right palm&#39;)&quot;</span> <span class="se">\</span>
        --o-filtered-table subject-1-filtered-table.qza

<span class="c1"># 联合过滤，支持SQL选择语句，AND，OR，AND NOT</span>
qiime feature-table filter-samples <span class="se">\</span>
        --i-table table.qza <span class="se">\</span>
        --m-metadata-file sample-metadata.tsv <span class="se">\</span>
        --p-where <span class="s2">&quot;Subject=&#39;subject-1&#39; AND BodySite=&#39;gut&#39;&quot;</span>
        --o-filtered-table subject-1-filtered-table.qza
</pre></div>


<ul>
<li>基于分类的过滤</li>
</ul>
<div class="codehilite"><pre><span></span>qiime taxa filter-table <span class="se">\</span>
        --i-table table.qza <span class="se">\</span>
        --i-taxonomy taxonomy.qza <span class="se">\</span>
        --p-exclude mitochondria <span class="se">\</span>
        --o-filtered-table table-no-mitochondria.qza

qiime taxa filter-table <span class="se">\</span>
        --i-table table.qza <span class="se">\</span>
        --i-taxonomy taxonomy.qza <span class="se">\</span>
        --p-include mitochondria, chloroplast <span class="se">\</span>
        --o-filtered-table table-no-mitochondria.qza

<span class="c1"># 保存有phylum-level注释的特征，从中保存的结果中去掉mitochondria,chloroplast</span>
qiime taxa filter-table <span class="se">\</span>
        --i-table table.qza <span class="se">\</span>
        --i-taxonomy taxonomy.qza <span class="se">\</span>
        --p-include p__ <span class="se">\</span>
        --p-exclude mitochondria,chloroplast <span class="se">\</span>
        --o-filtered-table table-with-phyla-no-mitochondria-no-chloroplast.qza

<span class="c1"># 通过指定匹配模式为extact进行过滤</span>
qiime taxa filter-table <span class="se">\</span>
        --i-table table.qza <span class="se">\</span>
        --i-taxonomy taxonomy.qza <span class="se">\</span>
        --p-mode exact <span class="se">\</span>
        --p-exclude <span class="s2">&quot;k__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rickettsiales; f__mitochondria&quot;</span> <span class="se">\</span>
        --o-filtered-table table-no-mitochondria-exact.qza
</pre></div>


<h3 id="32">3.2 过滤序列</h3>
<p>在q2-feature-table中也有个filter-seqs，你可以用它进行多条件过滤。q2-quality-control里面的exclude-seqs也能进行序列过滤。</p>
<div class="codehilite"><pre><span></span>qiime taxa filter-seqs <span class="se">\</span>
        --i-sequences sequences.qza <span class="se">\</span>
        --i-taxonomy taxonomy.qza <span class="se">\</span>
        --p-include p__ <span class="se">\</span>
        --p-exclude mitochondria,chloroplast <span class="se">\</span>
        --o-filtered-sequences sequences-with-phyla-no-mitochondria-no-chloroplast.qza
</pre></div>


<h3 id="33">3.3 过滤距离矩阵</h3>
<div class="codehilite"><pre><span></span>qiime diversity filter-distance-matrix <span class="se">\</span>
        --i-distance-matrix distance-matrix.qza <span class="se">\</span>
        --m-metadata-file samples-to-keep.tsv <span class="se">\</span>
        --o-filtered-distance-matrix identifier-filtered-distance-matrix.qza

qiime diversity filter-distance-matrix <span class="se">\</span>
        --i-distance-matrix distance-matrix.qza <span class="se">\</span>
        --m-metadata-file sample-metadata.tsv <span class="se">\</span>
        --p-where <span class="s2">&quot;Subject=&#39;subject-2&#39;&quot;</span> <span class="se">\</span>
        --o-filtered-distance-matrix subject-2-filtered-distance-matrix.qza
</pre></div>


<h2 id="4-qiime2">4 qiime2中特征分类器的训练</h2>
<p>我们使用greengenes的参考序列对Native Bayes classifier进行训练，然后对目标代表序列进行分类。我们已经预先训练好了<a href="https://docs.qiime2.org/2018.8/data-resources/">多个分类器</a>，你可以直接使用。<strong>要想训练分类器的话，有2个必须的东西：参考序列和物种分类</strong>。真实情况下，一般使用97%及以上的相似度的参考OTU数据集进行训练。参考序列和物种分类应该是要对应起来的额。比如你使用的是Greengenes 99% OTU sequences，那么你应该使用99% OTU taxonomy。</p>
<h3 id="41">4.1 下载和导入数据</h3>
<div class="codehilite"><pre><span></span><span class="c1"># 这里作为演示，用的是85 otu。真实情况不能使用。</span>
wget https://data.qiime2.org/2018.8/tutorials/training-feature-classifiers/85_otus.fasta
wget https://data.qiime2.org/2018.8/tutorials/training-feature-classifiers/85_otu_taxonomy.txt
wget https://data.qiime2.org/2018.8/tutorials/training-feature-classifiers/rep-seqs.qza

qiime tools import <span class="se">\</span>
        --type <span class="s1">&#39;FeatureData[Sequence]&#39;</span> <span class="se">\</span>
        --input-path 85_otus.fasta <span class="se">\</span>
        --output-path 85_otus.qza

qiime tools import <span class="se">\</span>
        --type <span class="s1">&#39;FeatureData[Taxonomy]&#39;</span> <span class="se">\</span>
        --input-path 85_otu_taxonomy.txt <span class="se">\</span>
        --output-path 85_otu_taxonomy.qza
</pre></div>


<h3 id="42-reads">4.2 提取参考reads</h3>
<p>在处理16s数据时，使用目标序列的特定区域训练朴素贝叶斯分类器会提高分类的准确度。在对ITS序列时，则不用这样，因为根据经验，在处理ITS的UNITE参考序列的预先训练时，进行提取或截断的效果并不是很好。所以对于ITS分析，使用全长更好。
在这里，我们先使用515F/806R primer把能匹配的reads提取出来，然后截断成120bp的长度。注意，只有当待分类的代表序列被截断成一致长度时，参考序列才需要截断成对应的长度(这里是120bp)。用于提取的primer应该是具有生物学意义的序列，不能是adapter、linker、barcode之类的，一般来讲，若是你的primer长于30 nt，肯定有非生物学序列在里面。</p>
<div class="codehilite"><pre><span></span>qiime feature-classifier extract-reads <span class="se">\</span>
        --i-sequences 85_otus.qza <span class="se">\</span>
        --p-f-primer GTGCCAGCMGCCGCGGTAA <span class="se">\</span>
        --p-r-primer GGACTACHVGGGTWTCTAAT <span class="se">\</span>
        --p-trunc-len <span class="m">120</span> <span class="se">\</span>
        --o-reads ref-seqs.qza
</pre></div>


<h3 id="43">4.3 训练分类器</h3>
<div class="codehilite"><pre><span></span>qiime feature-classifier fit-classifier-naive-bayes <span class="se">\</span>
        --i-reference-reads ref-seqs.qza <span class="se">\</span>
        --i-reference-taxonomy ref-taxonomy.qza <span class="se">\</span>
        --o-classifier classifier.qza
</pre></div>


<h3 id="44">4.4 测试分类器</h3>
<div class="codehilite"><pre><span></span>qiime feature-classifier classify-sklearn <span class="se">\</span>
        --i-classifier classifier.qza <span class="se">\</span>
        --i-reads rep-seqs.qza <span class="se">\</span>
        --o-classification taxonomy.qza

qiime metadata tabulate <span class="se">\</span>
        --m-input-file taxonomy.qza <span class="se">\</span>
        --o-visualization taxonomy.qzv
</pre></div>


<h2 id="5-q2-quality-control">5 使用q2-quality-control进行质控和过滤</h2>
<p>下面演示了q2-quality-control如何使用模拟群落/已知群落组成的已知样品来进行数据质控和过滤。</p>
<h3 id="51">5.1 下载数据</h3>
<div class="codehilite"><pre><span></span>wget https://data.qiime2.org/2018.8/tutorials/quality-control/query-seqs.qza
wget https://data.qiime2.org/2018.8/tutorials/quality-control/reference-seqs.qza
wget https://data.qiime2.org/2018.8/tutorials/quality-control/query-table.qza
wget https://data.qiime2.org/2018.8/tutorials/quality-control/qc-mock-3-expected.qza
wget https://data.qiime2.org/2018.8/tutorials/quality-control/qc-mock-3-observed.qza
</pre></div>


<h3 id="52">5.2 通过比对过滤序列</h3>
<p>exclude-seqs将查询序列和参考序列进行比对，根据多个比对标准区分能比对上和不能比对上的序列。你可以用本步骤来去除已知的污染序列、排除宿主序列、或这其他非目标序列。
exclude-seqs目前支持blast、vsearch、blastn-short作为比对方法，如果查询序列短于30nt，应当使用blastn-short。</p>
<div class="codehilite"><pre><span></span>qiime quality-control exclude-seqs <span class="se">\</span>
        --i-query-sequences query-seqs.qza <span class="se">\</span>
        --i-reference-sequences reference-seqs.qza <span class="se">\</span>
        --p-method blast <span class="se">\</span>
        --p-perc-identity <span class="m">0</span>.97 <span class="se">\</span>
        --p-perc-query-aligned <span class="m">0</span>.97 <span class="se">\</span>
        --o-sequence-hits hist.qza <span class="se">\</span>
        --o-sequence-misses misses.qza

qiime feature-table filter-features <span class="se">\</span>
        --i-table query-table.qza <span class="se">\</span>
        --m-metadata-file hits.qza <span class="se">\</span>
        --o-filtered-table no-hits-filtered-table.qza <span class="se">\</span>
        --p-exclude-ids
</pre></div>


<h3 id="53">5.3 用已知组成的样品进行质控</h3>
<p>模拟群落的群落组成和丰度都是已知的，可以用于生信的基准测试(benchmarking bioinformatics methods)，比如测定某个方法或流程的处理准确度。
quality-control的evaluate-composition可以验证分类器分类结果的准确度。现有的模拟群落可在<a href="https://github.com/caporaso-lab/mockrobiota/blob/master/inventory.tsv">mockrobiota查询</a>.
在结果中，预期和观测的物种丰度的TAR(taxon accuracy rate)、TDR(taxon detection rate)和linear regression scores都会计算，并根据每个物种水平进行作图。</p>
<div class="codehilite"><pre><span></span>qiime quality-control evaluate-composition <span class="se">\</span>
        --i-expected-features qc-mock-3-expected.qza <span class="se">\</span>
        --i-observed-features qc-mock-3-observed.qza <span class="se">\</span>
        --o-visualiation qc-mock-3-cmparison.qzv
</pre></div>


<h3 id="54">5.4 序列质量的验证</h3>
<p>evaluate-seqs将查询序列和参考序列进行比对验验证比对的质量。利用预测序列和观测序列可以验证desnoising或OTU picking结果的准确度。</p>
<div class="codehilite"><pre><span></span>qiime quality-control evaluate-seqs <span class="se">\</span>
  --i-query-sequences query-seqs.qza <span class="se">\</span>
  --i-reference-sequences reference-seqs.qza <span class="se">\</span>
  --o-visualization eval-seqs-test.qzv
</pre></div>


<h2 id="6-q2-sample-classifiermetadata">6 使用q2-sample-classifier预测metadata</h2>
<p>正如多数统计方法一样，sample-classifier的预测功能也需要一定量的样本才能产生有意义的结果，一般来说，需要至少50个样品；样本太少产生的结果将会导致不准确的模型或者错误。用于预测的类别型元数据列的每个值都至少需要10个sample；用于回归的连续型元数据列不能有太多离群值或者分布不均匀。</p>
<div class="codehilite"><pre><span></span>wget https://data.qiime2.org/2018.8/tutorials/moving-pictures/sample_metadata.tsv
wget https://data.qiime2.org/2018.8/tutorials/sample-classifier/moving-pictures-table.qza
</pre></div>


<h3 id="61">6.1 预测类别型样品数据</h3>
<p>这里我们将根据样品的微生物组成来预测这个样品时来源于哪个身体部位。按照classify-samples进行,内部步骤如下：
1. 把样品随机分成训练集和测试集；通过--p-test-size来指定测试集的大小。
2. 选择一个估值器(estimator)，设置参数进行训练。
3. K-fold cross-validation进行模型优化，通过--p-cv设置K-fold值。
4. 使用测试集进行预测
5. 计算预测模型准确度</p>
<h4 id="611">6.1.1 建立预测模型</h4>
<div class="codehilite"><pre><span></span>qiime sample-classfifier classify-samples <span class="se">\</span>
        --i-table moving-pictures-table.qza <span class="se">\</span>
        --m-metadata-file moving-pictures-sample-metadata.tsv <span class="se">\</span>
        --m-metadata-column BodySite <span class="se">\</span>
        --p-optimize-feature-selection <span class="se">\</span>
        --p-parameter-tuning <span class="se">\</span>
        --p-estimator RandomForestClassifier <span class="se">\</span>
        --p-n-estimators <span class="m">20</span> <span class="se">\</span>
        --output-dir moving-pictures-classifier
<span class="c1"># output:</span>
<span class="c1"># sample_estimator.qza, feature_importance.qza, predictions.qza, accuracy_result.qzv, model_summary.qzv</span>

qiime metadata tabulate <span class="se">\</span>
        --m-input-file moving-pictures-classifier/predictions.qza <span class="se">\</span>
        --o-visualization moving-pictures-classifier/predictions.qzv

qiime metadata tabulate <span class="se">\</span>
        --m-input-file moving-pictures-classifier/feature_importance.qza <span class="se">\</span>
        --o-visualization moving-pictures-classifier/feature_importance.qzv

<span class="c1"># 利用预测重要的feature过滤原始table</span>
qiime feature-table filter-features <span class="se">\</span>
        --i-table moving-pictures-table.qza <span class="se">\</span>
        --m-metadata-file moving-pictures-classifier/feature_importance.qza <span class="se">\</span>
        --o-filtered-table moving-pictures-classifier/important-feature-table.qza
</pre></div>


<h4 id="612">6.1.2 进行预测</h4>
<div class="codehilite"><pre><span></span><span class="c1"># 预测新的metadata</span>
qiime sample-classifier predict-classification <span class="se">\</span>
        --i-table new-table.qza <span class="se">\</span>
        --i-sample-estimator moving-pictures-classifier/sample_estimator.qza <span class="se">\</span>
        --o-predictions moving-pictures-classifier/new_predictions.qza


<span class="c1"># 用新的样品重新测试模型的准确性</span>
qiime sample-classifier confusion-matrix <span class="se">\</span>
        --i-predictions moving-pictures-classifier/new_predictions.qza <span class="se">\</span>
        --m-truth-file moving-pictures-sample-metadata.tsv <span class="se">\</span>
        --m-truth-column BodySite <span class="se">\</span>
        --o-visualization moving-pictures-classifier/new_confusion_matrix.qzv
</pre></div>


<h3 id="62">6.2 预测连续型样品数据</h3>
<div class="codehilite"><pre><span></span>wget https://data.qiime2.org/2018.8/tutorials/longitudinal/sample_metadata.tsv
wget https://data.qiime2.org/2018.8/tutorials/longitudinal/ecam_table_maturity.qza
</pre></div>


<h4 id="621">6.2.1 训练模型</h4>
<div class="codehilite"><pre><span></span>qiime sample-classifier regress-samples <span class="se">\</span>
        --i-table ecam-table.qza <span class="se">\</span>
        --m-metadata-file ecam-metadata.tsv <span class="se">\</span>
        --m-metadata-column month <span class="se">\</span>
        --p-estimator RandomForestRegressor <span class="se">\</span>
        --p-n-estimators <span class="m">20</span> <span class="se">\</span>
        --output-dir ecam-regressor

<span class="c1"># ouput</span>
<span class="c1"># sample_estimator.qza, feature_importance.qza, accuracy_results.qzv, model_summary.qzv</span>
</pre></div>


<h3 id="63-nested-cross-validation">6.3 使用nested cross-validation法进行预测</h3>
<h4 id="631">6.3.1 类别型变量：训练模型</h4>
<div class="codehilite"><pre><span></span>qiime sample-classifier classify-samples-ncv <span class="se">\</span>
        --i-table moving-pictures-table.qza <span class="se">\</span>
        --m-metadata-file moving-pictures-sample-metadata.tsv <span class="se">\</span>
        --m-metadata-column BodySite <span class="se">\</span>
        --p-estimator RandomForestClassifier <span class="se">\</span>
        --p-n-estimators <span class="m">20</span> <span class="se">\</span>
        --o-predictions BodySite-predictions-ncv.qza <span class="se">\</span>
        --o-feature-importance BodySite-importance-ncv.qza

<span class="c1"># 查看模型准确性</span>
qiime sample-classifier confusion-matrix <span class="se">\</span>
        --i-predictions BodySite-predictions-ncv.qza <span class="se">\</span>
        --m-truth-file moving-pictures-sample-metadata.tsv <span class="se">\</span>
        --m-truth-column BodySite <span class="se">\</span>
        --o-visualization ncv_confusion_matrix.qzv
</pre></div>


<h4 id="632">6.3.2 连续型变量：训练模型</h4>
<div class="codehilite"><pre><span></span>qiime sample-classifier regress-samples-ncv <span class="se">\</span>
        --i-table ecam-table.qzv <span class="se">\</span>
        --m-metadata ecam-metadata.tsv <span class="se">\</span>
        --m-metadata-column month <span class="se">\</span>
        --p-n-estimators <span class="m">20</span> <span class="se">\</span>
        --p-estimator RandomForestRegressor <span class="se">\</span>
        --o-predictions ecam-predictions-ncv.qza <span class="se">\</span>
        --o-feature-importance ecam-importance-ncv.qza

<span class="c1"># 查看结果准确性</span>
qiime sample-classifier scatterplot <span class="se">\</span>
        --i-predictions ecam-predictions-ncv.qza <span class="se">\</span>
        --m-truth-file ecam-metadata.tsv <span class="se">\</span>
        --m-truth-column month <span class="se">\</span>
        --o-visualization ecam-scatter.qzv
</pre></div>


<h3 id="64q2-sample-classifier">6.4.最佳实践：你不应该做q2-sample-classifier做的事</h3>
<ul>
<li>数据泄漏</li>
<li>过拟合</li>
</ul>
<h2 id="7-q2-longitudinal">7 使用q2-longitudinal进行纵向和成对差异比较</h2>
<p>q2-longitudinal可以用来进行纵向研究设计和成对样品的统计和可视化，以探究样品是如何在可观测的状态之间变化的。可观测的状态通常是时间或者环境梯度，也可以是不同时间点的成对分析比如成对的距离和差异分析；比如在治疗前后的临床研究。可观测的状态也可以是方法上的，在同个时间点上采用不同方法的结果的比较。比如q2-longitudinal可以比较不同样品收集方法、存储方法、DNA提取方法对单个样品的微生物群落特征组成影响。</p>
<p>在q2-longitudinal中的操作多数以某个度量值作为输入，这通常是metadata里面的某个列、alpha diversity vectors、PCoA results或者是feature ID。合法的度量值名字可以通过qiime metadata tabulate命令查看得到，feature name也可以通过qiime feature-data summarize命令查看得到。</p>
<h3 id="71">7.1 成对差异比较</h3>
<p>成对差异比较探究的是成对样品之间某个特定的度量值是否具有显著性变化；目前支持的比较类型有：
+ 特征丰度的比较(comparison of feature abundance): 比如在特征表中微生物序列变体或者变化率
+ 元数据值的比较(comparison of metadata values): 比如将alpha/beta diversity values作为度量值，结合某个元数据值进行比较</p>
<p>下面的命令是探究以delivery mode作为分组，不同的两个时间点上(state-column)两组的shannon diversity index(alpha diversity)是否发现显著变化。</p>
<div class="codehilite"><pre><span></span>qiime longitudinal pairwise-differences <span class="se">\</span>
        --m-metadata-file ecam-sample-metadata.tsv <span class="se">\</span>
        --m-metadata-file shannon.qza <span class="se">\</span>
        --p-metric shannon <span class="se">\</span>
        --p-group-column delivery <span class="se">\</span>
        --p-state-column month <span class="se">\</span>
        --p-state-1 <span class="m">0</span> <span class="se">\</span>
        --p-state-2 <span class="m">12</span> <span class="se">\</span>
        --p-individual-id-column studyid <span class="se">\</span>
        --p-replicate-handling random <span class="se">\</span>
        --o-visualization pairwise-differences.qzv
</pre></div>


<h3 id="72">7.2 成对距离比较</h3>
<p>成对距离比较探究的也是成对样品之间某个特定的度量值是否具有显著性变化，但是，相对于使用metadata的某个列或者qiime artifact作为输入，成对距离比较以距离矩阵作为输入，来探究成对样品在pre和post下的距离变化，并显示其差异是否显著。
下面的命令是探究以delivery作为分组，正常分娩和剖腹产在12个月后对微生物组成的稳定性的影响。</p>
<div class="codehilite"><pre><span></span>qiime longitudinal pairwise-distances <span class="se">\</span>
        --i-distance-matrix unweighted_unifrac_distance_matrix.qza <span class="se">\</span>
        --m-metadata-file ecam-sample-metadata.tsv <span class="se">\</span>
        --p-group-column delivery <span class="se">\</span>
        --p-state-column month <span class="se">\</span>
        --p-state-1 <span class="m">0</span> <span class="se">\</span>
        --p-state-2 <span class="m">12</span> <span class="se">\</span>
        --p-individual-id-column studyid <span class="se">\</span>
        --p-replicate-handling random <span class="se">\</span>
        --o-visualization pairwise-distances.qzv
</pre></div>


<h3 id="73">7.3 线性混合效应模型</h3>
<p>线性混合效应模型(linear mixed effects models, LME)探究单个因变量和一至多个自变量之间的关系；因变量的观测值通常在跨相关样本中获得，比如在重复测量的抽样实验中。该模型以至少一个数字型state-column(如时间)或者一至多个逗号分隔的group-columns作为自变量，然后绘出因变量(metric,某个度量)的回归曲线。此外，individual-id-column参数应该是个可以指明样品重复情况的metadata column。因变量可以是metadata的某列(column)，也可以是feature table的feature ID。逗号分隔的多个随机效应值可以作为该模型的输入；模型默认每个个体都有随机的截距；如果用户为每个个体提供随机的斜率(slope)的话，可以使用state-column值来作为random-effects的值。</p>
<p>注意，判断某个因素是为固定效应还是随机效应比较复杂。一般来说，如果一个因子的不同水平(metadata column values)可以代表所有的离散值的话，该因子应该是固定效应；例如delivery mode、sex、diet在下面的分析中就是固定效应。那相反地，如果一个因子的值是代表群体内的随机样品，那么这个因子就是随机效应；比如body-weight、daily-kcal-from-breastmilk等，这些值只能代表从人群中的某些人，并不能代表所有的情况。</p>
<p>这里我们使用LME来探究shannon diversity index是否会随时间、delivery mode、diet、sex发生变化。</p>
<div class="codehilite"><pre><span></span>qiime longitudinal linear-mixed-effects <span class="se">\</span>
        --m-metadata-file ecam-sample-metadata.tsv <span class="se">\</span>
        --m-metadata-file shannon.qza <span class="se">\</span>
        --p-metric shannon <span class="se">\</span>
        --p-group-columns delivery,diet,sex <span class="se">\</span>
        --p-state-column month <span class="se">\</span>
        --p-individual-id-column studyid <span class="se">\</span>
        --o-visualization linear-mixed-effects.qzv
</pre></div>


<p>产生的可视化文件包含多个图。首先是输入参数显示在最上方。然后是model summary显示LME模型的训练信息。主要的结果是第三个，所谓model results table。它显示了每个固定效应和效应之间的互作对因变量(shannon diversity)的影响，它还显示了其他信息如参数估计、估计标准差、z-score、p-values等。从该表可以看出shannon diversity受month、diet和其他几个交互因子影响明显。最后是依据各个group column制作的散点图和线性回归曲线。如果--p-lowess指定了，还会显示加权平均值。第一类散点图只是进行了一下快速统计，可以使用volatility进行交互式地统计和作图。第二类散点图显示每个样品的观测残差值和度量预测值之间的关系：度量预测值应该在0值附近，与残差值没有相关性。如果观察到U形或者非随机分布图，说明你的预测变量(group_columns, random_effects)不能很好地表征数据。</p>
<h3 id="74-volatility-anaylsis">7.4 波动分析(volatility anaylsis)</h3>
<p>波动分析(volatility analysis)可产生交互式的折线图以探究因变量在单组或多组之间的连续型自变量的依赖关系。多个metadata file(alpha,beta diversity)和FeatureTable[RelativeFrequency]可以作为输入，也可以选择不同的因变量作为图的y轴。
这里我们探究一下shannon diversity和其他metadata是如何随时间变化的。</p>
<div class="codehilite"><pre><span></span>qiime longitudinal volatility <span class="se">\</span>
        --m-metadata-file ecam-sample-metadata.tsv <span class="se">\</span>
        --m-metadata-file shannon.qza <span class="se">\</span>
        --p-default-metric shannon <span class="se">\</span>
        --p-default-group-column delivery <span class="se">\</span>
        --p-state-column month <span class="se">\</span>
        --p-individual-id-column studyid <span class="se">\</span>
        --o-visualization volatility.qzv
</pre></div>


<h3 id="75">7.5 用一阶差分来追踪变化率</h3>
<p>观察时间序列数据的另一个方法就是计算变化率随时间的变化，要实现这个目的就需要计算一阶差分，它表征了在连续时间点之间的变化大小。delta Yt = Y(t) -Y(t-1),这个变换在first-differences/first-distances中用到。对于生成的结果文件，你可以进行波动分析。</p>
<div class="codehilite"><pre><span></span><span class="c1"># alpha diversity or metadata column (metric)</span>
qiime longitudinal first-differences <span class="se">\</span>
  --m-metadata-file ecam-sample-metadata.tsv <span class="se">\</span>
  --m-metadata-file shannon.qza <span class="se">\</span>
  --p-state-column month <span class="se">\</span>
  --p-metric shannon <span class="se">\</span>
  --p-individual-id-column studyid <span class="se">\</span>
  --p-replicate-handling random <span class="se">\</span>
  --o-first-differences shannon-first-differences.qza
<span class="c1"># beta diversity (matrix)</span>
qiime longitudinal first-distances <span class="se">\</span>
  --i-distance-matrix unweighted_unifrac_distance_matrix.qza <span class="se">\</span>
  --m-metadata-file ecam-sample-metadata.tsv <span class="se">\</span>
  --p-state-column month <span class="se">\</span>
  --p-individual-id-column studyid <span class="se">\</span>
  --p-replicate-handling random <span class="se">\</span>
  --o-first-distances first-distances.qza
<span class="c1"># 经过计算，beta diversity也可以进行LME分析</span>
qiime longitudinal linear-mixed-effects <span class="se">\</span>
  --m-metadata-file first-distances.qza <span class="se">\</span>
  --m-metadata-file ecam-sample-metadata.tsv <span class="se">\</span>
  --p-metric Distance <span class="se">\</span>
  --p-state-column month <span class="se">\</span>
  --p-individual-id-column studyid <span class="se">\</span>
  --p-group-columns delivery,diet <span class="se">\</span>
  --o-visualization first-distances-LME.qzv
</pre></div>


<p>你也可以从固定时间点开始追踪变化率的大小变化。</p>
<div class="codehilite"><pre><span></span>qiime longitudinal first-distances <span class="se">\</span>
  --i-distance-matrix unweighted_unifrac_distance_matrix.qza <span class="se">\</span>
  --m-metadata-file ecam-sample-metadata.tsv <span class="se">\</span>
  --p-state-column month <span class="se">\</span>
  --p-individual-id-column studyid <span class="se">\</span>
  --p-replicate-handling random <span class="se">\</span>
  --p-baseline <span class="m">5</span> <span class="se">\</span>
  --o-first-distances first-distances-baseline-0.qza
</pre></div>


<h3 id="76">7.6 微生物相互依赖性非参数检验</h3>
<p>在微生物群落中，微生物种群并不会独立存在，而是以生态互作网存在。而这些互作网是否会在同组个体之家产生临时效应将指明不同的研究方向。微生物相互依赖性非参数检验(non-parametric microbial interdependence test, NMIT)将评估同组的同个群落内的特征相互依赖性(比如种类、序列变体、OTU)如何依据时间不同而不同。MNIT评估纵向样本相似性。对于每个个体，NMIT计算成对特征的相关性；然后基于NMIT矩阵计算成对个体的距离。该方法只对纵向数据有用，例如时间序列数据。为使检验结果鲁棒，我们建议每个个体最少要测5个时间点的数据；但它不要求所有的样品数据都在同个时间点测得。</p>
<p>NMIT的检验结果是一个距离矩阵，你可以像处理beta diversity的结果数据一样进行处理。</p>
<div class="codehilite"><pre><span></span>wget https://data.qiime2.org/2018.8/tutorials/longitudinal/ecam_table_taxa.qza

qiime longitudinal nmit <span class="se">\</span>
  --i-table ecam-table-taxa.qza <span class="se">\</span>
  --m-metadata-file ecam-sample-metadata.tsv <span class="se">\</span>
  --p-individual-id-column studyid <span class="se">\</span>
  --p-corr-method pearson <span class="se">\</span>
  --o-distance-matrix nmit-dm.qza

qiime diversity beta-group-significance <span class="se">\</span>
  --i-distance-matrix nmit-dm.qza <span class="se">\</span>
  --m-metadata-file ecam-sample-metadata.tsv <span class="se">\</span>
  --m-metadata-column delivery <span class="se">\</span>
  --o-visualization nmit.qzv

qiime diversity pcoa <span class="se">\</span>
  --i-distance-matrix nmit-dm.qza <span class="se">\</span>
  --o-pcoa nmit-pc.qza

qiime emperor plot <span class="se">\</span>
  --i-pcoa nmit-pc.qza <span class="se">\</span>
  --m-metadata-file ecam-sample-metadata.tsv <span class="se">\</span>
  --o-visualization nmit-emperor.qzv
</pre></div>


<h3 id="77">7.7 特征波动分析</h3>
<p>特征波动分析是一个监回归方法，可以确定某个数字型metadata column、state-column的值，并绘出不同state的相对频率图。state-column通常是个时间度量值，不过它也可以时任何数字型column；但是它不能和individual-id-column相同。</p>
<div class="codehilite"><pre><span></span>wget -O ecam-table.qza https://data.qiime2.org/2018.8/tutorials/longitudinal/ecam_table_maturity.qza

qiime longitudinal feature-volatility <span class="se">\</span>
  --i-table ecam-table.qza <span class="se">\</span>
  --m-metadata-file ecam-sample-metadata.tsv <span class="se">\</span>
  --p-state-column month <span class="se">\</span>
  --p-individual-id-column studyid <span class="se">\</span>
  --p-n-estimators <span class="m">10</span> <span class="se">\</span>
  --p-random-state <span class="m">123</span> <span class="se">\</span>
  --output-dir ecam-feat-volatility
</pre></div>


<h3 id="78">7.8 成熟指数预测</h3>
<p>这个分析在对各个时间点样品数目分布均匀时的分析结果较好; 对于某些时间点有缺失值的情况下无法进行分析；此外，本分析还需要大量数据才能正常分析，特别是对照组在各个时间点需要足够的生物学重复。这个分析基于特征数据(feature data)构建回归模型，计算微生物成熟指数，一次来预测给定连续型metadata column的值，例如给定微生物组成预测个体的年龄。这个分析域标准监督回归分析不同在于它量化了多个组之间随时间变化的相对速率。</p>
<p>下面我们比较在不同年龄下，剖腹产和自然分娩之间微生物群落的成熟度。</p>
<div class="codehilite"><pre><span></span>qiime longitudinal maturity-index <span class="se">\</span>
  --i-table ecam-table.qza <span class="se">\</span>
  --m-metadata-file ecam-sample-metadata.tsv <span class="se">\</span>
  --p-state-column month <span class="se">\</span>
  --p-group-by delivery <span class="se">\</span>
  --p-individual-id-column studyid <span class="se">\</span>
  --p-control Vaginal <span class="se">\</span>
  --p-test-size <span class="m">0</span>.4 <span class="se">\</span>
  --output-dir maturity
</pre></div>


<h2 id="8-qiime2">8 qiime2中嵌合体的去除</h2>
<p>qiime2中的嵌合体检测基于FeatureTable[Frequency]和FeatureData[Sequences]，它整合了Uchime de novo和vsearch这两个软件。</p>
<div class="codehilite"><pre><span></span>wget https://data.qiime2.org/2018.8/tutorials/chimera/atacama-table.qza
wget https://data.qiime2.org/2018.8/tutorials/chimera/atacama-rep-seqs.qza
</pre></div>


<h3 id="81-de-novo-chimera-checking">8.1 de novo chimera checking</h3>
<div class="codehilite"><pre><span></span>qiime vsearch uchime-denovo <span class="se">\</span>
        --i-table atacama-table.qza <span class="se">\</span>
        --i-sequences atacama-rep-seqs.qza <span class="se">\</span>
        --output-dir uchime-dn-out
<span class="c1"># out</span>
<span class="c1"># nonchimeras.qza, chimeras.qza, stats.qza</span>

<span class="c1"># 可视化统计结果</span>
qiime metadata tabulate <span class="se">\</span>
        --m-input-file uchime-dn-out/stats.qza <span class="se">\</span>
        --o-visualization uchime-dn-out/stats.qzv
</pre></div>


<h3 id="82-tables">8.2 对输入的tables和序列进行过滤</h3>
<ul>
<li>去除chimeras和borderline chimeras</li>
</ul>
<div class="codehilite"><pre><span></span>qiime feature-table filter-features <span class="se">\</span>
  --i-table atacama-table.qza <span class="se">\</span>
  --m-metadata-file uchime-dn-out/nonchimeras.qza <span class="se">\</span>
  --o-filtered-table uchime-dn-out/table-nonchimeric-wo-borderline.qza

qiime feature-table filter-seqs <span class="se">\</span>
  --i-data atacama-rep-seqs.qza <span class="se">\</span>
  --m-metadata-file uchime-dn-out/nonchimeras.qza <span class="se">\</span>
  --o-filtered-data uchime-dn-out/rep-seqs-nonchimeric-wo-borderline.qza

qiime feature-table summarize <span class="se">\</span>
  --i-table uchime-dn-out/table-nonchimeric-wo-borderline.qza <span class="se">\</span>
  --o-visualization uchime-dn-out/table-nonchimeric-wo-borderline.qzv
</pre></div>


<ul>
<li>去除嵌合体，但是保留 borderline chimeras</li>
</ul>
<div class="codehilite"><pre><span></span>qiime feature-table filter-features <span class="se">\</span>
  --i-table atacama-table.qza <span class="se">\</span>
  --m-metadata-file uchime-dn-out/chimeras.qza <span class="se">\</span>
  --p-exclude-ids <span class="se">\</span>
  --o-filtered-table uchime-dn-out/table-nonchimeric-w-borderline.qza

qiime feature-table filter-seqs <span class="se">\</span>
  --i-data atacama-rep-seqs.qza <span class="se">\</span>
  --m-metadata-file uchime-dn-out/chimeras.qza <span class="se">\</span>
  --p-exclude-ids <span class="se">\</span>
  --o-filtered-data uchime-dn-out/rep-seqs-nonchimeric-w-borderline.qza

qiime feature-table summarize <span class="se">\</span>
  --i-table uchime-dn-out/table-nonchimeric-w-borderline.qza <span class="se">\</span>
  --o-visualization uchime-dn-out/table-nonchimeric-w-borderline.qzv
</pre></div>


<h2 id="9-qiime2reads">9 qiime2中的reads合并与去噪</h2>
<p>这里描述的方法并不是说要代替DADA2中的合并序列、去噪过程。相反地，这里的方法的关注点在于分析 paired-end reads的其他方法。如果你对使用DADA2进行合并和去噪感兴趣的话，<a href="https://docs.qiime2.org/2018.8/tutorials/atacama-soils/">atacama soil</a>写明了如何使用qiime dada2 denoise-paired实现；如果使用dada2的话，请不要预先进行合并序列。</p>
<div class="codehilite"><pre><span></span>wget -O demux.qza https://data.qiime2.org/2018.8/tutorials/read-joining/atacama-seqs.qza
</pre></div>


<h3 id="91-joining-reads">9.1 joining reads</h3>
<div class="codehilite"><pre><span></span>qiime vsearch join-pairs <span class="se">\</span>
        --i-demultiplexed-seqs demux.qza <span class="se">\</span>
        --o-joined-sequences demux-joined.qza

qiime demux summarize <span class="se">\</span>
        --i-data demux-joined.qza <span class="se">\</span>
        --o-visualization demux-joined.qzv
</pre></div>


<h3 id="92-sequence-quality-control">9.2 sequence quality control</h3>
<div class="codehilite"><pre><span></span>qiime quality-filter q-score-joined <span class="se">\</span>
        --i-demux demux-joined.qza <span class="se">\</span>
        --o-filtered-sequences demux-joined-filtered.qza <span class="se">\</span>
        --o-filter-stats demux-joined-filter-stats.qza
</pre></div>


<h3 id="93-deblur">9.3 deblur</h3>
<p>在第二步之后，你可以选择使用deblur进一步质控，也可以进行去重复序列然后进行聚类。</p>
<div class="codehilite"><pre><span></span>qiime deblur denoise-16S <span class="se">\</span>
  --i-demultiplexed-seqs demux-joined-filtered.qza <span class="se">\</span>
  --p-trim-length <span class="m">250</span> <span class="se">\</span>
  --p-sample-stats <span class="se">\</span>
  --o-representative-sequences rep-seqs.qza <span class="se">\</span>
  --o-table table.qza <span class="se">\</span>
  --o-stats deblur-stats.qza

qiime feature-table summarize <span class="se">\</span>
  --i-table table.qza <span class="se">\</span>
  --o-visualization table.qzv
</pre></div>


<h3 id="94-reads">9.4 导入预合并的reads</h3>
<div class="codehilite"><pre><span></span>wget https://data.qiime2.org/2018.8/tutorials/read-joining/fj-joined.zip
unzip fj-joined.zip

qiime tools import <span class="se">\</span>
  --input-path fj-joined/manifest <span class="se">\</span>
  --output-path fj-joined-demux.qza <span class="se">\</span>
  --type SampleData<span class="o">[</span>JoinedSequencesWithQuality<span class="o">]</span> <span class="se">\</span>
  --input-format SingleEndFastqManifestPhred33

qiime demux summarize <span class="se">\</span>
  --i-data fj-joined-demux.qza <span class="se">\</span>
  --o-visualization fj-joined-demux.qzv
</pre></div>


<h2 id="10-qiime2otu">10 qiime2中的OTU聚类</h2>
<p>在qiime2中，de novo, closed-reference, open-reference clustering均已支持。在拆分序列、质控、去重复后，可以使用vsearch进行序列/特征聚类成OTUs。</p>
<div class="codehilite"><pre><span></span>wget https://data.qiime2.org/2018.8/tutorials/otu-clustering/seqs.fna
wget https://data.qiime2.org/2018.8/tutorials/otu-clustering/85_otus.qza

qiime tools import <span class="se">\</span>
        --input-path seqs.fna <span class="se">\</span>
        --output-path seqs.qza <span class="se">\</span>
        --type <span class="s1">&#39;SampleData[Sequences]&#39;</span>
</pre></div>


<h3 id="101">10.1 对序列进行去重复</h3>
<div class="codehilite"><pre><span></span>qiime vsearch dereplicate-sequences <span class="se">\</span>
        --i-sequences seqs.qza <span class="se">\</span>
        --o-dereplicated-table table.qza <span class="se">\</span>
        --o-dereplicated-sequences rep-seqs.qza
</pre></div>


<h3 id="102-clustering">10.2 clustering</h3>
<ul>
<li>de novo clustering</li>
</ul>
<div class="codehilite"><pre><span></span>qiime vsearch cluster-features-de-novo <span class="se">\</span>
        --i-table table.qza <span class="se">\</span>
        --i-sequences rep-seqs.qza <span class="se">\</span>
        --p-perc-identity <span class="m">0</span>.99 <span class="se">\</span>
        --o-clustered-table table-dn-99.qza <span class="se">\</span>
        --o-clustered-sequences rep-seqs-dn-99.qza
</pre></div>


<ul>
<li>closed-reference clustering</li>
</ul>
<div class="codehilite"><pre><span></span>qiime vsearch cluster-features-closed-reference <span class="se">\</span>
        --i-table tablq.qza <span class="se">\</span>
        --i-sequences rep-seqs.qza <span class="se">\</span>
        --i-reference-sequences 85_otus.qza <span class="se">\</span>
        --p-perc-identity <span class="m">0</span>.85 <span class="se">\</span>
        --o-clustered-table table-cr-85.qza <span class="se">\</span>
        --o-clustered-sequences  rep-seqs-cr-85.qza <span class="se">\</span>
        --o-unmatched-sequences unmatched-cr-85.qza
</pre></div>


<ul>
<li>open-reference clustering</li>
</ul>
<div class="codehilite"><pre><span></span>qiime vsearch cluster-features-open-reference <span class="se">\</span>
        --i-table tablq.qza <span class="se">\</span>
        --i-sequences rep-seqs.qza <span class="se">\</span>
        --i-reference-sequences 85_otus.qza <span class="se">\</span>
        --p-perc-identity <span class="m">0</span>.85 <span class="se">\</span>
        --o-clustered-table table-or-85.qza <span class="se">\</span>
        --o-clustered-sequences  rep-seqs-or-85.qza <span class="se">\</span>
        --o-new-reference-sequences new-ref-seqs-or-85.qza
</pre></div>
		</content>

	</main>
	<footer>
		<p> Powered by xizhihui </p>
		<p> Thanks to <a href="https://www.hifreud.com/">hifreud</a> the theme is learned from his site. </p>
	</footer>

<script type="text/javascript" src="/js/index.json"></script>
<script type="text/javascript" src="/js/category.json"></script>
<script type="text/javascript" src="/js/tags.json"></script>
<script type="text/javascript" src="/js/about.json"></script>
<script type="text/javascript" src="/js/index.js"></script>
<script type="text/javascript">
	// listen to click event on navigator in header
	// document
	// 	.querySelector("div#head_right ul")
	// 	.addEventListener("click", change_page)
	var table = [['h2', 'qiime2', '使用QIIME2进行扩增子分析'], ['h2', '1-qiime2', '1 qiime2中数据的导入与导出'], ['h3', '11', '1.1 导入含有测序质量的信息序列数据'], ['h4', '111-emp-protocol-multiplexed-fastq', "1.1.1 'EMP protocol' multiplexed fastq"], ['h4', '112-fastq-manifest-formats', "1.1.2 'Fastq manifest' formats"], ['h4', '113-casava-18-demultiplexed-fastq', '1.1.3 Casava 1.8 demultiplexed fastq'], ['h3', '12', '1.2 导入特征表数据'], ['h4', '121-biom', '1.2.1 biom'], ['h3', '13', '1.3 导入特征序列数据'], ['h3', '14', '1.4 导入无根系统发育树'], ['h3', '15', '1.5 导出特征表'], ['h3', '16', '1.6 导出发育树'], ['h2', '2-qiime2metadata', '2 qiime2中的metadata'], ['h3', '21-metadata', '2.1 可视化metadata'], ['h3', '22-metadata', '2.2 合并两个metadata'], ['h2', '3-qiime2', '3 qiime2中数据的过滤'], ['h3', '31', '3.1 过滤特征表'], ['h3', '32', '3.2 过滤序列'], ['h3', '33', '3.3 过滤距离矩阵'], ['h2', '4-qiime2', '4 qiime2中特征分类器的训练'], ['h3', '41', '4.1 下载和导入数据'], ['h3', '42-reads', '4.2 提取参考reads'], ['h3', '43', '4.3 训练分类器'], ['h3', '44', '4.4 测试分类器'], ['h2', '5-q2-quality-control', '5 使用q2-quality-control进行质控和过滤'], ['h3', '51', '5.1 下载数据'], ['h3', '52', '5.2 通过比对过滤序列'], ['h3', '53', '5.3 用已知组成的样品进行质控'], ['h3', '54', '5.4 序列质量的验证'], ['h2', '6-q2-sample-classifiermetadata', '6 使用q2-sample-classifier预测metadata'], ['h3', '61', '6.1 预测类别型样品数据'], ['h4', '611', '6.1.1 建立预测模型'], ['h4', '612', '6.1.2 进行预测'], ['h3', '62', '6.2 预测连续型样品数据'], ['h4', '621', '6.2.1 训练模型'], ['h3', '63-nested-cross-validation', '6.3 使用nested cross-validation法进行预测'], ['h4', '631', '6.3.1 类别型变量：训练模型'], ['h4', '632', '6.3.2 连续型变量：训练模型'], ['h3', '64q2-sample-classifier', '6.4.最佳实践：你不应该做q2-sample-classifier做的事'], ['h2', '7-q2-longitudinal', '7 使用q2-longitudinal进行纵向和成对差异比较'], ['h3', '71', '7.1 成对差异比较'], ['h3', '72', '7.2 成对距离比较'], ['h3', '73', '7.3 线性混合效应模型'], ['h3', '74-volatility-anaylsis', '7.4 波动分析(volatility anaylsis)'], ['h3', '75', '7.5 用一阶差分来追踪变化率'], ['h3', '76', '7.6 微生物相互依赖性非参数检验'], ['h3', '77', '7.7 特征波动分析'], ['h3', '78', '7.8 成熟指数预测'], ['h2', '8-qiime2', '8 qiime2中嵌合体的去除'], ['h3', '81-de-novo-chimera-checking', '8.1 de novo chimera checking'], ['h3', '82-tables', '8.2 对输入的tables和序列进行过滤'], ['h2', '9-qiime2reads', '9 qiime2中的reads合并与去噪'], ['h3', '91-joining-reads', '9.1 joining reads'], ['h3', '92-sequence-quality-control', '9.2 sequence quality control'], ['h3', '93-deblur', '9.3 deblur'], ['h3', '94-reads', '9.4 导入预合并的reads'], ['h2', '10-qiime2otu', '10 qiime2中的OTU聚类'], ['h3', '101', '10.1 对序列进行去重复'], ['h3', '102-clustering', '10.2 clustering']]
	var html = format_article_table(table)
	var table_ele = document.querySelector("main #table")
	table_ele.innerHTML = html
	// window.onload = evt => resize_nav(table_ele)
	adaption(document, window)

</script>
</body>
</html>