<!DOCTYPE HTML>
<html>
<head><meta name="generator" content="Hexo 3.8.0">
  <meta charset="utf-8">
  
  <title>生信基础 质控的那些事 | Xizhihui&#39;s Notes</title>
  <meta name="author" content="Xizhihui">
  
  <meta name="description" content="质控的方向就平常的生物实验而言，质控是要贯穿到实验的各个阶段的，比如那些阳性阴性对照就可以认为是在进行质控了。而在测序数据的分析过程中，也同样需要将质控贯穿到分析的各个阶段。质控基本上可以分为三个阶段，依据你的分析对象和分析目的不同而有所变化：raw sequence data, alignment, variant calling. 目前比较常见的质控多是局限在raw data的质控这个阶段。虽然如此，另外两个阶段的质控也应该是需要的，因为后续的许多分析也是基于这两个阶段。Raw data 质控的方向通常有以下几类：quality trimming, adapter removal, contaminant filtering. 而在alignment阶段，我们可以通过uniquely mapped reads, signed noise ratio 进行质控。在variant calling阶段，还在调查中……">
  
  
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">

  <meta property="og:title" content="生信基础 质控的那些事">
  <meta property="og:site_name" content="Xizhihui&#39;s Notes">

  
    <meta property="og:image" content="">
  

  
  
    <link href="/favicon.png" rel="icon">
  

  <!-- CSS -->
  <link rel="stylesheet" href="/css/themes/cosmo.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/font-awesome.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/style.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/responsive.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight-default.min.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/google-fonts.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/comment.css" media="screen" type="text/css">
  <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
	<script src="https://cdnjs.cloudflare.com/ajax/libs/es5-shim/4.5.9/es5-shim.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/es5-shim/4.5.7/es5-sham.min.js"></script>
  <![endif]-->

  <script src="/js/jquery-2.0.3.min.js"></script>
  
  <!-- analytics -->
  


<script>
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>


</head>
</html>
 <body>  
  <nav id="main-nav" class="navbar  navbar-default navbar-fixed-top" role="navigation">
    <div class="container">
      <button type="button" class="navbar-header navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
	<span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
       <a class="navbar-brand" href="/">Xizhihui&#39;s Notes</a>
      <div class="collapse navbar-collapse nav-menu">
		<ul class="nav navbar-nav">
		  
		  <li>
			<a href="/archives" title="All the articles.">
			  <i class="fa fa-archive"></i>Archives
			</a>
		  </li>
		  
		  <li>
			<a href="/categories" title="All the categories.">
			  <i class="fa fa-folder"></i>Categories
			</a>
		  </li>
		  
		  <li>
			<a href="/tags" title="All the tags.">
			  <i class="fa fa-tags"></i>Tags
			</a>
		  </li>
		  
		  <li>
			<a href="/about" title="About me.">
			  <i class="fa fa-user"></i>About
			</a>
		  </li>
		  
		</ul>
      </div>
    </div> <!-- container -->
</nav>
<div class="clearfix"></div>

  <div class="container">
  	<div class="content">
    	 


	
		<div class="page-header ">		
			<h1 class="title "> 生信基础 质控的那些事</h1>
		</div>		
	






<div class="row post">
	<!-- cols -->
	
	<div id="top_meta"></div>
	<div class="col-md-9">
	

	<!-- content -->
	<div class="mypage">		
	  		

	  <h2 id="质控的方向"><a href="#质控的方向" class="headerlink" title="质控的方向"></a>质控的方向</h2><p>就平常的生物实验而言，质控是要贯穿到实验的各个阶段的，比如那些阳性阴性对照就可以认为是在进行质控了。而在测序数据的分析过程中，也同样需要将质控贯穿到分析的各个阶段。质控基本上可以分为三个阶段，依据你的分析对象和分析目的不同而有所变化：raw sequence data, alignment, variant calling. 目前比较常见的质控多是局限在raw data的质控这个阶段。虽然如此，另外两个阶段的质控也应该是需要的，因为后续的许多分析也是基于这两个阶段。Raw data 质控的方向通常有以下几类：quality trimming, adapter removal, contaminant filtering. 而在alignment阶段，我们可以通过uniquely mapped reads, signed noise ratio 进行质控。在variant calling阶段，还在调查中……</p>
<a id="more"></a>
<h2 id="QC-of-Raw-data"><a href="#QC-of-Raw-data" class="headerlink" title="QC of Raw data"></a>QC of Raw data</h2><h3 id="FastQC质量统计"><a href="#FastQC质量统计" class="headerlink" title="FastQC质量统计"></a>FastQC质量统计</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">fastqc /path/*.fastq.gz -o /path/QC_result/</span><br></pre></td></tr></table></figure>
<h3 id="质控条件"><a href="#质控条件" class="headerlink" title="质控条件"></a>质控条件</h3><ul>
<li>去除非测序序列：adapter、primer、barcode、index</li>
<li>去除低质量reads：低于Q20、Q30的碱基占比达30%</li>
<li>去除含N过多reads: 占比达10%</li>
<li>去除重复序列：RNA-seq、16s-seq一般不做去除</li>
<li>去除3’端质量Q低于10的碱基，即碱基错误率为0.1</li>
<li>去除较短长度reads：低于25的去除</li>
<li>根据分析结果, 适当地重复上面某些步骤</li>
</ul>
<h3 id="使用Trimmomatic修剪reads"><a href="#使用Trimmomatic修剪reads" class="headerlink" title="使用Trimmomatic修剪reads"></a>使用Trimmomatic修剪reads</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">trimmomatic PE read1.fq.gz  read2.fq.gz \</span><br><span class="line">    clean_paired_forward.fq.gz clean_unpaired_forward.fq.gz \</span><br><span class="line">    clean_paired_reverse.fq.gz clean_unpaired_reverse.fq.gz \</span><br><span class="line">    ILLUMINACLIP:TruSeq3-PE.fa:2:30:10 \</span><br><span class="line">    LEADING:4 TRAILING:5 SLIDINGWINDOW:4:5 MINLEN:25 &amp;</span><br><span class="line"><span class="comment"># ILLUMINACLIP:adpater存在.fa文件中，允许2个错配；回文模式下匹配阈值;；简单模式下匹配阈值</span></span><br><span class="line"><span class="comment"># LEADING：切除首端质量值小于4的碱基</span></span><br><span class="line"><span class="comment"># TRAILING: 切除末端质量小于5的碱基</span></span><br></pre></td></tr></table></figure>
<h3 id="使用cutadapt-修剪-reads"><a href="#使用cutadapt-修剪-reads" class="headerlink" title="使用cutadapt 修剪 reads"></a>使用cutadapt 修剪 reads</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">cutadapt -a ADAPTER_FWD -A ADAPTER_REV -o out.1.fastq -p out.2.fastq reads.1.fastq reads.2.fastq</span><br></pre></td></tr></table></figure>
<h3 id="fastx-toolkit-个性化清洗和过滤"><a href="#fastx-toolkit-个性化清洗和过滤" class="headerlink" title="fastx_toolkit 个性化清洗和过滤"></a>fastx_toolkit 个性化清洗和过滤</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">fastq_quality_filter -q 30 -p 80 -z -i input.fastq -o out.fastq</span><br><span class="line"><span class="comment"># input.fastq不支持gz，可以如下,使用-c保留源文件</span></span><br><span class="line">gunzip -c input.fastq.gz | fastq_quality_filter ....etc....</span><br></pre></td></tr></table></figure>
<h3 id="multiqc"><a href="#multiqc" class="headerlink" title="multiqc"></a>multiqc</h3><h3 id="fastp"><a href="#fastp" class="headerlink" title="fastp"></a>fastp</h3><p>据说是一步到位</p>
<h2 id="另外两个阶段的质控"><a href="#另外两个阶段的质控" class="headerlink" title="另外两个阶段的质控"></a>另外两个阶段的质控</h2><p>在比对阶段和变异检测阶段的质控软件不多，已有的软件通常都包含了这两个阶段的质控，所以合并起来。</p>
<h3 id="使用RSeQC进行质控"><a href="#使用RSeQC进行质控" class="headerlink" title="使用RSeQC进行质控"></a>使用RSeQC进行质控</h3><p><a href="http://rseqc.sourceforge.net/#usage-information" target="_blank" rel="noopener">http://rseqc.sourceforge.net/#usage-information</a><br>该软件的input格式有SAM、BAM、Fasta、BED</p>
<table>
<thead>
<tr>
<th>1</th>
<th>2</th>
<th>3</th>
<th>4</th>
</tr>
</thead>
<tbody>
<tr>
<td>bam2fq.py</td>
<td>bam2wig.py</td>
<td>bam_stat.py</td>
<td>insertion_profile.py</td>
</tr>
<tr>
<td>clipping_profile.py</td>
<td>deletion_profile.py</td>
<td>divide_bam.py</td>
<td>infer_experiment.py</td>
</tr>
<tr>
<td>FPKM_count.py</td>
<td>eneBody_coverage.py</td>
<td>geneBody_coverage2.py</td>
<td>inner_distance.py   </td>
</tr>
<tr>
<td>junction_annotation.py</td>
<td>junction_saturation.py</td>
<td>mismatch_profile.py</td>
<td>overlay_bigwig.py</td>
</tr>
<tr>
<td>normalize_bigwig.py</td>
<td>read_distribution.py</td>
<td>read_duplication.py</td>
<td>read_GC.pyread_hexamer.py</td>
</tr>
<tr>
<td>read_NVC.py</td>
<td>read_quality.py</td>
<td>RNA_fragment_size.py</td>
<td>RPKM_count.py</td>
</tr>
<tr>
<td>RPKM_saturation.py</td>
<td>spilt_bam.py</td>
<td>split_paired_bam.py</td>
<td>none</td>
</tr>
</tbody>
</table>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">python scripts/bam2fq.py -i test_PairedEnd_StrandSpecific_hg19.sam  -o bam2fq_out1</span><br><span class="line"></span><br><span class="line">python clipping_profile.py -i Pairend_StrandSpecific_51mer_Human_hg19.bam -s <span class="string">"PE"</span> -o out</span><br></pre></td></tr></table></figure>
<h2 id="参考"><a href="#参考" class="headerlink" title="参考"></a>参考</h2><ul>
<li><a href="http://www.bio-info-trainee.com/1914.html" target="_blank" rel="noopener">用sickle软件来对双端测序数据过滤低质量reads</a>  </li>
<li><a href="https://doi.org/10.1016/j.ygeno.2014.03.006" target="_blank" rel="noopener">使用Q30进行质控</a></li>
<li><a href="http://www.biotrainee.com/thread-324-1-4.html" target="_blank" rel="noopener">论QC的重要性</a></li>
</ul>
	  
	</div>

	<!-- recommended posts -->
	

	<div>
  	<center>
	<div class="pagination">
<ul class="pagination">
	 
				
    	<li class="prev"><a href="/2018/10/12/生信基础-生信数据格式汇总/" class="alignleft prev"><i class="fa fa-arrow-circle-o-left"></i>Prev</a></li>
  		

        <li><a href="/archives"><i class="fa fa-archive"></i>Archive</a></li>

		
		   <li class="next"><a href="/2018/10/12/生信基础-链特异性/" class="alignright next">Next<i class="fa fa-arrow-circle-o-right"></i></a></li>         
        
	
</ul>
</div>

    </center>
	</div>

    <!-- share -->
    
        
    <div class="bdsharebuttonbox">
        <a href="#" class="bds_more" data-cmd="more"></a>
        <a href="#" class="bds_weixin" data-cmd="weixin" title="分享到微信"></a>
        <a href="#" class="bds_tsina" data-cmd="tsina" title="分享到新浪微博"></a>
        <a href="#" class="bds_fbook" data-cmd="fbook" title="分享到Facebook"></a>
        <a href="#" class="bds_twi" data-cmd="twi" title="分享到Twitter"></a>
        <a href="#" class="bds_linkedin" data-cmd="linkedin" title="分享到linkedin"></a>
        <a href="#" class="bds_evernotecn" data-cmd="evernotecn" title="分享到印象笔记"></a>
        <a href="#" class="bds_youdao" data-cmd="youdao" title="分享到有道云笔记"></a>
        <a href="#" class="bds_copy" data-cmd="copy" title="分享到复制网址"></a>
    </div>
    <script>
        window._bd_share_config={"common":{"bdSnsKey":{},"bdText":"","bdMini":"2","bdMiniList":false,"bdPic":"","bdStyle":"1","bdSize":"24"},"share":{}};
        with(document)0[(getElementsByTagName('head')[0]||body).appendChild(createElement('script')).src='http://bdimg.share.baidu.com/static/api/js/share.js?v=89860593.js?cdnversion='+~(-new Date()/36e5)];
    </script>


        

    
	
	<!-- comment -->
	
<section id="comment">
  <h2 class="title">Comments</h2>
  
</section>


	</div> <!-- col-md-9/col-md-12 -->
		
	
	<div id="side_meta">
		<div class="col-md-3" id="post_meta"> 

	<!-- date -->
	
	<div class="meta-widget">
	<i class="fa fa-clock-o"></i>
	2018-10-12 
	</div>
	

	<!-- categories -->
    
	<div class="meta-widget">
	<a data-toggle="collapse" data-target="#categorys"><i class="fa fa-folder"></i></a>	
    <ul id="categorys" class="tag_box list-unstyled collapse in">
          
  <li>
    </li><li><a href="/categories/Bioinformatics/">Bioinformatics<span>36</span></a></li>
  

    </ul>
	</div>
	

	<!-- tags -->
	
	<div class="meta-widget">
	<a data-toggle="collapse" data-target="#tags"><i class="fa fa-tags"></i></a>		  
    <ul id="tags" class="tag_box list-unstyled collapse in">	  
	    
  <li><a href="/tags/生信基础/">生信基础<span>13</span></a></li>
    </ul>
	</div>
	

	<!-- toc -->
	<div class="meta-widget">
	
	   <a data-toggle="collapse" data-target="#toc"><i class="fa fa-bars"></i></a>
	   <div id="toc" class="toc collapse in">
			<ol class="toc-article"><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#质控的方向"><span class="toc-article-text">质控的方向</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#QC-of-Raw-data"><span class="toc-article-text">QC of Raw data</span></a><ol class="toc-article-child"><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#FastQC质量统计"><span class="toc-article-text">FastQC质量统计</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#质控条件"><span class="toc-article-text">质控条件</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#使用Trimmomatic修剪reads"><span class="toc-article-text">使用Trimmomatic修剪reads</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#使用cutadapt-修剪-reads"><span class="toc-article-text">使用cutadapt 修剪 reads</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#fastx-toolkit-个性化清洗和过滤"><span class="toc-article-text">fastx_toolkit 个性化清洗和过滤</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#multiqc"><span class="toc-article-text">multiqc</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#fastp"><span class="toc-article-text">fastp</span></a></li></ol></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#另外两个阶段的质控"><span class="toc-article-text">另外两个阶段的质控</span></a><ol class="toc-article-child"><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#使用RSeQC进行质控"><span class="toc-article-text">使用RSeQC进行质控</span></a></li></ol></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#参考"><span class="toc-article-text">参考</span></a></li></ol>
		</div>
	
	</div>
	
    <hr>
	
</div><!-- col-md-3 -->

	</div>
		

</div><!-- row -->



	</div>
  </div>
  <div class="container-narrow">
  <footer> <p>
  &copy; 2018 Xizhihui
  
      with help from <a href="http://hexo.io/" target="_blank">Hexo</a> and <a href="http://getbootstrap.com/" target="_blank">Twitter Bootstrap</a>. Theme by <a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind</a>.    
</p> </footer>
</div> <!-- container-narrow -->
  


  
<a id="gotop" href="#">   
  <span>▲</span> 
</a>

<script src="/js/jquery.imagesloaded.min.js"></script>
<script src="/js/gallery.js"></script>
<script src="/js/bootstrap.min.js"></script>
<script src="/js/main.js"></script>
<script src="/js/search.js"></script> 


<link rel="stylesheet" href="/fancybox/jquery.fancybox.css" media="screen" type="text/css">
<script src="/fancybox/jquery.fancybox.pack.js"></script>
<script type="text/javascript">
(function($){
  $('.fancybox').fancybox();
})(jQuery);
</script>



   <script type="text/javascript">      
     var search_path = "search.xml";
	 if (search_path.length == 0) {
	 	search_path = "search.xml";
	 }
	 var path = "/" + search_path;
     searchFunc(path, 'local-search-input', 'local-search-result');
   </script>


<!-- syntax highlighting -->


</body>
   </html>
