<!DOCTYPE HTML>
<html>
<head><meta name="generator" content="Hexo 3.8.0">
  <meta charset="utf-8">
  
  <title>扩增子测序 mothur使用示例 | Xizhihui&#39;s Notes</title>
  <meta name="author" content="Xizhihui">
  
  <meta name="description" content="mothur, 搞那么长的命名后缀，让你想骂你 “mothur”">
  
  
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">

  <meta property="og:title" content="扩增子测序 mothur使用示例">
  <meta property="og:site_name" content="Xizhihui&#39;s Notes">

  
    <meta property="og:image" content="">
  

  
  
    <link href="/favicon.png" rel="icon">
  

  <!-- CSS -->
  <link rel="stylesheet" href="/css/themes/cosmo.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/font-awesome.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/style.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/responsive.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight-default.min.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/google-fonts.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/comment.css" media="screen" type="text/css">
  <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
	<script src="https://cdnjs.cloudflare.com/ajax/libs/es5-shim/4.5.9/es5-shim.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/es5-shim/4.5.7/es5-sham.min.js"></script>
  <![endif]-->

  <script src="/js/jquery-2.0.3.min.js"></script>
  
    <script src="/js/marked.js"></script>
    <script src="/js/comment.js"></script>
    <script src="/js/timeago.min.js"></script>
    <script src="/js/highlight.min.js"></script>
	<script src="/js/spin.min.js"></script>
  
  <!-- analytics -->
  



</head>
</html>
 <body>  
  <nav id="main-nav" class="navbar  navbar-default navbar-fixed-top" role="navigation">
    <div class="container">
      <button type="button" class="navbar-header navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
	<span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
       <a class="navbar-brand" href="/">Xizhihui&#39;s Notes</a>
      <div class="collapse navbar-collapse nav-menu">
		<ul class="nav navbar-nav">
		  
		  <li>
			<a href="/archives" title="All the articles.">
			  <i class="fa fa-archive"></i>Archives
			</a>
		  </li>
		  
		  <li>
			<a href="/categories" title="All the categories.">
			  <i class="fa fa-folder"></i>Categories
			</a>
		  </li>
		  
		  <li>
			<a href="/tags" title="All the tags.">
			  <i class="fa fa-tags"></i>Tags
			</a>
		  </li>
		  
		  <li>
			<a href="/about" title="About me.">
			  <i class="fa fa-user"></i>About
			</a>
		  </li>
		  
		</ul>
      </div>
    </div> <!-- container -->
</nav>
<div class="clearfix"></div>

  <div class="container">
  	<div class="content">
    	 


	
		<div class="page-header ">		
			<h1 class="title "> 扩增子测序 mothur使用示例</h1>
		</div>		
	






<div class="row post">
	<!-- cols -->
	
	<div id="top_meta"></div>
	<div class="col-md-9">
	

	<!-- content -->
	<div class="mypage">		
	  
		 <div class="alert alert-success description">
			<i class="fa fa-info-circle"></i> <p>mothur, 搞那么长的命名后缀，让你想骂你 “mothur”</p>
			
		 </div> <!-- alert -->
	  		

	  <h2 id="1-初始说明"><a href="#1-初始说明" class="headerlink" title="1 初始说明"></a>1 初始说明</h2><ul>
<li>测序数据类型</li>
</ul>
<p>Illumina Miseq paired-end reads</p>
<ul>
<li>实验设计</li>
</ul>
<p>断奶后365天(dpw 365)的小鼠排泄物，比较初始10天(dpw 10)和中间10天(dpw140-150)的排泄物的微生物组的稳定性(肠道微生物组的变化情况)。为了简化操作，只用到一只小鼠的十个时间点(前5后5)的数据。这里还有模拟了由21种细菌组成的菌群的全基因组测序数据。先用小鼠的排泄物测序数据学习分析微生物群落，然后用模拟的菌落判断分析的错误率和它在其他分析中的作用。</p>
<ul>
<li>关于软件</li>
</ul>
<p>mothur既提供交互模式(像python)，也提供命令行模式；后者可以进行批量操作。</p>
<a id="more"></a>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 解析模式一</span></span><br><span class="line">mothur <span class="comment"># 进入解析模式</span></span><br><span class="line">make.files(....)</span><br><span class="line">quit() <span class="comment"># 退出解析模式</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 命令行模式</span></span><br><span class="line">mothur <span class="string">'#make.config(file=stability.files, processors=8)'</span></span><br><span class="line"><span class="comment"># 批量操作,把多个命令写入cmd.batch文件里</span></span><br><span class="line">mothur cmd.batch</span><br></pre></td></tr></table></figure>
<h2 id="2-进入mothur交互模式"><a href="#2-进入mothur交互模式" class="headerlink" title="2 进入mothur交互模式"></a>2 进入mothur交互模式</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">mothur</span><br><span class="line">set.dir(tempdefault=MiSeq_SOP)</span><br><span class="line">set.dir(inputdir=MiSeq_SOP)</span><br><span class="line"><span class="built_in">help</span>(query_cmd)</span><br></pre></td></tr></table></figure>
<h2 id="3-stability-files"><a href="#3-stability-files" class="headerlink" title="3 stability.files"></a>3 stability.files</h2><p>这个文件记录了每个样品对应的read1和read2文件。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">make.file(inputdir=MiSeq_SOP, <span class="built_in">type</span>=fastq, prefix=metadata)</span><br></pre></td></tr></table></figure>
<h2 id="4-序列合并和质控：减少测序和PCR错误"><a href="#4-序列合并和质控：减少测序和PCR错误" class="headerlink" title="4 序列合并和质控：减少测序和PCR错误"></a>4 序列合并和质控：减少测序和PCR错误</h2><p>使用make.configs合并序列,合并逻辑是先对两个reads比对，找到overlap；根据overla的base quality score，如果是base-gap，那么base的质量值要大于25才取base否则认为该位置没有base；如果是base-base，base的质量值要大于6才取base；否则置为N。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 1. 合并序列</span></span><br><span class="line">make.contigs(file=metadata.files, processor=8)</span><br><span class="line">		<span class="comment"># MiSeq_SOP/metadata.trim.contigs.[fasta,qual]</span></span><br><span class="line">		<span class="comment"># MiSeq_SOP/metadata.scrap.contigs.[fasta, qual]</span></span><br><span class="line">		<span class="comment"># MiSeq_SOP/metadata.contigs.[report, groups]</span></span><br><span class="line"><span class="comment"># 2. 统计合并后的序列</span></span><br><span class="line">summary.seqs(fasta=metadata.trim.contigs.fasta)</span><br><span class="line">		<span class="comment"># MiSeq_SOP/metadata.trim.contigs.summary</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 3. 筛选合并后的序列, 排除含有N和长度大于275的序列</span></span><br><span class="line">screen.seqs(fasta=metadata.trim.contigs.fasta, </span><br><span class="line">        summary=metadata.trim.contigs.summary, </span><br><span class="line">        group=metadata.contigs.groups, </span><br><span class="line">        maxambig=0, maxlength=275)</span><br><span class="line">		<span class="comment"># MiSeq_SOP/metadata.contigs.[pick, good].groups</span></span><br><span class="line">		<span class="comment"># MiSeq_SOP/metadata.trim.contigs.[good.summary, good.fasta, bad.accnos]</span></span><br></pre></td></tr></table></figure>
<h2 id="5-序列的去冗余"><a href="#5-序列的去冗余" class="headerlink" title="5 序列的去冗余"></a>5 序列的去冗余</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 1.去测序冗余</span></span><br><span class="line">unique.seqs(fasta=metadata.trim.contigs.good.fasta)</span><br><span class="line">		<span class="comment"># MiSeq_SOP/metadata.trim.contigs.good.[names, unique.fasta]</span></span><br><span class="line"><span class="comment"># 2.生成count矩阵</span></span><br><span class="line">count.seqs(name=metadata.trim.contigs.good.names, </span><br><span class="line">        group=metadata.contigs.good.groups)</span><br><span class="line">		<span class="comment"># MiSeq_SOP/metadata.trim.contigs.good.count_table</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 3.对count矩阵进行统计</span></span><br><span class="line">summary.seqs(fasta=metadata.trim.contigs.good.unique.fasta, </span><br><span class="line">        count=metadata.trim.contigs.good.count_table)</span><br><span class="line">		<span class="comment"># MiSeq_SOP/metadata.trim.contigs.good.unique.summary</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 4.去非扩增区域冗余; 通过设置参考数据库中你的PCR扩增区域来去除非扩增区域的序列，以减少比对量</span></span><br><span class="line"><span class="comment">#   你可以使用oligos来指定primer文件进行去除，keepprimer</span></span><br><span class="line">pcr.seqs(fasta=../silva.bacteria/silva.bacteria.fasta, start=11894, end=25319, keepdots=F, processors=8)</span><br><span class="line">		<span class="comment"># silva.bacteria/silva.bacteria.pcr.fasta</span></span><br><span class="line">rename.file(input=../silva.bacteria/silva.bacteria.pcr.fasta, new=silva.v4.fasta)</span><br><span class="line">		<span class="comment"># MiSeq_SOP/silva.v4.fasta</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 5. 对silva.v4.fasta进行统计</span></span><br><span class="line">summar.seqs(fasta=silva.v4.fasta)</span><br><span class="line">		<span class="comment"># MiSeq_SOP/silva.v4.summary</span></span><br></pre></td></tr></table></figure>
<h2 id="6-序列的预聚类"><a href="#6-序列的预聚类" class="headerlink" title="6 序列的预聚类"></a>6 序列的预聚类</h2><p>要进行预聚类，我们需要先进行序列比对，获得相似序列。</p>
<h3 id="6-1-序列比对"><a href="#6-1-序列比对" class="headerlink" title="6.1 序列比对"></a>6.1 序列比对</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 1.比对并统计</span></span><br><span class="line">align.seqs(fasta=metadata.trim.contigs.good.unique.fasta, reference=silva.v4.fasta)</span><br><span class="line">		<span class="comment"># MiSeq_SOP/metadata.trim.contigs.good.unique.align.['', report]</span></span><br><span class="line">summary.seqs(fasta=metadata.trim.contigs.good.unique.align,</span><br><span class="line">        count=metadata.trim.contigs.good.count_table)</span><br><span class="line">		<span class="comment"># MiSeq_SOP/metadata.trim.contigs.good.unique.summary</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 2. 根据比对统计结果，大部分序列比对在1968-11550, polymer最大为8；再进行一次筛选序列</span></span><br><span class="line">screen.seqs(fasta=metadata.trim.contigs.good.unique.align, count=metadata.trim.contigs.good.count_table, summary=metadata.trim.contigs.good.unique.summary, start=1968, end=11550, maxhomop=8)</span><br><span class="line">		<span class="comment"># MiSeq_SOP/metadata.trim.contigs.good.unique.good.summary</span></span><br><span class="line">		<span class="comment"># MiSeq_SOP/metadata.trim.contigs.good.unique.good.align</span></span><br><span class="line">		<span class="comment"># MiSeq_SOP/metadata.trim.contigs.good.unique.good.accnos</span></span><br><span class="line">		<span class="comment"># MiSeq_SOP/metadata.trim.contigs.good.good.count_table</span></span><br><span class="line">summary.seqs(fasta=current, count=current)</span><br><span class="line">		<span class="comment"># MiSeq_SOP/metadata.trim.contigs.good.unique.good.summary</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 3. 把比对结果中未必对上部分序列和表示gap的-去掉</span></span><br><span class="line">filter.seqs(fasta=metadata.trim.contigs.good.unique.good.align, vertical=T, trump=.)</span><br><span class="line">		<span class="comment"># Length of filtered alignment: 376 </span></span><br><span class="line">		<span class="comment"># Number of columns removed: 13049</span></span><br><span class="line">		<span class="comment"># Length of the original alignment: 13425</span></span><br><span class="line">		<span class="comment"># Number of sequences used to construct filter: 16299</span></span><br><span class="line">		<span class="comment"># MiSeq_SOP/metadata.trim.contigs.good.unique.good.filter.fasta</span></span><br><span class="line">		<span class="comment"># MiSeq_SOP/metadata.filter</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 4. 去冗余, 在上一步剪切序列时可能产生</span></span><br><span class="line">unique.seqs(fasta=metadata.trim.contigs.good.unique.good.filter.fasta, count=metadata.trim.contigs.good.good.count_table)</span><br><span class="line">		<span class="comment"># MiSeq_SOP/metadata.trim.contigs.good.unique.good.filter.count_table</span></span><br><span class="line">		<span class="comment"># MiSeq_SOP/metadata.trim.contigs.good.unique.good.filter.unique.fasta</span></span><br></pre></td></tr></table></figure>
<h3 id="6-2-预聚类"><a href="#6-2-预聚类" class="headerlink" title="6.2 预聚类"></a>6.2 预聚类</h3><p>这里使用pre.cluster进行预聚类。它首先根据group对序列分组，然后依据序列丰度进行排序，最后把差异在2个核苷酸及以内的序列进行合并。一般是100bp取1个核苷酸的差异,这里的序列长度是250左右，所以为2。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">pre.cluster(fasta=metadata.trim.contigs.good.unique.good.filter.unique.fasta, count=metadata.trim.contigs.good.unique.good.filter.count_table, diffs=2)</span><br><span class="line">		<span class="comment"># MiSeq_SOP/metadata.trim.contigs.good.unique.good.filter.unique.precluster.[SampleID].map</span></span><br><span class="line">		<span class="comment"># MiSeq_SOP/metadata.trim.contigs.good.unique.good.filter.unique.precluster.[fasta,count_table]</span></span><br></pre></td></tr></table></figure>
<h2 id="7-去除嵌合体-chimeras"><a href="#7-去除嵌合体-chimeras" class="headerlink" title="7 去除嵌合体 chimeras"></a>7 去除嵌合体 chimeras</h2><p>chimera.vsearch将会把数据按样品分组，检测嵌合体的存在。我们倾向于使用高丰度序列作为reference。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 1. 检测嵌合体序列；输出的count_table依据去除了</span></span><br><span class="line">chimera.vsearch(fasta=metadata.trim.contigs.good.unique.good.filter.unique.precluster.fasta,</span><br><span class="line">        count=metadata.trim.contigs.good.unique.good.filter.unique.precluster.count_table, </span><br><span class="line">        dereplicate=T)</span><br><span class="line">		<span class="comment"># MiSeq_SOP/metadata.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.[pick.count_table, chimeras,accnos]</span></span><br><span class="line"><span class="comment"># 2. 去除嵌合体序列</span></span><br><span class="line">remove.seqs(fasta=metadata.trim.contigs.good.unique.good.filter.unique.precluster.fasta,</span><br><span class="line">        accnos=metadata.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.accnos)</span><br><span class="line">		<span class="comment"># MiSeq_SOP/metadata.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.fasta</span></span><br><span class="line"><span class="comment"># 3. 统计</span></span><br><span class="line">summary.seqs(fasta=current, count=current)</span><br><span class="line">	<span class="comment"># MiSeq_SOP/metadata.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.summary</span></span><br></pre></td></tr></table></figure>
<h2 id="8-进行聚类（分类）"><a href="#8-进行聚类（分类）" class="headerlink" title="8 进行聚类（分类）"></a>8 进行聚类（分类）</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 1. 分类</span></span><br><span class="line">classify.seqs(fasta=metadata.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.fasta,</span><br><span class="line">        count=metadata.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.count_table,</span><br><span class="line">        reference=../trainset9_032012.pds.fasta, </span><br><span class="line">        taxonomy=trainset9_032012.pds.tax, cutoff=80)</span><br><span class="line">	<span class="comment"># MiSeq_SOP/metadata.contigs.good.unique.good.filter.unique.precluster.pick.pds.wang.[taxonomy, tax.summary]</span></span><br><span class="line">	</span><br><span class="line"><span class="comment"># 2. 去除污染; 这里的unknown是在上一步分类时无法根据训练集和准确进行准确分类的一类</span></span><br><span class="line">remove.lineage(fasta=current, count=current, taxonomy=current, taxon=Chloroplast-Mitochondria-unknown-Archaea-Eukaryota)</span><br><span class="line">		<span class="comment"># MiSeq_SOP/metadata.trim.contigs.good.unique.good.filter.unique.precluster.pick.pds.wang.pick.taxonomy</span></span><br><span class="line">		<span class="comment"># MiSeq_SOP/metadata.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta</span></span><br><span class="line">		<span class="comment"># MiSeq_SOP/metadata.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.count_table</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 3. 统计</span></span><br><span class="line">summary.tax(taxonomy=current, count=current)</span><br><span class="line">		<span class="comment"># MiSeq_SOP/metadata.contigs.good.unique.good.filter.unique.precluster.pick.pds.wang.tax.summary</span></span><br></pre></td></tr></table></figure>
<h2 id="9-检测错误率"><a href="#9-检测错误率" class="headerlink" title="9 检测错误率"></a>9 检测错误率</h2><p>如果你在测序的时候有一些模拟群落的数据的话，你就可以检测一下你的数据的错误率。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 1. 提取出mock的分组</span></span><br><span class="line">get.groups(count=current, fasta=current, groups=Mock)</span><br><span class="line">		<span class="comment"># MiSeq_SOP/metadata.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count_table</span></span><br><span class="line">		<span class="comment"># MiSeq_SOP/metadata.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.fasta</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 2. 检测错误率</span></span><br><span class="line">seq.error(fasta=current, count=current, reference=HMP_MOCK.v35.fasta, aligned=F)</span><br><span class="line">		<span class="comment"># Overall error rate:	6.5108e-05</span></span><br><span class="line">	<span class="comment"># MiSeq_SOP/metadata.trim.contigs.good.unique.good.filter.unique.</span></span><br><span class="line">	<span class="comment"># precluster.pick.pick.pick.error.[summary,seq,chimera,seq.forward, seq.reverse, count, matrix, ref]</span></span><br></pre></td></tr></table></figure>
<h2 id="10-生成Mock-OTUs"><a href="#10-生成Mock-OTUs" class="headerlink" title="10 生成Mock OTUs"></a>10 生成Mock OTUs</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 1. 计算距离</span></span><br><span class="line">dist.seqs(fasta=metadata.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.fasta,</span><br><span class="line">    cutoff=0.03)</span><br><span class="line">	<span class="comment"># metadata.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.dist</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 2.进行聚类</span></span><br><span class="line">cluster(column=metadata.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.dist,</span><br><span class="line">    count=metadata.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count_table)</span><br><span class="line">	<span class="comment"># metadata.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.[list, steps, sensspec]</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 3. 生成shared文件</span></span><br><span class="line">make.shared(list=metadata.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.list,</span><br><span class="line">    count=metadata.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count_table, </span><br><span class="line">    label=0.03)</span><br><span class="line">	<span class="comment"># metadata.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.shared</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 4. 稀释</span></span><br><span class="line">rarefaction.single(shared=metadata.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.shared)</span><br><span class="line">		<span class="comment"># metadata.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.groups.rarefaction</span></span><br></pre></td></tr></table></figure>
<h2 id="11-准备分析"><a href="#11-准备分析" class="headerlink" title="11 准备分析"></a>11 准备分析</h2><p>接下来我们有2件事要做。第一件就是把序列与OTU对应起来，第二件是把序列和系统发育表型对应起来。<br>当前我们需要把mock数据从我们的数据集里面去掉。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">remove.groups(fasta=metadata.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta,</span><br><span class="line">    count=metadata.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.count_table,</span><br><span class="line">    taxonomy=metadata.contigs.good.unique.good.filter.unique.precluster.pick.pds.wang.pick.taxonomy, </span><br><span class="line">    groups=Mock)</span><br><span class="line">	<span class="comment"># metadata.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.fasta</span></span><br><span class="line">	<span class="comment"># metadata.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count_table</span></span><br><span class="line">	<span class="comment"># metadata.trim.contigs.good.unique.good.filter.unique.precluster.pick.pds.wang.pick.pick.taxonomy</span></span><br></pre></td></tr></table></figure>
<h2 id="12-生成OTUs"><a href="#12-生成OTUs" class="headerlink" title="12 生成OTUs"></a>12 生成OTUs</h2><p>对于较小的数据集，可以使用dist.seqs和cluster来进行，就像前面Mock OTUs一样。 对于大的数据集，使用cluster.split。在该方法中，会先根据分类信息对序列进行分箱，然后在各个箱内进行聚类。对于指定的分类水平，可以通过指定taxlevel=level来实现。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 1. 生成OTUs</span></span><br><span class="line"><span class="comment"># dist.seqs + cluster</span></span><br><span class="line">dist.seqs(fasta=curent, count=current)</span><br><span class="line">	<span class="comment"># metadata.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.dist</span></span><br><span class="line"></span><br><span class="line">cluster(column=metadata.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.dist, count=current)</span><br><span class="line">	<span class="comment"># metadata.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.[list,steps,sensspec]</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># cluster.split</span></span><br><span class="line">cluster.split(fasta=metadata.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.fasta,</span><br><span class="line">    count=metadata.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count_table,</span><br><span class="line">    taxonomy=metadata.trim.contigs.good.unique.good.filter.unique.precluster.pick.pds.wang.pick.pick.taxonomy, </span><br><span class="line">    splitmethod=classify, cutoff=0.03, taxlevel=4)</span><br><span class="line">	<span class="comment"># metadata.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.[list,steps,sensspec]</span></span><br></pre></td></tr></table></figure>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 2. 生成shared文件(每个样品为行，每个OTU为列，OTU数目矩阵)</span></span><br><span class="line">makd.shared(list=metadata.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.list,</span><br><span class="line">    count=metadata.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count_table, </span><br><span class="line">    label=0.03)</span><br><span class="line">	<span class="comment"># metadata.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.shared</span></span><br></pre></td></tr></table></figure>
<p>然后使用classify.otu给每个OTU同对应的物种分类联系起来。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 3. otu被观察到的次数（OTU丰度）和对应的物种分类信息</span></span><br><span class="line">classify.otu(list=metadata.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.list,</span><br><span class="line">    count=metadata.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count_table, </span><br><span class="line">    label=0.03)</span><br><span class="line">	<span class="comment"># metadata.trim.contigs.good.unique.good.filter.unique.precluster.</span></span><br><span class="line">	<span class="comment"># pick.pick.pick.opti_mcc.0.03.cons.[taxonomy, tax.summary]</span></span><br></pre></td></tr></table></figure>
<h2 id="13-生成phylotypes-系统发育表型信息"><a href="#13-生成phylotypes-系统发育表型信息" class="headerlink" title="13 生成phylotypes 系统发育表型信息"></a>13 生成phylotypes 系统发育表型信息</h2><p>你可以使用phylotype来对序列按照他们的物种分类分箱成系统发育表型。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 1. 生成系统发育表型表</span></span><br><span class="line">phylotype(taxnomy=metadata.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.taxonomy)	<span class="comment"># current</span></span><br><span class="line">		<span class="comment"># metadata.trim.contigs.good.unique.good.filter.unique.</span></span><br><span class="line">		<span class="comment"># precluster.pick.pds.wang.pick.pick.tx.[sabund, rabund, list]</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 2. 生成shared文件，这里生成level6，genus的;这里设置label是从1-7对应着genus-kingdom</span></span><br><span class="line">make.shared(list=metadata.trim.contigs.good.unique.good.filter.unique.precluster.pick.pds.wang.pick.pick.tx.list,</span><br><span class="line">    count=metadata.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count_table,</span><br><span class="line">    label=1)</span><br><span class="line">	<span class="comment"># metadata.trim.contigs.good.unique.good.filter.unique.precluster.pick.pds.wang.pick.pick.tx.shared</span></span><br><span class="line">	<span class="comment"># label   Group   numOtus Otu01   Otu02   Otu03   Otu04</span></span><br><span class="line">	<span class="comment"># 1       F3D0    62      1663    2626    216     57</span></span><br><span class="line">	<span class="comment"># 1       F3D1    62      1959    1249    246     35</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 3. 生成OTU丰度、对应的物种分类信息</span></span><br><span class="line">classify.otu(list=metadata.trim.contigs.good.unique.good.filter.unique.precluster.pick.pds.wang.pick.pick.tx.list,</span><br><span class="line">    count=metadata.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count_table, </span><br><span class="line">    label=1)</span><br><span class="line">	<span class="comment"># metadata.trim.contigs.good.unique.good.filter.unique.</span></span><br><span class="line">	<span class="comment"># precluster.pick.pds.wang.pick.pick.tx.1.cons.[taxonomy,tax.summary]</span></span><br><span class="line">	<span class="comment"># OTU     Size    Taxonomy</span></span><br><span class="line">	<span class="comment"># Otu01   22183   Bacteria(100);...;Lachnospiraceae_unclassified(100);</span></span><br><span class="line">	<span class="comment"># Otu02   54221   Bacteria(100);...;"Porphyromonadaceae"_unclassified(100);</span></span><br></pre></td></tr></table></figure>
<h2 id="14-生成系统发育树"><a href="#14-生成系统发育树" class="headerlink" title="14 生成系统发育树"></a>14 生成系统发育树</h2><p>我们可以使用clearcut来生成系统发育树，它既可以以比对后的序列作为输入，也可以以序列距离来作为输入。这里以序列距离作为输入。如果以比对后的序列作为输入，你需要指定序列是DNA还是蛋白。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 1. 计算距离; fasta和计算OTUs的是一致的</span></span><br><span class="line">dist.seqs(fasta=metadata.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.fasta, </span><br><span class="line">        output=lt, processors=8)</span><br><span class="line">		<span class="comment"># metadata.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.phylip.dist</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 2. 生成树</span></span><br><span class="line">clearcut(phylip=metadata.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.phylip.dist)	<span class="comment"># current</span></span><br><span class="line">		<span class="comment"># metadata.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.phylip.tre</span></span><br></pre></td></tr></table></figure>
<h2 id="15-准备分析-OTU-based"><a href="#15-准备分析-OTU-based" class="headerlink" title="15 准备分析: OTU-based"></a>15 准备分析: OTU-based</h2><p>下面我们要进行基于OTU的分析，基于phylotype的分析也是一样的操作. 为了简化，我们要把current的各个相关的默认参数指定为OTU的。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 这里这样设置后，...0.03.cons.taxonomy就成了metadata.taxonomy</span></span><br><span class="line">rename.files(taxonomy=metadata.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.0.03.cons.taxonomy,</span><br><span class="line">    shared=metadata.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.shared)</span><br></pre></td></tr></table></figure>
<p>如果要知道每个样品有多少条序列，可以这么做. 然后对样品进行稀释抽样</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 1. 对序列进行分组计数；得到样品的最小总序列数是2390</span></span><br><span class="line">count.groups(shared=metadata.opti_mcc.shared)</span><br><span class="line">		<span class="comment"># metadata.opti_mcc.count.summary</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 2. 进行稀释抽样，抽样大小是2390</span></span><br><span class="line">sub.sample(shared=metadata.opti_mcc.shared, size=2390)</span><br><span class="line">		<span class="comment"># metadata.opti_mcc.0.03.subsample.shared</span></span><br></pre></td></tr></table></figure>
<h2 id="16-基于OTU的分析-alpha-diversity"><a href="#16-基于OTU的分析-alpha-diversity" class="headerlink" title="16 基于OTU的分析: alpha diversity"></a>16 基于OTU的分析: alpha diversity</h2><ul>
<li>稀释曲线</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 生成的稀释表，需要你自己去画图</span></span><br><span class="line"><span class="comment"># calc指定计算哪个alpha diversity index, 有多个的话，sobs-chao-ace</span></span><br><span class="line"><span class="comment">#  ace, bootstrap, chao, coverage, default, heip, invsimpson, jack, npshannon, nseqs, shannon, shannoneven, shannonrange, simpson, simpsoneven, smithwilson, sobs</span></span><br><span class="line">rarefection.single(shared=metadata.opti_mcc.shared, calc=sobs, freq=100)</span><br><span class="line">		<span class="comment"># metadata.opti_mcc.groups.rarefaction</span></span><br></pre></td></tr></table></figure>
<ul>
<li>生成一个含序列数目、sample converage, observed OTUs, inverse simpson diversity estimation的表</li>
</ul>
<p>基于这里的ave-std.summary，你还可以进行repeated-measures ANOVA进行分组的组间差异比较</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">summary.single(shared=metadata.opti_mcc.shared, calc=nseqs-coverage-sobs-invsimpson, subsample=2390)</span><br><span class="line">		<span class="comment"># metadata.opti_mcc.groups.ave-std.summary</span></span><br><span class="line">		<span class="comment"># metadata.opti_mcc.groups.summary</span></span><br></pre></td></tr></table></figure>
<h2 id="17-基于OTU的分析-beta-diversity"><a href="#17-基于OTU的分析-beta-diversity" class="headerlink" title="17 基于OTU的分析: beta diversity"></a>17 基于OTU的分析: beta diversity</h2><p>现在我们要比较样品的关系和组成了。</p>
<ul>
<li>OTU 相对丰度热图</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 你可以直接使用metadata.opti_mcc.0.03.subsample.shared在R里面画热图</span></span><br><span class="line">heatmap.bin(shared=metadata.opti_mcc.0.03.subsample.shared, scale=log2, numotu=50)</span><br><span class="line">	<span class="comment"># metadata.opti_mcc.0.03.subsample.0.03.heatmap.bin.svg</span></span><br></pre></td></tr></table></figure>
<ul>
<li>计算样品的关系和组成相似性</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">dist.shared(shared=metadata.opti_mcc.shared, calc=thetayc-jclass, subsample=2390)</span><br><span class="line">		<span class="comment"># metadata.opti_mcc.thetayc.0.03.lt.[dist, ave.dist,std.dist]</span></span><br><span class="line">		<span class="comment"># metadata.opti_mcc.jclass.0.03.lt.[dist, ave.dist,std.dist]</span></span><br><span class="line"></span><br><span class="line">heatmap.sim(calc=jcalss-thetayc)</span><br><span class="line"><span class="comment">#heatmap.sim(phylip=metadata.opti_mcc.jclass.0.03.lt.ave.dist)</span></span><br><span class="line">		<span class="comment"># metadata.opti_mcc.thetayc.0.03.lt.ave.heatmap.sim.svg</span></span><br><span class="line">		<span class="comment"># metadata.opti_mcc.jclass.0.03.lt.ave.heatmap.sim.svg</span></span><br></pre></td></tr></table></figure>
<ul>
<li>venn图</li>
</ul>
<p>当组别在5以内时，利用veen才方便。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 利用生成的.sharedotus到R里面画图</span></span><br><span class="line">venn(shared=metadata.opti_mcc.0.03.subsample.shared, groups=F3D0-F3D1-F3D2-F3D3)</span><br><span class="line">		<span class="comment"># metadata.opti_mcc.0.03.subsample.0.03.sharedsobs.F3D0-F3D1-F3D2-F3D3.svg</span></span><br><span class="line">		<span class="comment"># metadata.opti_mcc.0.03.subsample.0.03.sharedsobs.F3D0-F3D1-F3D2-F3D3.sharedotus</span></span><br></pre></td></tr></table></figure>
<ul>
<li>系统发育树</li>
</ul>
<p>这里利用tree.shared来画系统发育树，以可视化样品之间的相似性。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 这里选择jclass</span></span><br><span class="line">tree.shared(phylip=metadata.opti_mcc.jclass.0.03.lt.ave.dist)</span><br><span class="line">		<span class="comment"># metadata.opti_mcc.jclass.0.03.lt.ave.tre</span></span><br></pre></td></tr></table></figure>
<ul>
<li>基于系统发育树的差异比较</li>
</ul>
<p>parsimony, unifrac.unweighted, unifrac.weighted.<br>我们需要构建一个design文件，以指示那个样品属于哪个组；然后进行计算</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">parsimony(tree=metadata.opti_mcc.jclass.0.03.lt.ave.tre, group=mouse.time.desin, groups=all)</span><br><span class="line">		<span class="comment"># Tree#   Groups  ParsScore       ParsSig</span></span><br><span class="line">		<span class="comment"># 1       Early-Late      1       &lt;0.001</span></span><br><span class="line">		<span class="comment"># metadata.opti_mcc.jclass.0.03.lt.ave.tre.parsimony</span></span><br><span class="line">		<span class="comment"># metadata.opti_mcc.jclass.0.03.lt.ave.tre.psummary</span></span><br></pre></td></tr></table></figure>
<ul>
<li>PCoA和nmds分析/可视化</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">pcoa(phylip=metadata.opti_mcc.jclass.0.03.lt.ave.dist)</span><br><span class="line">		<span class="comment"># metadata.opti_mcc.jclass.0.03.lt.ave.pcoa.[axes, loadings]</span></span><br><span class="line">		<span class="comment"># metadata.opti_mcc.jclass.0.03.lt.ave.pcoa.loadings</span></span><br><span class="line"></span><br><span class="line">nmds(phylip=metadata.opti_mcc.jclass.0.03.lt.ave.dist)</span><br><span class="line">		<span class="comment"># Number of dimensions:   2</span></span><br><span class="line">		<span class="comment"># Lowest stress : 0.142764</span></span><br><span class="line">		<span class="comment"># R-squared for configuration:    0.918996</span></span><br><span class="line">		<span class="comment"># metadata.opti_mcc.thetayc.0.03.lt.ave.nmds.[iters, stress, axes]</span></span><br></pre></td></tr></table></figure>
<ul>
<li>AMOVA分析/HOMOVA分析</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">amova(phylp=metadata.opti_mcc.thetayc.0.03.lt.ave.dist, design=mouse.time.design)</span><br><span class="line">		<span class="comment"># metadata.opti_mcc.thetayc.0.03.lt.ave.amova</span></span><br><span class="line"></span><br><span class="line">homova(phylip=metadata.opti_mcc.thetayc.0.03.lt.ave.dist, design=mouse.time.desin)</span><br><span class="line">		<span class="comment"># metadata.opti_mcc.thetayc.0.03.lt.ave.homova</span></span><br></pre></td></tr></table></figure>
<ul>
<li>核心OTU分析：导致差异的OTU</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 生成的corr.axes文件可用biplot图可视化</span></span><br><span class="line">corr.axes(axes=metadata.opti_mcc.jclass.0.03.lt.ave.pcoa.axes, </span><br><span class="line">        shared=metadata.opti_mcc.0.03.subsample.shared, </span><br><span class="line">        method=spearman, numaxes=3)</span><br><span class="line">		<span class="comment"># metadata.opti_mcc.0.03.subsample.spearman.corr.axes</span></span><br><span class="line"></span><br><span class="line">corr.axes(axes=metadata.opti_mcc.jclass.0.03.lt.ave.pcoa.axes, </span><br><span class="line">        metadata=mouse.dpw.metadata, </span><br><span class="line">        method=spearman, numaxes=3)</span><br><span class="line">		<span class="comment"># mouse.dpw.spearman.corr.axes</span></span><br></pre></td></tr></table></figure>
<ul>
<li>是否可以将数据分成不同的群落类型？</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 得到的表Laplace最小值对应k=2。所以可以分为2个类型</span></span><br><span class="line">get.communitytype(shared=metadata.opti_mcc.0.03.subsample.shared)</span><br><span class="line">		<span class="comment"># metadata.opti_mcc.0.03.subsample.0.03.dmm.mix.fit</span></span><br><span class="line">		<span class="comment"># metadata.opti_mcc.0.03.subsample.0.03.dmm.[1-5].mix.posterior</span></span><br><span class="line">		<span class="comment"># metadata.opti_mcc.0.03.subsample.0.03.dmm.[1-5].mix.relabund</span></span><br><span class="line">		<span class="comment"># metadata.opti_mcc.0.03.subsample.0.03.dmm.mix.design，指明各个样品所属类型</span></span><br><span class="line">		<span class="comment"># metadata.opti_mcc.0.03.subsample.0.03.dmm.mix.parameters</span></span><br><span class="line">		<span class="comment"># metadata.opti_mcc.0.03.subsample.0.03.dmm.mix.summary，指明主要是哪个OTU导致的分类型</span></span><br></pre></td></tr></table></figure>
<h2 id="18-群体水平的分析"><a href="#18-群体水平的分析" class="headerlink" title="18 群体水平的分析"></a>18 群体水平的分析</h2><ul>
<li>metastats</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">metastats(shared=metadata.opti_mcc.0.03.subsample.shared, </span><br><span class="line">        design=mouse.time.design)</span><br><span class="line">		<span class="comment"># metadata.opti_mcc.0.03.subsample.0.03.Late_Early.metastats</span></span><br></pre></td></tr></table></figure>
<ul>
<li>lefse</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">lefse(shared=metadata.opti_mcc.0.03.subsample.shared, design=mouse.time.design)</span><br></pre></td></tr></table></figure>
<h2 id="19-基于系统发育的分析-phylogeny-based-analysis"><a href="#19-基于系统发育的分析-phylogeny-based-analysis" class="headerlink" title="19 基于系统发育的分析 phylogeny-based analysis"></a>19 基于系统发育的分析 phylogeny-based analysis</h2><h3 id="19-1-alpha-diversity"><a href="#19-1-alpha-diversity" class="headerlink" title="19.1 alpha diversity"></a>19.1 alpha diversity</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">rename.file(tree=metadata.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.phylip.tre)</span><br><span class="line">rename.file(count=metadata.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count_table)</span><br><span class="line">phylo.diversity(tree=metadata.tre, count=metadata.count_table, rarefy=T)</span><br><span class="line">		<span class="comment"># metadata.1.phylodiv.summary</span></span><br><span class="line">		<span class="comment"># phylodiv.rarefaction</span></span><br></pre></td></tr></table></figure>
<h3 id="19-2-beta-diversity"><a href="#19-2-beta-diversity" class="headerlink" title="19.2 beta diversity"></a>19.2 beta diversity</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 生成.ave.dist是距离矩阵，可以像OTU-based analysis那样进行分析那些PCOA之类的。</span></span><br><span class="line">unifrac.unweighted(tree=metadata.tre, count=metadata.count_table, distance=lt, processors=2, random=F, subsample=2390)</span><br><span class="line">unifrac.weighted(tree=metadata.tre, count=metadata.count_table, distance=lt, processors=2, random=F, subsample=2390)</span><br><span class="line">		<span class="comment"># metadata.uwsummary</span></span><br><span class="line">		<span class="comment"># metadata.1.unweighted.ave.dist</span></span><br><span class="line">		<span class="comment"># metadata.1.unweighted.std.dist</span></span><br><span class="line">		<span class="comment"># metadata.tre1.unweighted.phylip.dist</span></span><br></pre></td></tr></table></figure>
	  
	</div>

	<!-- recommended posts -->
	

	<div>
  	<center>
	<div class="pagination">
<ul class="pagination">
	 
				
    	<li class="prev"><a href="/2018/10/15/扩增子测序-理论基础/" class="alignleft prev"><i class="fa fa-arrow-circle-o-left"></i>Prev</a></li>
  		

        <li><a href="/archives"><i class="fa fa-archive"></i>Archive</a></li>

		
		   <li class="next"><a href="/2018/10/14/基因组变异检测(Variance-Calling-with-GATK)/" class="alignright next">Next<i class="fa fa-arrow-circle-o-right"></i></a></li>         
        
	
</ul>
</div>

    </center>
	</div>

    <!-- share -->
    
        
    <div class="bdsharebuttonbox">
        <a href="#" class="bds_more" data-cmd="more"></a>
        <a href="#" class="bds_weixin" data-cmd="weixin" title="分享到微信"></a>
        <a href="#" class="bds_tsina" data-cmd="tsina" title="分享到新浪微博"></a>
        <a href="#" class="bds_fbook" data-cmd="fbook" title="分享到Facebook"></a>
        <a href="#" class="bds_twi" data-cmd="twi" title="分享到Twitter"></a>
        <a href="#" class="bds_linkedin" data-cmd="linkedin" title="分享到linkedin"></a>
        <a href="#" class="bds_evernotecn" data-cmd="evernotecn" title="分享到印象笔记"></a>
        <a href="#" class="bds_youdao" data-cmd="youdao" title="分享到有道云笔记"></a>
        <a href="#" class="bds_copy" data-cmd="copy" title="分享到复制网址"></a>
    </div>
    <script>
        window._bd_share_config={"common":{"bdSnsKey":{},"bdText":"","bdMini":"2","bdMiniList":false,"bdPic":"","bdStyle":"1","bdSize":"24"},"share":{}};
        with(document)0[(getElementsByTagName('head')[0]||body).appendChild(createElement('script')).src='http://bdimg.share.baidu.com/static/api/js/share.js?v=89860593.js?cdnversion='+~(-new Date()/36e5)];
    </script>


        

    
	
	<!-- comment -->
	
<section id="comment">
  <h2 class="title">Comments</h2>
    	 
	 <div id="comment-thread"></div>
	 <div id="loading-spin"></div>
	 <script type="text/javascript">
	   getComments({
           type: "github" ? "github" : "github",       
	       user: "xizhihui",
	       repo: "",
		   client_id: "",
           client_secret: "",
		   no_comment: "No comments yet. Press the button and go to comment now!",
		   go_to_comment: "Go to comment",
		   no_issue: "no_issue",
		   issue_title: "扩增子测序 mothur使用示例",
		   issue_id: "",
		   btn_class: "btn btn-primary",
		   comments_target: "#comment-thread",
		   loading_target: "#loading_spin"
		   });
	 </script>
  
</section>


	</div> <!-- col-md-9/col-md-12 -->
		
	
	<div id="side_meta">
		<div class="col-md-3" id="post_meta"> 

	<!-- date -->
	
	<div class="meta-widget">
	<i class="fa fa-clock-o"></i>
	2018-10-15 
	</div>
	

	<!-- categories -->
    
	<div class="meta-widget">
	<a data-toggle="collapse" data-target="#categorys"><i class="fa fa-folder"></i></a>	
    <ul id="categorys" class="tag_box list-unstyled collapse in">
          
  <li>
    </li><li><a href="/Categories/Bioinformatics/">Bioinformatics<span>36</span></a></li>
  

    </ul>
	</div>
	

	<!-- tags -->
	
	<div class="meta-widget">
	<a data-toggle="collapse" data-target="#tags"><i class="fa fa-tags"></i></a>		  
    <ul id="tags" class="tag_box list-unstyled collapse in">	  
	    
  <li><a href="/Tags/软件和包/">软件和包<span>22</span></a></li> <li><a href="/Tags/扩增子/">扩增子<span>5</span></a></li>
    </ul>
	</div>
	

	<!-- toc -->
	<div class="meta-widget">
	
	   <a data-toggle="collapse" data-target="#toc"><i class="fa fa-bars"></i></a>
	   <div id="toc" class="toc collapse in">
			<ol class="toc-article"><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#1-初始说明"><span class="toc-article-text">1 初始说明</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#2-进入mothur交互模式"><span class="toc-article-text">2 进入mothur交互模式</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#3-stability-files"><span class="toc-article-text">3 stability.files</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#4-序列合并和质控：减少测序和PCR错误"><span class="toc-article-text">4 序列合并和质控：减少测序和PCR错误</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#5-序列的去冗余"><span class="toc-article-text">5 序列的去冗余</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#6-序列的预聚类"><span class="toc-article-text">6 序列的预聚类</span></a><ol class="toc-article-child"><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#6-1-序列比对"><span class="toc-article-text">6.1 序列比对</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#6-2-预聚类"><span class="toc-article-text">6.2 预聚类</span></a></li></ol></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#7-去除嵌合体-chimeras"><span class="toc-article-text">7 去除嵌合体 chimeras</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#8-进行聚类（分类）"><span class="toc-article-text">8 进行聚类（分类）</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#9-检测错误率"><span class="toc-article-text">9 检测错误率</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#10-生成Mock-OTUs"><span class="toc-article-text">10 生成Mock OTUs</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#11-准备分析"><span class="toc-article-text">11 准备分析</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#12-生成OTUs"><span class="toc-article-text">12 生成OTUs</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#13-生成phylotypes-系统发育表型信息"><span class="toc-article-text">13 生成phylotypes 系统发育表型信息</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#14-生成系统发育树"><span class="toc-article-text">14 生成系统发育树</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#15-准备分析-OTU-based"><span class="toc-article-text">15 准备分析: OTU-based</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#16-基于OTU的分析-alpha-diversity"><span class="toc-article-text">16 基于OTU的分析: alpha diversity</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#17-基于OTU的分析-beta-diversity"><span class="toc-article-text">17 基于OTU的分析: beta diversity</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#18-群体水平的分析"><span class="toc-article-text">18 群体水平的分析</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#19-基于系统发育的分析-phylogeny-based-analysis"><span class="toc-article-text">19 基于系统发育的分析 phylogeny-based analysis</span></a><ol class="toc-article-child"><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#19-1-alpha-diversity"><span class="toc-article-text">19.1 alpha diversity</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#19-2-beta-diversity"><span class="toc-article-text">19.2 beta diversity</span></a></li></ol></li></ol>
		</div>
	
	</div>
	
    <hr>
	
</div><!-- col-md-3 -->

	</div>
		

</div><!-- row -->



	</div>
  </div>
  <div class="container-narrow">
  <footer> <p>
  &copy; 2018 Xizhihui
  
      with help from <a href="http://hexo.io/" target="_blank">Hexo</a> and <a href="http://getbootstrap.com/" target="_blank">Twitter Bootstrap</a>. Theme by <a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind</a>.    
</p> </footer>
</div> <!-- container-narrow -->
  


  
<a id="gotop" href="#">   
  <span>▲</span> 
</a>

<script src="/js/jquery.imagesloaded.min.js"></script>
<script src="/js/gallery.js"></script>
<script src="/js/bootstrap.min.js"></script>
<script src="/js/main.js"></script>
<script src="/js/search.js"></script> 


<link rel="stylesheet" href="/fancybox/jquery.fancybox.css" media="screen" type="text/css">
<script src="/fancybox/jquery.fancybox.pack.js"></script>
<script type="text/javascript">
(function($){
  $('.fancybox').fancybox();
})(jQuery);
</script>



   <script type="text/javascript">      
     var search_path = "search.xml";
	 if (search_path.length == 0) {
	 	search_path = "search.xml";
	 }
	 var path = "/" + search_path;
     searchFunc(path, 'local-search-input', 'local-search-result');
   </script>


<!-- syntax highlighting -->

  <script>
  marked.setOptions({
    highlight: function (code, lang) {
        return hljs.highlightAuto(code).value;
    }
  });
  function Highlighting(){
    var markdowns = document.getElementsByClassName('markdown');
    for(var i=0;i<markdowns.length;i++){
        if(markdowns[i].innerHTML) markdowns[i].innerHTML =marked(markdowns[i].innerHTML);
    }
  }
  window.addEventListener('DOMContentLoaded', Highlighting, false);
  window.addEventListener('load', Highlighting, false);
  </script>


</body>
   </html>
