<!DOCTYPE HTML>
<html>
<head><meta name="generator" content="Hexo 3.8.0">
  <meta charset="utf-8">
  
  <title>组装或定量 kallisto_sleuth | Xizhihui&#39;s Notes</title>
  <meta name="author" content="Xizhihui">
  
  <meta name="description" content="免比对的定量：kallistokallisto是一个align-free的测序结果定量工具。速度快得很.他可以在10min内完成index的构建，然后花3min完成30 million的human reads的定量。它不仅快速而且准确。
Pseudoalignment保存了用于定量的关键信息。An important feature of kallisto is that it outputs bootstraps along with the estimates of transcript abundances">
  
  
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">

  <meta property="og:title" content="组装或定量 kallisto_sleuth">
  <meta property="og:site_name" content="Xizhihui&#39;s Notes">

  
    <meta property="og:image" content="">
  

  
  
    <link href="/favicon.png" rel="icon">
  

  <!-- CSS -->
  <link rel="stylesheet" href="/css/themes/cosmo.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/font-awesome.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/style.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/responsive.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight-default.min.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/google-fonts.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/comment.css" media="screen" type="text/css">
  <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
	<script src="https://cdnjs.cloudflare.com/ajax/libs/es5-shim/4.5.9/es5-shim.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/es5-shim/4.5.7/es5-sham.min.js"></script>
  <![endif]-->

  <script src="/js/jquery-2.0.3.min.js"></script>
  
  <!-- analytics -->
  


<script>
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>


</head>
</html>
 <body>  
  <nav id="main-nav" class="navbar  navbar-default navbar-fixed-top" role="navigation">
    <div class="container">
      <button type="button" class="navbar-header navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
	<span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
       <a class="navbar-brand" href="/">Xizhihui&#39;s Notes</a>
      <div class="collapse navbar-collapse nav-menu">
		<ul class="nav navbar-nav">
		  
		  <li>
			<a href="/archives" title="All the articles.">
			  <i class="fa fa-archive"></i>Archives
			</a>
		  </li>
		  
		  <li>
			<a href="/categories" title="All the categories.">
			  <i class="fa fa-folder"></i>Categories
			</a>
		  </li>
		  
		  <li>
			<a href="/tags" title="All the tags.">
			  <i class="fa fa-tags"></i>Tags
			</a>
		  </li>
		  
		  <li>
			<a href="/about" title="About me.">
			  <i class="fa fa-user"></i>About
			</a>
		  </li>
		  
		</ul>
      </div>
    </div> <!-- container -->
</nav>
<div class="clearfix"></div>

  <div class="container">
  	<div class="content">
    	 


	
		<div class="page-header ">		
			<h1 class="title "> 组装或定量 kallisto_sleuth</h1>
		</div>		
	






<div class="row post">
	<!-- cols -->
	
	<div id="top_meta"></div>
	<div class="col-md-9">
	

	<!-- content -->
	<div class="mypage">		
	  		

	  <h2 id="免比对的定量：kallisto"><a href="#免比对的定量：kallisto" class="headerlink" title="免比对的定量：kallisto"></a>免比对的定量：kallisto</h2><p><a href="https://pachterlab.github.io/kallisto/manual.html" target="_blank" rel="noopener">kallisto</a>是一个align-free的测序结果定量工具。速度快得很.他可以在10min内完成index的构建，然后花3min完成30 million的human reads的定量。它不仅快速而且准确。</p>
<p>Pseudoalignment保存了用于定量的关键信息。An important feature of kallisto is that it outputs bootstraps along with the estimates of transcript abundances</p>
<a id="more"></a>
<h3 id="流程概览"><a href="#流程概览" class="headerlink" title="流程概览"></a>流程概览</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 建立一个索引</span></span><br><span class="line">kallisto index -i transcripts.idx transcripts.fasta.gz</span><br><span class="line"><span class="comment"># 进行定量</span></span><br><span class="line">kallisto quant -i transcripts.idx -o output -b 100 reads_1.fastq.gz reads_2.fastq.gz</span><br><span class="line"><span class="comment"># single-end</span></span><br><span class="line"><span class="comment"># kallisto quant -i transcripts.idx -o output -b 100 --single -l 180 -s 20 reads.fastq.gz</span></span><br></pre></td></tr></table></figure>
<h3 id="0-安装"><a href="#0-安装" class="headerlink" title="0. 安装"></a>0. 安装</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">conda install kallisto</span><br></pre></td></tr></table></figure>
<h3 id="1-kallisto-index-建立索引"><a href="#1-kallisto-index-建立索引" class="headerlink" title="1. kallisto index: 建立索引"></a>1. kallisto index: 建立索引</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">kallisto index [arguments] ref.fa[.gz]</span><br><span class="line"></span><br><span class="line"><span class="comment"># 必需参数</span></span><br><span class="line">-i, --index=STRING          存放索引的目录名字</span><br><span class="line"></span><br><span class="line"><span class="comment"># 可选参数</span></span><br><span class="line">-k, --kmer-size=INT         k-mer (odd) length (default: 31, max value: 31)</span><br><span class="line">    --make-unique           Replace repeated target names with unique names</span><br></pre></td></tr></table></figure>
<h3 id="2-kallisto-quant-进行定量"><a href="#2-kallisto-quant-进行定量" class="headerlink" title="2. kallisto quant: 进行定量"></a>2. kallisto quant: 进行定量</h3><h4 id="注意事项"><a href="#注意事项" class="headerlink" title="注意事项"></a>注意事项</h4><ul>
<li>每次的输入文件只能是一个样本的文件, 输入多个文件仅仅是因为这个样本的数据被分成了多个文件.</li>
<li>对于single-end,你必需通过-l参数指定fragment length</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br></pre></td><td class="code"><pre><span class="line">kallisto quant [arguments] reads.fastq</span><br><span class="line"></span><br><span class="line">示例: pair-end / single-end</span><br><span class="line">kallisto quant -i index -o output pairA_1.fastq pairA_2.fastq pairB_1.fastq pairB_2.fastq</span><br><span class="line">kallisto quant -i index -o output --single -l 200 -s 20 file1.fastq.gz file2.fastq.gz file3.fastq.gz</span><br><span class="line"></span><br><span class="line"><span class="comment"># 必需参数</span></span><br><span class="line">-i, --index=STRING            索引所在目录</span><br><span class="line">-o, --output-dir=STRING       输出目录。默认输出三个文件,abuncance.h5, abundance.tsv, run_info.json</span><br><span class="line"></span><br><span class="line">Optional arguments:</span><br><span class="line">    --bias                    Perform sequence based bias correction</span><br><span class="line">-b, --bootstrap-samples=INT   Number of bootstrap samples (default: 0)</span><br><span class="line">    --seed=INT                Seed <span class="keyword">for</span> the bootstrap sampling (default: 42)</span><br><span class="line">    --plaintext               指定输出纯文本文件,而非HDF5格式</span><br><span class="line">    --fusion                  搜索融合位点?(Search <span class="keyword">for</span> fusions <span class="keyword">for</span> Pizzly)</span><br><span class="line">    --single                  输入的文件是single-end reads</span><br><span class="line">    --single-overhang         对不再转录本上的<span class="built_in">read</span>也进行定量</span><br><span class="line">    --fr-stranded             Strand specific reads, first <span class="built_in">read</span> forward</span><br><span class="line">    --rf-stranded             Strand specific reads, first <span class="built_in">read</span> reverse</span><br><span class="line">-l, --fragment-length=DOUBLE  Estimated average fragment length</span><br><span class="line">-s, --sd=DOUBLE               Estimated standard deviation of fragment length</span><br><span class="line">                              (default: -l, -s values are estimated from paired</span><br><span class="line">                               end data, but are required when using --single)</span><br><span class="line">-t, --threads=INT             Number of threads to use (default: 1)</span><br><span class="line">    --pseudobam               Save pseudoalignments to transcriptome to BAM file</span><br><span class="line">    --genomebam               Project pseudoalignments to genome sorted BAM file</span><br><span class="line">-g, --gtf                     GTF file <span class="keyword">for</span> transcriptome information</span><br><span class="line">                              (required <span class="keyword">for</span> --genomebam)</span><br><span class="line">-c, --chromosomes             Tab separated file with chrosome names and lengths</span><br><span class="line">                              (optional <span class="keyword">for</span> --genomebam, but recommended)</span><br></pre></td></tr></table></figure>
<h3 id="3-kallisto-preudo-伪比对"><a href="#3-kallisto-preudo-伪比对" class="headerlink" title="3. kallisto preudo: 伪比对"></a>3. kallisto preudo: 伪比对</h3><p>这个功能是进行伪比对这一步，主要是用于单细胞RNA-seq的。它的命令在形式上与目的上与kallisto quant相似，但是它不会利用EM-算法来计算reads abundance.而且它还有个选项指定多个细胞到一个批文件里面.</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line">kallisto pseudo [arguments] FASTQ-files</span><br><span class="line"></span><br><span class="line"><span class="comment"># 必需参数</span></span><br><span class="line">-i, --index=STRING            Filename <span class="keyword">for</span> the kallisto index to be used <span class="keyword">for</span></span><br><span class="line">                              pseudoalignment</span><br><span class="line">-o, --output-dir=STRING       Directory to write output to</span><br><span class="line"></span><br><span class="line"><span class="comment"># 可选参数</span></span><br><span class="line">-u  --umi                     First file <span class="keyword">in</span> pair is a UMI file</span><br><span class="line">-b  --batch=FILE              Process files listed <span class="keyword">in</span> FILE</span><br><span class="line">    --single                  Quantify single-end reads</span><br><span class="line">-l, --fragment-length=DOUBLE  Estimated average fragment length</span><br><span class="line">-s, --sd=DOUBLE               Estimated standard deviation of fragment length</span><br><span class="line">                              (default: -l, -s values are estimated from paired</span><br><span class="line">                               end data, but are required when using --single)</span><br><span class="line">-t, --threads=INT             Number of threads to use (default: 1)</span><br></pre></td></tr></table></figure>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># batch file的形式, single-end只有一个fastq,2变成1</span></span><br><span class="line">&#123;	</span><br><span class="line">	<span class="comment">#id file1 file 2</span></span><br><span class="line">	cell1 cell1_1.fastq.gz cell1_1.fastq.gz</span><br><span class="line">	cell2 cell2_1.fastq.gz cell2_1.fastq.gz</span><br><span class="line">	cell3 cell3_1.fastq.gz cell3_1.fastq.gz</span><br><span class="line">&#125;</span><br><span class="line"><span class="comment"># 对于--umi指定后的形式</span></span><br><span class="line">&#123;</span><br><span class="line">	<span class="comment">#id umi-file file-1</span></span><br><span class="line">	cell1 cell_1.umi cell_1.fastq.gz</span><br><span class="line">	cell2 cell_2.umi cell_2.fastq.gz</span><br><span class="line">	cell3 cell_3.umi cell_3.fastq.gz</span><br><span class="line">&#125;</span><br></pre></td></tr></table></figure>
<h3 id="4-kallisto-h5dump-把HDF5文件转为纯文本格式"><a href="#4-kallisto-h5dump-把HDF5文件转为纯文本格式" class="headerlink" title="4. kallisto h5dump: 把HDF5文件转为纯文本格式"></a>4. kallisto h5dump: 把HDF5文件转为纯文本格式</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">kallisto h5dump -o output_path abundance.h5</span><br></pre></td></tr></table></figure>
<h2 id="利用sleuth进行差异分析"><a href="#利用sleuth进行差异分析" class="headerlink" title="利用sleuth进行差异分析"></a>利用sleuth进行差异分析</h2><p>sleuth可以与kallisto进行无缝对接，进行表达定量的后续差异分析；它在RStudio上工作，让你能交互式地进行数据探索。</p>
<h3 id="安装"><a href="#安装" class="headerlink" title="安装"></a>安装</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 通过bioconductor安装</span></span><br><span class="line"><span class="keyword">source</span>(<span class="string">'http://bioconductor.org/biocLite.R'</span>)</span><br><span class="line">biocLite(<span class="string">'devtools'</span>)</span><br><span class="line">biocLite(<span class="string">'pachterlab/sleuth'</span>)</span><br><span class="line"><span class="comment"># 通过conda安装</span></span><br><span class="line"><span class="comment"># conda install --channel bioconda r-sleuth</span></span><br></pre></td></tr></table></figure>
<h3 id="特性"><a href="#特性" class="headerlink" title="特性"></a>特性</h3><ul>
<li>既适用于transcript-level的分析，也适合gene-level的分析</li>
<li>与kallisto兼容，让你的分析流程更快</li>
<li>使用bootstraps来检测和校正实验的技术性变异</li>
<li>用于数据分析的可交互应用</li>
</ul>
<h3 id="get-started-with-RStudio"><a href="#get-started-with-RStudio" class="headerlink" title="get started with RStudio"></a><a href="https://pachterlab.github.io/sleuth_walkthroughs/trapnell/analysis.html" target="_blank" rel="noopener">get started with RStudio</a></h3><h4 id="1-构建实验设计表格"><a href="#1-构建实验设计表格" class="headerlink" title="1. 构建实验设计表格"></a>1. 构建实验设计表格</h4><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 不打印messages</span></span><br><span class="line">suppressMessages(&#123;</span><br><span class="line">	<span class="keyword">library</span>(<span class="string">'sleuth'</span>)</span><br><span class="line">&#125;)</span><br><span class="line">options(stringsAsFactors=<span class="literal">F</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># 找到kallisto处理结果</span></span><br><span class="line">sample_id &lt;- dir(file.path(<span class="string">'..'</span>, <span class="string">'results'</span>))</span><br><span class="line">kal_dirs &lt;- file.path(<span class="string">'..'</span>, <span class="string">'results'</span>, sample_id, <span class="string">'kallisto'</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># 找到对于的实验设计文件</span></span><br><span class="line">s2c &lt;- read.table(file.path(<span class="string">'..'</span>, <span class="string">'metadata'</span>, <span class="string">'hiseq_info.txt'</span>), header=<span class="literal">T</span>)</span><br><span class="line">s2c &lt;- dplyr::select(s2c, sample=run_accession, condition)</span><br><span class="line"><span class="comment"># 将path作为列添加上去，否则会报错</span></span><br><span class="line">s2c &lt;- dplyr::mutate(s2c, path=kal_dirs)</span><br><span class="line"></span><br><span class="line"><span class="comment">##      sample condition                          path</span></span><br><span class="line"><span class="comment">## 1 SRR493366  scramble ../results/SRR493366/kallisto</span></span><br><span class="line"><span class="comment">## 2 SRR493367  scramble ../results/SRR493367/kallisto</span></span><br><span class="line"><span class="comment">## 3 SRR493368  scramble ../results/SRR493368/kallisto</span></span><br><span class="line"><span class="comment">## 4 SRR493369   HOXA1KD ../results/SRR493369/kallisto</span></span><br><span class="line"><span class="comment">## 5 SRR493370   HOXA1KD ../results/SRR493370/kallisto</span></span><br><span class="line"><span class="comment">## 6 SRR493371   HOXA1KD ../results/SRR493371/kallisto</span></span><br></pre></td></tr></table></figure>
<h4 id="2-构建‘sleuth’对象"><a href="#2-构建‘sleuth’对象" class="headerlink" title="2. 构建‘sleuth’对象"></a>2. 构建‘sleuth’对象</h4><p>sleuth对象不仅包含了实验信息，也包含了用于差异分析的模型和结果。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 1. 添加kallisto处理好的数据</span></span><br><span class="line">so &lt;- sleuth_prep(s2c, extra_bootstrap_summary=<span class="literal">T</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># 2. 确定sleuth应答错误衡量(full)模型的参数, condition是前面构建的实验设计表格里面的</span></span><br><span class="line"><span class="comment"># 详细讲就是~condition作为formula进行线性回归，然后对样本数据进行平滑</span></span><br><span class="line">so &lt;- sleuth_fit(so, ~condition, <span class="string">'full'</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># 3. 确定sleuth归并模型的参数;前提是假定各个条件下的表达量是一致的</span></span><br><span class="line">so &lt;- sleuth_fit(so, ~<span class="number">1</span>, <span class="string">'reduced'</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># 4. 使用似然率确定表达差异</span></span><br><span class="line">so &lt;- sleuth_lrt(so, <span class="string">'reduced'</span>, <span class="string">'full'</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># 查看使用了哪些回归模型</span></span><br><span class="line">models(so)</span><br><span class="line"></span><br><span class="line"><span class="comment"># 5. 进行差异分析</span></span><br><span class="line">sleuth_table &lt;- sleuth_results(so, <span class="string">'reduced:full'</span>, <span class="string">'lrt'</span>, show_all=<span class="literal">F</span>)</span><br><span class="line">sleuth_significant &lt;- dplyr::filter(sleuth_table, qval&lt;=<span class="number">0.05</span>)</span><br><span class="line">head(sleuth_significant, <span class="number">20</span>)</span><br></pre></td></tr></table></figure>
<h4 id="3-基因名注释"><a href="#3-基因名注释" class="headerlink" title="3. 基因名注释"></a>3. 基因名注释</h4><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 注意,这里使用的ENSEMBL human transcriptome, 所以使用biomaRt</span></span><br><span class="line"><span class="keyword">source</span>(<span class="string">'http://bioconductor.org/biocLite.R'</span>)</span><br><span class="line">biocLite(<span class="string">'biomaRt'</span>)</span><br></pre></td></tr></table></figure>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br></pre></td><td class="code"><pre><span class="line">mart &lt;- biomaRt::useMart(biomart=<span class="string">'ENSEMBL_MART_ENSEMBL'</span>,</span><br><span class="line">						dataset=<span class="string">'hsapiens_gene_ensembl'</span>,</span><br><span class="line">						host=<span class="string">'ensembl.org'</span>)</span><br><span class="line">t2g &lt;- biomaRt::getBM(attributes=c(<span class="string">'ensembl_transcript_id'</span>,</span><br><span class="line">									<span class="string">'ensembl_gene_id'</span>,</span><br><span class="line">									<span class="string">'extenal_gene_name'</span>),</span><br><span class="line">					  mart=mart)</span><br><span class="line">t2g &lt;- dplyr::rename(t2g, </span><br><span class="line">					 target_id=ensembl_transcript_id,</span><br><span class="line">					 ens_gene=ensembl_gene_id,</span><br><span class="line">					 ext_gene=external_gene_name)</span><br><span class="line"></span><br><span class="line"><span class="comment"># 添加到sleuth对象上</span></span><br><span class="line">so &lt;- sleuth_prep(s2c, target_mapping=t2g)</span><br><span class="line"></span><br><span class="line"><span class="comment"># 再进行一次计算</span></span><br><span class="line">so &lt;- sleuth_fit(so, ~condition, <span class="string">'full'</span>)</span><br><span class="line">so &lt;- sleuth_fit(so, ~<span class="number">1</span>, <span class="string">'reduced'</span>)</span><br><span class="line">so &lt;- sleuth_rlt(so, <span class="string">'reduced'</span>, <span class="string">'full'</span>)</span><br></pre></td></tr></table></figure>
<h4 id="4-可视化结果"><a href="#4-可视化结果" class="headerlink" title="4. 可视化结果"></a>4. 可视化结果</h4><ul>
<li>对结果的查看和交互可以通过产生sleuth live site实现</li>
</ul>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">sleuth_live(so)</span><br></pre></td></tr></table></figure>
<ul>
<li>箱图</li>
</ul>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">plot_bootstrap(so, <span class="string">'ENST00000264734'</span>, units=<span class="string">'est_counts'</span>, color_by=<span class="string">'condition'</span>)</span><br></pre></td></tr></table></figure>
<ul>
<li>PCA plot<br>以实验条件的pca条件</li>
</ul>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">plot_pca(so, color_by=<span class="string">'condition'</span>, text_labels=<span class="literal">T</span>)</span><br></pre></td></tr></table></figure>
<ul>
<li>样品counts分布图</li>
</ul>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">plot_group_density(so, use_filtered=<span class="literal">T</span>, units=<span class="string">'est_counts'</span>, trans=<span class="string">'log'</span>,grouping=setdiff(colnames(so$sample_to_corvariates), <span class="string">'sample'</span>),offset=<span class="number">1</span>)</span><br></pre></td></tr></table></figure>
<ul>
<li>MA-plot</li>
</ul>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">plot_ma(so, test=<span class="string">'condition'</span>)</span><br></pre></td></tr></table></figure>
<ul>
<li>均值方差分析</li>
</ul>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">plot_mean_var(so)</span><br></pre></td></tr></table></figure>
<ul>
<li>热图</li>
</ul>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">library</span>(pheatmap)</span><br><span class="line">plot_transcript_heatmap(so, transcripts=sleuth_table$target_id[<span class="number">1</span>:<span class="number">20</span>])</span><br></pre></td></tr></table></figure>
<ul>
<li>火山图</li>
</ul>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">plot_volcano(so, test=<span class="string">'condition'</span>)</span><br></pre></td></tr></table></figure>
	  
	</div>

	<!-- recommended posts -->
	

	<div>
  	<center>
	<div class="pagination">
<ul class="pagination">
	 
				
    	<li class="prev"><a href="/2018/10/12/组装或定量-eXpress/" class="alignleft prev"><i class="fa fa-arrow-circle-o-left"></i>Prev</a></li>
  		

        <li><a href="/archives"><i class="fa fa-archive"></i>Archive</a></li>

		
		   <li class="next"><a href="/2018/10/12/组装或定量-stringTie/" class="alignright next">Next<i class="fa fa-arrow-circle-o-right"></i></a></li>         
        
	
</ul>
</div>

    </center>
	</div>

    <!-- share -->
    
        
    <div class="bdsharebuttonbox">
        <a href="#" class="bds_more" data-cmd="more"></a>
        <a href="#" class="bds_weixin" data-cmd="weixin" title="分享到微信"></a>
        <a href="#" class="bds_tsina" data-cmd="tsina" title="分享到新浪微博"></a>
        <a href="#" class="bds_fbook" data-cmd="fbook" title="分享到Facebook"></a>
        <a href="#" class="bds_twi" data-cmd="twi" title="分享到Twitter"></a>
        <a href="#" class="bds_linkedin" data-cmd="linkedin" title="分享到linkedin"></a>
        <a href="#" class="bds_evernotecn" data-cmd="evernotecn" title="分享到印象笔记"></a>
        <a href="#" class="bds_youdao" data-cmd="youdao" title="分享到有道云笔记"></a>
        <a href="#" class="bds_copy" data-cmd="copy" title="分享到复制网址"></a>
    </div>
    <script>
        window._bd_share_config={"common":{"bdSnsKey":{},"bdText":"","bdMini":"2","bdMiniList":false,"bdPic":"","bdStyle":"1","bdSize":"24"},"share":{}};
        with(document)0[(getElementsByTagName('head')[0]||body).appendChild(createElement('script')).src='http://bdimg.share.baidu.com/static/api/js/share.js?v=89860593.js?cdnversion='+~(-new Date()/36e5)];
    </script>


        

    
	
	<!-- comment -->
	
<section id="comment">
  <h2 class="title">Comments</h2>
  
</section>


	</div> <!-- col-md-9/col-md-12 -->
		
	
	<div id="side_meta">
		<div class="col-md-3" id="post_meta"> 

	<!-- date -->
	
	<div class="meta-widget">
	<i class="fa fa-clock-o"></i>
	2018-10-12 
	</div>
	

	<!-- categories -->
    
	<div class="meta-widget">
	<a data-toggle="collapse" data-target="#categorys"><i class="fa fa-folder"></i></a>	
    <ul id="categorys" class="tag_box list-unstyled collapse in">
          
  <li>
    </li><li><a href="/categories/Bioinformatics/">Bioinformatics<span>36</span></a></li>
  

    </ul>
	</div>
	

	<!-- tags -->
	
	<div class="meta-widget">
	<a data-toggle="collapse" data-target="#tags"><i class="fa fa-tags"></i></a>		  
    <ul id="tags" class="tag_box list-unstyled collapse in">	  
	    
  <li><a href="/tags/软件和包/">软件和包<span>22</span></a></li> <li><a href="/tags/定量/">定量<span>3</span></a></li>
    </ul>
	</div>
	

	<!-- toc -->
	<div class="meta-widget">
	
	   <a data-toggle="collapse" data-target="#toc"><i class="fa fa-bars"></i></a>
	   <div id="toc" class="toc collapse in">
			<ol class="toc-article"><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#免比对的定量：kallisto"><span class="toc-article-text">免比对的定量：kallisto</span></a><ol class="toc-article-child"><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#流程概览"><span class="toc-article-text">流程概览</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#0-安装"><span class="toc-article-text">0. 安装</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#1-kallisto-index-建立索引"><span class="toc-article-text">1. kallisto index: 建立索引</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#2-kallisto-quant-进行定量"><span class="toc-article-text">2. kallisto quant: 进行定量</span></a><ol class="toc-article-child"><li class="toc-article-item toc-article-level-4"><a class="toc-article-link" href="#注意事项"><span class="toc-article-text">注意事项</span></a></li></ol></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#3-kallisto-preudo-伪比对"><span class="toc-article-text">3. kallisto preudo: 伪比对</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#4-kallisto-h5dump-把HDF5文件转为纯文本格式"><span class="toc-article-text">4. kallisto h5dump: 把HDF5文件转为纯文本格式</span></a></li></ol></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#利用sleuth进行差异分析"><span class="toc-article-text">利用sleuth进行差异分析</span></a><ol class="toc-article-child"><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#安装"><span class="toc-article-text">安装</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#特性"><span class="toc-article-text">特性</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#get-started-with-RStudio"><span class="toc-article-text">get started with RStudio</span></a><ol class="toc-article-child"><li class="toc-article-item toc-article-level-4"><a class="toc-article-link" href="#1-构建实验设计表格"><span class="toc-article-text">1. 构建实验设计表格</span></a></li><li class="toc-article-item toc-article-level-4"><a class="toc-article-link" href="#2-构建‘sleuth’对象"><span class="toc-article-text">2. 构建‘sleuth’对象</span></a></li><li class="toc-article-item toc-article-level-4"><a class="toc-article-link" href="#3-基因名注释"><span class="toc-article-text">3. 基因名注释</span></a></li><li class="toc-article-item toc-article-level-4"><a class="toc-article-link" href="#4-可视化结果"><span class="toc-article-text">4. 可视化结果</span></a></li></ol></li></ol></li></ol>
		</div>
	
	</div>
	
    <hr>
	
</div><!-- col-md-3 -->

	</div>
		

</div><!-- row -->



	</div>
  </div>
  <div class="container-narrow">
  <footer> <p>
  &copy; 2018 Xizhihui
  
      with help from <a href="http://hexo.io/" target="_blank">Hexo</a> and <a href="http://getbootstrap.com/" target="_blank">Twitter Bootstrap</a>. Theme by <a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind</a>.    
</p> </footer>
</div> <!-- container-narrow -->
  


  
<a id="gotop" href="#">   
  <span>▲</span> 
</a>

<script src="/js/jquery.imagesloaded.min.js"></script>
<script src="/js/gallery.js"></script>
<script src="/js/bootstrap.min.js"></script>
<script src="/js/main.js"></script>
<script src="/js/search.js"></script> 


<link rel="stylesheet" href="/fancybox/jquery.fancybox.css" media="screen" type="text/css">
<script src="/fancybox/jquery.fancybox.pack.js"></script>
<script type="text/javascript">
(function($){
  $('.fancybox').fancybox();
})(jQuery);
</script>



   <script type="text/javascript">      
     var search_path = "search.xml";
	 if (search_path.length == 0) {
	 	search_path = "search.xml";
	 }
	 var path = "/" + search_path;
     searchFunc(path, 'local-search-input', 'local-search-result');
   </script>


<!-- syntax highlighting -->


</body>
   </html>
