<!DOCTYPE html>
<html>
<head>
	<title>XZH's Notes</title>
	<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
	<meta name="author" content="xizhihui" />
	<meta name="viewport" content="width=device-width, initial-scale=1" />
	<link rel="stylesheet" type="text/css" href="/statics/reset.css">
	<link rel="stylesheet" type="text/css" href="/statics/header_footer.css"></link>
	<link rel="stylesheet" type="text/css" href="/statics/default.css">
	<link rel="stylesheet" type="text/css" href="/statics/main.css">
</head>
<body>
	<header class="Header">
		<nav>
			<h3>XZH</h3>
			<ul>
				<li><a href="/index.html#Timeline">Timeline</a></li><li><a href="/index.html#Category">Category</a></li><li><a href="/index.html#Tags">Tags</a></li><li><a href="/index.html#About">About</a></li>
			</ul>
		</nav>
	</header>
	<div class="Motto">
		<h2>XZH's Notes</h2>
		<h4>The future comes for you ~ ~ ~</h4>
	</div>

	<main>
		<nav id="table"></nav>
		<content id="content" class="content">
			<h2 id="ballgown">Ballgown</h2>
<p>Ballgown是一款灵活的用于RNA-seq数据差异分析的软件，除了差异分析，他还可以进行转录本的组织、可视化和分析表达程度。</p>
<h3 id="preprocessing">preprocessing</h3>
<p>在正式使用之前,你应该做完以下几步：</p>
<ul>
<li>RNA-seq的reads应该比对到参考基因组上了</li>
<li>转录本也组装完成，或者你有一个参考转录本</li>
<li>用于分析的表达矩阵应该是ballgown的可读格式</li>
</ul>
<p>推荐的两个pipeline: </p>
<ul>
<li>
<p>pipeline1: TopHat2 -&gt; stringtie -&gt; ballgown</p>
</li>
<li>
<p>pipeline2: TopHat2 -&gt; Cufflinks -&gt; Tablemaker -&gt; ballgown</p>
</li>
</ul>
<p>Tablemaker:     https://github.com/leekgroup/tablemaker</p>
<div class="codehilite"><pre><span></span><span class="c1"># pipeline1</span>
tophat2 -G ref.gff -o output_path -p <span class="m">6</span> ref_index reads.fastq
stringtie -B -G ref.gff -p <span class="m">6</span> accepted_hist.bam -o stringtie.gff

<span class="c1"># pipeline2</span>
tophat2 -G ref.gff -o output_path -p <span class="m">6</span> ref_index reads.fastq
cufflinks -g ref.gff -o output_path accepted_hits.bam
<span class="c1"># tablemaker调用cufflinks</span>
</pre></div>


<h3 id="_1">安装</h3>
<div class="codehilite"><pre><span></span><span class="kn">source</span><span class="p">(</span><span class="s">&#39;http://bioconductor.org/biocLite.R&#39;</span><span class="p">)</span>
biocLite<span class="p">(</span><span class="s">&#39;ballgown&#39;</span><span class="p">)</span>
</pre></div>


<h3 id="ballgown_1">Ballgown需求的格式</h3>
<p>https://github.com/alyssafrazee/ballgown#ballgown-readable-expression-output</p>
<div class="codehilite"><pre><span></span><span class="c1"># 文件目录</span>
extdata/
    sample01/
        e2t.ctab
        e_data.ctab
        i2t.ctab
        i_data.ctab
        t_data.ctab

<span class="c1"># e_data.ctab: exon-level expression measurements, one row per exon</span>
<span class="c1"># e_id  chr strand  start   end rcount  ucount  mrcount cov cov_sd  mcov    mcov_sd</span>

<span class="c1"># i_data.ctab: intro-(ie. junction-)level expression measurements, one row per intron</span>
<span class="c1"># e_id  chr strand  start   end rcount  ucount  mrcount</span>

<span class="c1"># t_data.ctab: transcript-level expression measurements, one row per transcript</span>
<span class="c1"># t_id  chr strand  start   end t_name  num_exons   length  gene_id gene_name   cov     FPKM</span>

<span class="c1"># e2t.ctab: which exons belong to which transcripts</span>
<span class="c1"># e_id t_id</span>

<span class="c1"># i2t.ctab: which introns belong to which transcripts</span>
<span class="c1"># i_data    t_data</span>
</pre></div>


<h3 id="_2">使用</h3>
<h4 id="1">1. 加载数据</h4>
<div class="codehilite"><pre><span></span><span class="kn">library</span><span class="p">(</span>ballgown<span class="p">)</span>

<span class="c1"># 1.寻找数据目录</span>
data_directory <span class="o">=</span> <span class="kp">system.file</span><span class="p">(</span><span class="s">&#39;extdata&#39;</span><span class="p">,</span> package<span class="o">=</span><span class="s">&#39;ballgown&#39;</span><span class="p">)</span>     <span class="c1"># 找到Ballgown所在的目录下extdata目录</span>
data_directory

<span class="c1"># 1.1 番外：自己构造目录文件</span>
samples_path <span class="o">=</span> <span class="kt">data.frame</span><span class="p">(</span><span class="kt">c</span><span class="p">(</span><span class="s">&#39;sample01_path&#39;</span><span class="p">,</span> <span class="s">&#39;sample02_path&#39;</span><span class="p">,</span> <span class="kc">...</span><span class="p">))</span>

<span class="c1"># 2. 构造Ballgown对象</span>
bg <span class="o">=</span> ballgown<span class="p">(</span>dataDir<span class="o">=</span>data_directory<span class="p">,</span> samplePattern<span class="o">=</span><span class="s">&#39;sample&#39;</span><span class="p">,</span> meas<span class="o">=</span><span class="s">&#39;all&#39;</span><span class="p">)</span>
<span class="c1"># meas的参数是那些.ctab文件不重复列的列名</span>
<span class="c1"># bg = ballgown(samples=samples_path, samplePattern=&#39;sample&#39;, meas=&#39;all&#39;)</span>
bg

<span class="c1"># 如果数据量太大</span>
<span class="c1"># in load.R</span>
<span class="kn">library</span><span class="p">(</span>ballgown<span class="p">)</span>
data_directory <span class="o">=</span> <span class="kp">system.file</span><span class="p">(</span><span class="s">&#39;extdata&#39;</span><span class="p">,</span> package<span class="o">=</span><span class="s">&#39;ballgown&#39;</span><span class="p">)</span>
bg <span class="o">=</span> ballgown<span class="p">(</span>dataDir<span class="o">=</span>data_directory<span class="p">,</span> samplePattern<span class="o">=</span><span class="s">&#39;sample&#39;</span><span class="p">,</span> meas<span class="o">=</span><span class="s">&#39;all&#39;</span><span class="p">)</span>
<span class="kp">save</span><span class="p">(</span>bg<span class="p">,</span> file<span class="o">=</span><span class="s">&#39;bg.rda&#39;</span><span class="p">)</span>
<span class="c1"># 然后在每次使用时,运行以下命令,然后通过load()加载bg.rda</span>
R CMD BATCH load.R
</pre></div>


<h4 id="2">2. 处理组装数据</h4>
<p>一个Ballgown对象有6个分支, structure, expr, indexes, dirs, mergeDate, meas.</p>
<div class="codehilite"><pre><span></span><span class="c1"># structure十分依赖GenomicRanges包。它指定了组装的转录本的基因位置，外显子内含子转录本的关系。</span>
<span class="kp">structure</span><span class="p">(</span>bg<span class="p">)</span><span class="o">$</span>exon
<span class="kp">structure</span><span class="p">(</span>bg<span class="p">)</span><span class="o">$</span>intron
<span class="kp">structure</span><span class="p">(</span>bg<span class="p">)</span><span class="o">$</span>trans

<span class="c1"># expr包含了表达数据</span>
<span class="c1"># *expr(bg, meas_name)</span>
transcript_fpkm <span class="o">=</span> texpr<span class="p">(</span>bg<span class="p">,</span> <span class="s">&#39;FPKM&#39;</span><span class="p">)</span>
transcript_cov <span class="o">=</span> texpr<span class="p">(</span>bg<span class="p">,</span> <span class="s">&#39;cov&#39;</span><span class="p">)</span>
whole_tx_table <span class="o">=</span> texpr<span class="p">(</span>bg<span class="p">,</span> <span class="s">&#39;all&#39;</span><span class="p">)</span>
exon_mcov <span class="o">=</span> eexpr<span class="p">(</span>bg<span class="p">,</span> <span class="s">&#39;mcov&#39;</span><span class="p">)</span>
junction_rcount <span class="o">=</span> iexpr<span class="p">(</span>bg<span class="p">)</span>
whole_intron_table <span class="o">=</span> iexpr<span class="p">(</span>bg<span class="p">,</span> <span class="s">&#39;all&#39;</span><span class="p">)</span>
gene_expression <span class="o">=</span> gexpr<span class="p">(</span>bg<span class="p">)</span>

<span class="c1"># indexes</span>
indexes<span class="p">(</span>bg<span class="p">)</span>
pData<span class="p">(</span>bg<span class="p">)</span> <span class="o">=</span> <span class="kt">data.frame</span><span class="p">(</span>id<span class="o">=</span>sampleNames<span class="p">(</span>bg<span class="p">),</span> group<span class="o">=</span><span class="kp">rep</span><span class="p">(</span><span class="kt">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="m">0</span><span class="p">),</span> each<span class="o">=</span><span class="m">10</span><span class="p">))</span>
exon_transcript_table <span class="o">=</span> indexes<span class="p">(</span>bg<span class="p">)</span><span class="o">$</span>e2t
transcript_gene_table <span class="o">=</span> indexes<span class="p">(</span>bg<span class="p">)</span><span class="o">$</span>t2g
phenotype_table <span class="o">=</span> pData<span class="p">(</span>bg<span class="p">)</span>
<span class="kp">head</span><span class="p">(</span>transcript_gene_table<span class="p">)</span>

<span class="c1"># 其他</span>
<span class="kp">head</span><span class="p">(</span>bg<span class="o">@</span>dirs<span class="p">)</span>
bg<span class="o">@</span>mergeDate
bg<span class="o">@</span>meas
</pre></div>


<h4 id="3">3.数据特征可视化</h4>
<ul>
<li>可视化组装的转录本结构</li>
</ul>
<div class="codehilite"><pre><span></span>plotTranscripts<span class="p">(</span>gene<span class="o">=</span><span class="s">&#39;XLOC_000454&#39;</span><span class="p">,</span>
                gown<span class="o">=</span>bg<span class="p">,</span> samples<span class="o">=</span><span class="s">&#39;sample12&#39;</span><span class="p">,</span>
                meas<span class="o">=</span><span class="s">&#39;FPKM&#39;</span><span class="p">,</span>
                colorby<span class="o">=</span><span class="s">&#39;transcript&#39;</span><span class="p">,</span>
                main<span class="o">=</span><span class="s">&#39;transcripts from gene XLOC_000454: sample12, FPKM&#39;</span><span class="p">)</span>
<span class="c1"># 多个样本同时画图</span>
plotTranscripts<span class="p">(</span><span class="s">&#39;XLOC_00054&#39;</span><span class="p">,</span> bg<span class="p">,</span>
                samples<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="s">&#39;sample01&#39;</span><span class="p">,</span> <span class="s">&#39;sample02&#39;</span><span class="p">),</span>
                meas<span class="o">=</span><span class="s">&#39;FPKM&#39;</span><span class="p">,</span>
                colorby<span class="o">=</span><span class="s">&#39;transcript&#39;</span><span class="p">)</span>
</pre></div>


<ul>
<li>进行分组比较表达量</li>
</ul>
<div class="codehilite"><pre><span></span>plotMeans<span class="p">(</span><span class="s">&#39;XLOC_00054&#39;</span><span class="p">,</span> bg<span class="p">,</span>
           groupvar<span class="o">=</span><span class="s">&#39;group&#39;</span><span class="p">,</span>
           meas<span class="o">=</span><span class="s">&#39;FPKM&#39;</span><span class="p">,</span>
           colorby<span class="o">=</span><span class="s">&#39;transcript&#39;</span><span class="p">)</span>
</pre></div>


<h4 id="4">4. 差异分析</h4>
<p>ballgown默认使用parametric F-test comparing nested linear model进行统计.它同时使用两个模型,一个模型是包含感兴趣协变量,如case/control status, 如time;另一个不包含协变量.显著性的p-value表明包含协变量的模型得到的结果要比不包含的模型结果更为显著，也就说明具有差异。而q-value&lt;0.05表明FDR被控制在5%。stattest函数可自动处理两组比较和多组比较和时间系列比较。</p>
<ul>
<li>组组比较，group分组</li>
</ul>
<div class="codehilite"><pre><span></span>stat_results <span class="o">=</span> stattest<span class="p">(</span>bg<span class="p">,</span> feature<span class="o">=</span><span class="s">&#39;transcript&#39;</span><span class="p">,</span> meas<span class="o">=</span><span class="s">&#39;FPKM&#39;</span><span class="p">,</span> covariate<span class="o">=</span><span class="s">&#39;group&#39;</span><span class="p">)</span>
<span class="kp">head</span><span class="p">(</span>stat_resuts<span class="p">)</span>
</pre></div>


<ul>
<li>时间系列比较</li>
</ul>
<div class="codehilite"><pre><span></span><span class="c1"># 如果你的时间点较少，比如低于5，建议将其作为分组变量进行组组比较</span>
pData<span class="p">(</span>bg<span class="p">)</span> <span class="o">=</span> <span class="kt">data.frame</span><span class="p">(</span>pData<span class="p">(</span>bg<span class="p">),</span> time<span class="o">=</span><span class="kp">rep</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="m">10</span><span class="p">,</span><span class="m">2</span><span class="p">))</span>     <span class="c1"># dummy time corvariate</span>
timecourse_results <span class="o">=</span> stattest<span class="p">(</span>bg<span class="p">,</span> feature<span class="o">=</span><span class="s">&#39;transcript&#39;</span><span class="p">,</span> meas<span class="o">=</span><span class="s">&#39;FPKM&#39;</span><span class="p">,</span> covariate<span class="o">=</span><span class="s">&#39;time&#39;</span><span class="p">,</span> timecourse<span class="o">=</span><span class="bp">T</span><span class="p">)</span>
</pre></div>


<ul>
<li>调整混杂因素</li>
</ul>
<div class="codehilite"><pre><span></span><span class="c1"># 你可以通过pData调整任何混杂因素,也可以像下面一样添加</span>
group_adj_timecourse_results <span class="o">=</span> stattest<span class="p">(</span>bg<span class="p">,</span> feature<span class="o">=</span><span class="s">&#39;transcript&#39;</span><span class="p">,</span>
                                        meas<span class="o">=</span><span class="s">&#39;FPKM&#39;</span><span class="p">,</span> covariate<span class="o">=</span><span class="s">&#39;time&#39;</span><span class="p">,</span> 
                                        timecourse<span class="o">=</span><span class="bp">T</span><span class="p">,</span> adjustvars<span class="o">=</span><span class="s">&#39;group&#39;</span><span class="p">)</span>
</pre></div>


<ul>
<li>自定义比较模型 </li>
</ul>
<div class="codehilite"><pre><span></span><span class="c1"># sex,age,group,time vs. group,time</span>
    <span class="c1"># 模拟数据</span>
<span class="kp">set.seed</span><span class="p">(</span><span class="m">43</span><span class="p">)</span>
sex <span class="o">=</span> <span class="kp">sample</span><span class="p">(</span><span class="kt">c</span><span class="p">(</span><span class="s">&#39;M&#39;</span><span class="p">,</span> <span class="s">&#39;F&#39;</span><span class="p">),</span> size<span class="o">=</span><span class="kp">nrow</span><span class="p">(</span>pData<span class="p">(</span>bg<span class="p">)),</span> replace<span class="o">=</span><span class="bp">T</span><span class="p">)</span>
age <span class="o">=</span> <span class="kp">sample</span><span class="p">(</span><span class="m">21</span><span class="o">:</span><span class="m">52</span><span class="p">,</span> size<span class="o">=</span><span class="kp">nrow</span><span class="p">(</span>pData<span class="p">(</span>bg<span class="p">)),</span> replace<span class="o">=</span><span class="bp">T</span><span class="p">)</span>
    <span class="c1"># 创建design matrices</span>
mod <span class="o">=</span> model.matrix<span class="p">(</span><span class="o">~</span> sex <span class="o">+</span> age <span class="o">+</span> pData<span class="p">(</span>bg<span class="p">)</span><span class="o">$</span>group <span class="o">+</span> pData<span class="p">(</span>bg<span class="p">)</span><span class="o">$</span>time<span class="p">)</span>
mod0 <span class="o">=</span> model.matrix<span class="p">(</span><span class="o">~</span> pData<span class="p">(</span>bg<span class="p">)</span><span class="o">$</span>group <span class="o">+</span> pData<span class="p">(</span>bg<span class="p">)</span><span class="o">$</span>time<span class="p">)</span>
    <span class="c1"># 进行差异分析</span>
adjusted_results <span class="o">=</span> stattest<span class="p">(</span>bg<span class="p">,</span> feature<span class="o">=</span><span class="s">&#39;transcript&#39;</span><span class="p">,</span> meas<span class="o">=</span><span class="s">&#39;FPKM&#39;</span><span class="p">,</span> mod0<span class="o">=</span>mod0<span class="p">,</span> mod<span class="o">=</span>mod<span class="p">)</span>
<span class="kp">head</span><span class="p">(</span>adjusted_results<span class="p">)</span>
</pre></div>


<ul>
<li>输出到其他差异统计软件如limma,limma voom, DESeq, DEXSeq, EdgeR, etc.</li>
</ul>
<div class="codehilite"><pre><span></span><span class="c1"># 见2.处理组装数据</span>
<span class="o">*</span>expr<span class="p">(</span>bg<span class="p">,</span> meas_name<span class="p">)</span>
</pre></div>


<ul>
<li>简单地转录聚类
如此做是为了避免一个基因有多个相似的转录用作组装，会导致表达差异检测不准确。可以进行一下聚类看看。聚类使用的聚类是Jaccard distance;你也可以使用k-means clustering或者hierarchical clustering。</li>
</ul>
<div class="codehilite"><pre><span></span>clusterTranscripts<span class="p">(</span>gene<span class="o">=</span><span class="s">&#39;XLOC_000454&#39;</span><span class="p">,</span> gown<span class="o">=</span>bg<span class="p">,</span> k<span class="o">=</span><span class="m">2</span><span class="p">,</span> method<span class="o">=</span><span class="s">&#39;kmeans&#39;</span><span class="p">)</span>
<span class="c1"># 也可以可视化</span>
plotLatentTranscripts<span class="p">(</span>gene<span class="o">=</span><span class="s">&#39;XLOC_00054&#39;</span><span class="p">,</span> gown<span class="o">=</span>bg<span class="p">,</span> k<span class="o">=</span><span class="m">2</span><span class="p">,</span> method<span class="o">=</span><span class="s">&#39;kmeans&#39;</span><span class="p">,</span> returncluster<span class="o">=</span><span class="bp">F</span><span class="p">)</span>
</pre></div>


<ul>
<li>针对基因进行聚合表达量</li>
</ul>
<div class="codehilite"><pre><span></span>agg <span class="o">=</span> collapseTranscripts<span class="p">(</span>gene<span class="o">=</span><span class="s">&#39;XLOC_00054&#39;</span><span class="p">,</span> gown<span class="o">=</span>bg<span class="p">,</span> k<span class="o">=</span><span class="m">2</span><span class="p">,</span> method<span class="o">=</span><span class="s">&#39;kmeans&#39;</span><span class="p">)</span>
stattest<span class="p">(</span>gowntable<span class="o">=</span>agg<span class="o">$</span>tab<span class="p">,</span> pData<span class="o">=</span>pData<span class="p">(</span>bg<span class="p">),</span> feature<span class="o">=</span><span class="s">&#39;transcript_cluster&#39;</span><span class="p">,</span>
        covariate<span class="o">=</span><span class="s">&#39;group&#39;</span><span class="p">,</span> libadjust<span class="o">=</span><span class="bp">F</span><span class="p">)</span>
</pre></div>
		</content>

	</main>
	<footer>
		<p> Powered by xizhihui </p>
		<p> Thanks to <a href="https://www.hifreud.com/">hifreud</a> the theme is learned from his site. </p>
	</footer>

<script type="text/javascript" src="/js/index.json"></script>
<script type="text/javascript" src="/js/category.json"></script>
<script type="text/javascript" src="/js/tags.json"></script>
<script type="text/javascript" src="/js/about.json"></script>
<script type="text/javascript" src="/js/index.js"></script>
<script type="text/javascript">
	// listen to click event on navigator in header
	// document
	// 	.querySelector("div#head_right ul")
	// 	.addEventListener("click", change_page)
	var table = [['h2', 'ballgown', 'Ballgown'], ['h3', 'preprocessing', 'preprocessing'], ['h3', '_1', '安装'], ['h3', 'ballgown_1', 'Ballgown需求的格式'], ['h3', '_2', '使用'], ['h4', '1', '1. 加载数据'], ['h4', '2', '2. 处理组装数据'], ['h4', '3', '3.数据特征可视化'], ['h4', '4', '4. 差异分析']]
	var html = format_article_table(table)
	var table_ele = document.querySelector("main #table")
	table_ele.innerHTML = html
	// window.onload = evt => resize_nav(table_ele)
	adaption(document, window)

</script>
</body>
</html>