<!DOCTYPE html>
<html>
<head>
	<title>XZH's Notes</title>
	<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
	<meta name="author" content="xizhihui" />
	<meta name="viewport" content="width=device-width, initial-scale=1" />
	<link rel="stylesheet" type="text/css" href="/statics/reset.css">
	<link rel="stylesheet" type="text/css" href="/statics/header_footer.css"></link>
	<link rel="stylesheet" type="text/css" href="/statics/default.css">
	<link rel="stylesheet" type="text/css" href="/statics/main.css">
</head>
<body>
	<header class="Header">
		<nav>
			<h3>XZH</h3>
			<ul>
				<li><a href="/index.html#Timeline">Timeline</a></li><li><a href="/index.html#Category">Category</a></li><li><a href="/index.html#Tags">Tags</a></li><li><a href="/index.html#About">About</a></li>
			</ul>
		</nav>
	</header>
	<div class="Motto">
		<h2>XZH's Notes</h2>
		<h4>The future comes for you ~ ~ ~</h4>
	</div>

	<main>
		<nav id="table"></nav>
		<content id="content" class="content">
			<h2 id="chip-seq">ChIP-seq</h2>
<p>ChIP-seq是使用抗体捕获富集DNA片段和高通量测序技术来获得某些marker与DNA的结合位点的一项综合技术。ChIP是染色质免疫共沉淀, 通过特异抗体将DNA结合蛋白免疫沉淀, 用于捕获蛋白质的DNA靶点, 比如转录因子啊, 组蛋白修饰啊. 它主要分为以下四步：cross-linking、sonication、IP、Sequencing。在DNA与蛋白交联以后, 通过超声的方式随机打断染色体, 在利用抗体将目的交联物筛选出来, 再反交联获取DNA,最后上机测序.获取到测序数据后,典型的分析流程如图.</p>
<p><img alt="ChIP分析流程" src="http://mmbiz.qpic.cn/mmbiz_jpg/MPBFtnFrw4m2jWmOk9ic1ZkZ8VrQBrr8oCMwfEZDZG75OY3ic9qMqHhibpiafgZVfJaCSp7gqI87ETibVltjV4GgXkQ/640?wx_fmt=jpeg&amp;tp=webp&amp;wxfrom=5&amp;wx_lazy=1" /></p>
<p>文字版: rawdata -&gt; QC -&gt; mapping -&gt; peak calling(binding site) -&gt; visualization/annotation/MotifAnalysis.
在peak calling完成之后,我们首先会可视化看一下数据，接下来就会想知道这些peak都和什么样的基因有关. </p>
<h3 id="11">1.1 抗体的种类</h3>
<ul>
<li>转录因子或抑制因子(transcription factors/repressors)如CTCT</li>
<li>组蛋白或组蛋白修饰(histones/histone modifications)如H3/H3K4me3</li>
<li>DNA修饰(DNA modfications)如DNA甲基化</li>
<li>染色质重塑蛋白(chromatin remodelling proteins)如BMI1</li>
<li>其他转录机制相关蛋白(transcription machinery)如Pol2</li>
</ul>
<h3 id="12">1.2 实验结果简要展示</h3>
<p><img alt="ChIP结果简要展示" src="https://s1.ax1x.com/2018/10/14/iUEKwq.png" /></p>
<h3 id="13">1.3 富集的类型</h3>
<p>经过抗体富集的DNA可以是单个结合点，也可以是较宽的结合区域，当然也可能是任何地方。</p>
<p><img alt="type of enrichment" src="https://s1.ax1x.com/2018/10/14/iUElkV.png" /></p>
<h3 id="14">1.4. 表征结果的形式</h3>
<p>ChIP测的是富集, 而且是相对富集, 就是指A/B两个区域的富集信号的富集差别.如果要得到绝对值的话,那就要进行归一化或者校正.影响富集的因素有,比如mark结合的起始位点, 能结合的位点种类或数目,富集时的信号强度.</p>
<h2 id="step1">二、step1: 数据预处理</h2>
<h3 id="21-chip">2.1 ChIP文库</h3>
<ul>
<li>potential technical problems</li>
<li>
<ul>
<li>adapter contimination</li>
</ul>
</li>
<li>
<ul>
<li>PCR duplication</li>
</ul>
</li>
<li>potential biological problems</li>
<li>
<ul>
<li>lack of enrichment</li>
</ul>
</li>
<li>
<ul>
<li>other selection bias</li>
</ul>
</li>
</ul>
<h3 id="22">2.2 质控</h3>
<ul>
<li>QC of reads: fastqc, multiqc</li>
<li>QC of alignment: multiqc</li>
<li>Filtering of alignment with MAPQ: samtools</li>
<li>Deduplication: picard</li>
<li>基于read density去除outliers</li>
</ul>
<div class="codehilite"><pre><span></span><span class="c1"># 用MAPQ进行过滤其值低于20的</span>
samtools view -q <span class="m">20</span> -b -o filtered.bam input.bam
</pre></div>


<h4 id="what-about-deduplication">what about deduplication?为何不直接去掉所有的重复？</h4>
<p>重复可以是由PCR引起的，也可能是由enrich引起的同一序列存在多个的情况，一般来说，重复使得富集更为明显，对识别富集区域(可能的结合区域/位点)有一定帮助。基于此，我们不能一股脑地去掉所有重复。</p>
<h2 id="step2peak-calling">三、step2：peak calling</h2>
<h3 id="31-peak-callers-workflow">3.1 Peak callers workflow</h3>
<p><img alt="Peak callers workflow" src="https://s1.ax1x.com/2018/10/14/iUVn3D.png" /></p>
<ul>
<li>优化初始数据：校正forwar/reverse peak offset, deduplication</li>
<li>构建模型：peak = Observed + Model</li>
<li>滑动窗口: 窗口大小=fragmentSize/2, 保留counts数大于Model的窗口</li>
<li>校正：若合并后的窗口counts数大于合并后model的counts，则合并相邻窗口生成总的候选peak set</li>
</ul>
<h3 id="32-peak-calling">3.2 Peak calling 失败的原因</h3>
<p>Peak calling是富集程度、背景值、序列总数的综合结果。
+ 用于Observed的data是进行Model的data的子集(比如处理无效果，无处理的两组)
+ Observed data有peak的地方，Model data连数据都没有(With no input the region around the peak is used to model the background)</p>
<h3 id="33">3.3 结果可靠吗？</h3>
<ul>
<li>多数ChIP enrichment不是链特异性的，你应该在两条链上都能看到富集</li>
<li>重复之间的结果应该是一致的</li>
</ul>
<h3 id="34">3.4 下游分析</h3>
<ul>
<li>composition and motif analysis: 前者分析peak所在DNA组成，后者分析序列潜在生物学功能</li>
<li>GO：peak所在基因的GO富集</li>
</ul>
<h2 id="step3peak-data">四、step3：探索peak data</h2>
<h3 id="41-peak">4.1 可视化peak在序列上的分布情况</h3>
<ul>
<li>Is there any enrichment?</li>
<li>What is the size / patterning of enrichment?</li>
<li>How well are my controls behaving?</li>
<li>What is the best way to quantitate this data?</li>
<li>Are there any technical artefacts?</li>
<li>Are my peaks narrow or broad</li>
<li>Do peak positions obviously correspond to existing features(like TSS)?</li>
</ul>
<h4 id="igvenrichment">IGV查看enrichment</h4>
<blockquote>
<p>请注意，因为你需要使用IGV可视化查看enrichment的分布情况，这里IGV的input bam文件是sorted bam文件。</p>
</blockquote>
<div class="codehilite"><pre><span></span>samtools sort -O bam -o <span class="nv">$name</span>.sorted.bam <span class="nv">$name</span>.bam
</pre></div>


<h3 id="42">4.2 检查对照</h3>
<p>如果有使用IgG或Mock IP，不同的片段化DNA的方法的话，检查它们各自的peak中，peak coverage是均匀的吗？peak的分布模式是一致的吗？如果不是，那么：
+ Low coverage：Repetitive unmappable regions, Holes in the assembly
+ High coverage: Mismapped reads from outside the assembly
+ Biases: DNA stability, GC content, Segmental Duplication</p>
<h3 id="43-compare-samples">4.3 compare samples</h3>
<ul>
<li>可视化比较peak分布</li>
<li>scatterplot比较input/chip，input/input，chip/chip</li>
<li>correlation metrix, correlation tree, pca-plot, tSNE plot</li>
<li>Cumulative Distribution Plot, Q-Q plot</li>
</ul>
<h3 id="44-enrichmentpeaktrend-plots">4.4 把enrichment/peak同序列特征相联系：trend plots</h3>
<p>可视化peak在features(Gene body,promoter,CpG islands, Tss)的分布情况，查看是否符合一般性质(比如甲基化修饰应该在CpG islands富集), 然后分析可能的feature.</p>
<ul>
<li>
<p>查看总的enrich情况，看是否有peak在某个feature
<img alt="overall average" src="https://s1.ax1x.com/2018/10/14/iUZSat.png" /></p>
</li>
<li>
<p>查看不同sample在单个feature的enrichment
<img alt="enrichment of single feature" src="https://s1.ax1x.com/2018/10/14/iUZAMQ.md.png" /></p>
</li>
</ul>
<h2 id="macs2peak-calling">五、使用MACS2进行peak calling</h2>
<div class="codehilite"><pre><span></span><span class="c1"># 第一个方法,使用python的pip</span>
<span class="c1"># 1.先用conda info --envs查看当前的环境, 我的是有base(python3.7)和env_name(python2.7)</span>
<span class="c1"># 2.如果没有python2.7的环境,可以创建一个名为py27的环境</span>
conda create -n py27 <span class="nv">python</span><span class="o">=</span><span class="m">2</span>.7
<span class="nb">source</span> activate py27
pip install MACS2
<span class="nb">source</span> deactivate py27
</pre></div>


<div class="codehilite"><pre><span></span><span class="c1"># 1. 进入python2.7环境</span>
<span class="nb">source</span> activate env_name
macs2 callpeak -c control.bam -t treat.bam <span class="se">\</span>
        -m <span class="m">10</span> <span class="m">30</span> -p 1e-5 <span class="se">\</span>
        -f BAM -g mm -n treat
    <span class="c1"># -c &lt;bam&gt;      进行peak calling的对照</span>
    <span class="c1"># -t &lt;bam&gt;      进行peak calling的目标</span>
    <span class="c1"># -m 10 30      建立双峰模型的参数</span>
    <span class="c1"># -p &lt;pvalue&gt;   显著性阈值,1e-5</span>
    <span class="c1"># -f BAM        输入的文件类型</span>
    <span class="c1"># -g &lt;organism&gt; 进行peak calling的物种, hg(人), mm(小鼠), rn(rat)</span>
    <span class="c1"># -n &lt;string&gt;   输出文件的前缀</span>
nohup macs2 callpeak -c ../02alignment/IgGold.sorted.bam -t ../02alignment/suz12.sorted.bam -m <span class="m">10</span> <span class="m">30</span> -p 1e-5 -f BAM -g mm -n suz12 <span class="m">2</span>&gt;suz12.masc2.log <span class="p">&amp;</span>

<span class="c1"># 每个样本共输出四个文件:</span>
    <span class="c1"># 1. name_peaks.xls         存放peak信息,格式与bed一致;坐标从1开始,而bed从0开始.</span>
    <span class="c1"># 2. name_model.r           peak模型,可直接进行R作图.</span>
    <span class="c1"># 3. name_peaks.narrowpeak  peak矩阵,与.broadpeak相似.可导入R等进行数据分析.BED6+4 format file.</span>
    <span class="c1"># 4. name_summits.bed       每个peak的peak summits(极值点的位置),用此寻找结合位点的motif.</span>
<span class="c1"># 查看peaks数目</span>
ls *.bed <span class="p">|</span> xargs -i wc -l <span class="o">{}</span> <span class="c1"># 6514 suz12_summits.bed</span>
</pre></div>


<h2 id="chipseeker">六、使用ChIPseeker进行可视化</h2>
<h3 id="60-bed">6.0 BED 文件格式</h3>
<table>
<thead>
<tr>
<th>1</th>
<th>2</th>
<th>3</th>
<th>4</th>
<th>5</th>
<th>6</th>
<th>7</th>
<th>8</th>
<th>9</th>
<th>10</th>
<th>11</th>
<th>12</th>
</tr>
</thead>
<tbody>
<tr>
<td>chrom</td>
<td>start</td>
<td>end</td>
<td>name</td>
<td>score</td>
<td>strand</td>
<td>thickStart</td>
<td>thickEnd</td>
<td>itemRgb</td>
<td>blockCount</td>
<td>blockSizes</td>
<td>blockStarts</td>
</tr>
</tbody>
</table>
<ul>
<li>在这12列中,前3列是必需的字段.</li>
<li>score是在genome browser中上色的分值(0-1000),就是peak峰的峰高(summit height of fragment pileup)</li>
<li>strand值链的正负性, 但是在ChIP中我们无法区分链的正负性</li>
<li>thickStart/End是在genome browser画矩形的起点和中点</li>
<li>itemRgb是上色的颜色</li>
<li>block是子元件,比如外显子内含子,5'utr之类的.</li>
</ul>
<h3 id="61">6.1 包和文件</h3>
<p>TxDb数据库依据你的ChIP实验动物的物种下载对应物种的数据库.也可以自己通过makeTxDbFromBiomart/makeTxDbFromUCSC包来准备TxDb类型的数据库.hg38(TxDb.Hsapiens.UCSC.hg38.knownGene)、hg19(TxDb.Hsapiens.UCSC.hg19.knownGene)、mm10(TxDb.Mmusculus.UCSC.mm10.knownGene)、mm9(TxDb.Mmusculus.UCSC.mm9.knownGene)</p>
<div class="codehilite"><pre><span></span><span class="kn">library</span><span class="p">(</span>ChIPseeker<span class="p">)</span>                         <span class="c1"># for all</span>
<span class="kn">library</span><span class="p">(</span>TxDb.Hsapiens.UCSC.hg19.knownGene<span class="p">)</span>  <span class="c1"># for 6.3</span>
<span class="kn">library</span><span class="p">(</span>org.Hs.eg.db<span class="p">)</span>                       <span class="c1"># for 6.3,6.5</span>
<span class="kn">library</span><span class="p">(</span>clusterProfiler<span class="p">)</span>                    <span class="c1"># for 6.5,6.6</span>
<span class="kn">library</span><span class="p">(</span>ReactomePA<span class="p">)</span>                         <span class="c1"># for 6.5,6.6</span>
txdb <span class="o">=</span> TxDb.Hsapiens.UCSC.hg19.knownGene
peak <span class="o">=</span> readPeakFile<span class="p">(</span>filename<span class="p">)</span>
</pre></div>


<h3 id="62-peakpeaks-coverage-plot">6.2 可视化peak在基因组上的富集区域(peaks coverage plot)</h3>
<p>除了peak文件, GRangeList也支持作为出入,来比较不同bed文件的peak情况。</p>
<div class="codehilite"><pre><span></span>covplot<span class="p">(</span>peak<span class="p">,</span> weightCol<span class="o">=</span><span class="s">&#39;V5&#39;</span><span class="p">)</span>       <span class="c1"># V5是peakfile的第五列(scores)</span>
<span class="c1"># 指定染色体区域</span>
covplot<span class="p">(</span>peak<span class="p">,</span> weightCol<span class="o">=</span><span class="s">&#39;V5&#39;</span><span class="p">,</span> chrs<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="s">&#39;chr10&#39;</span><span class="p">,</span> <span class="s">&#39;chr12&#39;</span><span class="p">),</span> xlim<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="m">4.5e7</span><span class="p">,</span> <span class="m">5e7</span><span class="p">))</span>
</pre></div>


<h3 id="63-tsspeaks">6.3 结合到TSS区域的peaks分析</h3>
<h4 id="631-peaks">6.3.1 peaks热图分析</h4>
<p>为了分析结合到TSS区域的peaks, 我们需要准备好表示TSS regions的文件, 然后再进行计算.注意,我们这里指定了TSS上下游3000bp的区域, 你也可以指定其他区域(getBioRegion和getTagMatrix联合使用)</p>
<div class="codehilite"><pre><span></span>promoter <span class="o">=</span> getPromoter<span class="p">(</span>TxDb<span class="o">=</span>txdb<span class="p">,</span> upstream<span class="o">=</span><span class="m">3000</span><span class="p">,</span> downstream<span class="o">=</span><span class="m">3000</span><span class="p">,</span> by<span class="o">=</span><span class="s">&#39;gene&#39;</span><span class="p">)</span> <span class="c1"># by=gene/transcript</span>
<span class="c1"># getBioRegion(TxDb = NULL, upstream = 1000, downstream = 1000, by = &quot;gene&quot;) # by=gene/transcript/exon/intron</span>
tagMatrix <span class="o">=</span> getTagMatrix<span class="p">(</span>peak<span class="p">,</span> windows<span class="o">=</span>promoter<span class="p">)</span>
tagHeatmap<span class="p">(</span>tagMatrix<span class="p">,</span> xlim<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="m">-3000</span><span class="p">,</span> <span class="m">3000</span><span class="p">),</span> color<span class="o">=</span><span class="s">&#39;red&#39;</span><span class="p">)</span>
<span class="c1"># 一步法热图</span>
peakHeatmap<span class="p">(</span>filename<span class="p">,</span> TxDb<span class="o">=</span>txdb<span class="p">,</span> upstream<span class="o">=</span><span class="m">3000</span><span class="p">,</span> downstream<span class="o">=</span><span class="m">3000</span><span class="p">,</span> color<span class="o">=</span><span class="s">&#39;red&#39;</span><span class="p">)</span>
</pre></div>


<h4 id="632-peaksread-count-frequency">6.3.2 peaks的平均read count frequency</h4>
<div class="codehilite"><pre><span></span>plotAvgProf<span class="p">(</span>tagMatrix<span class="p">,</span> xlim<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="m">-3000</span><span class="p">,</span><span class="m">3000</span><span class="p">),</span> 
            xlab<span class="o">=</span><span class="s">&quot;Genomic Region (5&#39;-&gt;3&#39;)&quot;</span><span class="p">,</span> ylab<span class="o">=</span><span class="s">&quot;Read Count Frequency&quot;</span><span class="p">,</span>
            conf<span class="o">=</span><span class="m">0.95</span><span class="p">,</span> resample<span class="o">=</span><span class="m">1000</span><span class="p">)</span>   <span class="c1"># 添加置信区间</span>
<span class="c1"># 由文件名一步到位</span>
plotAvgProf2<span class="p">(</span>filename<span class="p">,</span> TxDb<span class="o">=</span>txdb<span class="p">,</span> upstream<span class="o">=</span><span class="m">3000</span><span class="p">,</span> 
            downstream<span class="o">=</span><span class="m">3000</span><span class="p">,</span> xlab<span class="o">=</span><span class="s">&quot;Genomic Region (5&#39;-&gt;3&#39;)&quot;</span><span class="p">,</span> 
            ylab<span class="o">=</span><span class="s">&quot;Read Count Frequency&quot;</span><span class="p">)</span>
</pre></div>


<h3 id="64">6.4 注释及其可视化</h3>
<ul>
<li>获取peak附近最近的基因(6.4.2)</li>
<li>注释peak所在的基因组区域(6.4.1)</li>
</ul>
<p>就像拿到fastq进行比对,你需要提供全基因组的参考序列, peak的注释同样需要参考信息, 也就是注释信息, 这些信息包含基因的起始和结束位置, 在基因的哪个区域是内含子, 哪里是外显子等信息.ChIP支持所有的有基因位置注释信息的物种, 这些注释我们要存储在TxDb对象里面.  </p>
<p>注释有好几种种, 一种是genomic annotation(annotation列),另一种是nearest gene annotation(其他咧).第一种genomic annotation的注释peak的位置,在基因的什么地方,比如UTR,外显子内含子之类的, 这个位置可能就是调控的根本, 可变剪切的调控就属于这类.而nearest gene annotation的注释是最近的基因, 是离peak的距离最近的TSS, 这个TSS所在的基因,这个关注的是启动子区域, 这个基因最有可能被调控. 然后由于一个基因有多个TSS, 多个转录本, 所以有一列transcriptId的列. 还有第三种注释,那就是注释peak区域上下游某个范围, 看这个范围的基因都有些啥.</p>
<div class="codehilite"><pre><span></span><span class="c1"># peakAnno是个csAnno实例, 可以用as.GRanges把它转化成GRanges实例.</span>
<span class="c1"># as.data.frame可以把csAnno转换成data.frame,以供写入文件</span>
<span class="c1"># annoDb可选,用以添加各种ID的列. 物种要对应正确</span>
peakAnno <span class="o">=</span> annotatePeak<span class="p">(</span>filename<span class="p">,</span> tssRegion<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="m">-3000</span><span class="p">,</span> <span class="m">3000</span><span class="p">),</span> TxDb<span class="o">=</span>txdb<span class="p">,</span> annoDb<span class="o">=</span><span class="s">&#39;org.Hs.eg.db&#39;</span><span class="p">)</span>
</pre></div>


<h4 id="641-peaks">6.4.1 peaks分布区域的饼图, 柱形图</h4>
<div class="codehilite"><pre><span></span>plotAnnoPie<span class="p">(</span>peakAnno<span class="p">)</span>
plotAnnoBar<span class="p">(</span>peakAnno<span class="p">)</span>
<span class="c1"># vennpie(peakAnno)</span>
upsetplot<span class="p">(</span>peakAnno<span class="p">,</span> vennpie<span class="o">=</span><span class="bp">T</span><span class="p">)</span>
</pre></div>


<h4 id="642-peakbinding-sitetss">6.4.2 peak(binding site)与最近基因的TSS距离</h4>
<div class="codehilite"><pre><span></span>plotDistToTSS<span class="p">(</span>peakAnno<span class="p">,</span> title<span class="o">=</span><span class="s">&#39;Distribution of transcription factor-binding loci\n relative to TSS&#39;</span><span class="p">)</span>
</pre></div>


<h3 id="65">6.5 进行功能富集分析</h3>
<p>一旦我们掌握peak区域(TF-binding site)最近的基因信息后,就可以对这些基因进行富集分析啦, 看看它们有什么功能咯.这就有好多方向.Go、KEGG、DO(基因和人类疾病,DOSE)、Reactome（基因与pathways和reactions,ReactomePA).为了进行这些富集分析,就可能要用到seq2gene函数了.它可以以多对多的形式把基因组区域同基因联系起来.</p>
<div class="codehilite"><pre><span></span>pathway1 <span class="o">=</span> ReactomePA<span class="o">::</span>enrichPathway<span class="p">(</span><span class="kp">as.data.frame</span><span class="p">(</span>peakAnno<span class="p">)</span><span class="o">$</span>geneId<span class="p">)</span>
<span class="kp">head</span><span class="p">(</span>pathway1<span class="p">)</span>
gene <span class="o">=</span> seq2gene<span class="p">(</span>peak<span class="p">,</span> tssRegion<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="m">-1000</span><span class="p">,</span> <span class="m">1000</span><span class="p">),</span> flankDistance<span class="o">=</span><span class="m">3000</span><span class="p">,</span> TxDb<span class="o">=</span>txdb<span class="p">)</span>
pathway2 <span class="o">=</span> ReactomePA<span class="o">::</span>enrichPathway<span class="p">(</span>gene<span class="p">)</span>
<span class="kp">head</span><span class="p">(</span>pathway2<span class="p">,</span> <span class="m">2</span><span class="p">)</span>
dotplot<span class="p">(</span>pathway2<span class="p">)</span>
</pre></div>


<h3 id="66-peak-data-set-peak">6.6 比较不同的peak data set (多个peak结果一起分析咯)</h3>
<h4 id="661-tss-regionpeak">6.6.1 结合到TSS region的peak分析: 频率和热图</h4>
<p>plotAvgProf,tagHeatmap都支持tagMatrixList的输入, plotAvgProf2和peakHeatmap也接受filenameList</p>
<div class="codehilite"><pre><span></span>peaks <span class="o">=</span> <span class="kp">sapply</span><span class="p">(</span>filenames<span class="p">,</span> readPeakFile<span class="p">)</span>
tagMatrixList <span class="o">=</span> <span class="kp">lapply</span><span class="p">(</span>peaks<span class="p">,</span> getTagMatrix<span class="p">,</span> window<span class="o">=</span>promoter<span class="p">)</span>
plotAvgProf<span class="p">(</span>tagMatrixList<span class="p">,</span> xlim<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="m">-3000</span><span class="p">,</span><span class="m">3000</span><span class="p">))</span>      <span class="c1"># average profiles of ChIP peaks among different experiments</span>
plotAvfProf<span class="p">(</span>tagMatrixList<span class="p">,</span> xlim<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="m">-3000</span><span class="p">,</span><span class="m">3000</span><span class="p">),</span> conf<span class="o">=</span><span class="m">0.95</span><span class="p">,</span> resample<span class="o">=</span><span class="m">500</span><span class="p">,</span> facet<span class="o">=</span><span class="s">&#39;row&#39;</span><span class="p">)</span>    <span class="c1"># 置信区间和按行作图</span>
tagHeatmap<span class="p">(</span>tagMatrixList<span class="p">,</span> xlim<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="m">-3000</span><span class="p">,</span><span class="m">3000</span><span class="p">),</span> color<span class="o">=</span><span class="kc">NULL</span><span class="p">)</span>   <span class="c1"># heatmap of peaks among different experiments</span>
</pre></div>


<h4 id="662-peak">6.6.2 peak注释比较</h4>
<p>多个annotatePeak输出的结果整合成列表, 也可以输入到plotAnnoBar和plotDistToTSS里面去.</p>
<div class="codehilite"><pre><span></span>peakAnnoList <span class="o">=</span> <span class="kp">lapply</span><span class="p">(</span>filenames<span class="p">,</span> annotatePeak<span class="p">,</span> TxDb<span class="o">=</span>txdb<span class="p">,</span> tssRegion<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="m">-3000</span><span class="p">,</span> <span class="m">3000</span><span class="p">),</span> verbose<span class="o">=</span><span class="bp">F</span><span class="p">)</span>
plotAnnoBar<span class="p">(</span>peakAnnoList<span class="p">)</span>   <span class="c1"># Genomic Annotation among different ChIPseq data</span>
plotDistToTSS<span class="p">(</span>peakAnnoList<span class="p">)</span> <span class="c1"># Distribution of Binding Sites among different ChIPseq data</span>
</pre></div>


<h4 id="663">6.6.3 功能富集比较</h4>
<div class="codehilite"><pre><span></span>genes <span class="o">=</span> <span class="kp">lapply</span><span class="p">(</span>peakAnnoList<span class="p">,</span> <span class="kr">function</span><span class="p">(</span>x<span class="p">)</span> <span class="kp">as.data.frame</span><span class="p">(</span>x<span class="p">)</span><span class="o">$</span>geneId<span class="p">)</span>
<span class="kp">names</span><span class="p">(</span>genes<span class="p">)</span> <span class="o">=</span> <span class="kp">sub</span><span class="p">(</span><span class="s">&#39;_&#39;</span><span class="p">,</span> <span class="s">&#39;\n&#39;</span><span class="p">,</span> <span class="kp">names</span><span class="p">(</span>genes<span class="p">))</span>
compKEGG <span class="o">=</span> clusterProfiler<span class="o">::</span>compareCluster<span class="p">(</span>geneCluster<span class="o">=</span><span class="s">&#39;genes&#39;</span><span class="p">,</span> fun<span class="o">=</span><span class="s">&#39;enrichKEGG&#39;</span><span class="p">,</span>
                            <span class="c1"># One of &quot;groupGO&quot;, &quot;enrichGO&quot;, &quot;enrichKEGG&quot;, &quot;enrichDO&quot; or &quot;enrichPathway&quot;</span>
                            pvalueCutoff<span class="o">=</span><span class="m">0.05</span><span class="p">,</span>
                            pAdjustMethod<span class="o">=</span><span class="s">&#39;BH&#39;</span><span class="p">)</span>
dotplot<span class="p">(</span>compKEGG<span class="p">,</span> showCategory<span class="o">=</span><span class="m">15</span><span class="p">,</span> title<span class="o">=</span><span class="s">&#39;KEGG Pathway Enrichment Analysis&#39;</span><span class="p">)</span>
</pre></div>


<h4 id="664-peak">6.6.4 peak和注释后基因的重叠</h4>
<p>这个可以用于实验重复/样品重复peak结果的一致性. 也可以看不同样品peak结果的交叉点.</p>
<div class="codehilite"><pre><span></span>genes <span class="o">=</span> <span class="kp">lapply</span><span class="p">(</span>peakAnnoList<span class="p">,</span> <span class="kr">function</span><span class="p">(</span>x<span class="p">)</span> <span class="kp">as.data.frame</span><span class="p">(</span>x<span class="p">)</span><span class="o">$</span>geneId<span class="p">)</span>
vennplot<span class="p">(</span>genes<span class="p">)</span>
</pre></div>


<h3 id="67-chip-seq-peak">6.7 ChIP-seq peak结果重叠的统计分析</h3>
<p>如果两个不同的ChIP实验的peak结果有很大重叠的话, 表明这两个实验中用于IP的蛋白极有可能是分子伴侣(cooperative in regulation). 所以,通过对重叠进行统计分析可以确认这个事实的显著性.<br />
当然,这里面有一些理论前提.我们把基因组位置随机打断的话, 在把这些片段进行peak calling, 我们会得到一个peak的概率分布. 基于此, 我们可以统计ChIP-seq peak calling的显著性.</p>
<div class="codehilite"><pre><span></span>p <span class="o">=</span> GRanges<span class="p">(</span>seqnames<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="s">&#39;chr1&#39;</span><span class="p">,</span> <span class="s">&#39;chr3&#39;</span><span class="p">),</span> ranges<span class="o">=</span>IRanges<span class="p">(</span>start<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="m">100</span><span class="p">),</span> end<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="m">50</span><span class="p">,</span><span class="m">130</span><span class="p">)))</span>
shuffle<span class="p">(</span>p<span class="p">,</span> TxDb<span class="o">=</span>txdb<span class="p">)</span>
<span class="c1"># 基于基因组坐标计算的overlap显著性</span>
<span class="c1">#（overlap significant of ChIP experiments based on the genome coordinations)</span>
enrichPeakOverlap<span class="p">(</span>queryPeak<span class="o">=</span>files<span class="p">[[</span><span class="m">5</span><span class="p">]],</span> targetPeak<span class="o">=</span><span class="kp">unlist</span><span class="p">(</span>files<span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">]),</span>
                  TxDb<span class="o">=</span>txdb<span class="p">,</span> pAdjustMethod<span class="o">=</span><span class="s">&#39;BH&#39;</span><span class="p">,</span>
                  nShuffle<span class="o">=</span><span class="m">50</span><span class="p">,</span> chainFile<span class="o">=</span><span class="kc">NULL</span><span class="p">,</span>
                  verbose<span class="o">=</span><span class="bp">F</span><span class="p">)</span>

<span class="c1"># 基于最近TSS的基因注释计算overlap显著性</span>
<span class="c1"># overlap sig. calculated based on their nearest gene annotation</span>
enrichAnnoOverlap<span class="p">(</span>queryPeak <span class="o">=</span> filename<span class="p">,</span> targetPeak <span class="o">=</span> filename<span class="p">(</span>s<span class="p">),</span>
                  TxDb<span class="o">=</span>txdb<span class="p">,</span> pAdjustMethod<span class="o">=</span><span class="s">&#39;BH&#39;</span><span class="p">,</span>
                  chainFile<span class="o">=</span><span class="kc">NULL</span><span class="p">,</span> distanceToTSS_cutoff<span class="o">=</span><span class="kc">NULL</span><span class="p">)</span>
</pre></div>


<h3 id="68-geochip-seq">6.8 GEO中ChIP-seq数据挖掘</h3>
<p>我们已经可以计算不同ChIP-seq实验peak数据的重叠显著性了.而在GEO上有相当多的数据,那么我们可以拿自己的数据与里面的peak数据去找重叠,如果找到了新的蛋白-蛋白复合转录体,又是个新发现了.
这个包里面涵盖了17000个GEO bed文件, 可以通过getGEOspecies来获取这些文件的概要.</p>
<div class="codehilite"><pre><span></span>getGEOspecies<span class="p">()</span> <span class="c1"># 基于物种</span>
getGEOgenomeVersion<span class="p">()</span> <span class="c1"># 基于基因组版本</span>
<span class="c1"># 获取详细信息</span>
hg19 <span class="o">=</span> getGEOInfo<span class="p">(</span>genome<span class="o">=</span><span class="s">&#39;hg19&#39;</span><span class="p">,</span> simplify<span class="o">=</span><span class="bp">T</span><span class="p">)</span>
<span class="c1"># 下载数据</span>
downloadGEObedFiles<span class="p">(</span>genome<span class="o">=</span><span class="s">&#39;hg19&#39;</span><span class="p">,</span> destDir<span class="o">=</span><span class="s">&#39;hg19&#39;</span><span class="p">)</span>  <span class="c1"># 根据基因组版本</span>
downloadGSMbedFiles<span class="p">(</span>gsm_vector<span class="p">,</span> destDir<span class="o">=</span><span class="s">&#39;hg19&#39;</span><span class="p">)</span>     <span class="c1"># 根据gsm accession来下载</span>
</pre></div>


<h2 id="deeptools">七、使用deeptools进行可视化</h2>
<p>deeptools可以用于处理比对结果的多项质控, 并基于比对文件生成校正后的bed或bigwig格式的覆盖度文件(normalized coverage file),这就允许多个处理组的比较了. 当然,利用这些文件, 你还可以进行多种可视化.</p>
<p><img alt="deeptools in ChIP-seq" src="http://deeptools.readthedocs.io/en/latest/_images/start_workflow.png" /></p>
<ul>
<li>多个Bam文件的相关性: multiBamSummary, plotCorrelation, multiBigWigSummary</li>
<li>比对后read的覆盖度: plotCoverage(--ignoreDuplicates is useful), bamPEFragmentSize(for PE)</li>
<li>GC-bias: computeGCBias, correctGCBias</li>
<li>估测ChIP信号强度:</li>
</ul>
<h3 id="71">7.1 质控和数据处理</h3>
<ol>
<li>从bam文件生成bigwig
<a href="http://deeptools.readthedocs.io/en/latest/content/tools/bamCoverage.html">bamCoverage</a></li>
</ol>
<div class="codehilite"><pre><span></span>bamCoverage -b reads.bam -o coverage.bw <span class="se">\</span>
            --binSize <span class="m">10</span> <span class="se">\</span>
            --normalizeUsing RPGC <span class="se">\</span>
            --extendReads <span class="se">\</span>
<span class="c1"># -b,--bam          输入的bam文件</span>
<span class="c1"># -o                输出文件名</span>
<span class="c1"># -of               输出格式, bigwig, bedgraph</span>
</pre></div>


<ol>
<li>检测多个测序结果之间的可重复性</li>
</ol>
<div class="codehilite"><pre><span></span>multiBamSummary, plotCorrelation
</pre></div>


<ol>
<li>检测和校正 GC bias</li>
</ol>
<div class="codehilite"><pre><span></span>computeGCBias
</pre></div>


<ol>
<li>以特定文件校正ChIP的覆盖率度</li>
</ol>
<div class="codehilite"><pre><span></span>bamCompare with <span class="nv">ChIP</span> <span class="o">=</span> treatment, <span class="nv">input</span><span class="o">=</span>control
</pre></div>


<ol>
<li>检测不同ChIP实验的信号强度</li>
</ol>
<div class="codehilite"><pre><span></span>plotFingerprint
</pre></div>


<ol>
<li>得到TSS(转录起始位点)富集基因的覆盖度热图</li>
</ol>
<div class="codehilite"><pre><span></span>computeMatrix, plotHeatmap
</pre></div>


<ol>
<li>比较基因在常染色体与性染色体之间的信号差别</li>
</ol>
<div class="codehilite"><pre><span></span><span class="m">1</span>. 过滤出欲比较的基因
<span class="m">2</span>. computeMatrix
<span class="m">3</span>. plotProfile
<span class="m">4</span>. plotting the summary plots <span class="k">for</span> multiple samples
</pre></div>


<h3 id="72">7.2 函数的使用</h3>
<h4 id="721-computematrix">7.2.1 computeMatrix</h4>
<p>对基因组上的任何一个区域进行打分(peak 数),生成供plotHeatMap和plotProfiles用的中间文件。它的输入的是多个打分文件如bigwig文件,bed文件。当然，这个函数还可以根据输入的bw/bed文件对基因组上的区域进行筛选.<br />
它有两种模式,参考点模式(reference-point)和大区域模式(scale-regions).</p>
<p>bw文件可以通过bamCoverage/bamCompare函数得到.</p>
<ul>
<li>computeMatrix scale-regions模式</li>
</ul>
<div class="codehilite"><pre><span></span>computeMatrix scale-regions -S &lt;bw/bed files&gt; -R &lt;bed file&gt; -b <span class="m">1000</span>

<span class="c1"># -R,--regionsFileName          你想可视化的区域的文件名,以BED/GTF格式.</span>
<span class="c1">#                               多个文件的话,可以使用#进行分组.分组的文件将画在一起.</span>
<span class="c1"># -S,--scoreFileName            打分文件, 以bw,bed格式.看这里,就需要打分文件啦.bigwig.</span>
<span class="c1"># -o,-out,--outFileName         输出.gz文件以供plotHeatMap和plotProfile使用</span>
<span class="c1"># --outFileNameMatrix           输出peak calling矩阵,可以进行差异分析</span>
<span class="c1"># --outFileSortedRegions        在你设置一个筛选条件后输出的筛选后文件名</span>
<span class="c1"># -a,--downstream               转录起始位点下游的bp数</span>
<span class="c1"># -b,--upstream                 转录起始位点上游的bp数</span>
<span class="c1"># 具体筛选参数设定见handbook</span>

computeMatrix scale-regions <span class="se">\</span>
                -R genes_chr19_firstHalf.bed genes_chr19_secondHalf.bed <span class="se">\ </span><span class="c1"># separate multiple files with spaces</span>
                -S testFiles/log2ratio_*.bw  <span class="se">\ </span>or use the wild card approach
                -b <span class="m">3000</span> -a <span class="m">3000</span> <span class="se">\</span>
                --regionBodyLength <span class="m">5000</span> <span class="se">\</span>
                --skipZeros -o matrix2_multipleBW_l2r_twoGroups_scaled.gz <span class="se">\</span>
                --outFileNameMatrix matrix2_multipleBW_l2r_twoGroups_scaled.tab <span class="se">\</span>
                --outFileSortedRegions regions2_multipleBW_l2r_twoGroups_genes.bed
</pre></div>


<ul>
<li>computeMatrix reference-point模式</li>
</ul>
<div class="codehilite"><pre><span></span>computedMatrix reference-point -S &lt;bw/bed file<span class="o">(</span>s<span class="o">)</span>&gt; -R &lt;bed file&gt; -a <span class="m">3000</span> -b <span class="m">3000</span>

<span class="c1"># 参数可见scale-regions模式</span>
<span class="c1"># --referencePoint      指定参考点,可选的参考点为TSS,TES,center.TSS(region start) TES(region end), center(region center)</span>
                        不管你用的啥参考点, 可视化时都是用TSS作为默认标签。

computeMatrix reference-point --referencePoint TSS <span class="se">\ </span><span class="c1"># alternatives: TES, center</span>
       -b <span class="m">3000</span> -a <span class="m">10000</span> <span class="se">\ </span><span class="c1"># define the region you are interested in</span>
       -R testFiles/genes.bed <span class="se">\</span>
       -S testFiles/log2ratio_H3K4Me3_chr19.bw  <span class="se">\</span>
       --skipZeros <span class="se">\</span>
       -o matrix1_H3K4me3_l2r_TSS.gz <span class="se">\ </span><span class="c1"># to be used with plotHeatmap and plotProfile</span>
       --outFileSortedRegions regions1_H3K4me3_l2r_genes.bed
</pre></div>


<h4 id="722-multibamsummary">7.2.2 multiBamSummary</h4>
<p>multiBamSummary计算多个bam文件的基因组区域的read覆盖率,这个可以通过指定‘bins’模式来实现。你也可以通过‘BED-file’模式来计算指定区域的read覆盖度。它的标准输出是压缩后的numpy array(.npz). 可以使用plotCorrelation函数直接计算和可视化成对相关性。你还可以用plotPCA函数来对整个numpy文件进行主成分分析。我们并不推荐你只输入单个bw文件, 如果你只是想生成bedGraph文件的话, 倒是可以使用,但你要指定--outRawCounts选项.</p>
<ul>
<li>multiBamSummary bins</li>
</ul>
<div class="codehilite"><pre><span></span>multiBamSummary bins --bamfiles file1.bam file2.bam ... -o results.npz

<span class="c1"># --bamfiles            你输入的bam文件,用空格隔开</span>
<span class="c1"># -o                    输出文件名, read覆盖度矩阵. .npz文件</span>
<span class="c1"># -l,--labels           指定输入文件的标签,应该是在覆盖度矩阵里作为列名</span>
<span class="c1"># --smartLabels         自动指定标签, 使用文件名(basename filename)</span>
<span class="c1"># -r,--region           指定计算覆盖度的区域.形式为-region chr10, -region chr10:456700:891000</span>
<span class="c1"># --ignoreDuplicates    忽略重复的reads</span>
<span class="c1"># --outRawCounts        把counts per region以\t分隔的形式写入文件</span>
<span class="c1"># -e,--extendReads      当read的长度不足,允许read延伸到与fragment一致的长度.</span>
                        不推荐在含有spliced-read的数据里使用, 如RNA-seq。
<span class="c1"># --minMappingQuality   最低的比对质量分数,低于此的reads不进行计算</span>
<span class="c1"># --centerReads         指定reads要覆盖到fragment长度的中心</span>
<span class="c1"># 其他设定fragment长短的选项见handbook</span>
</pre></div>


<ul>
<li>multiBamSummary BED-file </li>
</ul>
<div class="codehilite"><pre><span></span>multiBamSummary BED-file --BED selection.bed --bamfiles file1.bam file2.bam -o results.npz

<span class="c1"># -b,--bamfiles             索引后的bam文件,以空格分隔</span>
<span class="c1"># -o                        生成覆盖度的矩阵文件名</span>
<span class="c1"># --BED                     指定计算覆盖度的区域文件</span>
</pre></div>
		</content>

	</main>
	<footer>
		<p> Powered by xizhihui </p>
		<p> Thanks to <a href="https://www.hifreud.com/">hifreud</a> the theme is learned from his site. </p>
	</footer>

<script type="text/javascript" src="/js/index.json"></script>
<script type="text/javascript" src="/js/category.json"></script>
<script type="text/javascript" src="/js/tags.json"></script>
<script type="text/javascript" src="/js/about.json"></script>
<script type="text/javascript" src="/js/index.js"></script>
<script type="text/javascript">
	// listen to click event on navigator in header
	// document
	// 	.querySelector("div#head_right ul")
	// 	.addEventListener("click", change_page)
	var table = [['h2', 'chip-seq', 'ChIP-seq'], ['h3', '11', '1.1 抗体的种类'], ['h3', '12', '1.2 实验结果简要展示'], ['h3', '13', '1.3 富集的类型'], ['h3', '14', '1.4. 表征结果的形式'], ['h2', 'step1', '二、step1: 数据预处理'], ['h3', '21-chip', '2.1 ChIP文库'], ['h3', '22', '2.2 质控'], ['h4', 'what-about-deduplication', 'what about deduplication?为何不直接去掉所有的重复？'], ['h2', 'step2peak-calling', '三、step2：peak calling'], ['h3', '31-peak-callers-workflow', '3.1 Peak callers workflow'], ['h3', '32-peak-calling', '3.2 Peak calling 失败的原因'], ['h3', '33', '3.3 结果可靠吗？'], ['h3', '34', '3.4 下游分析'], ['h2', 'step3peak-data', '四、step3：探索peak data'], ['h3', '41-peak', '4.1 可视化peak在序列上的分布情况'], ['h4', 'igvenrichment', 'IGV查看enrichment'], ['h3', '42', '4.2 检查对照'], ['h3', '43-compare-samples', '4.3 compare samples'], ['h3', '44-enrichmentpeaktrend-plots', '4.4 把enrichment/peak同序列特征相联系：trend plots'], ['h2', 'macs2peak-calling', '五、使用MACS2进行peak calling'], ['h2', 'chipseeker', '六、使用ChIPseeker进行可视化'], ['h3', '60-bed', '6.0 BED 文件格式'], ['h3', '61', '6.1 包和文件'], ['h3', '62-peakpeaks-coverage-plot', '6.2 可视化peak在基因组上的富集区域(peaks coverage plot)'], ['h3', '63-tsspeaks', '6.3 结合到TSS区域的peaks分析'], ['h4', '631-peaks', '6.3.1 peaks热图分析'], ['h4', '632-peaksread-count-frequency', '6.3.2 peaks的平均read count frequency'], ['h3', '64', '6.4 注释及其可视化'], ['h4', '641-peaks', '6.4.1 peaks分布区域的饼图, 柱形图'], ['h4', '642-peakbinding-sitetss', '6.4.2 peak(binding site)与最近基因的TSS距离'], ['h3', '65', '6.5 进行功能富集分析'], ['h3', '66-peak-data-set-peak', '6.6 比较不同的peak data set (多个peak结果一起分析咯)'], ['h4', '661-tss-regionpeak', '6.6.1 结合到TSS region的peak分析: 频率和热图'], ['h4', '662-peak', '6.6.2 peak注释比较'], ['h4', '663', '6.6.3 功能富集比较'], ['h4', '664-peak', '6.6.4 peak和注释后基因的重叠'], ['h3', '67-chip-seq-peak', '6.7 ChIP-seq peak结果重叠的统计分析'], ['h3', '68-geochip-seq', '6.8 GEO中ChIP-seq数据挖掘'], ['h2', 'deeptools', '七、使用deeptools进行可视化'], ['h3', '71', '7.1 质控和数据处理'], ['h3', '72', '7.2 函数的使用'], ['h4', '721-computematrix', '7.2.1 computeMatrix'], ['h4', '722-multibamsummary', '7.2.2 multiBamSummary']]
	var html = format_article_table(table)
	var table_ele = document.querySelector("main #table")
	table_ele.innerHTML = html
	// window.onload = evt => resize_nav(table_ele)
	adaption(document, window)

</script>
</body>
</html>