<!DOCTYPE HTML>
<html>
<head><meta name="generator" content="Hexo 3.8.0">
  <meta charset="utf-8">
  
  <title>组装或定量 stringTie | Xizhihui&#39;s Notes</title>
  <meta name="author" content="Xizhihui">
  
  <meta name="description" content="stringTie 是用于 RNA-seq 的转录本组装和定量软件">
  
  
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">

  <meta property="og:title" content="组装或定量 stringTie">
  <meta property="og:site_name" content="Xizhihui&#39;s Notes">

  
    <meta property="og:image" content="">
  

  
  
    <link href="/favicon.png" rel="icon">
  

  <!-- CSS -->
  <link rel="stylesheet" href="/css/themes/cosmo.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/font-awesome.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/style.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/responsive.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight-default.min.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/google-fonts.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/comment.css" media="screen" type="text/css">
  <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
	<script src="https://cdnjs.cloudflare.com/ajax/libs/es5-shim/4.5.9/es5-shim.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/es5-shim/4.5.7/es5-sham.min.js"></script>
  <![endif]-->

  <script src="/js/jquery-2.0.3.min.js"></script>
  
    <script src="/js/marked.js"></script>
    <script src="/js/comment.js"></script>
    <script src="/js/timeago.min.js"></script>
    <script src="/js/highlight.min.js"></script>
	<script src="/js/spin.min.js"></script>
  
  <!-- analytics -->
  



</head>
</html>
 <body>  
  <nav id="main-nav" class="navbar  navbar-default navbar-fixed-top" role="navigation">
    <div class="container">
      <button type="button" class="navbar-header navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
	<span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
       <a class="navbar-brand" href="/">Xizhihui&#39;s Notes</a>
      <div class="collapse navbar-collapse nav-menu">
		<ul class="nav navbar-nav">
		  
		  <li>
			<a href="/archives" title="All the articles.">
			  <i class="fa fa-archive"></i>Archives
			</a>
		  </li>
		  
		  <li>
			<a href="/categories" title="All the categories.">
			  <i class="fa fa-folder"></i>Categories
			</a>
		  </li>
		  
		  <li>
			<a href="/tags" title="All the tags.">
			  <i class="fa fa-tags"></i>Tags
			</a>
		  </li>
		  
		  <li>
			<a href="/about" title="About me.">
			  <i class="fa fa-user"></i>About
			</a>
		  </li>
		  
		</ul>
      </div>
    </div> <!-- container -->
</nav>
<div class="clearfix"></div>

  <div class="container">
  	<div class="content">
    	 


	
		<div class="page-header ">		
			<h1 class="title "> 组装或定量 stringTie</h1>
		</div>		
	






<div class="row post">
	<!-- cols -->
	
	<div id="top_meta"></div>
	<div class="col-md-9">
	

	<!-- content -->
	<div class="mypage">		
	  		

	  <p>stringTie 是用于 RNA-seq 的转录本组装和定量软件</p>
<a id="more"></a>
<h2 id="基本使用"><a href="#基本使用" class="headerlink" title="基本使用"></a>基本使用</h2><p>输入文件是BAM格式的比对结果文件，该文件必需经过排序，排序的方式基因组位置。这些文件可以是来源于Tophat比对的结果文件accepted_hits.bam，也可以是hisat2的结果文件经过转换和排序的文件(使用samtools)。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br></pre></td><td class="code"><pre><span class="line">stringtie &lt;aligned_reads.bam&gt; [options]*</span><br><span class="line"></span><br><span class="line"><span class="comment"># 选项</span></span><br><span class="line">-o [&lt;path/&gt;]&lt;out.gtf&gt; 			设置输出文件;组装的结果将写入out.gtf</span><br><span class="line">-p &lt;int&gt;						指定threads</span><br><span class="line">-G &lt;ref_ann.gff&gt;				指定参考注释文件</span><br><span class="line">--rf 							说明是fr-firststrand的文库</span><br><span class="line">--fr 							说明是fr-secondstrand的文库</span><br><span class="line">-l &lt;label&gt;						设定输出转录本的前缀,默认是STRG</span><br><span class="line">-m &lt;int&gt; 						最短的预测转录本长度</span><br><span class="line">-A &lt;gene_abund.tab&gt;				将基因表达量输出到gene_abund.tab文件(tab delimited format)</span><br><span class="line">-C &lt;cov_refs.gtf&gt;				输出reads能完全覆盖的参考序列区域的所有转录本</span><br><span class="line">-a &lt;int&gt;						没有剪切后reads比对上的junction最短长度</span><br><span class="line">-j &lt;<span class="built_in">float</span>&gt;						可比对到junction的剪切reads的最少个数</span><br><span class="line">-t 								禁止在组装转录本的两端进行trim</span><br><span class="line">-c &lt;<span class="built_in">float</span>&gt; 						设置进行转录本预测的最小<span class="built_in">read</span>覆盖度</span><br><span class="line">-g &lt;int&gt;						locus的最小间隔值,如果reads比对区域距离小于该值,则会被合并</span><br><span class="line">-B 								允许生成Ballgown的输入文件</span><br><span class="line">-b &lt;path&gt; 						指定-B生成文件的存储路径</span><br><span class="line">-e 								指定只进行比对文件的定量和输出能比对到参考转录本的组装转录本</span><br><span class="line">-M &lt;0.0-1.0&gt;					设定多位置比对reads在给定位置序列的最大占比</span><br><span class="line">-x &lt;seqid_list&gt;					忽略这里设置的参考序列区域,可以是逗号分隔的染色体. -x <span class="string">'chrM,chrX,chrY'</span></span><br><span class="line">--merge							指定使用转录合并模式,在本模式中，将使用GTF/GFF文件作为输入，然后把这些转录合并成非冗余的转录集合。</span><br><span class="line">								主要是在多样本的RNA-seq数据结果中使用.</span><br><span class="line">								本模式下有效的选项:-G, -o, -c,-m,-F,-T,-f,-i,-l</span><br></pre></td></tr></table></figure>
<blockquote>
<p>使用hisat2结果作为输入时, 首先在比对时要指定–dta选项,其次需要进行排序</p>
</blockquote>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">samtools view -Su alns.sam | samtools sort - alns.sorted</span><br></pre></td></tr></table></figure>
<h2 id="输出文件"><a href="#输出文件" class="headerlink" title="输出文件"></a>输出文件</h2><p>主要的输出:</p>
<ul>
<li>.GTF文件, 组装的转录本</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#seqname source      feature     start   end     score   strand  frame attributes</span></span><br><span class="line"><span class="comment">#chrX    StringTie   transcript  281394  303355  1000    +       .     gene_id "ERR188044.1"; transcript_id "ERR188044.1.1"; reference_id "NM_018390"; ref_gene_id "NM_018390";ref_gene_name "PLCXD1"; cov "101.256691"; FPKM "530.078918"; TPM "705.667908";</span></span><br><span class="line"><span class="comment">#chrX    StringTie   exon        281394  281684  1000    +       .     gene_id "ERR188044.1"; transcript_id "ERR188044.1.1"; exon_number "1"; reference_id "NM_018390";ref_gene_id "NM_018390"; ref_gene_name "PLCXD1"; cov "116.270836";</span></span><br></pre></td></tr></table></figure>
<ul>
<li>.tab文件, 基因表达量</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#Gene ID     Gene Name   Reference   Strand  Start   End     Coverage    FPKM        TPM</span></span><br><span class="line"><span class="comment">#NM_000451   SHOX        chrX        +       624344  646823  0.000000    0.000000    0.000000</span></span><br><span class="line"><span class="comment">#NM_006883   SHOX        chrX        +       624344  659411  0.000000    0.000000    0.000000</span></span><br></pre></td></tr></table></figure>
<ul>
<li><p>.GTF文件, 能与参考注释匹配的完全覆盖转录<br>同组装的转录本</p>
</li>
<li><p>用于Ballgown进行下游差异分析的文件<br>总共有5个文件, e2t.ctab, e_data.ctab, i2t.ctab, i_data.ctab, t_data.ctab</p>
</li>
<li><p>在merge mode, merged_gtf文件<br>如果stringtie在merge mode下运行的话，会将输入的一系列GTF/GFF文件合并组装成无冗余的转录本。输出的文件只是包含转录本，没有其他的数据如coverage，FPKM，TPM。stringtie可以用这个新的转录本重新计算表达量，但是你得设置-e参数。如果你要寻找新的转录本或者寻找转录本的来源，你可以使用gffcompare软件来实现。</p>
</li>
</ul>
<h2 id="结合Ballgown进行差异分析"><a href="#结合Ballgown进行差异分析" class="headerlink" title="结合Ballgown进行差异分析"></a>结合Ballgown进行差异分析</h2><h3 id="完全的差异分析"><a href="#完全的差异分析" class="headerlink" title="完全的差异分析"></a>完全的差异分析</h3><p><img src="https://ccb.jhu.edu/software/stringtie/DE_pipeline.png" alt="差异分析"></p>
<p>推荐的流程:</p>
<ol>
<li><p>对每个样本, 使用hisat2 –dta选项与参考基因组进行比对。<br>在比对时，我们强烈推荐加入参考注释信息。这可以通过hisat2 –ss/–exon实现,也可以通过hisat2的–know-splicesite-infile实现。值得注意的是，你一定要对输出的比对文件进行排序和转换成BAM文件。</p>
</li>
<li><p>对得到的每个样本的比对文件，用strintie进行组装。<br>我们推荐使用-G参数提供参考注释(如果有的话.)</p>
</li>
<li><p>用生成组装文件，使用stringtie –merge生成无冗余的转录本<br>同样的，如果有参考注释，也推荐使用</p>
</li>
<li><p>对每个样本的比对文件, 使用stringtie -B/-b, -G和-e选项计算表达量和生成Ballgown所需的table文件。<br>-G选项指定第三步生成无冗余转录本，这是唯一没有使用参考注释的情况。虽然说这里的-e选项不是必需的，但是使用它可以得到一个更准确的结果。</p>
</li>
<li><p>使用Ballgown进行差异分析</p>
</li>
</ol>
<h3 id="简化的差异分析"><a href="#简化的差异分析" class="headerlink" title="简化的差异分析"></a>简化的差异分析</h3><p><img src="https://ccb.jhu.edu/software/stringtie/DE_pipeline_refonly.png" alt="简化的差异分析"></p>
<hr>
<h2 id="stringtie结合DESeq2和edgeR的差异分析"><a href="#stringtie结合DESeq2和edgeR的差异分析" class="headerlink" title="stringtie结合DESeq2和edgeR的差异分析"></a>stringtie结合DESeq2和edgeR的差异分析</h2><ol>
<li>使用stringtie -e得到定量结果</li>
<li>使用<a href="https://ccb.jhu.edu/software/stringtie/dl/prepDE.py" target="_blank" rel="noopener">“prepDE.py”</a>提取定量结果生成counts矩阵</li>
</ol>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># prepDE.py [options]</span></span><br><span class="line"><span class="comment"># -i &lt;input&gt;	 	指定gtfs文件的路径和样品ID的txt文件</span></span><br><span class="line"><span class="comment"># -g G 			指定gene count matrix的输出路径</span></span><br><span class="line"><span class="comment"># -t T 			指定transcripts count matrix的输出路径</span></span><br><span class="line"><span class="comment"># -l length 		指定平均read长度</span></span><br><span class="line"><span class="comment"># -p pattern		选择样品子路径的正则表达式</span></span><br><span class="line"><span class="comment"># -c 			是否合并重复的gene,虽然它们有不同的gene ID</span></span><br><span class="line"><span class="comment"># -s string 		指定stringtie添加geneID前缀的字符串,默认为MSTRG</span></span><br><span class="line"><span class="comment"># -k Key 		如果-c指定,则指定本脚本添加到gneIDs的前缀, 默认为prepG</span></span><br><span class="line"><span class="comment"># --legend=Legend 如果-c指定,转录比对到geneIDs的表头文件的输出路径,默认为legend.csv</span></span><br><span class="line"></span><br><span class="line"><span class="comment">#prepDE.py -i sample_lst.txt</span></span><br><span class="line"></span><br><span class="line"> <span class="comment">## sample_lst.txt</span></span><br><span class="line"> <span class="comment">## ERR188021 &lt;PATH_TO_ERR188021.gtf&gt;</span></span><br><span class="line"> <span class="comment">## ERR188023 &lt;PATH_TO_ERR188023.gtf&gt;</span></span><br><span class="line"> <span class="comment">## ERR188024 &lt;PATH_TO_ERR188024.gtf&gt;</span></span><br></pre></td></tr></table></figure>
<ol start="3">
<li>使用DESeq2进行差异分析</li>
</ol>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">countData &lt;- as.matrix(read.csv(<span class="string">'gene_count_matrix.csv'</span>, row.names=<span class="string">'gene_id'</span>))</span><br><span class="line">colData &lt;- read.csv(PHENO_DATA, sep=<span class="string">'\t'</span>, row.names=<span class="number">1</span>) 	<span class="comment"># 这里是你的表型文件</span></span><br><span class="line">all(rownames(colData) %<span class="keyword">in</span>% colnames(countData))			<span class="comment"># 检测是否一致</span></span><br><span class="line">countData &lt;- countData[, rownames(colData)]</span><br><span class="line">all(rownames(colData) == colnames(countData))</span><br><span class="line"></span><br><span class="line">dds &lt;- DESeqDataSetFromMatrix(countData = countData,</span><br><span class="line">							  colData = colData,</span><br><span class="line">							  design=~CHOOSE_FEATURE)</span><br><span class="line">dds &lt;- DESeq(dds)</span><br><span class="line">res &lt;- results(dds)</span><br></pre></td></tr></table></figure>
<hr>
<h2 id="组装super-reads"><a href="#组装super-reads" class="headerlink" title="组装super-reads"></a>组装super-reads</h2><p>stringtie还可以把短的transcripts组装成长的contigs，我们把这个叫做super-reads。这一步可以省略，但我们还是推荐使用这一步，因为他可以提高转录本组装的正确性。不过你需要安装MaSuRCA genome assembler包。</p>
<p>1.super-reads的生成</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">superreads.pl reads_1.fastq reads_2.fastq &lt;masurca_path&gt; [options]*</span><br><span class="line"></span><br><span class="line"><span class="comment"># 选项</span></span><br><span class="line">-t &lt;num_threads&gt;</span><br><span class="line">-j &lt;jf_size&gt; 			MasuRCA需要运行Jellyfish,这里设置后者使用的<span class="built_in">hash</span> size</span><br><span class="line">-s &lt;step&gt;				打印运行成功的步骤</span><br><span class="line">-r &lt;paired_read_prefix&gt; 设置paired <span class="built_in">read</span>的前缀</span><br><span class="line">-f &lt;fragment_size&gt;		设置mean library insert length</span><br><span class="line">-d &lt;stdev&gt;				设置insert length的标准偏差</span><br><span class="line">-l &lt;name&gt; 				设置组装的super-reads文件名</span><br><span class="line">-u &lt;prefix&gt;				设置为组装的reads的文件名</span><br></pre></td></tr></table></figure>
<ol start="2">
<li>super-reads比对到参考基因组上<br>可以使用你喜欢的比对软件，如tophat，hisat，bowtie等。比如：</li>
</ol>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">tophat [options]* &lt;genome_index_base&gt; PE_reads_1.notAssembled.fq.gz,super_reads.fq PE_reads_2.notAssembled.fq.gz</span><br></pre></td></tr></table></figure>
	  
	</div>

	<!-- recommended posts -->
	

	<div>
  	<center>
	<div class="pagination">
<ul class="pagination">
	 
				
    	<li class="prev"><a href="/2018/10/12/芯片专题-AffyMetrix/" class="alignleft prev"><i class="fa fa-arrow-circle-o-left"></i>Prev</a></li>
  		

        <li><a href="/archives"><i class="fa fa-archive"></i>Archive</a></li>

		
		   <li class="next"><a href="/2018/10/12/质控软件-raw_data质控/" class="alignright next">Next<i class="fa fa-arrow-circle-o-right"></i></a></li>         
        
	
</ul>
</div>

    </center>
	</div>

    <!-- share -->
    
        
    <div class="bdsharebuttonbox">
        <a href="#" class="bds_more" data-cmd="more"></a>
        <a href="#" class="bds_weixin" data-cmd="weixin" title="分享到微信"></a>
        <a href="#" class="bds_tsina" data-cmd="tsina" title="分享到新浪微博"></a>
        <a href="#" class="bds_fbook" data-cmd="fbook" title="分享到Facebook"></a>
        <a href="#" class="bds_twi" data-cmd="twi" title="分享到Twitter"></a>
        <a href="#" class="bds_linkedin" data-cmd="linkedin" title="分享到linkedin"></a>
        <a href="#" class="bds_evernotecn" data-cmd="evernotecn" title="分享到印象笔记"></a>
        <a href="#" class="bds_youdao" data-cmd="youdao" title="分享到有道云笔记"></a>
        <a href="#" class="bds_copy" data-cmd="copy" title="分享到复制网址"></a>
    </div>
    <script>
        window._bd_share_config={"common":{"bdSnsKey":{},"bdText":"","bdMini":"2","bdMiniList":false,"bdPic":"","bdStyle":"1","bdSize":"24"},"share":{}};
        with(document)0[(getElementsByTagName('head')[0]||body).appendChild(createElement('script')).src='http://bdimg.share.baidu.com/static/api/js/share.js?v=89860593.js?cdnversion='+~(-new Date()/36e5)];
    </script>


        

    
	
	<!-- comment -->
	
<section id="comment">
  <h2 class="title">Comments</h2>
    	 
	 <div id="comment-thread"></div>
	 <div id="loading-spin"></div>
	 <script type="text/javascript">
	   getComments({
           type: "github" ? "github" : "github",       
	       user: "xizhihui",
	       repo: "",
		   client_id: "",
           client_secret: "",
		   no_comment: "No comments yet. Press the button and go to comment now!",
		   go_to_comment: "Go to comment",
		   no_issue: "no_issue",
		   issue_title: "组装或定量 stringTie",
		   issue_id: "",
		   btn_class: "btn btn-primary",
		   comments_target: "#comment-thread",
		   loading_target: "#loading_spin"
		   });
	 </script>
  
</section>


	</div> <!-- col-md-9/col-md-12 -->
		
	
	<div id="side_meta">
		<div class="col-md-3" id="post_meta"> 

	<!-- date -->
	
	<div class="meta-widget">
	<i class="fa fa-clock-o"></i>
	2018-10-12 
	</div>
	

	<!-- categories -->
    
	<div class="meta-widget">
	<a data-toggle="collapse" data-target="#categorys"><i class="fa fa-folder"></i></a>	
    <ul id="categorys" class="tag_box list-unstyled collapse in">
          
  <li>
    </li><li><a href="/categories/Bioinformatics/">Bioinformatics<span>36</span></a></li>
  

    </ul>
	</div>
	

	<!-- tags -->
	
	<div class="meta-widget">
	<a data-toggle="collapse" data-target="#tags"><i class="fa fa-tags"></i></a>		  
    <ul id="tags" class="tag_box list-unstyled collapse in">	  
	    
  <li><a href="/tags/软件和包/">软件和包<span>22</span></a></li> <li><a href="/tags/定量/">定量<span>3</span></a></li> <li><a href="/tags/组装/">组装<span>1</span></a></li>
    </ul>
	</div>
	

	<!-- toc -->
	<div class="meta-widget">
	
	   <a data-toggle="collapse" data-target="#toc"><i class="fa fa-bars"></i></a>
	   <div id="toc" class="toc collapse in">
			<ol class="toc-article"><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#基本使用"><span class="toc-article-text">基本使用</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#输出文件"><span class="toc-article-text">输出文件</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#结合Ballgown进行差异分析"><span class="toc-article-text">结合Ballgown进行差异分析</span></a><ol class="toc-article-child"><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#完全的差异分析"><span class="toc-article-text">完全的差异分析</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#简化的差异分析"><span class="toc-article-text">简化的差异分析</span></a></li></ol></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#stringtie结合DESeq2和edgeR的差异分析"><span class="toc-article-text">stringtie结合DESeq2和edgeR的差异分析</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#组装super-reads"><span class="toc-article-text">组装super-reads</span></a></li></ol>
		</div>
	
	</div>
	
    <hr>
	
</div><!-- col-md-3 -->

	</div>
		

</div><!-- row -->



	</div>
  </div>
  <div class="container-narrow">
  <footer> <p>
  &copy; 2018 Xizhihui
  
      with help from <a href="http://hexo.io/" target="_blank">Hexo</a> and <a href="http://getbootstrap.com/" target="_blank">Twitter Bootstrap</a>. Theme by <a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind</a>.    
</p> </footer>
</div> <!-- container-narrow -->
  


  
<a id="gotop" href="#">   
  <span>▲</span> 
</a>

<script src="/js/jquery.imagesloaded.min.js"></script>
<script src="/js/gallery.js"></script>
<script src="/js/bootstrap.min.js"></script>
<script src="/js/main.js"></script>
<script src="/js/search.js"></script> 


<link rel="stylesheet" href="/fancybox/jquery.fancybox.css" media="screen" type="text/css">
<script src="/fancybox/jquery.fancybox.pack.js"></script>
<script type="text/javascript">
(function($){
  $('.fancybox').fancybox();
})(jQuery);
</script>



   <script type="text/javascript">      
     var search_path = "search.xml";
	 if (search_path.length == 0) {
	 	search_path = "search.xml";
	 }
	 var path = "/" + search_path;
     searchFunc(path, 'local-search-input', 'local-search-result');
   </script>


<!-- syntax highlighting -->

  <script>
  marked.setOptions({
    highlight: function (code, lang) {
        return hljs.highlightAuto(code).value;
    }
  });
  function Highlighting(){
    var markdowns = document.getElementsByClassName('markdown');
    for(var i=0;i<markdowns.length;i++){
        if(markdowns[i].innerHTML) markdowns[i].innerHTML =marked(markdowns[i].innerHTML);
    }
  }
  window.addEventListener('DOMContentLoaded', Highlighting, false);
  window.addEventListener('load', Highlighting, false);
  </script>


</body>
   </html>
