<!DOCTYPE html>
<html>
<head>
	<title>XZH's Notes</title>
	<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
	<meta name="author" content="xizhihui" />
	<meta name="viewport" content="width=device-width, initial-scale=1" />
	<link rel="stylesheet" type="text/css" href="/statics/reset.css">
	<link rel="stylesheet" type="text/css" href="/statics/header_footer.css"></link>
	<link rel="stylesheet" type="text/css" href="/statics/default.css">
	<link rel="stylesheet" type="text/css" href="/statics/main.css">
</head>
<body>
	<header class="Header">
		<nav>
			<h3>XZH</h3>
			<ul>
				<li><a href="/index.html#Timeline">Timeline</a></li><li><a href="/index.html#Category">Category</a></li><li><a href="/index.html#Tags">Tags</a></li><li><a href="/index.html#About">About</a></li>
			</ul>
		</nav>
	</header>
	<div class="Motto">
		<h2>XZH's Notes</h2>
		<h4>The future comes for you ~ ~ ~</h4>
	</div>

	<main>
		<nav id="table"></nav>
		<content id="content" class="content">
			<h2 id="deseq2">DESeq2</h2>
<blockquote>
<p>与DESeq类似的包有：edgeR、limma、DSS、EBSeq、baySeq</p>
</blockquote>
<h2 id="_1">一、核心逻辑代码</h2>
<div class="codehilite"><pre><span></span><span class="c1"># countData: 以样品为列名,基因为行名的表达矩阵</span>
<span class="c1"># colData的形式: 以样品作为行名,列是样品对应的分组类型</span>
<span class="c1"># 生成count matrix</span>
dds <span class="o">&lt;-</span> DESeqDataSetFromMatrix<span class="p">(</span>countData <span class="o">=</span> cts<span class="p">,</span>
                              colData <span class="o">=</span> coldata<span class="p">,</span>
                              design <span class="o">=</span> <span class="o">~</span> batch <span class="o">+</span> condition<span class="p">)</span>
<span class="c1"># 生成DESeq数据集</span>
dds <span class="o">&lt;-</span> DESeq<span class="p">(</span>dds<span class="p">)</span>
<span class="c1"># 进行比较，获得结果</span>
res <span class="o">&lt;-</span> results<span class="p">(</span>dds<span class="p">,</span> contrast<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="s">&#39;condition&#39;</span><span class="p">,</span> <span class="s">&#39;treat&#39;</span><span class="p">,</span> <span class="s">&#39;ctrl&#39;</span><span class="p">))</span>
resultsNames<span class="p">(</span>dds<span class="p">)</span>
res <span class="o">&lt;-</span> lfcShrink<span class="p">(</span>dds<span class="p">,</span> coef<span class="o">=</span><span class="m">2</span><span class="p">)</span>

<span class="c1">#  DESeqDataSetFromTximport:   由Salmon, Saifish, kallisto生成</span>
<span class="c1">#  DESeqDataSetFromHTSeq:      由htseq-counts生成</span>
<span class="c1">#  DESeqDataSet:               由RangedSummarizedExpriment生成</span>
</pre></div>


<p>DESeq2为count数据提供两类变换方法，使不同均值的方差趋于稳定，rlog和vst，这两个函数可以用于处理含有色散平均趋势的负二项数据（类如RNA-seq）。由于rlog计算量很大，与vst的效果相近，那么，在数据集小于30的时候，使用rlog；大数据集使用vst可以加快速度。但做这样的变换不是用于差异分析，而是用于pca，WGCNA，clustering等与聚类相关的分析才用得到。</p>
<h2 id="standard-workflow">二、标准流程 standard workflow</h2>
<h3 id="21-input-data">2.1 input data</h3>
<ul>
<li>DESeq的input data是不需要预先进行标准化的，因为软件包内部会自己根据文库大小等进行标准化。  </li>
<li>input data有四类，分别对应四个读取函数，见核心逻辑代码部分。</li>
<li>DESeqDataSet是包中存取read counts和统计中间值的对象，通常简写为dds。其参数必需要有一个design formula，以被后续估计dispersion和log2 fold changes使用。</li>
</ul>
<h3 id="22-deseqdatasetfromtximporttxi">2.2 DESeqDataSetFromTximport：.txi文件</h3>
<blockquote>
<p>这类数据是由Salmon、Saifish、kallisto、RSEM等软件产生，数据应该是estimated gene counts。<br />
    使用这些软件的好处在于：<br />
        1. 它们会修正样品间gene length的可能改变。<br />
        2. 它们使用的速度更快、占用CPU内存更少（相比于基于比对的软件来讲）。<br />
        3. 它们可以避免丢弃多匹配片段（基因同源序列），提高灵敏度。</p>
</blockquote>
<h3 id="23-deseqdatasetfrommatrix-count-matrix">2.3 使用DESeqDataSetFromMatrix: count matrix文件</h3>
<blockquote>
<p>这类数据来源于Rsubread包的featureCounts函数。使用DESeqDataSetFromMatrix，需要提供counts matrix、数据框格式的样品信息、design formula。</p>
</blockquote>
<h3 id="24-input-htseq-count">2.4 input: htseq-count</h3>
<blockquote>
<p>具体的使用可见前面，以pasilla来源数据为例</p>
</blockquote>
<h3 id="25-input-summarizedexperiment">2.5 input: SummarizedExperiment</h3>
<div class="codehilite"><pre><span></span><span class="o">//</span> 使用airway包来源的数据进行示例
<span class="kn">library</span><span class="p">(</span><span class="s">&#39;airway&#39;</span><span class="p">)</span>
data<span class="p">(</span><span class="s">&#39;airway&#39;</span><span class="p">)</span>
se <span class="o">&lt;-</span> airway

<span class="kn">library</span><span class="p">(</span><span class="s">&#39;DESeq2&#39;</span><span class="p">)</span>
ddsSE <span class="o">&lt;-</span> DESeqDataSet<span class="p">(</span>se<span class="p">,</span> design <span class="o">=</span> <span class="o">~</span>cell <span class="o">+</span> dex<span class="p">)</span>
</pre></div>


<h3 id="26">2.6 预过滤</h3>
<div class="codehilite"><pre><span></span><span class="c1"># 为了减少内存占用，先把count数很低的数据剔除</span>
keep <span class="o">&lt;-</span> <span class="kp">rowSums</span><span class="p">(</span>counts<span class="p">(</span>dds<span class="p">))</span> <span class="o">&gt;=</span> <span class="m">10</span>
dds <span class="o">&lt;-</span> dds<span class="p">[</span>keep<span class="p">,]</span>
</pre></div>


<h3 id="27-factor-levels">2.7 指定factor levels</h3>
<blockquote>
<p>R 将基于字母顺序默认参考水平，但实际通常是根据对照组作为参考水平。因此有必要时要设置</p>
</blockquote>
<div class="codehilite"><pre><span></span>dds<span class="o">$</span>condition <span class="o">&lt;-</span> <span class="kp">factor</span><span class="p">(</span>dds<span class="o">$</span>condition<span class="p">,</span> levels<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="s">&#39;untreated&#39;</span><span class="p">,</span> <span class="s">&#39;treated&#39;</span><span class="p">)</span>
<span class="c1"># 也可以直接指定</span>
dds<span class="o">$</span>condition <span class="o">&lt;-</span> relevel<span class="p">(</span>dds<span class="o">$</span>condition<span class="p">,</span> ref<span class="o">=</span><span class="s">&#39;untreated&#39;</span><span class="p">)</span>
<span class="c1"># 如果有时对dds取子集时，导致某些水平不含数据，那么这个水平就可以丢弃</span>
dds<span class="o">$</span>condition <span class="o">&lt;-</span> <span class="kp">droplevels</span><span class="p">(</span>dds<span class="o">$</span>condition<span class="p">)</span>
</pre></div>


<h3 id="28-collapsereplicates">2.8 合并技术性重复：collapseReplicates函数</h3>
<h2 id="_2">三、差异表达分析</h2>
<blockquote>
<p>函数DESeq用来进行差异表达分析，所有计算都整合在该函数里面；生成一个结果对象。你需要使用results函数对该结果对象取值，才能获取到包含了log2FoldChange、p-value、p-adj-value等的结果。</p>
</blockquote>
<div class="codehilite"><pre><span></span>dds <span class="o">&lt;-</span> DESeq<span class="p">(</span>dds<span class="p">)</span>
res <span class="o">&lt;-</span> results<span class="p">(</span>dds<span class="p">)</span>
</pre></div>


<blockquote>
<p>如果你想修改coeffficient的数量的话，可以这样设置：</p>
</blockquote>
<div class="codehilite"><pre><span></span>res <span class="o">&lt;-</span> results<span class="p">(</span>dds<span class="p">,</span> contrast<span class="o">=</span><span class="s">&#39;2&#39;</span><span class="p">)</span>
<span class="c1"># 或者</span>
resFC <span class="o">&lt;-</span> lfcShrink<span class="p">(</span>dds<span class="p">,</span> coef<span class="o">=</span><span class="m">2</span><span class="p">)</span>
</pre></div>


<blockquote>
<p>如果有太多样本参与比较，可以并行计算</p>
</blockquote>
<div class="codehilite"><pre><span></span><span class="kn">library</span><span class="p">(</span><span class="s">&#39;BiocParallel&#39;</span><span class="p">)</span>
register<span class="p">(</span>MulticoreParam<span class="p">(</span><span class="m">4</span><span class="p">))</span>
dds <span class="o">&lt;-</span> DESeq<span class="p">(</span>dds<span class="p">)</span>
res <span class="o">&lt;-</span> results<span class="p">(</span>dds<span class="p">)</span>
resFc <span class="o">&lt;-</span> lfcShrink<span class="p">(</span>dds<span class="p">,</span> coef<span class="o">=</span><span class="m">2</span><span class="p">)</span>
</pre></div>


<blockquote>
<p>按照p-value排序结果、取摘要值、统计个数、调整p-value值</p>
</blockquote>
<div class="codehilite"><pre><span></span>resOrdered <span class="o">&lt;-</span> res<span class="p">[</span><span class="kp">order</span><span class="p">(</span>res<span class="o">$</span>pvalue<span class="p">),]</span>
<span class="kp">summary</span><span class="p">(</span>res<span class="p">)</span>
<span class="kp">sum</span><span class="p">(</span>res<span class="o">$</span>padj <span class="o">&lt;</span> <span class="m">0.1</span><span class="p">,</span> na.rm<span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
res0.05 <span class="o">&lt;-</span> results<span class="p">(</span>dds<span class="p">,</span> alpha<span class="o">=</span><span class="m">0.05</span><span class="p">)</span>
</pre></div>


<h2 id="_3">四、探索和导出结果</h2>
<h3 id="ma-plot">MA-plot</h3>
<p>收缩估计(shrinkage estimates)通过在回归时缩小对结果影响较小的系数值来达到改善结果的目的.有多种收缩方法可供选择,如apeglm, ashr, normal.前二者需要安装同名bioconductor包,后者是DESeq2自带的.</p>
<div class="codehilite"><pre><span></span>plotMA<span class="p">(</span>res<span class="p">,</span> ylim<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="m">-2</span><span class="p">,</span><span class="m">2</span><span class="p">))</span>
<span class="c1"># 使用lfcShrink来改变shrinkage estimators</span>
resApe <span class="o">&lt;-</span> lfcShrink<span class="p">(</span>dds<span class="p">,</span> coef<span class="o">=</span><span class="m">2</span><span class="p">,</span> type<span class="o">=</span><span class="s">&#39;apeglm&#39;</span><span class="p">)</span>
resAsh <span class="o">&lt;-</span> lfcShrink<span class="p">(</span>dds<span class="p">,</span> coef<span class="o">=</span><span class="m">2</span><span class="p">,</span> type<span class="o">=</span><span class="s">&#39;ashr&#39;</span><span class="p">)</span>
par<span class="p">(</span>mfrow<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="m">3</span><span class="p">),</span> mar<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="m">4</span><span class="p">,</span><span class="m">4</span><span class="p">,</span><span class="m">2</span><span class="p">,</span><span class="m">1</span><span class="p">))</span>
xlim <span class="o">&lt;-</span> <span class="kt">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="m">1e5</span><span class="p">);</span> ylim <span class="o">&lt;-</span> <span class="kt">c</span><span class="p">(</span><span class="m">-3</span><span class="p">,</span><span class="m">3</span><span class="p">)</span>   <span class="c1">#改变yx轴的值范围</span>
plotMA<span class="p">(</span>resLFC<span class="p">,</span> xlim<span class="o">=</span>xlim<span class="p">,</span> ylim<span class="o">=</span>ylim<span class="p">,</span> main<span class="o">=</span><span class="s">&quot;normal&quot;</span><span class="p">)</span>
plotMA<span class="p">(</span>resApe<span class="p">,</span> xlim<span class="o">=</span>xlim<span class="p">,</span> ylim<span class="o">=</span>ylim<span class="p">,</span> main<span class="o">=</span><span class="s">&quot;apeglm&quot;</span><span class="p">)</span>
plotMA<span class="p">(</span>resAsh<span class="p">,</span> xlim<span class="o">=</span>xlim<span class="p">,</span> ylim<span class="o">=</span>ylim<span class="p">,</span> main<span class="o">=</span><span class="s">&quot;ashr&quot;</span><span class="p">)</span>
</pre></div>


<h3 id="plot-counts">plot counts</h3>
<blockquote>
<p>计算一个基因的reads counts在各个处理组的情况</p>
</blockquote>
<div class="codehilite"><pre><span></span>plotCounts<span class="p">(</span>dds<span class="p">,</span> gene<span class="o">=</span><span class="kp">which.min</span><span class="p">(</span>res<span class="o">$</span>padj<span class="p">),</span> intgroup<span class="o">=</span><span class="s">&#39;condition&#39;</span><span class="p">)</span>
<span class="c1"># 输出数据框以供ggplot使用</span>
d <span class="o">&lt;-</span> plotCounts<span class="p">(</span>dds<span class="p">,</span> gene<span class="o">=</span><span class="kp">which.min</span><span class="p">(</span>res<span class="o">$</span>padj<span class="p">),</span>intgroup<span class="o">=</span><span class="s">&#39;condition&#39;</span><span class="p">,</span> returnData<span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
<span class="kn">library</span><span class="p">(</span><span class="s">&#39;ggplot2&#39;</span><span class="p">)</span>
ggplot<span class="p">(</span>d<span class="p">,</span> aes<span class="p">(</span>x<span class="o">=</span>condition<span class="p">,</span>y<span class="o">=</span>count<span class="p">))</span> <span class="o">+</span>
    geom_point<span class="p">(</span>position<span class="o">=</span>position_jitter<span class="p">(</span>w<span class="o">=</span><span class="m">0.1</span><span class="p">,</span>h<span class="o">=</span><span class="m">0</span><span class="p">))</span> <span class="o">+</span>
    scale_y_log10<span class="p">(</span>breaks<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="m">25</span><span class="p">,</span><span class="m">100</span><span class="p">,</span><span class="m">400</span><span class="p">))</span>
</pre></div>


<h3 id="_4">更多有关结果的信息</h3>
<blockquote>
<p>p-value是NA？原因有三：某行(row)所有样品都是0 counts；某行的样品值极端大或者小；自动独立过滤引起的低矫正后counts，p-adj会被设为NA。</p>
</blockquote>
<div class="codehilite"><pre><span></span>mcols<span class="p">(</span>res<span class="p">)</span><span class="o">$</span>description
</pre></div>


<h3 id="_5">富可视化、生成报告</h3>
<blockquote>
<p>可以使用Reporting Tools、regionReport、Glimma、pcaExplore输出结果</p>
</blockquote>
<div class="codehilite"><pre><span></span>write.csv<span class="p">(</span><span class="kp">as.data.frame</span><span class="p">(</span>resOrdered<span class="p">),</span> file<span class="o">=</span><span class="s">&#39;results.csv&#39;</span><span class="p">)</span>
resSig <span class="o">&lt;-</span> <span class="kp">subset</span><span class="p">(</span>resOrdered<span class="p">,</span> padj <span class="o">&lt;</span> <span class="m">0.1</span><span class="p">)</span>
write.csv<span class="p">(</span><span class="kp">as.data.frame</span><span class="p">(</span>resSig<span class="p">),</span> file<span class="o">=</span><span class="s">&#39;sig_results.csv&#39;</span><span class="p">)</span>
</pre></div>


<h3 id="_6">多因子设计</h3>
<blockquote>
<p>重点在于colData的生成和类似formula(~type+condition)的生成</p>
</blockquote>
<div class="codehilite"><pre><span></span><span class="c1">## DataFrame with 7 rows and 3 columns</span>
<span class="c1">##            condition        type sizeFactor</span>
<span class="c1">##             &lt;factor&gt;    &lt;factor&gt;  &lt;numeric&gt;</span>
<span class="c1">## treated1     treated single-read  1.6355014</span>
<span class="c1">## treated2     treated  paired-end  0.7612159</span>
<span class="c1">## treated3     treated  paired-end  0.8326603</span>
<span class="c1">## untreated1 untreated single-read  1.1383376</span>
<span class="c1">## untreated2 untreated single-read  1.7935406</span>
<span class="c1">## untreated3 untreated  paired-end  0.6494828</span>
<span class="c1">## untreated4 untreated  paired-end  0.7516005</span>
colData<span class="p">(</span>dds<span class="p">)</span> <span class="c1"># 存起备用 </span>
ddsMF <span class="o">&lt;-</span> dds 
<span class="kp">levels</span><span class="p">(</span>ddsMF<span class="o">$</span>type<span class="p">)</span>  <span class="c1"># 生成新的level </span>
<span class="c1"># re-run DESeq</span>
design<span class="p">(</span>ddsMF<span class="p">)</span> <span class="o">&lt;-</span> formula<span class="p">(</span><span class="o">~</span> type <span class="o">+</span> condition<span class="p">)</span> 
ddsMF <span class="o">&lt;-</span> DESeq<span class="p">(</span>ddsMF<span class="p">)</span>
resMF <span class="o">&lt;-</span> results<span class="p">(</span>ddsMF<span class="p">)</span>
<span class="c1"># 还可以这样干</span>
resMFType <span class="o">&lt;-</span> results<span class="p">(</span>ddsMF<span class="p">,</span> contrast<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="s">&#39;type&#39;</span><span class="p">,</span> <span class="s">&#39;single&#39;</span><span class="p">,</span> <span class="s">&#39;paired&#39;</span><span class="p">))</span>
</pre></div>


<h2 id="_7">五、通过聚类可视化表征数据质量</h2>
<h3 id="_8">表达矩阵的热图</h3>
<div class="codehilite"><pre><span></span><span class="c1"># ntd,vsd,rld分别是norm,vst,rlog变换后的数据</span>

<span class="kn">library</span><span class="p">(</span><span class="s">&#39;pheatmap&#39;</span><span class="p">)</span>
select <span class="o">&lt;-</span> <span class="kp">order</span><span class="p">(</span><span class="kp">rowMeans</span><span class="p">(</span>counts<span class="p">(</span>dds<span class="p">,</span>normalized<span class="o">=</span><span class="kc">TRUE</span><span class="p">)),</span> decreasing<span class="o">=</span><span class="kc">TRUE</span><span class="p">)[</span><span class="m">1</span><span class="o">:</span><span class="m">20</span><span class="p">]</span>
df <span class="o">&lt;-</span> <span class="kp">as.data.frame</span><span class="p">(</span>colData<span class="p">(</span>dds<span class="p">)[,</span><span class="kt">c</span><span class="p">(</span><span class="s">&#39;condition&#39;</span><span class="p">,</span> <span class="s">&#39;type&#39;</span><span class="p">)])</span>
pheatmap<span class="p">(</span>assay<span class="p">(</span>ntd<span class="p">)[</span>select<span class="p">,],</span> cluster_rows<span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span> show_rownames<span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span> cluster_cols<span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span> annotation_col<span class="o">=</span>df<span class="p">)</span>
pheatmap<span class="p">(</span>assay<span class="p">(</span>vsd<span class="p">)[</span>select<span class="p">,],</span> cluster_rows<span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span> show_rownames<span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span> cluster_cols<span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span> annotation_col<span class="o">=</span>df<span class="p">)</span>
pheatmap<span class="p">(</span>assay<span class="p">(</span>rld<span class="p">)[</span>select<span class="p">,],</span> cluster_rows<span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span> show_rownames<span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span> cluster_cols<span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span> annotation_col<span class="o">=</span>df<span class="p">)</span>
</pre></div>


<h3 id="_9">样品距离热图</h3>
<div class="codehilite"><pre><span></span><span class="o">//</span> 这类图直观地显示了样品间的相似度和差异性
sampleDists <span class="o">&lt;-</span> dist<span class="p">(</span><span class="kp">t</span><span class="p">(</span>assay<span class="p">(</span>vsd<span class="p">)))</span>

<span class="kn">library</span><span class="p">(</span><span class="s">&#39;RColorBrewer&#39;</span><span class="p">)</span>
sampleDistMatrix <span class="o">&lt;-</span> <span class="kp">as.matrix</span><span class="p">(</span>sampleDists<span class="p">)</span>
<span class="kp">rownames</span><span class="p">(</span>sampleDistMatrix<span class="p">)</span> <span class="o">&lt;-</span> <span class="kp">paste</span><span class="p">(</span>vsd<span class="o">$</span>condition<span class="p">,</span> vsd<span class="o">$</span>type<span class="p">,</span> sep<span class="o">=</span><span class="s">&#39;-&#39;</span><span class="p">)</span>
<span class="kp">colnames</span><span class="p">(</span>sampleDistMatrix<span class="p">)</span> <span class="o">&lt;-</span> <span class="kc">NULL</span>
colors <span class="o">&lt;-</span> colorRampPalette<span class="p">(</span> <span class="kp">rev</span><span class="p">(</span>brewer.pal<span class="p">(</span><span class="m">9</span><span class="p">,</span> <span class="s">&#39;Blues&#39;</span><span class="p">))</span> <span class="p">)(</span><span class="m">255</span><span class="p">)</span>
pheatmap<span class="p">(</span>sampleDistMatrix<span class="p">,</span>
        clustering_distance_rows<span class="o">=</span>sampleDists<span class="p">,</span>
        clustering_distance_cols<span class="o">=</span>sampleDists<span class="p">,</span>
        col<span class="o">=</span>colors<span class="p">)</span>
</pre></div>


<h3 id="_10">样品的主成分分析</h3>
<div class="codehilite"><pre><span></span>vsd <span class="o">=</span> vst<span class="p">(</span>dds<span class="p">)</span>
plotPCA<span class="p">(</span>vsd<span class="p">,</span> intgroup<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="s">&#39;condition&#39;</span><span class="p">,</span> <span class="s">&#39;type&#39;</span><span class="p">))</span>   <span class="c1"># 这里condition,type都是分组信息</span>
<span class="c1"># returnData可以返回主成分分析数据,进而使用其他方法画图</span>
pcaData <span class="o">&lt;-</span> plotPCA<span class="p">(</span>vsd<span class="p">,</span> intgroup<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="s">&quot;condition&quot;</span><span class="p">,</span> <span class="s">&quot;type&quot;</span><span class="p">),</span> returnData<span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
percentVar <span class="o">&lt;-</span> <span class="kp">round</span><span class="p">(</span><span class="m">100</span> <span class="o">*</span> <span class="kp">attr</span><span class="p">(</span>pcaData<span class="p">,</span> <span class="s">&quot;percentVar&quot;</span><span class="p">))</span>
ggplot<span class="p">(</span>pcaData<span class="p">,</span> aes<span class="p">(</span>PC1<span class="p">,</span> PC2<span class="p">,</span> color<span class="o">=</span>condition<span class="p">,</span> shape<span class="o">=</span>type<span class="p">))</span> <span class="o">+</span>
  geom_point<span class="p">(</span>size<span class="o">=</span><span class="m">3</span><span class="p">)</span> <span class="o">+</span>
  xlab<span class="p">(</span><span class="kp">paste0</span><span class="p">(</span><span class="s">&quot;PC1: &quot;</span><span class="p">,</span>percentVar<span class="p">[</span><span class="m">1</span><span class="p">],</span><span class="s">&quot;% variance&quot;</span><span class="p">))</span> <span class="o">+</span>
  ylab<span class="p">(</span><span class="kp">paste0</span><span class="p">(</span><span class="s">&quot;PC2: &quot;</span><span class="p">,</span>percentVar<span class="p">[</span><span class="m">2</span><span class="p">],</span><span class="s">&quot;% variance&quot;</span><span class="p">))</span> <span class="o">+</span> 
  coord_fixed<span class="p">()</span>
</pre></div>


<h2 id="_11">标准流程变体</h2>
<p>略.</p>
<h2 id="faqs">FAQs</h2>
<h3 id="_12">分析成对的样品</h3>
<blockquote>
<p>使用多因子设计即可。</p>
</blockquote>
<h3 id="_13">含有多个组，共同分析还是分开分析？</h3>
<blockquote>
<p>共同分析，两两比较时，使用result函数的contrast指定要比较的两个</p>
</blockquote>
<h3 id="without-replicate">可以分析无重复（without replicate）的数据吗？</h3>
<blockquote>
<p>可以，但是只推荐用于探索性的分析，不适用于比较分析。</p>
</blockquote>
		</content>

	</main>
	<footer>
		<p> Powered by xizhihui </p>
		<p> Thanks to <a href="https://www.hifreud.com/">hifreud</a> the theme is learned from his site. </p>
	</footer>

<script type="text/javascript" src="/js/index.json"></script>
<script type="text/javascript" src="/js/category.json"></script>
<script type="text/javascript" src="/js/tags.json"></script>
<script type="text/javascript" src="/js/about.json"></script>
<script type="text/javascript" src="/js/index.js"></script>
<script type="text/javascript">
	// listen to click event on navigator in header
	// document
	// 	.querySelector("div#head_right ul")
	// 	.addEventListener("click", change_page)
	var table = [['h2', 'deseq2', 'DESeq2'], ['h2', '_1', '一、核心逻辑代码'], ['h2', 'standard-workflow', '二、标准流程 standard workflow'], ['h3', '21-input-data', '2.1 input data'], ['h3', '22-deseqdatasetfromtximporttxi', '2.2 DESeqDataSetFromTximport：.txi文件'], ['h3', '23-deseqdatasetfrommatrix-count-matrix', '2.3 使用DESeqDataSetFromMatrix: count matrix文件'], ['h3', '24-input-htseq-count', '2.4 input: htseq-count'], ['h3', '25-input-summarizedexperiment', '2.5 input: SummarizedExperiment'], ['h3', '26', '2.6 预过滤'], ['h3', '27-factor-levels', '2.7 指定factor levels'], ['h3', '28-collapsereplicates', '2.8 合并技术性重复：collapseReplicates函数'], ['h2', '_2', '三、差异表达分析'], ['h2', '_3', '四、探索和导出结果'], ['h3', 'ma-plot', 'MA-plot'], ['h3', 'plot-counts', 'plot counts'], ['h3', '_4', '更多有关结果的信息'], ['h3', '_5', '富可视化、生成报告'], ['h3', '_6', '多因子设计'], ['h2', '_7', '五、通过聚类可视化表征数据质量'], ['h3', '_8', '表达矩阵的热图'], ['h3', '_9', '样品距离热图'], ['h3', '_10', '样品的主成分分析'], ['h2', '_11', '标准流程变体'], ['h2', 'faqs', 'FAQs'], ['h3', '_12', '分析成对的样品'], ['h3', '_13', '含有多个组，共同分析还是分开分析？'], ['h3', 'without-replicate', '可以分析无重复（without replicate）的数据吗？']]
	var html = format_article_table(table)
	var table_ele = document.querySelector("main #table")
	table_ele.innerHTML = html
	// window.onload = evt => resize_nav(table_ele)
	adaption(document, window)

</script>
</body>
</html>