<!DOCTYPE HTML>
<html>
<head><meta name="generator" content="Hexo 3.8.0">
  <meta charset="utf-8">
  
  <title>比对 Bowtie2 | Xizhihui&#39;s Notes</title>
  <meta name="author" content="Xizhihui">
  
  <meta name="description" content="bowtie2是个超快的、内存占用少的序列比对工具，善于比对相对较长的基因组。bowtie2有gapped、pair-end和local比对模式，可以多线程进行。它是许多pipeline的首个步骤，例如变异检测，CHIP-seq，RNA-seq，BS-seq等等。bowtie2不像常规目的的比对工具如MUMmer，Blast等。它在大的参考基因组的比对上表现更好，因为它针对当前各个测序平台的测序reads进行过优化。如果你的目的是比对很大的两个序列，比如基因组之间的比对，你应考虑使用MUMmer。如果你的目的是比对相对较短的序列如大肠杆菌的基因组，用bowtie2可以大大减少你的时间。">
  
  
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">

  <meta property="og:title" content="比对 Bowtie2">
  <meta property="og:site_name" content="Xizhihui&#39;s Notes">

  
    <meta property="og:image" content="">
  

  
  
    <link href="/favicon.png" rel="icon">
  

  <!-- CSS -->
  <link rel="stylesheet" href="/css/themes/cosmo.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/font-awesome.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/style.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/responsive.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight-default.min.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/google-fonts.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/comment.css" media="screen" type="text/css">
  <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
	<script src="https://cdnjs.cloudflare.com/ajax/libs/es5-shim/4.5.9/es5-shim.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/es5-shim/4.5.7/es5-sham.min.js"></script>
  <![endif]-->

  <script src="/js/jquery-2.0.3.min.js"></script>
  
  <!-- analytics -->
  


<script>
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>


</head>
</html>
 <body>  
  <nav id="main-nav" class="navbar  navbar-default navbar-fixed-top" role="navigation">
    <div class="container">
      <button type="button" class="navbar-header navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
	<span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
       <a class="navbar-brand" href="/">Xizhihui&#39;s Notes</a>
      <div class="collapse navbar-collapse nav-menu">
		<ul class="nav navbar-nav">
		  
		  <li>
			<a href="/archives" title="All the articles.">
			  <i class="fa fa-archive"></i>Archives
			</a>
		  </li>
		  
		  <li>
			<a href="/categories" title="All the categories.">
			  <i class="fa fa-folder"></i>Categories
			</a>
		  </li>
		  
		  <li>
			<a href="/tags" title="All the tags.">
			  <i class="fa fa-tags"></i>Tags
			</a>
		  </li>
		  
		  <li>
			<a href="/about" title="About me.">
			  <i class="fa fa-user"></i>About
			</a>
		  </li>
		  
		</ul>
      </div>
    </div> <!-- container -->
</nav>
<div class="clearfix"></div>

  <div class="container">
  	<div class="content">
    	 


	
		<div class="page-header ">		
			<h1 class="title "> 比对 Bowtie2</h1>
		</div>		
	






<div class="row post">
	<!-- cols -->
	
	<div id="top_meta"></div>
	<div class="col-md-9">
	

	<!-- content -->
	<div class="mypage">		
	  		

	  <p>bowtie2是个超快的、内存占用少的序列比对工具，善于比对相对较长的基因组。bowtie2有gapped、pair-end和local比对模式，可以多线程进行。它是许多pipeline的首个步骤，例如变异检测，CHIP-seq，RNA-seq，BS-seq等等。<br>bowtie2不像常规目的的比对工具如MUMmer，Blast等。它在大的参考基因组的比对上表现更好，因为它针对当前各个测序平台的测序reads进行过优化。如果你的目的是比对很大的两个序列，比如基因组之间的比对，你应考虑使用MUMmer。如果你的目的是比对相对较短的序列如大肠杆菌的基因组，用bowtie2可以大大减少你的时间。</p>
<a id="more"></a>
<h2 id="bowtie1和bowtie2的区别"><a href="#bowtie1和bowtie2的区别" class="headerlink" title="bowtie1和bowtie2的区别"></a>bowtie1和bowtie2的区别</h2><ul>
<li>对于长于50bp的序列,bowtie2速度更快也更灵敏，内存占用更少；但是Bowtie1在短于50bp的比对上有时会更快或更灵敏</li>
<li>bowtie2通过gap penalty支持gapped alignment</li>
<li>bowtie2支持局部比对(soft clipped, not end-to-end)</li>
<li>bowtie1的read length上限是1000bp，而bowtie2则没有上限</li>
<li>bowtie2支持参考基因组的跨Ns比对</li>
<li>对于pair-end reads,如果比对不上,bowtie2尝试将单个reads进行比对</li>
<li>bowtie2不支持colorspace reads的比对</li>
</ul>
<h2 id="Getting-started"><a href="#Getting-started" class="headerlink" title="Getting started"></a>Getting started</h2><h3 id="1-对参考基因组建立索引"><a href="#1-对参考基因组建立索引" class="headerlink" title="1. 对参考基因组建立索引"></a>1. 对参考基因组建立索引</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># prefix是输出的索引文件前缀, .1.bt2, .2.bt2, .3.bt2, .4.bt2, .rev.1.bt2, .rev.2.bt2</span></span><br><span class="line">bowtie2-build path/to/ref.fa prefix</span><br></pre></td></tr></table></figure>
<h3 id="2-进行比对"><a href="#2-进行比对" class="headerlink" title="2. 进行比对"></a>2. 进行比对</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 对于单端测序(single end)</span></span><br><span class="line">bowtie2 -x prefix -U Single_end.fq -S align.sam &gt; align.log</span><br><span class="line"><span class="comment"># 对于双端测序(paire-end)</span></span><br><span class="line">bowtie2 -x prefix -1 read1.fq -2 read2.fq -S align.sam &gt; align.log</span><br><span class="line"><span class="comment"># 如果要使用局部比对模式</span></span><br><span class="line">bowtie2 --<span class="built_in">local</span> -x prefix -U sigle_end.fq -S align.sam &gt; align.log</span><br></pre></td></tr></table></figure>
<h3 id="3-从index里获取原始ref-fa"><a href="#3-从index里获取原始ref-fa" class="headerlink" title="3. 从index里获取原始ref.fa"></a>3. 从index里获取原始ref.fa</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">bowtie2-inspect prefix</span><br></pre></td></tr></table></figure>
<h2 id="构建好的index下载"><a href="#构建好的index下载" class="headerlink" title="构建好的index下载"></a>构建好的index下载</h2><table>
<thead>
<tr>
<th>总卷</th>
<th>分卷</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="ftp://ftp.ccb.jhu.edu/pub/data/bowtie2_indexes/hg18.zip" target="_blank" rel="noopener">H. sapiens, UCSC hg18</a></td>
<td><a href="ftp://ftp.ccb.jhu.edu/pub/data/bowtie2_indexes/hg18.1.zip" target="_blank" rel="noopener">part1</a>, <a href="ftp://ftp.ccb.jhu.edu/pub/data/bowtie2_indexes/hg18.2.zip" target="_blank" rel="noopener">part2</a>, <a href="ftp://ftp.ccb.jhu.edu/pub/data/bowtie2_indexes/hg18.3.zip" target="_blank" rel="noopener">part3</a></td>
</tr>
<tr>
<td><a href="ftp://ftp.ccb.jhu.edu/pub/data/bowtie2_indexes/hg19.zip" target="_blank" rel="noopener">H. sapiens, UCSC hg19</a></td>
<td><a href="ftp://ftp.ccb.jhu.edu/pub/data/bowtie2_indexes/hg19.1.zip" target="_blank" rel="noopener">part1</a>, <a href="ftp://ftp.ccb.jhu.edu/pub/data/bowtie2_indexes/hg19.2.zip" target="_blank" rel="noopener">part2</a>, <a href="ftp://ftp.ccb.jhu.edu/pub/data/bowtie2_indexes/hg19.3.zip" target="_blank" rel="noopener">part3</a></td>
</tr>
<tr>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/archive/old_genbank/Eukaryotes/vertebrates_mammals/Homo_sapiens/GRCh38/seqs_for_alignment_pipelines/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.bowtie_index.tar.gz" target="_blank" rel="noopener">H. sapiens, NCBI GRCh38</a></td>
<td>none</td>
</tr>
<tr>
<td><a href="ftp://ftp.ccb.jhu.edu/pub/data/bowtie2_indexes/mm10.zip" target="_blank" rel="noopener">M. musculus, UCSC mm10</a></td>
<td><a href="ftp://ftp.ccb.jhu.edu/pub/data/bowtie2_indexes/mm10.1.zip" target="_blank" rel="noopener">part1</a>, <a href="ftp://ftp.ccb.jhu.edu/pub/data/bowtie2_indexes/mm10.2.zip" target="_blank" rel="noopener">part2</a>, <a href="ftp://ftp.ccb.jhu.edu/pub/data/bowtie2_indexes/mm10.3.zip" target="_blank" rel="noopener">part3</a></td>
</tr>
<tr>
<td><a href="ftp://ftp.ccb.jhu.edu/pub/data/bowtie2_indexes/mm9.zip" target="_blank" rel="noopener">M. musculus, UCSC mm9</a></td>
<td><a href="ftp://ftp.ccb.jhu.edu/pub/data/bowtie2_indexes/mm9.1.zip" target="_blank" rel="noopener">part1</a>, <a href="ftp://ftp.ccb.jhu.edu/pub/data/bowtie2_indexes/mm9.2.zip" target="_blank" rel="noopener">part2</a>, <a href="ftp://ftp.ccb.jhu.edu/pub/data/bowtie2_indexes/mm9.3.zip" target="_blank" rel="noopener">part3</a></td>
</tr>
<tr>
<td><a href="ftp://ftp.ccb.jhu.edu/pub/data/bowtie2_indexes/rn4.zip" target="_blank" rel="noopener">R. norvegicus, UCSC rn4</a></td>
<td><a href="ftp://ftp.ccb.jhu.edu/pub/data/bowtie2_indexes/rn4.1.zip" target="_blank" rel="noopener">part1</a>, <a href="ftp://ftp.ccb.jhu.edu/pub/data/bowtie2_indexes/rn4.2.zip" target="_blank" rel="noopener">part2</a>, <a href="ftp://ftp.ccb.jhu.edu/pub/data/bowtie2_indexes/rn4.3.zip" target="_blank" rel="noopener">part3</a></td>
</tr>
</tbody>
</table>
<h2 id="bowtie2相关术语解读"><a href="#bowtie2相关术语解读" class="headerlink" title="bowtie2相关术语解读"></a>bowtie2相关术语解读</h2><h3 id="1-比对模式"><a href="#1-比对模式" class="headerlink" title="1. 比对模式"></a>1. 比对模式</h3><p>bowtie2里面有两种比对模式：end-to-end, local。前者在比对时要read的首尾比对上,中间没有比对上的作为gap进行罚分.而local比对模式在比对时，优先保证read内部而非两端的比对，就比对效果上看，local模式下的read两端没有比对上，类似切了两端，所以又称为soft clipped。</p>
<h3 id="2-scores-higher-more-similar"><a href="#2-scores-higher-more-similar" class="headerlink" title="2. scores: higher = more similar"></a>2. scores: higher = more similar</h3><p>比对得分指示着read与参考序列的相似性，得分越高相似性越高。比对得分的控制可以通过指定各个情况的得分实现。如下：</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">--ma:	match bonus,比对上的得分</span><br><span class="line">--mp:   mismatch penalty,比对不上的罚分</span><br><span class="line">--np:	<span class="built_in">read</span>/ref序列中有N的罚分</span><br><span class="line">--rdg:	affine <span class="built_in">read</span> gap penalty,比对时<span class="built_in">read</span>出现gap的罚分;注意gap与mismatch的不同</span><br><span class="line">--rfg:	affine reference gap penalty,比对时ref上出现gap的罚分</span><br><span class="line">--score-min:	指定的最低得分,高于该得分的比对才算比对成功</span><br></pre></td></tr></table></figure>
<h3 id="3-mapping-quality-higher-more-unique"><a href="#3-mapping-quality-higher-more-unique" class="headerlink" title="3. mapping quality: higher = more unique"></a>3. mapping quality: higher = more unique</h3><p>比对质量指示着read比对到参考序列上的唯一性，得分越高，越是唯一比对。如果一个read有多种比对情况,那么某个比对情况score越高，我们就说这个比对越唯一。</p>
<h3 id="4-mixed-mode"><a href="#4-mixed-mode" class="headerlink" title="4. mixed mode"></a>4. mixed mode</h3><p>如果pair-end的reads比对不上，bowtie2会尝试使用单个read进行比对，这个情况称为mixed mode。这样的结果就是得到的比对率可能会比不使用该模式的比对率高辣么一点点。你可以指定–no-mixed参数禁用。</p>
<h3 id="5-reporting"><a href="#5-reporting" class="headerlink" title="5. reporting"></a>5. reporting</h3><p>bowtie2有三种结果模式,默认模式是搜索多个比对情况,然后输出最佳比对。-a模式会搜索并输出所有比对情况。-k模式则是搜索1至多个比对情况，并输出。如果多个比对情况得分相同的话，bowtie2会随机选一个。</p>
	  
	</div>

	<!-- recommended posts -->
	

	<div>
  	<center>
	<div class="pagination">
<ul class="pagination">
	 
				
    	<li class="prev"><a href="/2018/10/12/比对软件-Hisat2/" class="alignleft prev"><i class="fa fa-arrow-circle-o-left"></i>Prev</a></li>
  		

        <li><a href="/archives"><i class="fa fa-archive"></i>Archive</a></li>

		
		   <li class="next"><a href="/2018/10/12/加权共表达基因网络分析-WGCNA/" class="alignright next">Next<i class="fa fa-arrow-circle-o-right"></i></a></li>         
        
	
</ul>
</div>

    </center>
	</div>

    <!-- share -->
    
        
    <div class="bdsharebuttonbox">
        <a href="#" class="bds_more" data-cmd="more"></a>
        <a href="#" class="bds_weixin" data-cmd="weixin" title="分享到微信"></a>
        <a href="#" class="bds_tsina" data-cmd="tsina" title="分享到新浪微博"></a>
        <a href="#" class="bds_fbook" data-cmd="fbook" title="分享到Facebook"></a>
        <a href="#" class="bds_twi" data-cmd="twi" title="分享到Twitter"></a>
        <a href="#" class="bds_linkedin" data-cmd="linkedin" title="分享到linkedin"></a>
        <a href="#" class="bds_evernotecn" data-cmd="evernotecn" title="分享到印象笔记"></a>
        <a href="#" class="bds_youdao" data-cmd="youdao" title="分享到有道云笔记"></a>
        <a href="#" class="bds_copy" data-cmd="copy" title="分享到复制网址"></a>
    </div>
    <script>
        window._bd_share_config={"common":{"bdSnsKey":{},"bdText":"","bdMini":"2","bdMiniList":false,"bdPic":"","bdStyle":"1","bdSize":"24"},"share":{}};
        with(document)0[(getElementsByTagName('head')[0]||body).appendChild(createElement('script')).src='http://bdimg.share.baidu.com/static/api/js/share.js?v=89860593.js?cdnversion='+~(-new Date()/36e5)];
    </script>


        

    
	
	<!-- comment -->
	
<section id="comment">
  <h2 class="title">Comments</h2>
  
</section>


	</div> <!-- col-md-9/col-md-12 -->
		
	
	<div id="side_meta">
		<div class="col-md-3" id="post_meta"> 

	<!-- date -->
	
	<div class="meta-widget">
	<i class="fa fa-clock-o"></i>
	2018-10-12 
	</div>
	

	<!-- categories -->
    
	<div class="meta-widget">
	<a data-toggle="collapse" data-target="#categorys"><i class="fa fa-folder"></i></a>	
    <ul id="categorys" class="tag_box list-unstyled collapse in">
          
  <li>
    </li><li><a href="/categories/Bioinformatics/">Bioinformatics<span>36</span></a></li>
  

    </ul>
	</div>
	

	<!-- tags -->
	
	<div class="meta-widget">
	<a data-toggle="collapse" data-target="#tags"><i class="fa fa-tags"></i></a>		  
    <ul id="tags" class="tag_box list-unstyled collapse in">	  
	    
  <li><a href="/tags/软件和包/">软件和包<span>22</span></a></li>
    </ul>
	</div>
	

	<!-- toc -->
	<div class="meta-widget">
	
	   <a data-toggle="collapse" data-target="#toc"><i class="fa fa-bars"></i></a>
	   <div id="toc" class="toc collapse in">
			<ol class="toc-article"><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#bowtie1和bowtie2的区别"><span class="toc-article-text">bowtie1和bowtie2的区别</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#Getting-started"><span class="toc-article-text">Getting started</span></a><ol class="toc-article-child"><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#1-对参考基因组建立索引"><span class="toc-article-text">1. 对参考基因组建立索引</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#2-进行比对"><span class="toc-article-text">2. 进行比对</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#3-从index里获取原始ref-fa"><span class="toc-article-text">3. 从index里获取原始ref.fa</span></a></li></ol></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#构建好的index下载"><span class="toc-article-text">构建好的index下载</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#bowtie2相关术语解读"><span class="toc-article-text">bowtie2相关术语解读</span></a><ol class="toc-article-child"><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#1-比对模式"><span class="toc-article-text">1. 比对模式</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#2-scores-higher-more-similar"><span class="toc-article-text">2. scores: higher = more similar</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#3-mapping-quality-higher-more-unique"><span class="toc-article-text">3. mapping quality: higher = more unique</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#4-mixed-mode"><span class="toc-article-text">4. mixed mode</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#5-reporting"><span class="toc-article-text">5. reporting</span></a></li></ol></li></ol>
		</div>
	
	</div>
	
    <hr>
	
</div><!-- col-md-3 -->

	</div>
		

</div><!-- row -->



	</div>
  </div>
  <div class="container-narrow">
  <footer> <p>
  &copy; 2018 Xizhihui
  
      with help from <a href="http://hexo.io/" target="_blank">Hexo</a> and <a href="http://getbootstrap.com/" target="_blank">Twitter Bootstrap</a>. Theme by <a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind</a>.    
</p> </footer>
</div> <!-- container-narrow -->
  


  
<a id="gotop" href="#">   
  <span>▲</span> 
</a>

<script src="/js/jquery.imagesloaded.min.js"></script>
<script src="/js/gallery.js"></script>
<script src="/js/bootstrap.min.js"></script>
<script src="/js/main.js"></script>
<script src="/js/search.js"></script> 


<link rel="stylesheet" href="/fancybox/jquery.fancybox.css" media="screen" type="text/css">
<script src="/fancybox/jquery.fancybox.pack.js"></script>
<script type="text/javascript">
(function($){
  $('.fancybox').fancybox();
})(jQuery);
</script>



   <script type="text/javascript">      
     var search_path = "search.xml";
	 if (search_path.length == 0) {
	 	search_path = "search.xml";
	 }
	 var path = "/" + search_path;
     searchFunc(path, 'local-search-input', 'local-search-result');
   </script>


<!-- syntax highlighting -->


</body>
   </html>
