<!DOCTYPE html>
<html>
<head>
	<title>XZH's Notes</title>
	<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
	<meta name="author" content="xizhihui" />
	<meta name="viewport" content="width=device-width, initial-scale=1" />
	<link rel="stylesheet" type="text/css" href="/statics/reset.css">
	<link rel="stylesheet" type="text/css" href="/statics/header_footer.css"></link>
	<link rel="stylesheet" type="text/css" href="/statics/default.css">
	<link rel="stylesheet" type="text/css" href="/statics/main.css">
</head>
<body>
	<header class="Header">
		<nav>
			<h3>XZH</h3>
			<ul>
				<li><a href="/index.html#Timeline">Timeline</a></li><li><a href="/index.html#Category">Category</a></li><li><a href="/index.html#Tags">Tags</a></li><li><a href="/index.html#About">About</a></li>
			</ul>
		</nav>
	</header>
	<div class="Motto">
		<h2>XZH's Notes</h2>
		<h4>The future comes for you ~ ~ ~</h4>
	</div>

	<main>
		<nav id="table"></nav>
		<content id="content" class="content">
			<h2 id="rsem">RSEM</h2>
<p>RSEM利用的是transcripts而非genome。我们有两种方式来构建RSEM转录参考，其一是利用参考基因组来构建；另外一种方式是从许多转录本中构建。</p>
<h3 id="1">1. 下载对应的基因组和注释文件</h3>
<div class="codehilite"><pre><span></span>ftp://ftp.ensembl.org/pub/release-82/fasta/mus_musculus/dna/Mus_musculus.GRCm38.dna.toplevel.fa.gz
ftp://ftp.ensembl.org/pub/release-82/gtf/mus_musculus/Mus_musculus.GRCm38.82.chr.gtf.gz
</pre></div>


<h3 id="2">2. 构建索引</h3>
<div class="codehilite"><pre><span></span><span class="c1"># 使用bowtie2索引</span>
gunzip Mus_musculus.GRCm38.dna.toplevel.fa.gz
gunzip ref/Mus_musculus.GRCm38.82.chr.gtf.gz
<span class="c1"># 从基因组构建</span>
rsem-prepare-reference --gtf Mus_musculus.GRCm38.82.chr.gtf <span class="se">\</span>
                       --bowtie2 --bowtie2-path bowtie2 <span class="se">\</span>
                       Mus_musculus.GRCm38.dna.toplevel.fa mouse_ref

<span class="c1"># 从转录本构建</span>
rsem-prepare-reference --transcript-to-gene-map mouse_ref_mapping.txt <span class="se">\</span>
                       --bowtie2 --bowtie2-path bowtie2 <span class="se">\</span>
                       ref/mouse_ref.fa ref/mouse_ref
</pre></div>


<h3 id="3">3. 表达定量</h3>
<div class="codehilite"><pre><span></span>rsem-calculate-expression -p <span class="m">8</span> --paired-end <span class="se">\</span>
                        --bowtie2 --bowtie2-path bowtie2 <span class="se">\</span>
                        --estimate-rspd <span class="se">\</span>
                        --append-names <span class="se">\</span>
                        --output-genome-bam <span class="se">\</span>
                        read1.fastq read2.fastq <span class="se">\</span>
                        ref/mouse_ref path/to/output/prefix
<span class="c1"># output =&gt; prefix.genes.results, prefix.genome.bam, prefix.genome.sorted.bam,</span>
<span class="c1">#           prefix.genome.sorted.bam.bai, prefix.isoforms.results, prefix.stat,</span>
<span class="c1">#           prefix.transcript.bam, prefix.transcript.sorted.bam, prefix.transcript.sorted.bam.bai</span>

<span class="c1"># --output-genome-bam   只在由基因组构建索引时有效</span>
<span class="c1"># --bam                 当输入不是fastq文件而是bam文件时,指定该参数</span>
<span class="c1"># --calc-ci             指定了会计算CQV值, coefficient of quartile variation，这将回答该基因的测序深度是否足够用于差异检测</span>
<span class="c1"># --no-bam-output       不输出bam文件</span>
<span class="c1"># --single-cell-prior   使用针对单细胞的预设条件运行</span>
</pre></div>


<h3 id="4">4. 结果输出</h3>
<div class="codehilite"><pre><span></span><span class="c1"># in R</span>
data <span class="o">=</span> read.table<span class="p">(</span><span class="s">&#39;prefix.gene.results&#39;</span><span class="p">,</span> header<span class="o">=</span><span class="bp">T</span><span class="p">,</span> stringsAsFactors<span class="o">=</span><span class="bp">F</span><span class="p">)</span>
idx <span class="o">=</span> <span class="kp">order</span><span class="p">(</span>data<span class="p">[,</span><span class="s">&#39;TPM&#39;</span><span class="p">],</span> decreasing<span class="o">=</span><span class="bp">T</span><span class="p">)</span>
data<span class="p">[</span>idx<span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">10</span><span class="p">],</span> <span class="kt">c</span><span class="p">(</span><span class="s">&#39;gene_id&#39;</span><span class="p">,</span> <span class="s">&#39;expected_count&#39;</span><span class="p">,</span> <span class="s">&#39;TPM&#39;</span><span class="p">)]</span>
</pre></div>


<div class="codehilite"><pre><span></span><span class="c1"># in bash</span>
rsem-plot-model /path/to/output/prefix prefix.diagnostic.pdf
rsem-plot-transcript-wiggles --gene-list --show-unique <span class="se">\</span>
                            prefix gene_ids.txt aim_gene_transcript_wiggle.pdf
<span class="c1"># gene_ids.txt: 含有gene identifier of aim_gene</span>

rsem-bam2wig prefix.genome.sorted.bam prefix.wig prefix
</pre></div>


<h3 id="5">5. 生成表达矩阵</h3>
<div class="codehilite"><pre><span></span>rsem-generate-data-matrix sample1.genes.results sample2.genes.results <span class="se">\</span>
                          sample3.genes.results ... <span class="se">\</span>
                          sampleN.genes.results &gt; gene_matrix.txt
rsem-run-ebseq gene_matrix.txt NumberOfgroup1,numberofgroup2 gene_matrix.ebseq.results
rsem-control-fdr gene_matrix.ebseq.result <span class="m">0</span>.05 gene_matrix.de.txt
</pre></div>


<h3 id="6-differentially-expressed-isoforms">6. 检测转录表达差异(differentially expressed isoforms)</h3>
<div class="codehilite"><pre><span></span>rsem-generate-ngvector mouse_ref.transcripts.fa mouse_ref
<span class="c1"># mouse_ref.transcripts.fa是mouse的转录组参考序列</span>
<span class="c1"># output=&gt; mouse_ref.ngvec</span>
rsem-generate-data-matrix sample1.isoforms.results <span class="se">\</span>
                          sample2.isoforms.results sample3.isoforms.results <span class="se">\</span>
                          ...
                          sampleN.isoforms.results &gt; isoform_matrix.txt
rsem-run-ebseq --ngvector mouse_ref.ngvec isoform_matrix.txt control_sample_num,treate_sample_num <span class="se">\</span>
                          isoform_ebseq.results
rsem-control-fdr isoform_ebseq.results <span class="m">0</span>.05 isoform.de.txt
</pre></div>


<h3 id="7">7. 检测测序深度是否足够</h3>
<div class="codehilite"><pre><span></span><span class="c1"># 1. 在rsem-calculate-expression中指定--calc-ci</span>
<span class="c1"># 2. 如果基因的cov值高于0.05表明测序深度不够,需要多少？</span>
rsem-simulate-reads ../ref/mouse_ref prefix.stat/prefix.model <span class="se">\</span>
                    prefix.isoforms.results <span class="m">0</span>.36 <span class="m">20000000</span> prefix_sim_2M <span class="se">\</span>
                    --seed <span class="m">0</span>
<span class="c1"># ../ref/mouse_ref指定参考文件的位置</span>
<span class="c1"># 0.36是预设的背景噪音比率</span>
<span class="c1"># output =&gt; prefix_sim_2M_1.fastq, prefix_sim_2M_2.fastq</span>

rsem-calculate-expression -p <span class="m">8</span> --paired-end <span class="se">\</span>
                          --bowtie2 --bowtie2-path bowtie2 <span class="se">\</span>
                          --estimate-rspd <span class="se">\</span>
                          --append-names <span class="se">\</span>
                          --no-bam-output <span class="se">\</span>
                          --calc-ci <span class="se">\</span>
                          prefix_sim_2M_1.fastq prefix_sim_2M_2.fastq <span class="se">\</span>
                          ../ref/mouse_ref prefix_sim_2M
<span class="c1"># 查看output的六个文件中prefix_sim_2m.genes.results,如果基因的cov低于0.05,表明20000000的测序深度足够.</span>
</pre></div>
		</content>

	</main>
	<footer>
		<p> Powered by xizhihui </p>
		<p> Thanks to <a href="https://www.hifreud.com/">hifreud</a> the theme is learned from his site. </p>
	</footer>

<script type="text/javascript" src="/js/index.json"></script>
<script type="text/javascript" src="/js/category.json"></script>
<script type="text/javascript" src="/js/tags.json"></script>
<script type="text/javascript" src="/js/about.json"></script>
<script type="text/javascript" src="/js/index.js"></script>
<script type="text/javascript">
	// listen to click event on navigator in header
	// document
	// 	.querySelector("div#head_right ul")
	// 	.addEventListener("click", change_page)
	var table = [['h2', 'rsem', 'RSEM'], ['h3', '1', '1. 下载对应的基因组和注释文件'], ['h3', '2', '2. 构建索引'], ['h3', '3', '3. 表达定量'], ['h3', '4', '4. 结果输出'], ['h3', '5', '5. 生成表达矩阵'], ['h3', '6-differentially-expressed-isoforms', '6. 检测转录表达差异(differentially expressed isoforms)'], ['h3', '7', '7. 检测测序深度是否足够']]
	var html = format_article_table(table)
	var table_ele = document.querySelector("main #table")
	table_ele.innerHTML = html
	// window.onload = evt => resize_nav(table_ele)
	adaption(document, window)

</script>
</body>
</html>