<!DOCTYPE HTML>
<html>
<head><meta name="generator" content="Hexo 3.8.0">
  <meta charset="utf-8">
  
  <title>Page 3 | Bioinformatics | Xizhihui&#39;s Notes</title>
  <meta name="author" content="Xizhihui">
  
  
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">

  
  <meta property="og:site_name" content="Xizhihui&#39;s Notes">

  
    <meta property="og:image" content="">
  

  
  
    <link href="/favicon.png" rel="icon">
  

  <!-- CSS -->
  <link rel="stylesheet" href="/css/themes/cosmo.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/font-awesome.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/style.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/responsive.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight-default.min.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/google-fonts.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/comment.css" media="screen" type="text/css">
  <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
	<script src="https://cdnjs.cloudflare.com/ajax/libs/es5-shim/4.5.9/es5-shim.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/es5-shim/4.5.7/es5-sham.min.js"></script>
  <![endif]-->

  <script src="/js/jquery-2.0.3.min.js"></script>
  
    <script src="/js/marked.js"></script>
    <script src="/js/comment.js"></script>
    <script src="/js/timeago.min.js"></script>
    <script src="/js/highlight.min.js"></script>
	<script src="/js/spin.min.js"></script>
  
  <!-- analytics -->
  



</head>
</html>
 <body>  
  <nav id="main-nav" class="navbar  navbar-default navbar-fixed-top" role="navigation">
    <div class="container">
      <button type="button" class="navbar-header navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
	<span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
       <a class="navbar-brand" href="/">Xizhihui&#39;s Notes</a>
      <div class="collapse navbar-collapse nav-menu">
		<ul class="nav navbar-nav">
		  
		  <li>
			<a href="/archives" title="All the articles.">
			  <i class="fa fa-archive"></i>Archives
			</a>
		  </li>
		  
		  <li>
			<a href="/categories" title="All the categories.">
			  <i class="fa fa-folder"></i>Categories
			</a>
		  </li>
		  
		  <li>
			<a href="/tags" title="All the tags.">
			  <i class="fa fa-tags"></i>Tags
			</a>
		  </li>
		  
		  <li>
			<a href="/about" title="About me.">
			  <i class="fa fa-user"></i>About
			</a>
		  </li>
		  
		</ul>
      </div>
    </div> <!-- container -->
</nav>
<div class="clearfix"></div>

  <div class="container">
  	<div class="content">
    	 

<!-- title -->
<div class="page-header ">
  <h1 class="archive-title-category title ">Bioinformatics</h1>
</div>

<div class="row page">
  <!-- cols -->
  
  <div class="col-md-9">
	

	  <div id="top_search"></div>

      
         <!-- display as entry -->
	     <div class="mypage">
	       
		     
	
	<!-- display as entry -->	
		<h3 class="title">
			<div class="date"> 2018-10-12 </div>
			<div class="article-title"><a href="/2018/10/12/生信基础-unique比对的获取/">生信基础 unique比对的获取</a></div>						
		</h3>
	


		     <div class="entry">
  <div class="row">
	
	
		<h2 id="1-Sam文件各标签含义（tophat-hisat2"><a href="#1-Sam文件各标签含义（tophat-hisat2" class="headerlink" title="1.Sam文件各标签含义（tophat/hisat2)"></a>1.Sam文件各标签含义（tophat/hisat2)</h2><ul>
<li>NH:i:<n>: N=1时 为unique。常用于tophat/hisat2产生的sam文件unique read筛选。</n></li>
<li>CC:Z: 当为‘=’为map到同一条基因上，一般在map基因组时由于内含子存在而容易出现，他只代表两种不同的方式，计数时应记为1。此处一般为其他基因的名字。CP:i 和HI：i标签为map到第i条基因及起始位置。</li>
<li>YT:Z:S 代表的含义与bowtie产生的sam也不同。具体还未知！其他标签AS，XN,XM,XO,XG,NM,MD等如下图可以看出都相同。</li>
</ul>
	
	</div>
  <a type="button" href="/2018/10/12/生信基础-unique比对的获取/#more" class="btn btn-default more">Read More</a>
</div>

	       
		     
	
	<!-- display as entry -->	
		<h3 class="title">
			<div class="date"> 2018-10-12 </div>
			<div class="article-title"><a href="/2018/10/12/生信基础-常用gene_ID的转换/">生信基础 常用gene_ID的转换</a></div>						
		</h3>
	


		     <div class="entry">
  <div class="row">
	
	
		<h2 id="ID-类型"><a href="#ID-类型" class="headerlink" title="ID 类型"></a>ID 类型</h2><table>
<thead>
<tr>
<th>ID 示例</th>
<th>ID 来源</th>
</tr>
</thead>
<tbody>
<tr>
<td>ENSG00000116717</td>
<td>Ensemble ID</td>
</tr>
<tr>
<td>GA45A_HUMAN</td>
<td>UniProtKB/Swiss-Prot, entry name</td>
</tr>
<tr>
<td>A5PJB2_BOVIN</td>
<td>UniProtKB/TrEMBL, entry name</td>
</tr>
<tr>
<td>A2BC19, P12345,</td>
<td>A0A022YWF9    UniProt, accession number</td>
</tr>
<tr>
<td>GLA, GLB, UGT1A1</td>
<td>HGNC Gene Symbol</td>
</tr>
<tr>
<td>U12345, AF123456</td>
<td>GenBank, NCBI, accession number</td>
</tr>
<tr>
<td>NT_123456, NM_123456, NP_123456</td>
<td>RefSeq, NCBI, accession number</td>
</tr>
<tr>
<td>10598, 717</td>
<td>Entrez ID, NCBI</td>
</tr>
<tr>
<td>uc001ett, uc031tla.1</td>
<td>UCSC ID</td>
</tr>
</tbody>
</table>
	
	</div>
  <a type="button" href="/2018/10/12/生信基础-常用gene_ID的转换/#more" class="btn btn-default more">Read More</a>
</div>

	       
		     
	
	<!-- display as entry -->	
		<h3 class="title">
			<div class="date"> 2018-10-12 </div>
			<div class="article-title"><a href="/2018/10/12/生信基础-各种数据的下载/">生信基础 各种数据的下载</a></div>						
		</h3>
	


		     <div class="entry">
  <div class="row">
	
	
		<h2 id="三大数据库版本对应情况"><a href="#三大数据库版本对应情况" class="headerlink" title="三大数据库版本对应情况"></a>三大数据库版本对应情况</h2><table>
<thead>
<tr>
<th>NCBI</th>
<th>UCSC</th>
<th>Ensemble</th>
</tr>
</thead>
<tbody>
<tr>
<td>GRCh36</td>
<td>hg18</td>
<td>ENSEMBL release_52</td>
</tr>
<tr>
<td>GRCh37</td>
<td>hg19</td>
<td>ENSEMBL release_59/61/64/68/69/75</td>
</tr>
<tr>
<td>GRCh38</td>
<td>hg38</td>
<td>ENSEMBL release_76/77/78/80/81/82</td>
</tr>
</tbody>
</table>
	
	</div>
  <a type="button" href="/2018/10/12/生信基础-各种数据的下载/#more" class="btn btn-default more">Read More</a>
</div>

	       
		     
	
	<!-- display as entry -->	
		<h3 class="title">
			<div class="date"> 2018-10-12 </div>
			<div class="article-title"><a href="/2018/10/12/生信基础-测序原理/">生信基础 测序原理</a></div>						
		</h3>
	


		     <div class="entry">
  <div class="row">
	
	
		<p>在生信分析中，我们通常接触的都是下机数据，也就是测序的结果数据，但是这些数据是怎么产生的呢？这就要讲到测序的原理。其实，测序的本质原理就是DNA链的合成咯，通过合成新的 DNA 链我们从而知道 DNA 链的序列组成。一代测序的 sanger 法是通过每步添加一种特定标记的双脱氧核糖核苷酸( ddNTP )来合成 DNA 链，由于 ddNTP 会导致 DNA 链合成的中断，那就可以得到各个长度大小的 DNA 片段，通过跑胶分离则可知道不同DNA长度的链上分别是什么标记，由此知道其标记对应的 ddNTP，从而推出 DNA 的序列组成；这类一代测序法过程比较耗时。而二代测序的改进在于，它是边合成边测序的，无需后续的跑胶等实现；DNA 合成的数量巨大，也就使得测序过程也更为快速。在如今市场上，二代测序虽然说有许多平台，如 Illumina，Roche 454、Ion Torrent，但市场上一直是 Illumina 占据主导地位。所以我们这里将说明一下 Illumina 的测序过程。</p>
	
	</div>
  <a type="button" href="/2018/10/12/生信基础-测序原理/#more" class="btn btn-default more">Read More</a>
</div>

	       
		     
	
	<!-- display as entry -->	
		<h3 class="title">
			<div class="date"> 2018-10-12 </div>
			<div class="article-title"><a href="/2018/10/12/生信基础-无重复样本的差异分析/">生信基础 无重复样本的差异分析</a></div>						
		</h3>
	


		     <div class="entry">
  <div class="row">
	
	
		<ul>
<li>edgeR</li>
<li>DESeq2</li>
<li>GFOLD</li>
</ul>
	
	</div>
  <a type="button" href="/2018/10/12/生信基础-无重复样本的差异分析/#more" class="btn btn-default more">Read More</a>
</div>

	       
		     
	
	<!-- display as entry -->	
		<h3 class="title">
			<div class="date"> 2018-10-12 </div>
			<div class="article-title"><a href="/2018/10/12/生信基础-生信数据格式汇总/">生信基础 生信数据格式汇总</a></div>						
		</h3>
	


		     <div class="entry">
  <div class="row">
	
	
		<h2 id="Fastq和Fasta格式"><a href="#Fastq和Fasta格式" class="headerlink" title="Fastq和Fasta格式"></a>Fastq和Fasta格式</h2><h3 id="FASTA"><a href="#FASTA" class="headerlink" title="FASTA"></a>FASTA</h3><p>Fasta格式是存储序列相关信息的一种格式，包含两个部分。Fasta第一个部分是以“&gt;”开头的一行。以序列来源、序列ID为值对，用竖线“|”进行分隔。在最后竖线处，以空格分开，添加序列相关描述。第二部分是序列的详细信息，多数为序列组成。其基本形式如下：</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">&gt;ENSEMBL|geneID|source2|geneID description</span><br><span class="line">序列ATCG</span><br></pre></td></tr></table></figure>
	
	</div>
  <a type="button" href="/2018/10/12/生信基础-生信数据格式汇总/#more" class="btn btn-default more">Read More</a>
</div>

	       
		     
	
	<!-- display as entry -->	
		<h3 class="title">
			<div class="date"> 2018-10-12 </div>
			<div class="article-title"><a href="/2018/10/12/生信基础-链特异性/">生信基础 链特异性</a></div>						
		</h3>
	


		     <div class="entry">
  <div class="row">
	
	
		<h2 id="1-链特异性的基本知识"><a href="#1-链特异性的基本知识" class="headerlink" title="1.链特异性的基本知识"></a>1.链特异性的基本知识</h2><ul>
<li><a href="http://www.genek.tv/article/27" target="_blank" rel="noopener">http://www.genek.tv/article/27</a></li>
<li><a href="http://databeauty.com/blog/opinion/2016/09/21/RNA-seq-strand-issues.html" target="_blank" rel="noopener">http://databeauty.com/blog/opinion/2016/09/21/RNA-seq-strand-issues.html</a></li>
</ul>
<p>在建库的时候,选择的建库方法可以是链特异性的,链特异性的文库可以清楚地分出reads的方向是否与转录本相同;常用的链特异性文库方法是基于dUTP的方法.</p>
<table>
<thead>
<tr>
<th>文库类型</th>
<th>PE</th>
<th>SE</th>
<th>建库方法</th>
</tr>
</thead>
<tbody>
<tr>
<td>非特异性</td>
<td>无</td>
<td>无</td>
<td>Standard Illumina</td>
</tr>
<tr>
<td>fr-firststrand</td>
<td>RF</td>
<td>R</td>
<td>dUTP,NSR,NNSR</td>
</tr>
<tr>
<td>fr-secondstrand</td>
<td>FR</td>
<td>F</td>
<td>Ligation,Standard SOLiD</td>
</tr>
</tbody>
</table>
	
	</div>
  <a type="button" href="/2018/10/12/生信基础-链特异性/#more" class="btn btn-default more">Read More</a>
</div>

	       
		     
	
	<!-- display as entry -->	
		<h3 class="title">
			<div class="date"> 2018-10-12 </div>
			<div class="article-title"><a href="/2018/10/12/生信基础-质控的那些事/">生信基础 质控的那些事</a></div>						
		</h3>
	


		     <div class="entry">
  <div class="row">
	
	
		<h2 id="质控的方向"><a href="#质控的方向" class="headerlink" title="质控的方向"></a>质控的方向</h2><p>就平常的生物实验而言，质控是要贯穿到实验的各个阶段的，比如那些阳性阴性对照就可以认为是在进行质控了。而在测序数据的分析过程中，也同样需要将质控贯穿到分析的各个阶段。质控基本上可以分为三个阶段，依据你的分析对象和分析目的不同而有所变化：raw sequence data, alignment, variant calling. 目前比较常见的质控多是局限在raw data的质控这个阶段。虽然如此，另外两个阶段的质控也应该是需要的，因为后续的许多分析也是基于这两个阶段。Raw data 质控的方向通常有以下几类：quality trimming, adapter removal, contaminant filtering. 而在alignment阶段，我们可以通过uniquely mapped reads, signed noise ratio 进行质控。在variant calling阶段，还在调查中……</p>
	
	</div>
  <a type="button" href="/2018/10/12/生信基础-质控的那些事/#more" class="btn btn-default more">Read More</a>
</div>

	       
		     
	
	<!-- display as entry -->	
		<h3 class="title">
			<div class="date"> 2018-10-12 </div>
			<div class="article-title"><a href="/2018/10/12/组装或定量-eXpress/">组装或定量 eXpress</a></div>						
		</h3>
	


		     <div class="entry">
  <div class="row">
	
	
		<a id="more"></a>
<blockquote>
<p><a href="https://pachterlab.github.io/eXpress/downloads/express-1.5.1/express-1.5.1-linux_x86_64.tgz" target="_blank" rel="noopener">https://pachterlab.github.io/eXpress/downloads/express-1.5.1/express-1.5.1-linux_x86_64.tgz</a></p>
</blockquote>
<h2 id="使用方法"><a href="#使用方法" class="headerlink" title="使用方法"></a>使用方法</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br></pre></td><td class="code"><pre><span class="line">express [options]* &lt;target_seqs.fasta&gt; &lt;aligned_reads.(sam/bam)&gt;</span><br><span class="line"></span><br><span class="line"><span class="comment"># &lt;target_seqs.fasta&gt;:	参考序列的fasta文件</span></span><br><span class="line"><span class="comment"># &lt;aligned_reads.(sam/bam)&gt;:	比对结果文件</span></span><br><span class="line"><span class="comment"># -o path: 	指定输出路径</span></span><br><span class="line"><span class="comment"># -B n:	指定进行batch EM rounds计算的次数,以时间为代价提高定量准确性</span></span><br><span class="line"><span class="comment"># -O n:	指定进行online EM rounds计算的次数,以时间为代价提高定量准确性</span></span><br><span class="line"><span class="comment"># -m n:	指定fragment的平均长度</span></span><br><span class="line"><span class="comment"># -s n：指定fragment长度的标准差</span></span><br><span class="line"><span class="comment"># -H str:	以逗号分隔的位置坐标,指定haplotypes的位置。利于等位基因的特异表达定量</span></span><br><span class="line"><span class="comment"># --output-align-samp:	</span></span><br><span class="line"><span class="comment"># --output-align-prob:</span></span><br><span class="line"><span class="comment"># --fr-stranded: 指定链特异性比对的方向,read1比对到forward链,read2比对到reverse链</span></span><br><span class="line"><span class="comment"># --rf-stranded：指定链特异性比对的方向,read1比对到reverse链,read2比对到forward链</span></span><br><span class="line"><span class="comment"># --f-stranded:	 single-end, read比对到forward链</span></span><br><span class="line"><span class="comment"># --r-stranded:	 single-end, read比对到reverse链</span></span><br><span class="line"><span class="comment">#</span></span><br></pre></td></tr></table></figure>
<h2 id="输入文件"><a href="#输入文件" class="headerlink" title="输入文件"></a>输入文件</h2><p>eXpress需要一个multi-fasta格式的输入文件,计算该文件序列的转录丰度。</p>

	
	</div>
  <a type="button" href="/2018/10/12/组装或定量-eXpress/#more" class="btn btn-default more">Read More</a>
</div>

	       
		     
	
	<!-- display as entry -->	
		<h3 class="title">
			<div class="date"> 2018-10-12 </div>
			<div class="article-title"><a href="/2018/10/12/比对软件-Hisat2/">比对软件 Hisat2</a></div>						
		</h3>
	


		     <div class="entry">
  <div class="row">
	
	
		<p>hisat2是快速灵敏的比对软件,可用于全基因组测序，转录组测序，外显子测序的数据比对.基于GCSA（bwt的拓展），我们设计了graph FM index用于比对。hisat2的比对结果是sam格式文件，你可以使用samtools，GATK等软件进行后续的分析.</p>
	
	</div>
  <a type="button" href="/2018/10/12/比对软件-Hisat2/#more" class="btn btn-default more">Read More</a>
</div>

	       
	     </div>
	     <div>
	       <center>
	         <div class="pagination">
<ul class="pagination">
	 
		
    	<li class="prev"><a href="/categories/Bioinformatics/page/2/" class="alignleft prev"><i class="fa fa-arrow-circle-o-left"></i> Prev</a></li>
  		

        <li><a href="/"><i class="fa fa-home"></i>Home</a></li>

		
		   <li class="next"> <a href="/categories/Bioinformatics/page/4/" class="alignright next">Next<i class="fa fa-arrow-circle-o-right"></i></a> </li>          
        
	
</ul>
</div>

	       </center>
	     </div>	
      

</div> <!-- col-md-9/col-md-12 -->


<div class="col-md-3">
	<div id="sidebar">
	
			
  <div id="site_search">
   <div class="form-group">
    <input type="text" id="local-search-input" name="q" results="0" placeholder="Search" class="st-search-input st-default-search-input form-control">
   </div>  
  <div id="local-search-result"></div>
  </div>


		
			
	<div class="widget">
		<h4>Categories</h4>
		<ul class="tag_box inline list-unstyled">
		
			<li><a href="/categories/Bioinformatics/">Bioinformatics<span>36</span></a></li>
		
			<li><a href="/categories/随想/">随想<span>1</span></a></li>
		
		</ul>
	</div>

		
			
	<div class="widget">
		<h4>Tag Cloud</h4>
		<ul class="tag_box inline list-unstyled">		
		
			<li><a href="/tags/定量/">定量<span>3</span></a></li>
		
			<li><a href="/tags/组装/">组装<span>1</span></a></li>
		
			<li><a href="/tags/随想/">随想<span>1</span></a></li>
		
			<li><a href="/tags/流程/">流程<span>1</span></a></li>
		
			<li><a href="/tags/富集分析/">富集分析<span>1</span></a></li>
		
			<li><a href="/tags/可视化/">可视化<span>1</span></a></li>
		
			<li><a href="/tags/扩增子/">扩增子<span>5</span></a></li>
		
			<li><a href="/tags/变异检测/">变异检测<span>1</span></a></li>
		
			<li><a href="/tags/数据处理/">数据处理<span>1</span></a></li>
		
			<li><a href="/tags/生信基础/">生信基础<span>13</span></a></li>
		
			<li><a href="/tags/软件和包/">软件和包<span>22</span></a></li>
		
			<li><a href="/tags/差异分析/">差异分析<span>4</span></a></li>
		
		 
		</ul>
	</div>


		
			
<div class="widget">
  <h4>Recent Posts</h4>
  <ul class="entry list-unstyled">
    
      <li>
        <a href="/2018/11/26/第一次写的文/" title="有了第一次写的文，还差第一个爱的人"><i class="fa fa-file-o"></i>第一次写的文</a>
      </li>
    
      <li>
        <a href="/2018/11/02/可视化-ggplot2入门/" title="听说你要画图？试试 ggplot2 吧，可以搭积木的。"><i class="fa fa-file-o"></i>可视化 ggplot2入门</a>
      </li>
    
      <li>
        <a href="/2018/11/02/数据处理-dplyr/"><i class="fa fa-file-o"></i>数据处理 dplyr</a>
      </li>
    
      <li>
        <a href="/2018/10/28/流程构建-Snakemake使用初步/" title="呦呦呦"><i class="fa fa-file-o"></i>流程构建 Snakemake使用初步</a>
      </li>
    
      <li>
        <a href="/2018/10/28/可变剪切-rMATS和rmats2sashimiplot/"><i class="fa fa-file-o"></i>可变剪切 rMATS和rmats2sashimiplot</a>
      </li>
    
  </ul>
</div>

		
			
<div class="widget">
	<h4>Links</h4>
	<ul class="blogroll list-unstyled">
	
		<li><i class="fa fa-github"></i><a href="https://github.com/wzpan/freemind/" title="Freemind's Github repository." target="_blank" ]);"="">Freemind</a></li>
	
		<li><i class="fa fa-github"></i><a href="http://www.github.com/xizhihui" title="My Github account." target="_blank" ]);"="">xizhihui</a></li>
	
	</ul>
</div>


		
	</div> <!-- sidebar -->
</div> <!-- col-md-3 -->



</div>
	</div>
  </div>
  <div class="container-narrow">
  <footer> <p>
  &copy; 2018 Xizhihui
  
      with help from <a href="http://hexo.io/" target="_blank">Hexo</a> and <a href="http://getbootstrap.com/" target="_blank">Twitter Bootstrap</a>. Theme by <a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind</a>.    
</p> </footer>
</div> <!-- container-narrow -->
  


  
<a id="gotop" href="#">   
  <span>▲</span> 
</a>

<script src="/js/jquery.imagesloaded.min.js"></script>
<script src="/js/gallery.js"></script>
<script src="/js/bootstrap.min.js"></script>
<script src="/js/main.js"></script>
<script src="/js/search.js"></script> 


<link rel="stylesheet" href="/fancybox/jquery.fancybox.css" media="screen" type="text/css">
<script src="/fancybox/jquery.fancybox.pack.js"></script>
<script type="text/javascript">
(function($){
  $('.fancybox').fancybox();
})(jQuery);
</script>



   <script type="text/javascript">      
     var search_path = "search.xml";
	 if (search_path.length == 0) {
	 	search_path = "search.xml";
	 }
	 var path = "/" + search_path;
     searchFunc(path, 'local-search-input', 'local-search-result');
   </script>


<!-- syntax highlighting -->

  <script>
  marked.setOptions({
    highlight: function (code, lang) {
        return hljs.highlightAuto(code).value;
    }
  });
  function Highlighting(){
    var markdowns = document.getElementsByClassName('markdown');
    for(var i=0;i<markdowns.length;i++){
        if(markdowns[i].innerHTML) markdowns[i].innerHTML =marked(markdowns[i].innerHTML);
    }
  }
  window.addEventListener('DOMContentLoaded', Highlighting, false);
  window.addEventListener('load', Highlighting, false);
  </script>


</body>
   </html>
