<!DOCTYPE HTML>
<html>
<head><meta name="generator" content="Hexo 3.8.0">
  <meta charset="utf-8">
  
  <title>Page 3 | Bioinformatics | Xizhihui&#39;s Notes</title>
  <meta name="author" content="Xizhihui">
  
  
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">

  
  <meta property="og:site_name" content="Xizhihui&#39;s Notes">

  
    <meta property="og:image" content="">
  

  
  
    <link href="/favicon.png" rel="icon">
  

  <!-- CSS -->
  <link rel="stylesheet" href="/css/themes/cosmo.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/font-awesome.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/style.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/responsive.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight-default.min.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/google-fonts.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/comment.css" media="screen" type="text/css">
  <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
	<script src="https://cdnjs.cloudflare.com/ajax/libs/es5-shim/4.5.9/es5-shim.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/es5-shim/4.5.7/es5-sham.min.js"></script>
  <![endif]-->

  <script src="/js/jquery-2.0.3.min.js"></script>
  
  <!-- analytics -->
  


<script>
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>


</head>
</html>
 <body>  
  <nav id="main-nav" class="navbar  navbar-default navbar-fixed-top" role="navigation">
    <div class="container">
      <button type="button" class="navbar-header navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
	<span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
       <a class="navbar-brand" href="/">Xizhihui&#39;s Notes</a>
      <div class="collapse navbar-collapse nav-menu">
		<ul class="nav navbar-nav">
		  
		  <li>
			<a href="/archives" title="All the articles.">
			  <i class="fa fa-archive"></i>Archives
			</a>
		  </li>
		  
		  <li>
			<a href="/categories" title="All the categories.">
			  <i class="fa fa-folder"></i>Categories
			</a>
		  </li>
		  
		  <li>
			<a href="/tags" title="All the tags.">
			  <i class="fa fa-tags"></i>Tags
			</a>
		  </li>
		  
		  <li>
			<a href="/about" title="About me.">
			  <i class="fa fa-user"></i>About
			</a>
		  </li>
		  
		</ul>
      </div>
    </div> <!-- container -->
</nav>
<div class="clearfix"></div>

  <div class="container">
  	<div class="content">
    	 

<!-- title -->
<div class="page-header ">
  <h1 class="archive-title-category title ">Bioinformatics</h1>
</div>

<div class="row page">
  <!-- cols -->
  
  <div class="col-md-9">
	

	  <div id="top_search"></div>

      
         <!-- display as entry -->
	     <div class="mypage">
	       
		     
	
	<!-- display as entry -->	
		<h3 class="title">
			<div class="date"> 2018-10-12 </div>
			<div class="article-title"><a href="/2018/10/12/差异分析-RSEM/" title="呦呦呦">差异分析 RSEM</a></div>						
		</h3>
	


		     <div class="entry">
  <div class="row">
	
	
		<p>RSEM利用的是transcripts而非genome。我们有两种方式来构建RSEM转录参考，其一是利用参考基因组来构建；另外一种方式是从许多转录本中构建。</p>
	
	</div>
  <a type="button" href="/2018/10/12/差异分析-RSEM/#more" class="btn btn-default more">Read More</a>
</div>

	       
		     
	
	<!-- display as entry -->	
		<h3 class="title">
			<div class="date"> 2018-10-12 </div>
			<div class="article-title"><a href="/2018/10/12/差异分析-edgeR/">差异分析 edgeR</a></div>						
		</h3>
	


		     <div class="entry">
  <div class="row">
	
	
		<p>下方的代码把edgeR的三种差异分析整合成一个函数, 调用时直接指定参数即可.</p>
<ul>
<li>classcial</li>
<li>glm: likelihood ratio test/ quasi-likelihood F-test</li>
<li><ul>
<li>quasi-likelihood(qlf): 推荐用于差异分析，因为他对错误率限制较好。</li>
</ul>
</li>
<li><ul>
<li>likelihood(lrt)：对与单细胞RNA-测序和没有重复的数据较好</li>
</ul>
</li>
</ul>
	
	</div>
  <a type="button" href="/2018/10/12/差异分析-edgeR/#more" class="btn btn-default more">Read More</a>
</div>

	       
		     
	
	<!-- display as entry -->	
		<h3 class="title">
			<div class="date"> 2018-10-12 </div>
			<div class="article-title"><a href="/2018/10/12/差异分析-DESeq2/">差异分析 DESeq2</a></div>						
		</h3>
	


		     <div class="entry">
  <div class="row">
	
	
		<blockquote>
<p>与DESeq类似的包有：edgeR、limma、DSS、EBSeq、baySeq</p>
</blockquote>
<h2 id="一、核心逻辑代码"><a href="#一、核心逻辑代码" class="headerlink" title="一、核心逻辑代码"></a>一、核心逻辑代码</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># countData: 以样品为列名,基因为行名的表达矩阵</span></span><br><span class="line"><span class="comment"># colData的形式: 以样品作为行名,列是样品对应的分组类型</span></span><br><span class="line"><span class="comment"># 生成count matrix</span></span><br><span class="line">dds &lt;- DESeqDataSetFromMatrix(countData = cts,</span><br><span class="line">                              colData = coldata,</span><br><span class="line">                              design = ~ batch + condition)</span><br><span class="line"><span class="comment"># 生成DESeq数据集</span></span><br><span class="line">dds &lt;- DESeq(dds)</span><br><span class="line"><span class="comment"># 进行比较，获得结果</span></span><br><span class="line">res &lt;- results(dds, contrast=c(<span class="string">'condition'</span>, <span class="string">'treat'</span>, <span class="string">'ctrl'</span>))</span><br><span class="line">resultsNames(dds)</span><br><span class="line">res &lt;- lfcShrink(dds, coef=<span class="number">2</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment">#  DESeqDataSetFromTximport:   由Salmon, Saifish, kallisto生成</span></span><br><span class="line"><span class="comment">#  DESeqDataSetFromHTSeq:      由htseq-counts生成</span></span><br><span class="line"><span class="comment">#  DESeqDataSet:               由RangedSummarizedExpriment生成</span></span><br></pre></td></tr></table></figure>
	
	</div>
  <a type="button" href="/2018/10/12/差异分析-DESeq2/#more" class="btn btn-default more">Read More</a>
</div>

	       
		     
	
	<!-- display as entry -->	
		<h3 class="title">
			<div class="date"> 2018-10-12 </div>
			<div class="article-title"><a href="/2018/10/12/生信基础-常用gene_ID的转换/">生信基础 常用gene_ID的转换</a></div>						
		</h3>
	


		     <div class="entry">
  <div class="row">
	
	
		<h2 id="ID-类型"><a href="#ID-类型" class="headerlink" title="ID 类型"></a>ID 类型</h2><table>
<thead>
<tr>
<th>ID 示例</th>
<th>ID 来源</th>
</tr>
</thead>
<tbody>
<tr>
<td>ENSG00000116717</td>
<td>Ensemble ID</td>
</tr>
<tr>
<td>GA45A_HUMAN</td>
<td>UniProtKB/Swiss-Prot, entry name</td>
</tr>
<tr>
<td>A5PJB2_BOVIN</td>
<td>UniProtKB/TrEMBL, entry name</td>
</tr>
<tr>
<td>A2BC19, P12345,</td>
<td>A0A022YWF9    UniProt, accession number</td>
</tr>
<tr>
<td>GLA, GLB, UGT1A1</td>
<td>HGNC Gene Symbol</td>
</tr>
<tr>
<td>U12345, AF123456</td>
<td>GenBank, NCBI, accession number</td>
</tr>
<tr>
<td>NT_123456, NM_123456, NP_123456</td>
<td>RefSeq, NCBI, accession number</td>
</tr>
<tr>
<td>10598, 717</td>
<td>Entrez ID, NCBI</td>
</tr>
<tr>
<td>uc001ett, uc031tla.1</td>
<td>UCSC ID</td>
</tr>
</tbody>
</table>
	
	</div>
  <a type="button" href="/2018/10/12/生信基础-常用gene_ID的转换/#more" class="btn btn-default more">Read More</a>
</div>

	       
		     
	
	<!-- display as entry -->	
		<h3 class="title">
			<div class="date"> 2018-10-12 </div>
			<div class="article-title"><a href="/2018/10/12/生信基础-reads-重复和比对重复/" title="呦呦呦">生信基础 reads 重复和比对重复</a></div>						
		</h3>
	


		     <div class="entry">
  <div class="row">
	
	
		<p>在说duplication之前,我们有必要说一下PCR bias和unique reads. PCR bias是指在PCR扩增的过程中,PCR引物会偏好性地同某条DNA链结合,导致的结果就是这条DNA链被扩增的数目要更多;而不是所有的DNA链被平行扩增.那么什么又是unique reads呢?在得到测序结果后,对于任意两条reads,只要其reads的起点,中间序列,终点这三点有一点不同,这两条reads就是互为unique reads. 所以我们来看, 建库后PCR扩增的测序结果里面,肯定有很多条reads不是unique reads,这些reads就是所谓的duplication, 这其中肯定也包含了PCR bias引起的额外重复.</p>
	
	</div>
  <a type="button" href="/2018/10/12/生信基础-reads-重复和比对重复/#more" class="btn btn-default more">Read More</a>
</div>

	       
		     
	
	<!-- display as entry -->	
		<h3 class="title">
			<div class="date"> 2018-10-12 </div>
			<div class="article-title"><a href="/2018/10/12/生信基础-unique比对的获取/">生信基础 unique比对的获取</a></div>						
		</h3>
	


		     <div class="entry">
  <div class="row">
	
	
		<h2 id="1-Sam文件各标签含义（tophat-hisat2"><a href="#1-Sam文件各标签含义（tophat-hisat2" class="headerlink" title="1.Sam文件各标签含义（tophat/hisat2)"></a>1.Sam文件各标签含义（tophat/hisat2)</h2><ul>
<li>NH:i:<n>: N=1时 为unique。常用于tophat/hisat2产生的sam文件unique read筛选。</n></li>
<li>CC:Z: 当为‘=’为map到同一条基因上，一般在map基因组时由于内含子存在而容易出现，他只代表两种不同的方式，计数时应记为1。此处一般为其他基因的名字。CP:i 和HI：i标签为map到第i条基因及起始位置。</li>
<li>YT:Z:S 代表的含义与bowtie产生的sam也不同。具体还未知！其他标签AS，XN,XM,XO,XG,NM,MD等如下图可以看出都相同。</li>
</ul>
	
	</div>
  <a type="button" href="/2018/10/12/生信基础-unique比对的获取/#more" class="btn btn-default more">Read More</a>
</div>

	       
		     
	
	<!-- display as entry -->	
		<h3 class="title">
			<div class="date"> 2018-10-12 </div>
			<div class="article-title"><a href="/2018/10/12/生信基础-各种数据的下载/">生信基础 各种数据的下载</a></div>						
		</h3>
	


		     <div class="entry">
  <div class="row">
	
	
		<h2 id="三大数据库版本对应情况"><a href="#三大数据库版本对应情况" class="headerlink" title="三大数据库版本对应情况"></a>三大数据库版本对应情况</h2><table>
<thead>
<tr>
<th>NCBI</th>
<th>UCSC</th>
<th>Ensemble</th>
</tr>
</thead>
<tbody>
<tr>
<td>GRCh36</td>
<td>hg18</td>
<td>ENSEMBL release_52</td>
</tr>
<tr>
<td>GRCh37</td>
<td>hg19</td>
<td>ENSEMBL release_59/61/64/68/69/75</td>
</tr>
<tr>
<td>GRCh38</td>
<td>hg38</td>
<td>ENSEMBL release_76/77/78/80/81/82</td>
</tr>
</tbody>
</table>
	
	</div>
  <a type="button" href="/2018/10/12/生信基础-各种数据的下载/#more" class="btn btn-default more">Read More</a>
</div>

	       
		     
	
	<!-- display as entry -->	
		<h3 class="title">
			<div class="date"> 2018-10-12 </div>
			<div class="article-title"><a href="/2018/10/12/生信基础-无重复样本的差异分析/">生信基础 无重复样本的差异分析</a></div>						
		</h3>
	


		     <div class="entry">
  <div class="row">
	
	
		<ul>
<li>edgeR</li>
<li>DESeq2</li>
<li>GFOLD</li>
</ul>
	
	</div>
  <a type="button" href="/2018/10/12/生信基础-无重复样本的差异分析/#more" class="btn btn-default more">Read More</a>
</div>

	       
		     
	
	<!-- display as entry -->	
		<h3 class="title">
			<div class="date"> 2018-10-12 </div>
			<div class="article-title"><a href="/2018/10/12/生信基础-生信数据格式汇总/">生信基础 生信数据格式汇总</a></div>						
		</h3>
	


		     <div class="entry">
  <div class="row">
	
	
		<h2 id="Fastq和Fasta格式"><a href="#Fastq和Fasta格式" class="headerlink" title="Fastq和Fasta格式"></a>Fastq和Fasta格式</h2><h3 id="FASTA"><a href="#FASTA" class="headerlink" title="FASTA"></a>FASTA</h3><p>Fasta格式是存储序列相关信息的一种格式，包含两个部分。Fasta第一个部分是以“&gt;”开头的一行。以序列来源、序列ID为值对，用竖线“|”进行分隔。在最后竖线处，以空格分开，添加序列相关描述。第二部分是序列的详细信息，多数为序列组成。其基本形式如下：</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">&gt;ENSEMBL|geneID|source2|geneID description</span><br><span class="line">序列ATCG</span><br></pre></td></tr></table></figure>
	
	</div>
  <a type="button" href="/2018/10/12/生信基础-生信数据格式汇总/#more" class="btn btn-default more">Read More</a>
</div>

	       
		     
	
	<!-- display as entry -->	
		<h3 class="title">
			<div class="date"> 2018-10-12 </div>
			<div class="article-title"><a href="/2018/10/12/加权共表达基因网络分析-WGCNA/" title="呦呦呦">加权共表达基因网络分析 WGCNA</a></div>						
		</h3>
	


		     <div class="entry">
  <div class="row">
	
	
		<p>在拿到高通量测序数据以后(特别是基因表达数据),通常要分析基因与表型之间的相关性,以探究基因对表型所起关键的调节功能；而加权基因共表达网络分析(Weighted Gene Coexpression Network Analysis)就是其中比较实用的一种分析方法。那么,什么是加权基因共表达网络分析呢？要了解这个,我们需要对以下前提有所了解。</p>
<ul>
<li>功能相关的基因,其表达水平/表达模式也基本上是相似的</li>
<li>表达水平高度相关的基因具有潜在的共有调控机制或参与相似的生物学过程</li>
<li>如果得到某个表达模式中涉及到许多基因,其中有些基因是已知的,基于上面两点,那么就可以通过该已知基因的功能推测未知基因的功能</li>
</ul>
<p>根据上面三点,我们就要得到有许多基因涉及的表达模式,而这个可以通过对基因的表达数据进行聚类得到.讲到这里,WGCNA实质上也是一种聚类方法.而在WGCNA之前,现在也在使用的另一种基因与表型的分析方法,那就是基因共表达网络分析.而WGCNA相较于后者来讲,有何优势?</p>
	
	</div>
  <a type="button" href="/2018/10/12/加权共表达基因网络分析-WGCNA/#more" class="btn btn-default more">Read More</a>
</div>

	       
	     </div>
	     <div>
	       <center>
	         <div class="pagination">
<ul class="pagination">
	 
		
    	<li class="prev"><a href="/categories/Bioinformatics/page/2/" class="alignleft prev"><i class="fa fa-arrow-circle-o-left"></i> Prev</a></li>
  		

        <li><a href="/"><i class="fa fa-home"></i>Home</a></li>

		
		   <li class="next"> <a href="/categories/Bioinformatics/page/4/" class="alignright next">Next<i class="fa fa-arrow-circle-o-right"></i></a> </li>          
        
	
</ul>
</div>

	       </center>
	     </div>	
      

</div> <!-- col-md-9/col-md-12 -->


<div class="col-md-3">
	<div id="sidebar">
	
			
  <div id="site_search">
   <div class="form-group">
    <input type="text" id="local-search-input" name="q" results="0" placeholder="Search" class="st-search-input st-default-search-input form-control">
   </div>  
  <div id="local-search-result"></div>
  </div>


		
			
	<div class="widget">
		<h4>Categories</h4>
		<ul class="tag_box inline list-unstyled">
		
			<li><a href="/categories/Bioinformatics/">Bioinformatics<span>38</span></a></li>
		
			<li><a href="/categories/Python/">Python<span>1</span></a></li>
		
			<li><a href="/categories/R/">R<span>1</span></a></li>
		
			<li><a href="/categories/杂货铺/">杂货铺<span>1</span></a></li>
		
			<li><a href="/categories/随想/">随想<span>1</span></a></li>
		
		</ul>
	</div>

		
			
	<div class="widget">
		<h4>Tag Cloud</h4>
		<ul class="tag_box inline list-unstyled">		
		
			<li><a href="/tags/定量/">定量<span>3</span></a></li>
		
			<li><a href="/tags/数据处理/">数据处理<span>1</span></a></li>
		
			<li><a href="/tags/软件和包-单细胞/">软件和包 单细胞<span>1</span></a></li>
		
			<li><a href="/tags/组装/">组装<span>1</span></a></li>
		
			<li><a href="/tags/软件和包/">软件和包<span>22</span></a></li>
		
			<li><a href="/tags/差异分析/">差异分析<span>4</span></a></li>
		
			<li><a href="/tags/R基础/">R基础<span>1</span></a></li>
		
			<li><a href="/tags/可视化/">可视化<span>1</span></a></li>
		
			<li><a href="/tags/变异检测/">变异检测<span>1</span></a></li>
		
			<li><a href="/tags/富集分析/">富集分析<span>1</span></a></li>
		
			<li><a href="/tags/扩增子/">扩增子<span>5</span></a></li>
		
			<li><a href="/tags/稀疏矩阵/">稀疏矩阵<span>1</span></a></li>
		
			<li><a href="/tags/生信基础/">生信基础<span>13</span></a></li>
		
			<li><a href="/tags/算法/">算法<span>1</span></a></li>
		
			<li><a href="/tags/bug/">bug<span>1</span></a></li>
		
			<li><a href="/tags/随想/">随想<span>1</span></a></li>
		
			<li><a href="/tags/流程/">流程<span>1</span></a></li>
		
		 
		</ul>
	</div>


		
			
<div class="widget">
  <h4>Recent Posts</h4>
  <ul class="entry list-unstyled">
    
      <li>
        <a href="/2018/12/23/稀疏矩阵的表示法/"><i class="fa fa-file-o"></i>稀疏矩阵的表示法</a>
      </li>
    
      <li>
        <a href="/2018/12/23/Seurat-与-Cellranger-之间互通的二三事/"><i class="fa fa-file-o"></i>Seurat 与 Cellranger 之间互通的二三事</a>
      </li>
    
      <li>
        <a href="/2018/12/06/动态编程/" title="算法的力量真强大。"><i class="fa fa-file-o"></i>动态编程</a>
      </li>
    
      <li>
        <a href="/2018/11/27/R-中的函数式/" title="函数式编程，纯碎的函数。"><i class="fa fa-file-o"></i>R 中的函数式</a>
      </li>
    
      <li>
        <a href="/2018/11/27/Bugs/" title="当你打开这里，你要承认，你无时无刻都在产生各种 Bugs，别人写代码，你写 Bug，whatever，这里给你提供杀虫剂。"><i class="fa fa-file-o"></i>杀虫剂</a>
      </li>
    
  </ul>
</div>

		
			
<div class="widget">
	<h4>Links</h4>
	<ul class="blogroll list-unstyled">
	
		<li><i class="fa fa-github"></i><a href="https://github.com/wzpan/freemind/" title="Freemind's Github repository." target="_blank" ]);"="">Freemind</a></li>
	
		<li><i class="fa fa-github"></i><a href="http://www.github.com/xizhihui" title="My Github account." target="_blank" ]);"="">xizhihui</a></li>
	
	</ul>
</div>


		
	</div> <!-- sidebar -->
</div> <!-- col-md-3 -->



</div>
	</div>
  </div>
  <div class="container-narrow">
  <footer> <p>
  &copy; 2018 Xizhihui
  
      with help from <a href="http://hexo.io/" target="_blank">Hexo</a> and <a href="http://getbootstrap.com/" target="_blank">Twitter Bootstrap</a>. Theme by <a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind</a>.    
</p> </footer>
</div> <!-- container-narrow -->
  


  
<a id="gotop" href="#">   
  <span>▲</span> 
</a>

<script src="/js/jquery.imagesloaded.min.js"></script>
<script src="/js/gallery.js"></script>
<script src="/js/bootstrap.min.js"></script>
<script src="/js/main.js"></script>
<script src="/js/search.js"></script> 


<link rel="stylesheet" href="/fancybox/jquery.fancybox.css" media="screen" type="text/css">
<script src="/fancybox/jquery.fancybox.pack.js"></script>
<script type="text/javascript">
(function($){
  $('.fancybox').fancybox();
})(jQuery);
</script>



   <script type="text/javascript">      
     var search_path = "search.xml";
	 if (search_path.length == 0) {
	 	search_path = "search.xml";
	 }
	 var path = "/" + search_path;
     searchFunc(path, 'local-search-input', 'local-search-result');
   </script>


<!-- syntax highlighting -->


</body>
   </html>
