<!DOCTYPE HTML>
<html>
<head><meta name="generator" content="Hexo 3.8.0">
  <meta charset="utf-8">
  
  <title>加权共表达基因网络分析 WGCNA | Xizhihui&#39;s Notes</title>
  <meta name="author" content="Xizhihui">
  
  <meta name="description" content="呦呦呦">
  
  
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">

  <meta property="og:title" content="加权共表达基因网络分析 WGCNA">
  <meta property="og:site_name" content="Xizhihui&#39;s Notes">

  
    <meta property="og:image" content="">
  

  
  
    <link href="/favicon.png" rel="icon">
  

  <!-- CSS -->
  <link rel="stylesheet" href="/css/themes/cosmo.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/font-awesome.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/style.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/responsive.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/highlight-default.min.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/google-fonts.css" media="screen" type="text/css">
  <link rel="stylesheet" href="/css/comment.css" media="screen" type="text/css">
  <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
	<script src="https://cdnjs.cloudflare.com/ajax/libs/es5-shim/4.5.9/es5-shim.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/es5-shim/4.5.7/es5-sham.min.js"></script>
  <![endif]-->

  <script src="/js/jquery-2.0.3.min.js"></script>
  
  <!-- analytics -->
  


<script>
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>


</head>
</html>
 <body>  
  <nav id="main-nav" class="navbar  navbar-default navbar-fixed-top" role="navigation">
    <div class="container">
      <button type="button" class="navbar-header navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
	<span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
       <a class="navbar-brand" href="/">Xizhihui&#39;s Notes</a>
      <div class="collapse navbar-collapse nav-menu">
		<ul class="nav navbar-nav">
		  
		  <li>
			<a href="/archives" title="All the articles.">
			  <i class="fa fa-archive"></i>Archives
			</a>
		  </li>
		  
		  <li>
			<a href="/categories" title="All the categories.">
			  <i class="fa fa-folder"></i>Categories
			</a>
		  </li>
		  
		  <li>
			<a href="/tags" title="All the tags.">
			  <i class="fa fa-tags"></i>Tags
			</a>
		  </li>
		  
		  <li>
			<a href="/about" title="About me.">
			  <i class="fa fa-user"></i>About
			</a>
		  </li>
		  
		</ul>
      </div>
    </div> <!-- container -->
</nav>
<div class="clearfix"></div>

  <div class="container">
  	<div class="content">
    	 


	
		<div class="page-header ">		
			<h1 class="title "> 加权共表达基因网络分析 WGCNA</h1>
		</div>		
	






<div class="row post">
	<!-- cols -->
	
	<div id="top_meta"></div>
	<div class="col-md-9">
	

	<!-- content -->
	<div class="mypage">		
	  
		 <div class="alert alert-success description">
			<i class="fa fa-info-circle"></i> <p>呦呦呦</p>
			
		 </div> <!-- alert -->
	  		

	  <p>在拿到高通量测序数据以后(特别是基因表达数据),通常要分析基因与表型之间的相关性,以探究基因对表型所起关键的调节功能；而加权基因共表达网络分析(Weighted Gene Coexpression Network Analysis)就是其中比较实用的一种分析方法。那么,什么是加权基因共表达网络分析呢？要了解这个,我们需要对以下前提有所了解。</p>
<ul>
<li>功能相关的基因,其表达水平/表达模式也基本上是相似的</li>
<li>表达水平高度相关的基因具有潜在的共有调控机制或参与相似的生物学过程</li>
<li>如果得到某个表达模式中涉及到许多基因,其中有些基因是已知的,基于上面两点,那么就可以通过该已知基因的功能推测未知基因的功能</li>
</ul>
<p>根据上面三点,我们就要得到有许多基因涉及的表达模式,而这个可以通过对基因的表达数据进行聚类得到.讲到这里,WGCNA实质上也是一种聚类方法.而在WGCNA之前,现在也在使用的另一种基因与表型的分析方法,那就是基因共表达网络分析.而WGCNA相较于后者来讲,有何优势?</p>
<a id="more"></a>
<h2 id="一、总述"><a href="#一、总述" class="headerlink" title="一、总述"></a>一、总述</h2><h3 id="1-1-相较于GCNA-WGCNA的优势"><a href="#1-1-相较于GCNA-WGCNA的优势" class="headerlink" title="1.1 相较于GCNA,WGCNA的优势"></a>1.1 相较于GCNA,WGCNA的优势</h3><ol>
<li>对Pearson coefficient的绝对值取β次幂构建相关性<br>这样做的优势在于,如果两个基因相关性较高的话,取幂后相关性更高;相关性较低,取幂后相关性更低.如此,就更加凸显两个基因的相关性.</li>
<li>相关性阈值的软设定<br>使用相关性软阈值设定的方法,进一步涵盖潜在的显著相关性基因;而非一刀切,因为你无法判定0.89与0.90之间,前者就相关性不显著,后者就显著相关.如果一刀切的话,你的结果则依据这个相关性阈值的设定不同,而显著不同</li>
<li>以TOM进一步表征基因的相关性<br>在基因两两相关而构建的网络（图）中，通过判断两个基因与其他基因相关性的相似程度来确定这两个基因的实际相关性（topological overlap），以此聚类得出的分组更符合前面所述的前提。</li>
</ol>
<h3 id="1-2-相较于差异分析-WGCNA优势"><a href="#1-2-相较于差异分析-WGCNA优势" class="headerlink" title="1.2 相较于差异分析,WGCNA优势"></a>1.2 相较于差异分析,WGCNA优势</h3><ul>
<li>从更高层次理解基因表达与表型之间的因果关系<br>相较于简单比较不同处理中基因表达的差异，WGCNA立足于基因模块（多个基因）的共有效应，更加符合实际情况，因为生命活动就是多中因素共同调节的结果。</li>
</ul>
<h3 id="1-3-基本术语"><a href="#1-3-基本术语" class="headerlink" title="1.3 基本术语"></a>1.3 基本术语</h3><table>
<thead>
<tr>
<th>术语</th>
<th>解析</th>
</tr>
</thead>
<tbody>
<tr>
<td>共表达网络(co-expression network)</td>
<td>无向的、加权的表达网络。无符号的a(i,j) =</td>
<td>cor(xi, xj)</td>
<td>^β,有符号的a(i,j) =</td>
<td>(1 + cor(xi, xj))/2</td>
<td>^β</td>
</tr>
<tr>
<td>模块(module)</td>
<td>高度相关的基因集</td>
</tr>
<tr>
<td>连接性(connectivity)</td>
<td>网络中某个基因与其他基因连接值的总和</td>
</tr>
<tr>
<td>模块内部连接性(intramodular connectivity)</td>
<td>模块中某个基因与同模块其他基因的连接值总和</td>
</tr>
<tr>
<td>模块特征(module eigengene)</td>
<td>给定模块的第一个主成分</td>
</tr>
<tr>
<td>特征显著性(eigengene significance)</td>
<td>模块与表型之间的相关系数</td>
</tr>
<tr>
<td>模块关系(基于模块特征的连接性)</td>
<td>给定模块的模块特征与基因表达的相关性</td>
</tr>
<tr>
<td>关键基因(hub gene)</td>
<td>给定模块中具有较大连接性的基因</td>
</tr>
<tr>
<td>基因显著性(gene significance)</td>
<td>基因表达的显著性(-log(p-value))</td>
</tr>
<tr>
<td>模块显著性(module significant)</td>
<td>给定模块中所有基因显著性的绝对值的平均值</td>
</tr>
</tbody>
</table>
<h3 id="1-4-基本过程"><a href="#1-4-基本过程" class="headerlink" title="1.4 基本过程"></a>1.4 基本过程</h3><ol>
<li>计算基因的相关性(Pearson correlation的β次幂，connectivity, K)</li>
<li>依据connectivity计算两两基因的topological overlap（relative connectivity）</li>
<li>根据拓扑重叠值构建基因表达网络</li>
<li>对此进行层次聚类，确定共表达模块（表达模式）</li>
<li>选取感兴趣的模块进行后续分析：模块的功能富集、模块与性状的相关性、模块与样本间的相关系数</li>
<li>挖掘模块的关键信息：寻找模块核心基因（hub gene)、预测基因功能</li>
</ol>
<h2 id="二、分析过程"><a href="#二、分析过程" class="headerlink" title="二、分析过程"></a>二、分析过程</h2><h2 id="2-1-导入数据、预处理"><a href="#2-1-导入数据、预处理" class="headerlink" title="2.1 导入数据、预处理"></a>2.1 导入数据、预处理</h2><p>这里主要有两个目的，一个是检测数据集是否都满足WGCNA的要求，二是排除离群值，后者需要你根据数据产生的聚类树，选择合适的cutHeight来去除离群值。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br></pre></td><td class="code"><pre><span class="line">expr = read.table(<span class="string">'./clean_data/expr_matrix.txt'</span>)</span><br><span class="line">gsg = goodSamplesGenes(expr)</span><br><span class="line"><span class="keyword">if</span> (!gsg$allOK) expr = expr[gsg$goodSamples, gsg$goodGenes]</span><br><span class="line">sampleTree = hclust(as.dist(expr), method=<span class="string">'average'</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># [ Visualization 1 ], 根据画出的图确定cutHeight,这里假设是120</span></span><br><span class="line">pdf(file=<span class="string">'visual_01_sampleTree_with_outlier.pdf'</span>)</span><br><span class="line">plot(sampleTree,</span><br><span class="line">	main=<span class="string">'Sample Clustering to detect outliers'</span>,</span><br><span class="line">	xlab=<span class="string">''</span>, sub=<span class="string">''</span>)</span><br><span class="line">abline(h=<span class="number">120</span>, col=<span class="string">'red'</span>)</span><br><span class="line">dev.off()</span><br><span class="line"></span><br><span class="line">cutHeight = <span class="number">120</span></span><br><span class="line">sample_cut = cutreeStatic(sampleTree, cutHeight=cutHeight)</span><br><span class="line">expr = expr[sample_cut == <span class="number">1</span>, ]</span><br><span class="line">sampleTreeCut = hclust(as.dist(expr), method=<span class="string">'average'</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># [ Visualization 2], 可视化样本聚类树和表型数据</span></span><br><span class="line">traits = read.table(file=<span class="string">'./clean_data/traits.txt'</span>)</span><br><span class="line">pdf(file=<span class="string">'visual_02_sampleCutTree_with_traits.pdf'</span>)</span><br><span class="line">colors = numbers2colors(traits)</span><br><span class="line">plotDendroAndColors(sampleTreeCut, colors,</span><br><span class="line">					groupLabels=colnames(traits),</span><br><span class="line">                    main=<span class="string">'Sample Clustering and traits'</span>)</span><br><span class="line">dev.off()</span><br><span class="line"></span><br><span class="line">save(sampleTree, sampleTreeCut, expr, file=<span class="string">'01_sampleTree_cut_outlier.RData'</span>)</span><br></pre></td></tr></table></figure>
<h2 id="2-2-获取软阈值"><a href="#2-2-获取软阈值" class="headerlink" title="2.2 获取软阈值"></a>2.2 获取软阈值</h2><p>这里的软阈值选取power significance 大于0.8或0.85,0.9的power值。一般来说都在7-10之间。如果数据较好，可以直接通过”beta=soft$EstiamteSoft”获取，否则通过可视化获取。如果数据展示出的scale-free network不太好，那就多选取一些数据，而非仅仅基于有差异表达的基因表达矩阵。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><span class="line">powers = <span class="number">1</span>:<span class="number">20</span></span><br><span class="line">powerSigR2 = <span class="number">0.8</span></span><br><span class="line">soft = pickSoftThreshold(expr, powerVectors=powers, verbose=<span class="number">3</span>)</span><br><span class="line">beta = soft$EstimateSoft</span><br><span class="line"></span><br><span class="line"><span class="comment"># [ visualizaiton 3], 可视化软阈值的选取</span></span><br><span class="line">pdf(file=<span class="string">'visual_03_soft_pick_threshold.pdf'</span>)</span><br><span class="line">par(mfrow=c(<span class="number">1</span>,<span class="number">2</span>))</span><br><span class="line">plot(soft$fitIndices$Power,  -sign(soft$fitIndices$slope) * soft$fitIndices$SFT.R.sq, </span><br><span class="line">     xlab=<span class="string">'Soft Threshold (power)'</span>,</span><br><span class="line">     ylab=<span class="string">'Scale Free Topology Model Fit, signed R^2'</span>, type=<span class="string">'n'</span>,</span><br><span class="line">     main=paste(<span class="string">'Scale Independce'</span>))</span><br><span class="line">text(soft$fitIndices$Power, -sign(soft$fitIndices$slope) * soft$fitIndices$SFT.R.sq,</span><br><span class="line">     labels=soft$fitIndices$Power, cex=<span class="number">0.9</span>, col=<span class="string">'red'</span>)</span><br><span class="line">abline(h=<span class="number">0.8</span>, col=<span class="string">'blue'</span>)</span><br><span class="line">plot(soft$fitIndices$Power, soft$fitIndices$mean.k.,</span><br><span class="line">     xlab=<span class="string">'Soft Threshold (power)'</span>,</span><br><span class="line">     ylab=<span class="string">'Mean connectivity'</span>, type=<span class="string">'n'</span>,</span><br><span class="line">     main=paste(<span class="string">'Mean connectivity'</span>))</span><br><span class="line">text(soft$fitIndices$Power, soft$fitIndices$mean.k.,</span><br><span class="line">     labels=soft$fitIndices$Power, cex=<span class="number">0.9</span>, col=<span class="string">'red'</span>)</span><br><span class="line">dev.off()</span><br><span class="line"></span><br><span class="line">save(soft, beta, file=<span class="string">'02_soft_threshold.RData'</span>)</span><br></pre></td></tr></table></figure>
<h2 id="2-3-构建网络"><a href="#2-3-构建网络" class="headerlink" title="2.3 构建网络"></a>2.3 构建网络</h2><p>构建网络有一步法和多步法，由于后续分析和可视化的需要，这里使用多步法。在我们通过拓扑网络计算节点距离进行聚类后，还可以根据聚类后的合并模块特征靠近的或者较小的模块。</p>
<h3 id="2-3-3-计算拓扑网络和内部连接性，基于此构建聚类树"><a href="#2-3-3-计算拓扑网络和内部连接性，基于此构建聚类树" class="headerlink" title="2.3.3 计算拓扑网络和内部连接性，基于此构建聚类树"></a>2.3.3 计算拓扑网络和内部连接性，基于此构建聚类树</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><span class="line">softConnect = softConnectivity(expr, power=beta)-<span class="number">1</span></span><br><span class="line">adjacency = adjacency(expr, power=beta)</span><br><span class="line">TOM = TOMsimilarity(adjacency) <span class="comment"># 一步到位 TOM = TOMsimilarityFromExpr(expr,beta)</span></span><br><span class="line">dissTOM = <span class="number">1</span> - TOM <span class="comment"># dissTOM = TOMdist(adjacency)</span></span><br><span class="line">geneTree = hclust(as.dist(dissTOM), method=<span class="string">'average'</span>)</span><br><span class="line"><span class="comment"># [visualization 4], 可视化拓扑图</span></span><br><span class="line">	pdf(file=<span class="string">'visual_04_scale_free_topology_plot.pdf'</span>)</span><br><span class="line">	scaleFreePlot(softConnect, truncated=<span class="literal">F</span>,</span><br><span class="line">		main=paste(<span class="string">'Scale Free Topology, power='</span>, beta,sep=<span class="string">''</span>))</span><br><span class="line">	dev.off()</span><br><span class="line"><span class="comment"># [visualization 5], 可视化拓扑网络聚类树</span></span><br><span class="line">	pdf(file=<span class="string">'visual_05_raw_gene_tree_on_dissTOM.pdf'</span>)</span><br><span class="line">	plot(geneTree, xlab=<span class="string">''</span>, sub=<span class="string">''</span>,</span><br><span class="line">		    main=<span class="string">'Gene Clustering on TOM-based dissimilarity'</span>,</span><br><span class="line">		    labels=<span class="literal">F</span>, hang=<span class="number">0.04</span>)</span><br><span class="line">	dev.off()</span><br><span class="line"><span class="comment"># [visualization 5.1], 可视化模块与模块之间的拓扑相异性，揭示基于拓扑重叠进行聚类的效果</span></span><br><span class="line">	pdf(file=<span class="string">'visual_05_dissTOM_between_module.pdf'</span>)</span><br><span class="line">	TOMplot(dissTOM, geneTree)</span><br><span class="line">	dev.off()</span><br><span class="line">save(softConnect, adjacency, TOM, geneTree, file=<span class="string">'03_conn_adj_TOM.RData'</span>)</span><br><span class="line">rm(c(soft, softConnect))</span><br></pre></td></tr></table></figure>
<h3 id="2-3-2-把拓扑网络聚类树进行动态修剪，产生初始模块"><a href="#2-3-2-把拓扑网络聚类树进行动态修剪，产生初始模块" class="headerlink" title="2.3.2 把拓扑网络聚类树进行动态修剪，产生初始模块"></a>2.3.2 把拓扑网络聚类树进行动态修剪，产生初始模块</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">minModuleSize=<span class="number">30</span></span><br><span class="line">dissTOM_cut = cutreeDynamic(dendro=geneTree, distM=dissTOM,</span><br><span class="line">                            deepSplit=<span class="number">2</span>, pamRespectsDendro=<span class="literal">F</span>,</span><br><span class="line">                            minClusterSize=minModuleSize)</span><br><span class="line">dynamicColors = labels2colors(dissTOM_cut)</span><br><span class="line"><span class="comment"># [visualization 6] 可视化拓扑重叠相异性的多维尺度图(mds-plot)</span></span><br><span class="line">	pdf(file=<span class="string">'visual_06_MDS_plot_on_dissTOM_cut.pdf'</span>)</span><br><span class="line">	dimension_num = <span class="number">2</span></span><br><span class="line">	cmd = cmdscale(as.dist(dissTOM_cut),dimensional_num)</span><br><span class="line">	plot(cmd, col=as.character(dynamicColors), main=<span class="string">'MDS plot'</span>,</span><br><span class="line">		xlab=<span class="string">'Scaling Dimension 1'</span>, ylab=<span class="string">'Scaling Dimension 2'</span>)</span><br><span class="line">	dev.off()</span><br></pre></td></tr></table></figure>
<h3 id="2-3-3-计算模块特征，合并特征相似或靠近的模块，产生最终模块"><a href="#2-3-3-计算模块特征，合并特征相似或靠近的模块，产生最终模块" class="headerlink" title="2.3.3 计算模块特征，合并特征相似或靠近的模块，产生最终模块"></a>2.3.3 计算模块特征，合并特征相似或靠近的模块，产生最终模块</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br></pre></td><td class="code"><pre><span class="line">MEs = moduleEigengenes(expr, dynamicColors)$eigengenes</span><br><span class="line">dissMEs = <span class="number">1</span> - cor(MEs)</span><br><span class="line">dissMEs_tree = hclust(as.dist(dissMEs), method=<span class="string">'average'</span>)</span><br><span class="line">MEDissThreshold = <span class="number">0.25</span></span><br><span class="line">mergedMEs = mergeCloseModules(expr, dynamicColors,</span><br><span class="line">							cutHeight=MEDissThreshold, verbose=<span class="number">3</span>)</span><br><span class="line"><span class="comment"># [visualization 6], 可视化拓扑网络聚类树的动态修剪和合并产生的模块比较</span></span><br><span class="line">	pdf(file=<span class="string">'visual_06_raw_tree_with_dynamicCut_merge.pdf'</span>)</span><br><span class="line">	mergedColors = mergedMEs$colors</span><br><span class="line">	plotDendroAndColors(geneTree, cbind(dynamicColors, mergedColors),</span><br><span class="line">							c(<span class="string">'Dynamic Tree cut'</span>, <span class="string">'Dynamic Merged'</span>),</span><br><span class="line">							dendroLabels=<span class="literal">F</span>, hang=<span class="number">0.03</span>,</span><br><span class="line">	                    	addGuide=<span class="literal">T</span>, guideHang=<span class="number">0.05</span>,</span><br><span class="line">	                    	main=<span class="string">'Gene Dendrogram with Dynamic cut &amp; Merge'</span>)</span><br><span class="line">	dev.off()</span><br><span class="line"><span class="comment"># [visualization 7], 可视化模块特征聚类图</span></span><br><span class="line">	pdf(file=<span class="string">'visual_07_module_eigengenes_cluster.pdf'</span>)</span><br><span class="line">	plot(dissMEs_tree, main=<span class="string">'Module Eigengenes Clustering'</span>)</span><br><span class="line">	dev.off()</span><br><span class="line">save(MEs, dissMEs_tree, mergedMEs, file=<span class="string">'04_rawMEs_dissMEs_tree_mergedMEs.RData'</span>)</span><br><span class="line">rm(c(MEs, dissMEs_tree))</span><br></pre></td></tr></table></figure>
<h2 id="2-4-关联分析"><a href="#2-4-关联分析" class="headerlink" title="2.4 关联分析"></a>2.4 关联分析</h2><p>从“2.1-2.3”是我们进行WGCNA的分析过程，下面是一些具体的关联分析。所谓关联分析就是把我们得到的最终的聚类模块同其他数据如表型数据计算相关性，看有哪些模块与这些数据相关，那么模块中的基因就与这些数据相关。下面是一些示例。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 导入相关数据</span></span><br><span class="line">traits = read.table(<span class="string">'./clean_data/traits.txt'</span>)</span><br><span class="line"><span class="comment"># expr 如果不存在就load('01_sampleTree_cut_outlier.RData')</span></span><br><span class="line"><span class="comment"># mergedMEs如果不存在，就load('04_rawMEs_dissMEs_tree_mergedMEs.RData')</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 重新计算最终的模块特征, 并据此排序</span></span><br><span class="line">mergedColors = mergedMEs$colors</span><br><span class="line">MEs = moduleEigengenes(expr, mergedColors)$eigengenes</span><br><span class="line">MEs = orderMEs(MEs)</span><br><span class="line">nGenes = ncol(expr)</span><br><span class="line">nSamples = nrow(expr)</span><br><span class="line">modNames = substring(names(MEs), <span class="number">3</span>)</span><br><span class="line">moduleColors = mergedColors</span><br></pre></td></tr></table></figure>
<h3 id="2-4-1-模块与模块相关性"><a href="#2-4-1-模块与模块相关性" class="headerlink" title="2.4.1 模块与模块相关性"></a>2.4.1 模块与模块相关性</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 方法一</span></span><br><span class="line"><span class="comment"># plotMEpairs(MEs)</span></span><br><span class="line"><span class="comment"># 方法二</span></span><br><span class="line">moduleModuleCor = cor(MEs, MEs, use=<span class="string">'p'</span>)</span><br><span class="line">moduleModulePvalue = corPvalueStudent(moduleModuleCor, nSamples)</span><br><span class="line">moduleModuleText = paste0(signif(moduleModuleCor,<span class="number">2</span>), <span class="string">'\n('</span>,</span><br><span class="line">						signif(moduleModulePvalue,<span class="number">2</span>), <span class="string">')'</span>)</span><br><span class="line">dim(moduleModuleText) = dim(moduleModuleCor)</span><br><span class="line"><span class="comment"># [visualization 8] 模型与模块之间的相关性</span></span><br><span class="line">pdf(file=<span class="string">'visual_09_module_module_relationship.pdf'</span>)</span><br><span class="line">labeledHeatmap(Matrix=moduleModuleCor, xLabels=names(MEs),</span><br><span class="line">				yLabels=names(MEs), ySymbols=names(MEs),</span><br><span class="line">				xSymbols=names(MEs), colorLabels=<span class="literal">F</span>,</span><br><span class="line">				colors=greenWhiteRed(<span class="number">50</span>), textMatrix=moduleModuleText,</span><br><span class="line">				setStdMargins=<span class="literal">F</span>, zlim=c(-<span class="number">1</span>,<span class="number">1</span>),</span><br><span class="line">				main=paste(<span class="string">'Relationships between Module Eigengenes'</span>))</span><br><span class="line">dev.off()</span><br><span class="line">MMdata = cbind(moduleModuleCor, moduleModulePvalue)</span><br><span class="line">colnames(MMdata) = c(paste0(colnames(moduleModuleCor), <span class="string">'.Cor'</span>),</span><br><span class="line">					paste0(colnames(moduleModulePvalue), <span class="string">'.Pvalue'</span>))</span><br><span class="line">write.csv(MMdata, file=<span class="string">'WGCNA_02_Modue_Module_relationship.csv'</span>)</span><br></pre></td></tr></table></figure>
<h3 id="2-4-2-模块与表型之间的相关性"><a href="#2-4-2-模块与表型之间的相关性" class="headerlink" title="2.4.2 模块与表型之间的相关性"></a>2.4.2 模块与表型之间的相关性</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><span class="line">moduleTraitCor = cor(MEs, traits, use=<span class="string">'p'</span>)</span><br><span class="line">moduleTraitPvalue = corPvalueStudent(moduleTraitCor, nSamples)</span><br><span class="line">moduleTraitText = paste0(signif(moduleTraitCor,<span class="number">2</span>), <span class="string">'\n('</span>,</span><br><span class="line">						signif(moduleTraitPvalue,<span class="number">2</span>), <span class="string">')'</span>)</span><br><span class="line">dim(moduleTraitText) = dim(moduleTraitCor)</span><br><span class="line"><span class="comment"># [visualization 9] 模型与表型之间的相关性</span></span><br><span class="line">pdf(file=<span class="string">'visual_08_module_traits_relationship.pdf'</span>)</span><br><span class="line">labeledHeatmap(Matrix=moduleTraitCor, xLabels=names(traits),</span><br><span class="line">				yLabels=names(MEs), ySymbols=names(MEs),</span><br><span class="line">				colorLabels=<span class="literal">F</span>, colors=greenWhiteRed(<span class="number">50</span>),</span><br><span class="line">				textMatrix=moduleTraitText, setStdMargins=<span class="literal">F</span>,</span><br><span class="line">				zlim=c(-<span class="number">1</span>,<span class="number">1</span>),</span><br><span class="line">				main=paste(<span class="string">'Relationships between module eigengenes and traits '</span>))</span><br><span class="line">dev.off()</span><br><span class="line">MTdata = cbind(moduleTraitCor, moduleTraitPvalue)</span><br><span class="line">colnames(MTdata) = c(paste0(colnames(moduleTraitCor), <span class="string">'.Cor'</span>),</span><br><span class="line">					paste0(colnames(moduleTraitPvalue), <span class="string">'.Pvalue'</span>))</span><br><span class="line">write.csv(MTdata, file=<span class="string">'WGCNA_01_Modue_Traits_relationship.csv'</span>)</span><br></pre></td></tr></table></figure>
<h3 id="2-4-3-模块与基因表达量相关性"><a href="#2-4-3-模块与基因表达量相关性" class="headerlink" title="2.4.3 模块与基因表达量相关性"></a>2.4.3 模块与基因表达量相关性</h3><p>通过检测模块与表型之间的相关性，我们可以得到高度相关的模块与表型。使用这个模块和表型，获取要分析的基因。假定我们的模块是“blue”和表型是“weight”。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br></pre></td><td class="code"><pre><span class="line">geneModuleCor = cor(expr, MEs, use=<span class="string">'p'</span>) </span><br><span class="line"><span class="comment"># modNames = substring(names(MEs), 3)</span></span><br><span class="line">geneModulePvalue = corPvalueStudent(geneModuleCor, nSamples)</span><br><span class="line">names(geneModuleCor) = paste(modNames, <span class="string">'.Cor'</span>, sep=<span class="string">''</span>)</span><br><span class="line">names(geneModulePvalue) = paste(modNames, <span class="string">'.Cor.pvalue'</span>, sep=<span class="string">''</span>)</span><br><span class="line">write.csv(cbind(geneModuleCor, geneModulePvalue),</span><br><span class="line">	file=<span class="string">'WGCNA_03_gene_module_relationship.csv'</span>)</span><br><span class="line"></span><br><span class="line">geneTraitCor = cor(expr, traits, use=<span class="string">'p'</span>)</span><br><span class="line">geneTraitPvalue = corPvalueStudent(geneTraitCor, nSamples)</span><br><span class="line">names(geneTraitCor) = paste(colnames(traits), <span class="string">'.Cor'</span>, sep=<span class="string">''</span>)</span><br><span class="line">names(geneTraitPvalue) = paste(colnames(traits), <span class="string">'.Cor.pvalue'</span>, sep=<span class="string">''</span>)</span><br><span class="line">write.csv(cbind(geneTraitCor, geneTraitPvalue),</span><br><span class="line">	file=<span class="string">'WGCNA_04_gene_trait_relationship.csv'</span>)</span><br><span class="line"></span><br><span class="line">module_aim = <span class="string">'blue'</span></span><br><span class="line">trait_aim = <span class="string">'weight'</span></span><br><span class="line">module_column = match(module_aim, modNames)</span><br><span class="line">trait_column = match(trait_aim, colnames(traits))</span><br><span class="line">geneModuleCorAim = geneModuleCor[moduleColors == module_aim, module_column]</span><br><span class="line">geneTraitCorAim = geneTraitCor[moduleColors == module_aim, trait_column]</span><br><span class="line"><span class="comment"># [visualization 10] 模块与基因表达量的相关性</span></span><br><span class="line">pdf(file=paste(<span class="string">'visual_10_gene_significance_MM_in_'</span>, module_aim, <span class="string">'.pdf'</span>, sep=<span class="string">''</span>))</span><br><span class="line"><span class="comment"># 这里应该也可以用labledHeatmap</span></span><br><span class="line">verboseScatterplot(abs(geneModuleCorAim),</span><br><span class="line">					abs(geneTraitCorAim),</span><br><span class="line">					xlab=paste(<span class="string">'Module Membership in'</span>, module_aim, <span class="string">' module'</span>),</span><br><span class="line">					ylab=paste(<span class="string">'Gene significance for'</span>, trait_aim),</span><br><span class="line">					main=paste(<span class="string">'Module membership ('</span>, module_aim, <span class="string">') vs Gene significance ('</span>, trait_aim, <span class="string">')'</span>),</span><br><span class="line">					col = module_aim)</span><br><span class="line">dev.off()</span><br></pre></td></tr></table></figure>
<h3 id="2-4-4-模块与模块内总体基因显著值"><a href="#2-4-4-模块与模块内总体基因显著值" class="headerlink" title="2.4.4 模块与模块内总体基因显著值"></a>2.4.4 模块与模块内总体基因显著值</h3><p>“we define a gene significance variable as minus log10 of the univarite Cox regression pvalue for predicting survival on the basis of the gene epxression info“, 有文献定义了个gene significance, 然后可视化各个模块内该gene significance的分布情况。这说gen significance的分布情况。这说明gene significance是由我们自己定义的。我们这里使用每个基因的平均表达量作为其基因显著特征</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">exprPerGene = colMeans(expr)</span><br><span class="line"><span class="comment"># [visualizaton 11] 模块与总体基因表达量柱状图</span></span><br><span class="line">pdf(file=<span class="string">'visual_11_total_geneSig_in_module.pdf'</span>)</span><br><span class="line">plotModuleSignificance(exprPerGene, moduleColors, main=<span class="string">"Module significance"</span>)</span><br><span class="line">dev.off()</span><br></pre></td></tr></table></figure>
<h3 id="2-4-5-各个模块内的基因表达模式图"><a href="#2-4-5-各个模块内的基因表达模式图" class="headerlink" title="2.4.5 各个模块内的基因表达模式图"></a>2.4.5 各个模块内的基因表达模式图</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">pdf(file=paste(<span class="string">'visual_12_gene_expression_pattern_in_all_modules.pdf'</span>, sep=<span class="string">''</span>), w=<span class="number">12</span>,h=<span class="number">9</span>)</span><br><span class="line"><span class="keyword">for</span>(moduleName <span class="keyword">in</span> modNames) &#123;</span><br><span class="line">    column = match(moduleName, modNames)</span><br><span class="line">    sizeGrWindow(<span class="number">12</span>,<span class="number">9</span>)</span><br><span class="line">    par(mfrow=c(<span class="number">2</span>,<span class="number">1</span>))</span><br><span class="line">    par(mar=c(<span class="number">1</span>,<span class="number">3</span>,<span class="number">6</span>,<span class="number">3</span>))</span><br><span class="line">    plotMat(t(expr[, moduleColors == moduleName]),</span><br><span class="line">            nrgcols=<span class="number">30</span>, clabels=rownames(traits));</span><br><span class="line">    par(mar=c(<span class="number">2</span>,<span class="number">3</span>,<span class="number">2</span>,<span class="number">1</span>))</span><br><span class="line">    barplot(MEs[, column], xlab = moduleName, ylab=<span class="string">'Gene Expression pattern'</span>, col=moduleName)</span><br><span class="line">&#125;</span><br><span class="line">dev.off()</span><br></pre></td></tr></table></figure>
<h3 id="2-4-6-模块内部基因连接性与基因显著的相关性"><a href="#2-4-6-模块内部基因连接性与基因显著的相关性" class="headerlink" title="2.4.6 模块内部基因连接性与基因显著的相关性"></a>2.4.6 模块内部基因连接性与基因显著的相关性</h3><p>这里的基因显著定义还是与2.4.4是一样的。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">exprPerGene = colMeans(expr)</span><br><span class="line">ConnectivityMeasure = intramodularConnectivity(adjacency, moduleColors)</span><br><span class="line">pdf(file=paste(<span class="string">'visual_13_intramodularConnect_gene_significance.pdf'</span>, sep=<span class="string">''</span>))</span><br><span class="line"><span class="keyword">for</span>(moduleName <span class="keyword">in</span> modNames) &#123;</span><br><span class="line">	verboseScatterplot(ConnectivityMeasure$kWithin[moduleName == moduleColors],</span><br><span class="line">						exprPerGene[moduleName == moduleColors],</span><br><span class="line">						col=moduleColors[moduleColors == moduleName],</span><br><span class="line">						main=paste(<span class="string">'module '</span>, moduleName, sep=<span class="string">''</span>),</span><br><span class="line">						ylab=<span class="string">'Gene significance'</span>,</span><br><span class="line">						xlab=<span class="string">'Intramodular K'</span>)</span><br><span class="line">&#125;</span><br><span class="line">dev.off()</span><br></pre></td></tr></table></figure>
<h2 id="2-5-导出数据"><a href="#2-5-导出数据" class="headerlink" title="2.5 导出数据"></a>2.5 导出数据</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br></pre></td><td class="code"><pre><span class="line">load(<span class="string">'01_sampleTree_cut_outlier.RData'</span>)</span><br><span class="line">load(<span class="string">'03_conn_adj_TOM.RData'</span>)</span><br><span class="line">load(<span class="string">'04_rawMEs_dissMEs_tree_mergedMEs.RData'</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># softConnect, adjacency, TOM, geneTree,MEs, dissMEs_tree, mergedMEs</span></span><br><span class="line">probes = colnames(expr)</span><br><span class="line"><span class="comment"># 想要导出的模块</span></span><br><span class="line">modules = c(<span class="string">'green'</span>, <span class="string">'black'</span>)</span><br><span class="line">inModule = is.finite(match(mergedColors, modules))</span><br><span class="line">modTOM = TOM[inModule, inModule]</span><br><span class="line">modProbes = probes[inModule]</span><br><span class="line">dimnames(modTOM) = list(modProbes, modProbes)</span><br><span class="line">IMConn = softConnectivity(expr[, modProbes]);</span><br><span class="line"></span><br><span class="line">nTop = <span class="number">30</span>;</span><br><span class="line">top = (rank(-IMConn) &lt;= nTop)</span><br><span class="line"></span><br><span class="line">cyt = exportNetworkToCytoscape(modTOM[top, top],</span><br><span class="line">        edgeFile=paste(<span class="string">'CytoscapeInput-edges-'</span>, paste(modules, collapse=<span class="string">'-'</span>), <span class="string">'.txt'</span>, sep=<span class="string">''</span>),</span><br><span class="line">        nodeFile=paste(<span class="string">'CytoscapeInput-nodes-'</span>, paste(modules, collapse=<span class="string">'-'</span>), <span class="string">'.txt'</span>, sep=<span class="string">''</span>),</span><br><span class="line">        weighted=<span class="literal">T</span>,</span><br><span class="line">        threshold = <span class="number">0.02</span>,</span><br><span class="line">        nodeNames = modProbes,</span><br><span class="line">        nodeAttr=mergedColors[inModule])</span><br><span class="line"></span><br><span class="line">vis = exportNetworkToVisANT(modTOM[top, top],</span><br><span class="line">                            file = paste(<span class="string">"VisANTInput-"</span>, module, <span class="string">".txt"</span>, sep=<span class="string">""</span>),</span><br><span class="line">                            weighted = <span class="literal">TRUE</span>,</span><br><span class="line">                            threshold = <span class="number">0</span>,</span><br><span class="line">                            probeToGene = data.frame(annot$substanceBXH, annot$gene_symbol) )</span><br></pre></td></tr></table></figure>
	  
	</div>

	<!-- recommended posts -->
	

	<div>
  	<center>
	<div class="pagination">
<ul class="pagination">
	 
				
    	<li class="prev"><a href="/2018/10/12/GO_KEGG富集分析-clusterProfiler/" class="alignleft prev"><i class="fa fa-arrow-circle-o-left"></i>Prev</a></li>
  		

        <li><a href="/archives"><i class="fa fa-archive"></i>Archive</a></li>

		
		   <li class="next"><a href="/2018/10/12/差异分析-Ballgown/" class="alignright next">Next<i class="fa fa-arrow-circle-o-right"></i></a></li>         
        
	
</ul>
</div>

    </center>
	</div>

    <!-- share -->
    
        
    <div class="bdsharebuttonbox">
        <a href="#" class="bds_more" data-cmd="more"></a>
        <a href="#" class="bds_weixin" data-cmd="weixin" title="分享到微信"></a>
        <a href="#" class="bds_tsina" data-cmd="tsina" title="分享到新浪微博"></a>
        <a href="#" class="bds_fbook" data-cmd="fbook" title="分享到Facebook"></a>
        <a href="#" class="bds_twi" data-cmd="twi" title="分享到Twitter"></a>
        <a href="#" class="bds_linkedin" data-cmd="linkedin" title="分享到linkedin"></a>
        <a href="#" class="bds_evernotecn" data-cmd="evernotecn" title="分享到印象笔记"></a>
        <a href="#" class="bds_youdao" data-cmd="youdao" title="分享到有道云笔记"></a>
        <a href="#" class="bds_copy" data-cmd="copy" title="分享到复制网址"></a>
    </div>
    <script>
        window._bd_share_config={"common":{"bdSnsKey":{},"bdText":"","bdMini":"2","bdMiniList":false,"bdPic":"","bdStyle":"1","bdSize":"24"},"share":{}};
        with(document)0[(getElementsByTagName('head')[0]||body).appendChild(createElement('script')).src='http://bdimg.share.baidu.com/static/api/js/share.js?v=89860593.js?cdnversion='+~(-new Date()/36e5)];
    </script>


        

    
	
	<!-- comment -->
	
<section id="comment">
  <h2 class="title">Comments</h2>
  
</section>


	</div> <!-- col-md-9/col-md-12 -->
		
	
	<div id="side_meta">
		<div class="col-md-3" id="post_meta"> 

	<!-- date -->
	
	<div class="meta-widget">
	<i class="fa fa-clock-o"></i>
	2018-10-12 
	</div>
	

	<!-- categories -->
    
	<div class="meta-widget">
	<a data-toggle="collapse" data-target="#categorys"><i class="fa fa-folder"></i></a>	
    <ul id="categorys" class="tag_box list-unstyled collapse in">
          
  <li>
    </li><li><a href="/categories/Bioinformatics/">Bioinformatics<span>36</span></a></li>
  

    </ul>
	</div>
	

	<!-- tags -->
	
	<div class="meta-widget">
	<a data-toggle="collapse" data-target="#tags"><i class="fa fa-tags"></i></a>		  
    <ul id="tags" class="tag_box list-unstyled collapse in">	  
	    
  <li><a href="/tags/软件和包/">软件和包<span>22</span></a></li> <li><a href="/tags/富集分析/">富集分析<span>1</span></a></li>
    </ul>
	</div>
	

	<!-- toc -->
	<div class="meta-widget">
	
	   <a data-toggle="collapse" data-target="#toc"><i class="fa fa-bars"></i></a>
	   <div id="toc" class="toc collapse in">
			<ol class="toc-article"><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#一、总述"><span class="toc-article-text">一、总述</span></a><ol class="toc-article-child"><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#1-1-相较于GCNA-WGCNA的优势"><span class="toc-article-text">1.1 相较于GCNA,WGCNA的优势</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#1-2-相较于差异分析-WGCNA优势"><span class="toc-article-text">1.2 相较于差异分析,WGCNA优势</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#1-3-基本术语"><span class="toc-article-text">1.3 基本术语</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#1-4-基本过程"><span class="toc-article-text">1.4 基本过程</span></a></li></ol></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#二、分析过程"><span class="toc-article-text">二、分析过程</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#2-1-导入数据、预处理"><span class="toc-article-text">2.1 导入数据、预处理</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#2-2-获取软阈值"><span class="toc-article-text">2.2 获取软阈值</span></a></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#2-3-构建网络"><span class="toc-article-text">2.3 构建网络</span></a><ol class="toc-article-child"><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#2-3-3-计算拓扑网络和内部连接性，基于此构建聚类树"><span class="toc-article-text">2.3.3 计算拓扑网络和内部连接性，基于此构建聚类树</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#2-3-2-把拓扑网络聚类树进行动态修剪，产生初始模块"><span class="toc-article-text">2.3.2 把拓扑网络聚类树进行动态修剪，产生初始模块</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#2-3-3-计算模块特征，合并特征相似或靠近的模块，产生最终模块"><span class="toc-article-text">2.3.3 计算模块特征，合并特征相似或靠近的模块，产生最终模块</span></a></li></ol></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#2-4-关联分析"><span class="toc-article-text">2.4 关联分析</span></a><ol class="toc-article-child"><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#2-4-1-模块与模块相关性"><span class="toc-article-text">2.4.1 模块与模块相关性</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#2-4-2-模块与表型之间的相关性"><span class="toc-article-text">2.4.2 模块与表型之间的相关性</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#2-4-3-模块与基因表达量相关性"><span class="toc-article-text">2.4.3 模块与基因表达量相关性</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#2-4-4-模块与模块内总体基因显著值"><span class="toc-article-text">2.4.4 模块与模块内总体基因显著值</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#2-4-5-各个模块内的基因表达模式图"><span class="toc-article-text">2.4.5 各个模块内的基因表达模式图</span></a></li><li class="toc-article-item toc-article-level-3"><a class="toc-article-link" href="#2-4-6-模块内部基因连接性与基因显著的相关性"><span class="toc-article-text">2.4.6 模块内部基因连接性与基因显著的相关性</span></a></li></ol></li><li class="toc-article-item toc-article-level-2"><a class="toc-article-link" href="#2-5-导出数据"><span class="toc-article-text">2.5 导出数据</span></a></li></ol>
		</div>
	
	</div>
	
    <hr>
	
</div><!-- col-md-3 -->

	</div>
		

</div><!-- row -->



	</div>
  </div>
  <div class="container-narrow">
  <footer> <p>
  &copy; 2018 Xizhihui
  
      with help from <a href="http://hexo.io/" target="_blank">Hexo</a> and <a href="http://getbootstrap.com/" target="_blank">Twitter Bootstrap</a>. Theme by <a href="http://github.com/wzpan/hexo-theme-freemind/">Freemind</a>.    
</p> </footer>
</div> <!-- container-narrow -->
  


  
<a id="gotop" href="#">   
  <span>▲</span> 
</a>

<script src="/js/jquery.imagesloaded.min.js"></script>
<script src="/js/gallery.js"></script>
<script src="/js/bootstrap.min.js"></script>
<script src="/js/main.js"></script>
<script src="/js/search.js"></script> 


<link rel="stylesheet" href="/fancybox/jquery.fancybox.css" media="screen" type="text/css">
<script src="/fancybox/jquery.fancybox.pack.js"></script>
<script type="text/javascript">
(function($){
  $('.fancybox').fancybox();
})(jQuery);
</script>



   <script type="text/javascript">      
     var search_path = "search.xml";
	 if (search_path.length == 0) {
	 	search_path = "search.xml";
	 }
	 var path = "/" + search_path;
     searchFunc(path, 'local-search-input', 'local-search-result');
   </script>


<!-- syntax highlighting -->


</body>
   </html>
